// @novastar/native/build/module/generated/AutoSetStepType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var AutoSetStepTypeEnum;
(function (AutoSetStepTypeEnum) {
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["Module"] = 0] = "Module";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["OE"] = 1] = "OE";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["ModuleDataGroup"] = 2] = "ModuleDataGroup";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["DataGroupSequence"] = 3] = "DataGroupSequence";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["RGBMapping"] = 4] = "RGBMapping";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["ScanType"] = 5] = "ScanType";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["ScanLineArray"] = 6] = "ScanLineArray";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["ScanLineArray_NoCode"] = 7] = "ScanLineArray_NoCode";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["UnusedScan"] = 8] = "UnusedScan";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["FirstScanLine"] = 9] = "FirstScanLine";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["PointDistribute"] = 10] = "PointDistribute";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["New_OE"] = 11] = "New_OE";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["New_RGBMapping"] = 12] = "New_RGBMapping";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["New_DataGroup"] = 13] = "New_DataGroup";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["New_IrregularDataGroup"] = 14] = "New_IrregularDataGroup";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["New_ScanType"] = 15] = "New_ScanType";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["New_PointDistribute"] = 16] = "New_PointDistribute";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["New_Complete"] = 17] = "New_Complete";
    AutoSetStepTypeEnum[AutoSetStepTypeEnum["None"] = 18] = "None";
})(AutoSetStepTypeEnum || (AutoSetStepTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link AutoSetStepTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:59257
 */
export const AutoSetStepType = EnumFromString(AutoSetStepTypeEnum, 'AutoSetStepType');
