// @novastar/native/build/module/generated/AutoAdjustFreqType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var AutoAdjustFreqTypeEnum;
(function (AutoAdjustFreqTypeEnum) {
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_50HZ"] = 192] = "Freq_50HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_60HZ"] = 193] = "Freq_60HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_75HZ"] = 194] = "Freq_75HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_120HZ"] = 195] = "Freq_120HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_48HZ"] = 196] = "Freq_48HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_30HZ"] = 197] = "Freq_30HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_85HZ"] = 198] = "Freq_85HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_100HZ"] = 199] = "Freq_100HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_24HZ"] = 200] = "Freq_24HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_72HZ"] = 201] = "Freq_72HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_25HZ"] = 202] = "Freq_25HZ";
    AutoAdjustFreqTypeEnum[AutoAdjustFreqTypeEnum["Freq_240HZ"] = 203] = "Freq_240HZ";
})(AutoAdjustFreqTypeEnum || (AutoAdjustFreqTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link AutoAdjustFreqTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1461
 */
export const AutoAdjustFreqType = EnumFromString(AutoAdjustFreqTypeEnum, 'AutoAdjustFreqType');
