// @novastar/native/build/module/generated/AutoAdjustRefreshRateType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var AutoAdjustRefreshRateTypeEnum;
(function (AutoAdjustRefreshRateTypeEnum) {
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50"] = 0] = "R_50";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60"] = 1] = "R_60";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75"] = 2] = "R_75";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120"] = 3] = "R_120";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48"] = 4] = "R_48";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30"] = 5] = "R_30";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85"] = 6] = "R_85";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100"] = 7] = "R_100";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24"] = 8] = "R_24";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72"] = 9] = "R_72";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25"] = 10] = "R_25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240"] = 11] = "R_240";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144"] = 12] = "R_144";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50_N05"] = 16] = "R_50_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60_N05"] = 17] = "R_60_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75_N05"] = 18] = "R_75_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120_N05"] = 19] = "R_120_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48_N05"] = 20] = "R_48_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30_N05"] = 21] = "R_30_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85_N05"] = 22] = "R_85_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100_N05"] = 23] = "R_100_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24_N05"] = 24] = "R_24_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72_N05"] = 25] = "R_72_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25_N05"] = 26] = "R_25_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240_N05"] = 27] = "R_240_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144_N05"] = 28] = "R_144_N05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50_N10"] = 32] = "R_50_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60_N10"] = 33] = "R_60_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75_N10"] = 34] = "R_75_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120_N10"] = 35] = "R_120_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48_N10"] = 36] = "R_48_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30_N10"] = 37] = "R_30_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85_N10"] = 38] = "R_85_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100_N10"] = 39] = "R_100_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24_N10"] = 40] = "R_24_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72_N10"] = 41] = "R_72_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25_N10"] = 42] = "R_25_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240_N10"] = 43] = "R_240_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144_N10"] = 44] = "R_144_N10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50_P05"] = 48] = "R_50_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60_P05"] = 49] = "R_60_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75_P05"] = 50] = "R_75_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120_P05"] = 51] = "R_120_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48_P05"] = 52] = "R_48_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30_P05"] = 53] = "R_30_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85_P05"] = 54] = "R_85_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100_P05"] = 55] = "R_100_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24_P05"] = 56] = "R_24_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72_P05"] = 57] = "R_72_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25_P05"] = 58] = "R_25_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240_P05"] = 59] = "R_240_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144_P05"] = 60] = "R_144_P05";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50_P10"] = 64] = "R_50_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60_P10"] = 65] = "R_60_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75_P10"] = 66] = "R_75_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120_P10"] = 67] = "R_120_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48_P10"] = 68] = "R_48_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30_P10"] = 69] = "R_30_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85_P10"] = 70] = "R_85_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100_P10"] = 71] = "R_100_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24_P10"] = 72] = "R_24_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72_P10"] = 73] = "R_72_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25_P10"] = 74] = "R_25_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240_P10"] = 75] = "R_240_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144_P10"] = 76] = "R_144_P10";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50_P15"] = 80] = "R_50_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60_P15"] = 81] = "R_60_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75_P15"] = 82] = "R_75_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120_P15"] = 83] = "R_120_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48_P15"] = 84] = "R_48_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30_P15"] = 85] = "R_30_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85_P15"] = 86] = "R_85_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100_P15"] = 87] = "R_100_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24_P15"] = 88] = "R_24_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72_P15"] = 89] = "R_72_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25_P15"] = 90] = "R_25_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240_P15"] = 91] = "R_240_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144_P15"] = 92] = "R_144_P15";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50_P20"] = 96] = "R_50_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60_P20"] = 97] = "R_60_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75_P20"] = 98] = "R_75_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120_P20"] = 99] = "R_120_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48_P20"] = 100] = "R_48_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30_P20"] = 101] = "R_30_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85_P20"] = 102] = "R_85_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100_P20"] = 103] = "R_100_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24_P20"] = 104] = "R_24_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72_P20"] = 105] = "R_72_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25_P20"] = 106] = "R_25_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240_P20"] = 107] = "R_240_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144_P20"] = 108] = "R_144_P20";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_50_P25"] = 112] = "R_50_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_60_P25"] = 113] = "R_60_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_75_P25"] = 114] = "R_75_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_120_P25"] = 115] = "R_120_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_48_P25"] = 116] = "R_48_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_30_P25"] = 117] = "R_30_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_85_P25"] = 118] = "R_85_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_100_P25"] = 119] = "R_100_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_24_P25"] = 120] = "R_24_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_72_P25"] = 121] = "R_72_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_25_P25"] = 122] = "R_25_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_240_P25"] = 123] = "R_240_P25";
    AutoAdjustRefreshRateTypeEnum[AutoAdjustRefreshRateTypeEnum["R_144_P25"] = 124] = "R_144_P25";
})(AutoAdjustRefreshRateTypeEnum || (AutoAdjustRefreshRateTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link AutoAdjustRefreshRateTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.LEDConfigAccessorBase.decompiled.cs:206
 */
export const AutoAdjustRefreshRateType = EnumFromString(AutoAdjustRefreshRateTypeEnum, 'AutoAdjustRefreshRateType');
