// @novastar/native/build/module/generated/BaudRateType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var BaudRateTypeEnum;
(function (BaudRateTypeEnum) {
    BaudRateTypeEnum[BaudRateTypeEnum["BT_115200bps"] = 0] = "BT_115200bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_57600bps"] = 1] = "BT_57600bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_38400bps"] = 2] = "BT_38400bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_19200bps"] = 3] = "BT_19200bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_9600bps"] = 4] = "BT_9600bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_4800bps"] = 5] = "BT_4800bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_2400bps"] = 6] = "BT_2400bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_1200bps"] = 7] = "BT_1200bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_Reserved"] = 8] = "BT_Reserved";
})(BaudRateTypeEnum || (BaudRateTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link BaudRateTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1091
 */
export const BaudRateType = EnumFromString(BaudRateTypeEnum, 'BaudRateType');
