// @novastar/native/build/module/generated/CardType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var CardTypeEnum;
(function (CardTypeEnum) {
    CardTypeEnum[CardTypeEnum["Scanner_Card"] = 0] = "Scanner_Card";
    CardTypeEnum[CardTypeEnum["FunctionCard_1"] = 1] = "FunctionCard_1";
    CardTypeEnum[CardTypeEnum["FunctionCard_2"] = 2] = "FunctionCard_2";
    CardTypeEnum[CardTypeEnum["FunctionCard_3"] = 3] = "FunctionCard_3";
    CardTypeEnum[CardTypeEnum["FunctionCard_4"] = 4] = "FunctionCard_4";
    CardTypeEnum[CardTypeEnum["FunctionCard_5"] = 5] = "FunctionCard_5";
    CardTypeEnum[CardTypeEnum["FunctionCard_6"] = 6] = "FunctionCard_6";
    CardTypeEnum[CardTypeEnum["FunctionCard_7"] = 7] = "FunctionCard_7";
    CardTypeEnum[CardTypeEnum["FunctionCard_8"] = 8] = "FunctionCard_8";
    CardTypeEnum[CardTypeEnum["FunctionCard_9"] = 9] = "FunctionCard_9";
    CardTypeEnum[CardTypeEnum["FunctionCard_10"] = 10] = "FunctionCard_10";
    CardTypeEnum[CardTypeEnum["FunctionCard_11"] = 11] = "FunctionCard_11";
    CardTypeEnum[CardTypeEnum["FunctionCard_12"] = 12] = "FunctionCard_12";
    CardTypeEnum[CardTypeEnum["FunctionCard_13"] = 13] = "FunctionCard_13";
    CardTypeEnum[CardTypeEnum["FunctionCard_14"] = 14] = "FunctionCard_14";
    CardTypeEnum[CardTypeEnum["FunctionCard_15"] = 15] = "FunctionCard_15";
})(CardTypeEnum || (CardTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link CardTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:953
 */
export const CardType = EnumFromString(CardTypeEnum, 'CardType');
