// @novastar/native/build/module/generated/Chip5253BRGBExtendPropety.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { Chip5253BExtendProperty } from './Chip5253BExtendProperty';
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5253BRGBExtendPropety}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:31904
 */
export const Chip5253BRGBExtendPropety = t.intersection([
    t.type({
        EnableLowImprovement: common.Bool_true,
        EnShadeTransitOptimize: common.Bool_true,
        EnGCLKMutiRate: common.Bool_true,
        RedRegValueConfigThird: common.withDefault(common.UInt16, 49251),
        GreenRegValueConfigThird: common.withDefault(common.UInt16, 28771),
        BlueRegValueConfigThird: common.withDefault(common.UInt16, 24675),
        RedRegValueConfigFour: common.UInt16_61440,
        GreenRegValueConfigFour: common.UInt16_61440,
        BlueRegValueConfigFour: common.UInt16_61440,
        VRedRegValueConfigFour: common.UInt16_61440,
        RedRegValueConfigFive: common.UInt16_8673,
        GreenRegValueConfigFive: common.UInt16_8673,
        BlueRegValueConfigFive: common.UInt16_8673,
        VRedRegValueConfigFive: common.UInt16_8673,
        RedRegValueConfigSix: common.UInt16_3822,
        GreenRegValueConfigSix: common.UInt16_3822,
        BlueRegValueConfigSix: common.UInt16_3822,
        VRedRegValueConfigSix: common.UInt16_3822,
    }),
    t.partial({
        RedProperty: Chip5253BExtendProperty,
        GreenProperty: Chip5253BExtendProperty,
        BlueProperty: Chip5253BExtendProperty,
        VRedProperty: Chip5253BExtendProperty,
        EnDoubleRefresh: common.Bool,
        FailureLEDElimination: common.Bool,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigSeven: common.UInt16,
        GreenRegValueConfigSeven: common.UInt16,
        BlueRegValueConfigSeven: common.UInt16,
        VRedRegValueConfigSeven: common.UInt16,
        RedRegValueConfigSpecial: common.UInt16,
        GreenRegValueConfigSpecial: common.UInt16,
        BlueRegValueConfigSpecial: common.UInt16,
        VRedRegValueConfigSpecial: common.UInt16,
    }),
], 'Chip5253BRGBExtendPropety');
