// @novastar/native/build/module/generated/ChipMBI5850AutoRefreshRataParam.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { AutoAdjustFreqConfigMode } from './AutoAdjustFreqConfigMode';
import { ChipAutoAdjustRefreshRateParamBase } from './ChipAutoAdjustRefreshRateParamBase';
import { ChipType } from './ChipType';
export const ChipMBI5850AutoRefreshRataParamBase = t.intersection([
    ChipAutoAdjustRefreshRateParamBase,
    t.partial({
        ChipType,
        ChipLibVersion: common.UInt8,
        ConfigMode: AutoAdjustFreqConfigMode,
        AutoAdjustParamAddress: common.UInt32,
        AutoAdjustParamDataLen: common.Int32,
    }),
], 'ChipMBI5850AutoRefreshRataParamBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5850AutoRefreshRataParam}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5850.decompiled.cs:83
 */
export const ChipMBI5850AutoRefreshRataParam = t.intersection([
    ChipMBI5850AutoRefreshRataParamBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5850AutoRefreshRataParam') }),
], 'ChipMBI5850AutoRefreshRataParam');
