// @novastar/native/build/module/generated/ChipMBI5253NewBRGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5253NewBExtendProperty } from './ChipMBI5253NewBExtendProperty';
import { PointDetectParameter } from './PointDetectParameter';
export const ChipMBI5253NewBRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_14,
    }),
    t.partial({
        RedProperty: ChipMBI5253NewBExtendProperty,
        GreenProperty: ChipMBI5253NewBExtendProperty,
        BlueProperty: ChipMBI5253NewBExtendProperty,
        VRedProperty: ChipMBI5253NewBExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FailureLEDElimination: common.Bool,
        GclkAddedNumer: common.UInt8,
        EnGCLKMutiRate: common.Bool,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        ScanCount: common.UInt8,
        IsAdvancedMode: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        PointDetectParameter,
    }),
], 'ChipMBI5253NewBRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5253NewBRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5253NewB.decompiled.cs:1484
 */
export const ChipMBI5253NewBRGBVExtendProperty = t.intersection([
    ChipMBI5253NewBRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5253NewBRGBVExtendProperty') }),
], 'ChipMBI5253NewBRGBVExtendProperty');
