// @novastar/native/build/module/generated/ChipMBI5251RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5251ExtendProperty } from './ChipMBI5251ExtendProperty';
export const ChipMBI5251RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipMBI5251ExtendProperty,
        GreenProperty: ChipMBI5251ExtendProperty,
        BlueProperty: ChipMBI5251ExtendProperty,
        VRedProperty: ChipMBI5251ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        IsAdvancedMode: common.Bool,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
        RedRegValueConfigSecond: common.UInt16,
        GreenRegValueConfigSecond: common.UInt16,
        BlueRegValueConfigSecond: common.UInt16,
        VRedRegValueConfigSecond: common.UInt16,
        RedRegValueConfigThird: common.UInt16,
        GreenRegValueConfigThird: common.UInt16,
        BlueRegValueConfigThird: common.UInt16,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigFour: common.UInt16,
        GreenRegValueConfigFour: common.UInt16,
        BlueRegValueConfigFour: common.UInt16,
        VRedRegValueConfigFour: common.UInt16,
        RedRegValueConfigFive: common.UInt16,
        GreenRegValueConfigFive: common.UInt16,
        BlueRegValueConfigFive: common.UInt16,
        VRedRegValueConfigFive: common.UInt16,
        RedRegValueConfigSix: common.UInt16,
        GreenRegValueConfigSix: common.UInt16,
        BlueRegValueConfigSix: common.UInt16,
        VRedRegValueConfigSix: common.UInt16,
    }),
], 'ChipMBI5251RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5251RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5251.decompiled.cs:1238
 */
export const ChipMBI5251RGBVExtendProperty = t.intersection([
    ChipMBI5251RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5251RGBVExtendProperty') }),
], 'ChipMBI5251RGBVExtendProperty');
