// @novastar/native/build/module/generated/ChipCS2017ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipCS2017ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        FirstRegistor: common.UInt16_7936,
    }),
    t.partial({
        LineEliminationEnable: common.Bool,
        ShadowEliminationLevel: common.UInt8,
        ResponseSpeed: common.UInt8,
        WorkMode: common.UInt8,
        SetCurrentPosition: common.UInt8,
        InflectionCurrent: common.UInt8,
        DMAWriteMode: common.UInt8,
        CurrentGain: common.UInt8,
    }),
], 'ChipCS2017ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2017ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2017.decompiled.cs:78
 */
export const ChipCS2017ExtendProperty = t.intersection([
    ChipCS2017ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2017ExtendProperty') }),
], 'ChipCS2017ExtendProperty');
