// @novastar/native/build/module/generated/ChipMBI5253NewBExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipMBI5253NewBExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_14,
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        DarkBlockCompensationOne: common.UInt8,
        DarkBlockCompensationTwo: common.UInt8,
        DarkBlockCompensationThree: common.UInt8,
        FirstSweepElimination: common.UInt8,
        UnderEliminateGhost: common.UInt8,
        GrayHorizontalStripesEliminateEn: common.Bool,
        LowAshColorCastCompensation: common.UInt8,
        AdvancedRLowAshColorCastCompensation1: common.UInt8,
        AdvancedRLowAshColorCastCompensation2: common.UInt8,
        AdvancedRLowAshColorCastCompensation3: common.UInt8,
        GradientCompensation: common.UInt8,
        GradientCompensation1: common.UInt8,
        DislodgeBadPointsGrade: common.UInt8,
        CurrentGain: common.UInt8,
    }),
], 'ChipMBI5253NewBExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5253NewBExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5253NewB.decompiled.cs:1235
 */
export const ChipMBI5253NewBExtendProperty = t.intersection([
    ChipMBI5253NewBExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5253NewBExtendProperty') }),
], 'ChipMBI5253NewBExtendProperty');
