// @novastar/native/build/module/generated/ChipDP3264ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipDP3264ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        ConstantCurrentOutputInflectionPointGrade: common.UInt8,
        LowGreyPittingOptimization: common.UInt8,
        GammaSmoothingLevel: common.UInt8,
        ShadowEliminationLevelEnable: common.Bool,
        ShadowEliminationLevel: common.UInt8,
        ShadowEliminationTime: common.UInt8,
        ShadowEliminationMoment: common.UInt8,
        CouplingOptimizationEnable: common.Bool,
        CouplingOptimizationEnhance: common.Bool,
        CouplingOptimizationGrade: common.UInt8,
        CouplingCoefficient: common.UInt8,
        CurrentGain: common.UInt16,
    }),
], 'ChipDP3264ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipDP3264ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipDP3264.decompiled.cs:6833
 */
export const ChipDP3264ExtendProperty = t.intersection([
    ChipDP3264ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipDP3264ExtendProperty') }),
], 'ChipDP3264ExtendProperty');
