// @novastar/native/build/module/generated/Chip2163ExtendPropety.js
import * as t from 'io-ts';
import * as common from '../lib/common';
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip2163ExtendPropety}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:50629
 */
export const Chip2163ExtendPropety = t.intersection([
    t.type({
        EnhancedModeSwitch: common.Bool_true,
        RLowAshPitsImprove: common.UInt8_1,
        GLowAshPitsImprove: common.UInt8_1,
        BLowAshPitsImprove: common.UInt8_1,
        RVanishingStrength: common.UInt8_31,
        GVanishingStrength: common.UInt8_28,
        BVanishingStrength: common.UInt8_23,
        RedDN: common.Bool_true,
        GreenDN: common.Bool_true,
        BlueDN: common.Bool_true,
        RdeGhost: common.Bool_true,
        GdeGhost: common.Bool_true,
        BdeGhost: common.Bool_true,
        RHeadDarkLineOpt: common.UInt8_1,
        GHeadDarkLineOpt: common.UInt8_1,
        BHeadDarkLineOpt: common.UInt8_1,
        RFirstLineSlantsDarkOptimization: common.UInt8_1,
        GFirstLineSlantsDarkOptimization: common.UInt8_1,
        BFirstLineSlantsDarkOptimization: common.UInt8_1,
    }),
    t.partial({
        EnhancedMode: common.UInt8,
        RAberrationOptimize: common.Bool,
        GAberrationOptimize: common.Bool,
        BAberrationOptimize: common.Bool,
        FailureLEDElimination: common.Bool,
        RColorAdjust: common.UInt8,
        GColorAdjust: common.UInt8,
        BColorAdjust: common.UInt8,
        RColorAdjustEn: common.Bool,
        GColorAdjustEn: common.Bool,
        BColorAdjustEn: common.Bool,
        RBlackingEnhance: common.Bool,
        GBlackingEnhance: common.Bool,
        BBlackingEnhance: common.Bool,
        RedRegValueConfigThird: common.UInt16,
        GreenRegValueConfigThird: common.UInt16,
        BlueRegValueConfigThird: common.UInt16,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigSpecial: common.UInt16,
        GreenRegValueConfigSpecial: common.UInt16,
        BlueRegValueConfigSpecial: common.UInt16,
        VRedRegValueConfigSpecial: common.UInt16,
    }),
], 'Chip2163ExtendPropety');
