// @novastar/native/build/module/generated/ChipSM16027ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipSM16027ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        IsAdvancedMode: common.Bool,
        RVanishingStrength: common.UInt8,
        GVanishingStrength: common.UInt8,
        BVanishingStrength: common.UInt8,
        RLowAshPitsImprove: common.UInt8,
        GLowAshPitsImprove: common.UInt8,
        BLowAshPitsImprove: common.UInt8,
        RLowAshImprovement: common.UInt8,
        GLowAshImprovement: common.UInt8,
        BLowAshImprovement: common.UInt8,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
    }),
], 'ChipSM16027ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16027ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:45530
 */
export const ChipSM16027ExtendProperty = t.intersection([
    ChipSM16027ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16027ExtendProperty') }),
], 'ChipSM16027ExtendProperty');
