// @novastar/native/build/module/generated/ChipMY9221ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { DriverCurrentSpeedType } from './DriverCurrentSpeedType';
export const ChipMY9221ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        DriverSpeed: common.withDefault(DriverCurrentSpeedType, 'Fast'),
        IsCurrentBreakUp: common.Bool_true,
        IsInternalOscillator: common.Bool_true,
        IsDriverLED: common.Bool_true,
        IsAutoRefreshPicture: common.Bool_true,
        IsPictrueRepeat: common.Bool_true,
    }),
    t.partial({}),
], 'ChipMY9221ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMY9221ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:43403
 */
export const ChipMY9221ExtendProperty = t.intersection([
    ChipMY9221ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMY9221ExtendProperty') }),
], 'ChipMY9221ExtendProperty');
