// @novastar/native/build/module/generated/ChipCS2017SExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipCS2017SExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        FirstRegistor: common.UInt16_7936,
    }),
    t.partial({
        LineEliminationEnable: common.Bool,
        ShadowEliminationLevel: common.UInt8,
        ResponseSpeed: common.UInt8,
        WorkMode: common.UInt8,
        SetCurrentPosition: common.UInt8,
        InflectionCurrent: common.UInt8,
        DMAWriteMode: common.UInt8,
        CurrentGain: common.UInt8,
    }),
], 'ChipCS2017SExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2017SExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2017S.decompiled.cs:81
 */
export const ChipCS2017SExtendProperty = t.intersection([
    ChipCS2017SExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2017SExtendProperty') }),
], 'ChipCS2017SExtendProperty');
