// @novastar/native/build/module/generated/ChipYY6018RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipYY6018RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        FirstRegData: new common.BufferFromBase64('FirstRegData', 8),
        SecondRegData: new common.BufferFromBase64('SecondRegData', 8),
        ThirdRegData: new common.BufferFromBase64('ThirdRegData', 8),
        FourthRegData: new common.BufferFromBase64('FourthRegData', 6),
        SixthRegData: new common.BufferFromBase64('SixthRegData', 6),
        SeventhRegData: new common.BufferFromBase64('SeventhRegData', 6),
        IsUseNewModule: common.Bool,
        LowGrayCompsention: common.UInt8,
        FirstLineDark: common.UInt8,
        LineShadowEliminationStrength: common.UInt8,
        ColumnShadowEliminationStrength: common.UInt8,
        PreFilledMode: common.UInt8,
        GrayMode: common.UInt8,
        DCM_M: common.UInt8,
        DCM_D: common.UInt8,
        WeightValueConfig: common.Base64,
        RedOpenDischargeEn: common.Bool,
        GreenOpenDischargeEn: common.Bool,
        BlueOpenDischargeEn: common.Bool,
        WhiteOpenDischargeEn: common.Bool,
        RedResistanceValue: common.UInt8,
        GreenResistanceValue: common.UInt8,
        BlueResistanceValue: common.UInt8,
        AlgorithmSelect: common.UInt8,
        PARI: common.UInt8,
        PAR2: common.UInt8,
        LEDStyle: common.UInt8,
        SynMode: common.UInt8,
        ICLKDIVNum: common.UInt8,
        ISYNCNUM: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        SixthDataLen: common.Int32,
        SixthStartIndex: common.Int32,
        SixthRegisterAddr: common.Int32,
        SeventhDataLen: common.Int32,
        SeventhStartIndex: common.Int32,
        SeventhRegisterAddr: common.Int32,
    }),
], 'ChipYY6018RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipYY6018RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipYY6018.decompiled.cs:274
 */
export const ChipYY6018RGBVExtendProperty = t.intersection([
    ChipYY6018RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipYY6018RGBVExtendProperty') }),
], 'ChipYY6018RGBVExtendProperty');
