// @novastar/native/build/module/generated/ChipICN2038SExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipICN2038SExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        RWhiteBalance: common.Int32_16,
        GWhiteBalance: common.Int32_16,
        BWhiteBalance: common.Int32_16,
        HandUpdata: common.Bool_true,
    }),
    t.partial({
        IsAdvancedMode: common.Bool,
        IsDeadPix: common.Bool,
        RGainPoint: common.Int32,
        GGainPoint: common.Int32,
        BGainPoint: common.Int32,
        RConstantCurrent: common.Int32,
        GConstantCurrent: common.Int32,
        BConstantCurrent: common.Int32,
        ROESignalSelect: common.Int32,
        GOESignalSelect: common.Int32,
        BOESignalSelect: common.Int32,
        REnableSignalSelect: common.Bool,
        GEnableSignalSelect: common.Bool,
        BEnableSignalSelect: common.Bool,
        RDelayTime: common.Int32,
        GDelayTime: common.Int32,
        BDelayTime: common.Int32,
        RBlankingSignalSelect: common.Int32,
        GBlankingSignalSelect: common.Int32,
        BBlankingSignalSelect: common.Int32,
        RFallingTimeSelect: common.Int32,
        GFallingTimeSelect: common.Int32,
        BFallingTimeSelect: common.Int32,
        RLATCHSelect: common.Int32,
        GLATCHSelect: common.Int32,
        BLATCHSelect: common.Int32,
        RAMSelect: common.Int32,
        GAMSelect: common.Int32,
        BAMSelect: common.Int32,
        RAMSignalSelect: common.Bool,
        GAMSignalSelect: common.Bool,
        BAMSignalSelect: common.Bool,
        RCurrentRangSelect: common.Int32,
        GCurrentRangSelect: common.Int32,
        BCurrentRangSelect: common.Int32,
        RCLKtoSDOSelect: common.Int32,
        GCLKtoSDOSelect: common.Int32,
        BCLKtoSDOSelect: common.Int32,
        ROEWidthSelect: common.Int32,
        GOEWidthSelect: common.Int32,
        BOEWidthSelect: common.Int32,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
        RedRegValueConfigSecond: common.UInt16,
        GreenRegValueConfigSecond: common.UInt16,
        BlueRegValueConfigSecond: common.UInt16,
        VRedRegValueConfigSecond: common.UInt16,
        BlankingMode: common.Int32,
    }),
], 'ChipICN2038SExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICN2038SExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:29185
 */
export const ChipICN2038SExtendProperty = t.intersection([
    ChipICN2038SExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICN2038SExtendProperty') }),
], 'ChipICN2038SExtendProperty');
