// @novastar/native/build/module/generated/ChipICND2159ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipICND2159ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ThirdRegValue: common.UInt16,
        FourthRegValue: common.UInt16,
        LowAshPitsImprove: common.UInt8,
        IsAberrationOptimize: common.Bool,
        IsElimateShadow: common.Bool,
        LowGrayAdjustGrade: common.UInt8,
        IsLowGrayAdjust: common.Bool,
        LowGrayCompensateGrade: common.UInt8,
        IsLowGrayCompensate: common.Bool,
        FirstLineSlantsDarkOpt: common.UInt8,
        CurrentGain: common.UInt16,
    }),
], 'ChipICND2159ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICND2159ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICND2159.decompiled.cs:6499
 */
export const ChipICND2159ExtendProperty = t.intersection([
    ChipICND2159ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICND2159ExtendProperty') }),
], 'ChipICND2159ExtendProperty');
