// @novastar/native/build/module/generated/ChipMBI5264RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5264ExtendProperty } from './ChipMBI5264ExtendProperty';
export const ChipMBI5264RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_16,
    }),
    t.partial({
        RedProperty: ChipMBI5264ExtendProperty,
        GreenProperty: ChipMBI5264ExtendProperty,
        BlueProperty: ChipMBI5264ExtendProperty,
        VRedProperty: ChipMBI5264ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FailureLEDElimination: common.Bool,
        GclkAddedNumer: common.UInt8,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        ScanCount: common.UInt8,
        CompsentionTime: common.UInt16,
        GradientTransitionOptimizationTime: common.UInt16,
        M: common.UInt8,
        N: common.UInt8,
        Div: common.UInt8,
        IsAdvancedMode: common.Bool,
        IsGrayEnhncedMode: common.Bool,
        DefaultRegisterType: common.UInt8,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
    }),
], 'ChipMBI5264RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5264RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5264.decompiled.cs:392
 */
export const ChipMBI5264RGBVExtendProperty = t.intersection([
    ChipMBI5264RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5264RGBVExtendProperty') }),
], 'ChipMBI5264RGBVExtendProperty');
