// @novastar/native/build/module/generated/ChipCNS7153RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipCNS7153ExtendProperty } from './ChipCNS7153ExtendProperty';
export const ChipCNS7153RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        RedProperty: ChipCNS7153ExtendProperty,
        GreenProperty: ChipCNS7153ExtendProperty,
        BlueProperty: ChipCNS7153ExtendProperty,
        VRedProperty: ChipCNS7153ExtendProperty,
        IsUseNewModule: common.Bool,
    }),
], 'ChipCNS7153RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCNS7153RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCNS7153.decompiled.cs:1972
 */
export const ChipCNS7153RGBVExtendProperty = t.intersection([
    ChipCNS7153RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCNS7153RGBVExtendProperty') }),
], 'ChipCNS7153RGBVExtendProperty');
