// @novastar/native/build/module/generated/Chip9748ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const Chip9748ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        LastGhostAbatement2: common.UInt8_2,
    }),
    t.partial({
        CurrentSet1: common.UInt8,
        CurrentSet2: common.UInt8,
        CurrentSet3: common.UInt8,
        CurrentGain: common.UInt8,
        LowGrayscaleCompensation: common.UInt8,
        FirstScanCompensation: common.UInt8,
        CompensationMode: common.UInt8,
        NextGhostAbatement: common.UInt8,
        Adjustment: common.UInt8,
        LastGhostAbatement1: common.UInt8,
        FailureLEDElimination: common.UInt8,
    }),
], 'Chip9748ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip9748ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:40206
 */
export const Chip9748ExtendProperty = t.intersection([Chip9748ExtendPropertyBase, t.partial({ '@_xsi:type': t.literal('Chip9748ExtendProperty') })], 'Chip9748ExtendProperty');
