// @novastar/native/build/module/generated/Chip5353ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const Chip5353ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        DeGhostLevel: common.UInt8,
        FirstSweepEliminateFirst: common.UInt8,
        FirstSweepEliminateTwo: common.UInt8,
        LowAshImprovementFirst: common.UInt8,
        LowAshImprovementTwo: common.UInt8,
        LowAshImprovementThrid: common.Bool,
        EnhancedMode: common.Bool,
        LowAshdimLineConpensentionY: common.Bool,
        OpenDetectPower: common.UInt8,
    }),
], 'Chip5353ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip5353ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:40710
 */
export const Chip5353ExtendProperty = t.intersection([Chip5353ExtendPropertyBase, t.partial({ '@_xsi:type': t.literal('Chip5353ExtendProperty') })], 'Chip5353ExtendProperty');
