// @novastar/native/build/module/generated/ChipCS2066RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipCS2066ExtendProperty } from './ChipCS2066ExtendProperty';
export const ChipCS2066RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipCS2066ExtendProperty,
        GreenProperty: ChipCS2066ExtendProperty,
        BlueProperty: ChipCS2066ExtendProperty,
        VRedProperty: ChipCS2066ExtendProperty,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        PLLFreqDivision: common.UInt8,
        PLLFreqDoubling: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        FifthDataLen: common.Int32,
        FifthStartIndex: common.Int32,
        FifthRegisterAddr: common.Int32,
        SixthDataLen: common.Int32,
        SixthStartIndex: common.Int32,
        SixthRegisterAddr: common.Int32,
        SeventhDataLen: common.Int32,
        SeventhStartIndex: common.Int32,
        SeventhRegisterAddr: common.Int32,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipCS2066RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2066RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2066.decompiled.cs:8337
 */
export const ChipCS2066RGBVExtendProperty = t.intersection([
    ChipCS2066RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2066RGBVExtendProperty') }),
], 'ChipCS2066RGBVExtendProperty');
