// @novastar/native/build/module/generated/ChipICND2055AutoRefreshRataParam.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { AutoAdjustFreqConfigMode } from './AutoAdjustFreqConfigMode';
import { ChipAutoAdjustRefreshRateParamBase } from './ChipAutoAdjustRefreshRateParamBase';
import { ChipType } from './ChipType';
export const ChipICND2055AutoRefreshRataParamBase = t.intersection([
    ChipAutoAdjustRefreshRateParamBase,
    t.partial({
        ChipType,
        ChipLibVersion: common.UInt8,
        ConfigMode: AutoAdjustFreqConfigMode,
        AutoAdjustParamAddress: common.UInt32,
        AutoAdjustParamDataLen: common.Int32,
    }),
], 'ChipICND2055AutoRefreshRataParamBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICND2055AutoRefreshRataParam}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICND2055.decompiled.cs:47
 */
export const ChipICND2055AutoRefreshRataParam = t.intersection([
    ChipICND2055AutoRefreshRataParamBase,
    t.partial({ '@_xsi:type': t.literal('ChipICND2055AutoRefreshRataParam') }),
], 'ChipICND2055AutoRefreshRataParam');
