// @novastar/native/build/module/generated/Chip5354ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const Chip5354ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        DeGhostLevel: common.UInt8,
        FirstSweepEliminateFirst: common.UInt8,
        FirstSweepEliminateTwo: common.UInt8,
        LowAshImprovementFirst: common.UInt8,
        LowAshImprovementTwo: common.UInt8,
        LowAshImprovementThrid: common.Bool,
        EnhancedMode: common.Bool,
        LowAshdimLineConpensentionY: common.Bool,
        OpenDetectPower: common.UInt8,
    }),
], 'Chip5354ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip5354ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:41297
 */
export const Chip5354ExtendProperty = t.intersection([Chip5354ExtendPropertyBase, t.partial({ '@_xsi:type': t.literal('Chip5354ExtendProperty') })], 'Chip5354ExtendProperty');
