// @novastar/native/build/module/generated/ChipFM6356ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipFM6356ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RegFirst: common.UInt16,
        RegSecond: common.UInt16,
        RegThird: common.UInt16,
        RegFourth: common.UInt16,
        RegFifth: common.UInt16,
        IsElimateShadow: common.Bool,
        Gain: common.UInt16,
        LowGrayAdjustGrade: common.UInt8,
        IsLowGrayAdjust: common.Bool,
        IsLowGrayCompensate: common.Bool,
        LowGrayCompensateGrade: common.UInt8,
        IsFirstLineSlantsDark: common.Bool,
        FirstLineSlantsDarkOptimization: common.UInt8,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
    }),
], 'ChipFM6356ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6356ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6356.decompiled.cs:127
 */
export const ChipFM6356ExtendProperty = t.intersection([
    ChipFM6356ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6356ExtendProperty') }),
], 'ChipFM6356ExtendProperty');
