// @novastar/native/build/module/generated/ChipFM6363ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipFM6363ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RegFirst: common.UInt16,
        RegSecond: common.UInt16,
        RegThird: common.UInt16,
        RegFourth: common.UInt16,
        RegFifth: common.UInt16,
        FirstLineSlantsDarkOptimization: common.UInt8,
        LowGrayAdjustGrade: common.UInt8,
        IsLowGrayAdjust: common.Bool,
        IsElimateShadow: common.Bool,
        IsLowGrayCompensate: common.Bool,
        LowGrayCompensateGrade: common.UInt8,
        Gain: common.UInt16,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
    }),
], 'ChipFM6363ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6363ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6363.decompiled.cs:858
 */
export const ChipFM6363ExtendProperty = t.intersection([
    ChipFM6363ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6363ExtendProperty') }),
], 'ChipFM6363ExtendProperty');
