// @novastar/native/build/module/generated/ChipMBI524ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipMBI524ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        ColorCompensate: common.UInt8_3,
        LowAshImproveEnable: common.Bool_true,
    }),
    t.partial({
        SetBlanking: common.UInt8,
        IsAdvancedMode: common.Bool,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
    }),
], 'ChipMBI524ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI524ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:42397
 */
export const ChipMBI524ExtendProperty = t.intersection([
    ChipMBI524ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI524ExtendProperty') }),
], 'ChipMBI524ExtendProperty');
