// @novastar/native/build/module/generated/Chip5153ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5153ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:31144
 */
export const Chip5153ExtendProperty = t.partial({
    EliminateAfterLow: common.UInt8,
    FirstSweepEliminationMaster: common.UInt8,
    FirstSweepEliminationMedium: common.Bool,
    FirstSweepEliminationSecond: common.UInt8,
    LowAshColorCompensationOne: common.UInt8,
    LowAshColorCompensationTwo: common.UInt8,
    LowAshEnhanceUniformity: common.Bool,
    GrayHorizontalStripesEliminate: common.Bool,
    EnhancedModeOne: common.Bool,
    SlowOpen: common.Bool,
    SwapErrorPoint: common.UInt8,
    EnhancedModeTwo: common.Bool,
    RedRegValueConfigThird: common.UInt16,
    GreenRegValueConfigThird: common.UInt16,
    BlueRegValueConfigThird: common.UInt16,
    VRedRegValueConfigThird: common.UInt16,
}, 'Chip5153ExtendProperty');
