// @novastar/native/build/module/generated/ChipMBI5268ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipMBI5268ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_16,
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        DarkBlockCompensationOneEn: common.Bool,
        DarkBlockCompensationOne: common.UInt8,
        DarkBlockCompensationTwo: common.UInt8,
        DarkBlockCompensationThree: common.UInt8,
        FirstSweepEliminationEn: common.Bool,
        FirstSweepElimination: common.UInt8,
        UnderEliminateGhost: common.UInt8,
        GrayHorizontalStripesEliminateEn: common.Bool,
        LowAshColorCastCompensation: common.UInt8,
        AdvancedRLowAshColorCastCompensation1: common.UInt8,
        AdvancedRLowAshColorCastCompensation2: common.UInt8,
        AdvancedRLowAshColorCastCompensation3: common.UInt8,
        GradientCompensationEnable: common.Bool,
        LowAshUniformityImproved1: common.UInt8,
        LowAshUniformityImproved2: common.UInt8,
        DislodgeBadPointsGrade: common.UInt8,
        CurrentGain: common.UInt8,
        VrextVoltageEnable: common.Bool,
    }),
], 'ChipMBI5268ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5268ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5268.decompiled.cs:8467
 */
export const ChipMBI5268ExtendProperty = t.intersection([
    ChipMBI5268ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5268ExtendProperty') }),
], 'ChipMBI5268ExtendProperty');
