// @novastar/native/build/module/generated/ChipICN2065AutoRefreshRataParam.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { AutoAdjustFreqConfigMode } from './AutoAdjustFreqConfigMode';
import { ChipAutoAdjustRefreshRateParamBase } from './ChipAutoAdjustRefreshRateParamBase';
import { ChipType } from './ChipType';
export const ChipICN2065AutoRefreshRataParamBase = t.intersection([
    ChipAutoAdjustRefreshRateParamBase,
    t.partial({
        ChipType,
        ChipLibVersion: common.UInt8,
        ConfigMode: AutoAdjustFreqConfigMode,
        AutoAdjustParamAddress: common.UInt32,
        AutoAdjustParamDataLen: common.Int32,
    }),
], 'ChipICN2065AutoRefreshRataParamBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICN2065AutoRefreshRataParam}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICND2065.decompiled.cs:47
 */
export const ChipICN2065AutoRefreshRataParam = t.intersection([
    ChipICN2065AutoRefreshRataParamBase,
    t.partial({ '@_xsi:type': t.literal('ChipICN2065AutoRefreshRataParam') }),
], 'ChipICN2065AutoRefreshRataParam');
