// @novastar/native/build/module/generated/ChipICND2126ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipICND2126ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ShortDetection: common.UInt8,
        ErrorSurvey_OnTime: common.Bool,
        SavingPowerMode: common.UInt8,
        CurrentGain: common.UInt16,
        ShadowEliminationEn: common.Bool,
        InflectionPiont: common.UInt8,
        ShadowEliminationElectric: common.UInt8,
    }),
], 'ChipICND2126ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICND2126ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICN2126.decompiled.cs:1268
 */
export const ChipICND2126ExtendProperty = t.intersection([
    ChipICND2126ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICND2126ExtendProperty') }),
], 'ChipICND2126ExtendProperty');
