// @novastar/native/build/module/generated/ChipSM16017SExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
/**
 * @category Codecs
 * @desc Codec for interface {@link ChipSM16017SExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:43154
 */
export const ChipSM16017SExtendProperty = t.intersection([
    t.type({
        REliminateAfterLow: common.UInt8_5,
        GEliminateAfterLow: common.UInt8_5,
        BEliminateAfterLow: common.UInt8_5,
        GDimCompensation: common.UInt8_3,
        BDimCompensation: common.UInt8_3,
        RLowAshImprovement: common.withDefault(common.UInt8, 6),
        GLowAshImprovement: common.UInt8_7,
        BLowAshImprovement: common.UInt8_7,
        RedRegValueConfigFirst: common.UInt16_41728,
        GreenRegValueConfigFirst: common.UInt16_44932,
        BlueRegValueConfigFirst: common.UInt16_44932,
        VRedRegValueConfigFirst: common.UInt16_41728,
    }),
    t.partial({
        RDimCompensation: common.UInt8,
        IsAdvancedMode: common.Bool,
    }),
], 'ChipSM16017SExtendProperty');
