// @novastar/native/build/module/generated/ChipRT5938SSRGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipRT5938SSExtendProperty } from './ChipRT5938SSExtendProperty';
import { PointDetectParameter } from './PointDetectParameter';
export const ChipRT5938SSRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_1,
    }),
    t.partial({
        RedProperty: ChipRT5938SSExtendProperty,
        GreenProperty: ChipRT5938SSExtendProperty,
        BlueProperty: ChipRT5938SSExtendProperty,
        VRedProperty: ChipRT5938SSExtendProperty,
        IsUseNewModule: common.Bool,
        IsAdvancedMode: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        PointDetectParameter,
    }),
], 'ChipRT5938SSRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipRT5938SSRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipRT5938SS.decompiled.cs:6754
 */
export const ChipRT5938SSRGBVExtendProperty = t.intersection([
    ChipRT5938SSRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipRT5938SSRGBVExtendProperty') }),
], 'ChipRT5938SSRGBVExtendProperty');
