// @novastar/native/build/module/generated/ChipFM6047ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipFM6047ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ThirdRegValue: common.UInt16,
        FourthRegValue: common.UInt16,
        OEWidth: common.UInt8,
        OEDelay: common.UInt8,
        Inflexion: common.UInt8,
        HiddenMode: common.UInt8,
        CurrentGain: common.UInt16,
        ShadowEliminationLevel: common.UInt8,
        OpenTestEnable: common.Bool,
        ErrCurrentGain: common.UInt16,
    }),
], 'ChipFM6047ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6047ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6047.decompiled.cs:6202
 */
export const ChipFM6047ExtendProperty = t.intersection([
    ChipFM6047ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6047ExtendProperty') }),
], 'ChipFM6047ExtendProperty');
