// @novastar/native/build/module/generated/ChipSM16227ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipSM16227ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RUnderEliminateGhostOne: common.Int32,
        GUnderEliminateGhostOne: common.Int32,
        BUnderEliminateGhostOne: common.Int32,
        RUnderEliminateGhostTwo: common.Int32,
        GUnderEliminateGhostTwo: common.Int32,
        BUnderEliminateGhostTwo: common.Int32,
        ROffsetCompsation: common.Int32,
        GOffsetCompsation: common.Int32,
        BOffsetCompsation: common.Int32,
        RFirstCompsation: common.Int32,
        GFirstCompsation: common.Int32,
        BFirstCompsation: common.Int32,
        IsAdvancedMode: common.Bool,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
    }),
], 'ChipSM16227ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16227ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:43987
 */
export const ChipSM16227ExtendProperty = t.intersection([
    ChipSM16227ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16227ExtendProperty') }),
], 'ChipSM16227ExtendProperty');
