// @novastar/native/build/module/generated/ChipHBS1910RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipHBS1910ExtendProperty } from './ChipHBS1910ExtendProperty';
export const ChipHBS1910RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipHBS1910ExtendProperty,
        GreenProperty: ChipHBS1910ExtendProperty,
        BlueProperty: ChipHBS1910ExtendProperty,
        VRedProperty: ChipHBS1910ExtendProperty,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
    }),
], 'ChipHBS1910RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipHBS1910RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipHBS1910.decompiled.cs:2600
 */
export const ChipHBS1910RGBVExtendProperty = t.intersection([
    ChipHBS1910RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipHBS1910RGBVExtendProperty') }),
], 'ChipHBS1910RGBVExtendProperty');
