// @novastar/native/build/module/generated/Chip2053ExtendPropety.js
import * as t from 'io-ts';
import * as common from '../lib/common';
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip2053ExtendPropety}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:38979
 */
export const Chip2053ExtendPropety = t.intersection([
    t.type({
        EnhancedModeSwitch: common.Bool_true,
        RLowAshPitsImprove: common.UInt8_1,
        GLowAshPitsImprove: common.UInt8_1,
        BLowAshPitsImprove: common.UInt8_1,
        RColorAdjust: common.UInt8_15,
        GColorAdjust: common.UInt8_15,
        BColorAdjust: common.UInt8_15,
        RVanishingStrength: common.UInt8_31,
        GVanishingStrength: common.withDefault(common.UInt8, 25),
        BVanishingStrength: common.UInt8_23,
        RdeGhost: common.Bool_true,
        GdeGhost: common.Bool_true,
        BdeGhost: common.Bool_true,
        RBlackingEnhance: common.Bool_true,
        GBlackingEnhance: common.Bool_true,
        BBlackingEnhance: common.Bool_true,
        RHeadDarkLineOpt: common.UInt8_1,
        GHeadDarkLineOpt: common.UInt8_1,
        BHeadDarkLineOpt: common.UInt8_1,
        RFirstLineSlantsDarkOptimization: common.UInt8_1,
        GFirstLineSlantsDarkOptimization: common.UInt8_1,
        BFirstLineSlantsDarkOptimization: common.UInt8_1,
        RRemoveBadPointGain: common.UInt16_1,
        GRemoveBadPointGain: common.UInt16_1,
        BRemoveBadPointGain: common.UInt16_1,
    }),
    t.partial({
        EnhancedMode: common.UInt8,
        RAberrationOptimize: common.Bool,
        GAberrationOptimize: common.Bool,
        BAberrationOptimize: common.Bool,
        FailureLEDElimination: common.Bool,
        RColorAdjustEn: common.Bool,
        GColorAdjustEn: common.Bool,
        BColorAdjustEn: common.Bool,
        RedRegValueConfigThird: common.UInt16,
        GreenRegValueConfigThird: common.UInt16,
        BlueRegValueConfigThird: common.UInt16,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigSpecial: common.UInt16,
        GreenRegValueConfigSpecial: common.UInt16,
        BlueRegValueConfigSpecial: common.UInt16,
        VRedRegValueConfigSpecial: common.UInt16,
        IsCustomMode: common.Bool,
        RedRegValueConfigFifth: common.UInt16,
        GreenRegValueConfigFifth: common.UInt16,
        BlueRegValueConfigFifth: common.UInt16,
    }),
], 'Chip2053ExtendPropety');
