// @novastar/native/build/module/generated/Chip5353RGBVExtendPropey.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { Chip5353ExtendProperty } from './Chip5353ExtendProperty';
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5353RGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:40987
 */
export const Chip5353RGBVExtendPropey = t.intersection([
    t.type({
        GlobalCurrentGainAdjust: common.UInt8_5,
    }),
    t.partial({
        RedProperty: Chip5353ExtendProperty,
        GreenProperty: Chip5353ExtendProperty,
        BlueProperty: Chip5353ExtendProperty,
        VRedProperty: Chip5353ExtendProperty,
        GlobalCurrentGainConst: common.UInt8,
        FailureLEDElimination: common.Bool,
        EnableLowImprovement: common.Bool,
        LowImprovementFirst: common.UInt8,
        LowImprovementTwo: common.UInt8,
        EnableLowAshUniformity: common.Bool,
        EnableEnergyConserve: common.Bool,
        EnableGCLKMutiplier: common.Bool,
        RedRegValueConfigThird: common.UInt16,
        GreenRegValueConfigThird: common.UInt16,
        BlueRegValueConfigThird: common.UInt16,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigSpecial: common.UInt16,
        GreenRegValueConfigSpecial: common.UInt16,
        BlueRegValueConfigSpecial: common.UInt16,
        VRedRegValueConfigSpecial: common.UInt16,
    }),
], 'Chip5353RGBVExtendPropey');
