// @novastar/native/build/module/generated/ChipDP3265RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipDP3265ExtendProperty } from './ChipDP3265ExtendProperty';
export const ChipDP3265RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_13,
        RefreshNum: common.Int32_32,
        MaxRefreshNum: common.Int32_128,
    }),
    t.partial({
        RedProperty: ChipDP3265ExtendProperty,
        GreenProperty: ChipDP3265ExtendProperty,
        BlueProperty: ChipDP3265ExtendProperty,
        VRedProperty: ChipDP3265ExtendProperty,
        RefreshNumPerVs: common.UInt8,
        LineGclkNum: common.UInt8,
        R: common.UInt8,
        LowGrayDisplayEnhance: common.Bool,
        BlackScreenDynamicEnergySaving: common.UInt8,
        HighGrayDataIndependentRefresh: common.UInt8,
        EnableToRemoveBadPoints: common.Bool,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        ErrRedGain: common.Int32,
        ErrGreenGain: common.Int32,
        ErrBlueGain: common.Int32,
        ErrVRedGain: common.Int32,
    }),
], 'ChipDP3265RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipDP3265RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipDP3265.decompiled.cs:7086
 */
export const ChipDP3265RGBVExtendProperty = t.intersection([
    ChipDP3265RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipDP3265RGBVExtendProperty') }),
], 'ChipDP3265RGBVExtendProperty');
