// @novastar/native/build/module/generated/ChipMBI5252BRGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5252BExtendProperty } from './ChipMBI5252BExtendProperty';
import { PointDetectParameter } from './PointDetectParameter';
export const ChipMBI5252BRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_14,
    }),
    t.partial({
        RedProperty: ChipMBI5252BExtendProperty,
        GreenProperty: ChipMBI5252BExtendProperty,
        BlueProperty: ChipMBI5252BExtendProperty,
        VRedProperty: ChipMBI5252BExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FailureLEDElimination: common.Bool,
        GclkAddedNumer: common.UInt8,
        EnGCLKMutiRate: common.Bool,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        ScanCount: common.UInt8,
        IsAdvancedMode: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        PointDetectParameter,
    }),
], 'ChipMBI5252BRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5252BRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5252B.decompiled.cs:1484
 */
export const ChipMBI5252BRGBVExtendProperty = t.intersection([
    ChipMBI5252BRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5252BRGBVExtendProperty') }),
], 'ChipMBI5252BRGBVExtendProperty');
