// @novastar/native/build/module/generated/Chip5253RGBExtendPropety.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { Chip5253ExtendProperty } from './Chip5253ExtendProperty';
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5253RGBExtendPropety}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:31426
 */
export const Chip5253RGBExtendPropety = t.partial({
    RedProperty: Chip5253ExtendProperty,
    GreenProperty: Chip5253ExtendProperty,
    BlueProperty: Chip5253ExtendProperty,
    VRedProperty: Chip5253ExtendProperty,
    EnableLowImprovement: common.Bool,
    EnShadeTransitOptimize: common.Bool,
    EnDoubleRefresh: common.Bool,
    EnGCLKMutiRate: common.Bool,
    FailureLEDElimination: common.Bool,
    RedRegValueConfigThird: common.UInt16,
    GreenRegValueConfigThird: common.UInt16,
    BlueRegValueConfigThird: common.UInt16,
    VRedRegValueConfigThird: common.UInt16,
    RedRegValueConfigFour: common.UInt16,
    GreenRegValueConfigFour: common.UInt16,
    BlueRegValueConfigFour: common.UInt16,
    VRedRegValueConfigFour: common.UInt16,
    RedRegValueConfigFive: common.UInt16,
    GreenRegValueConfigFive: common.UInt16,
    BlueRegValueConfigFive: common.UInt16,
    VRedRegValueConfigFive: common.UInt16,
    RedRegValueConfigSix: common.UInt16,
    GreenRegValueConfigSix: common.UInt16,
    BlueRegValueConfigSix: common.UInt16,
    VRedRegValueConfigSix: common.UInt16,
    RedRegValueConfigSpecial: common.UInt16,
    GreenRegValueConfigSpecial: common.UInt16,
    BlueRegValueConfigSpecial: common.UInt16,
    VRedRegValueConfigSpecial: common.UInt16,
}, 'Chip5253RGBExtendPropety');
