// @novastar/native/build/module/generated/ChipMBI5850RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5850CustomExtendProperty } from './ChipMBI5850CustomExtendProperty';
import { ChipMBI5850ExtendProperty } from './ChipMBI5850ExtendProperty';
export const ChipMBI5850RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        ChipMBI5850ExtendProperty,
        ChipMBI5850CustomExtendProperty,
        IsUseNewModule: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipMBI5850RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5850RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5850.decompiled.cs:308
 */
export const ChipMBI5850RGBVExtendProperty = t.intersection([
    ChipMBI5850RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5850RGBVExtendProperty') }),
], 'ChipMBI5850RGBVExtendProperty');
