// @novastar/native/build/module/generated/ChipSM16207SExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipSM16207SExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REliminateAfterLow1: common.UInt8_5,
        GEliminateAfterLow1: common.UInt8_5,
        BEliminateAfterLow1: common.UInt8_5,
        RDimCompensation: common.UInt8_3,
        GDimCompensation: common.UInt8_3,
        BDimCompensation: common.UInt8_3,
        REliminateAfterLow2: common.UInt8_5,
        GEliminateAfterLow2: common.UInt8_5,
        BEliminateAfterLow2: common.UInt8_5,
    }),
    t.partial({
        RLowAshImprovement: common.UInt8,
        GLowAshImprovement: common.UInt8,
        BLowAshImprovement: common.UInt8,
        IsAdvancedMode: common.Bool,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
    }),
], 'ChipSM16207SExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16207SExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:44711
 */
export const ChipSM16207SExtendProperty = t.intersection([
    ChipSM16207SExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16207SExtendProperty') }),
], 'ChipSM16207SExtendProperty');
