// @novastar/native/build/module/generated/ChipDM413RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipDM413ExtendProperty } from './ChipDM413ExtendProperty';
export const ChipDM413RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipDM413ExtendProperty,
        GreenProperty: ChipDM413ExtendProperty,
        BlueProperty: ChipDM413ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
    }),
], 'ChipDM413RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipDM413RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipDM413.decompiled.cs:80
 */
export const ChipDM413RGBVExtendProperty = t.intersection([
    ChipDM413RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipDM413RGBVExtendProperty') }),
], 'ChipDM413RGBVExtendProperty');
