// @novastar/native/build/module/generated/ChipCFD455AExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipCFD455AExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ThirdRegValue: common.UInt16,
        FourthRegValue: common.UInt16,
        LowAshEnhanceUniformity: common.UInt8,
        LineBlankStrength: common.UInt8,
        LineBlankEnhanceModeEn: common.Bool,
        CouplingStrengthAdjust: common.UInt8,
        OpenTestPointAdjust: common.UInt8,
        CouplingOptimizationEn: common.Bool,
        LowGrayCompsentionTimeSettingMode: common.UInt8,
        CouplingRangeAdjust: common.UInt8,
        PerDriverEnable: common.Bool,
        OpenProtectEnable: common.Bool,
        ScanMode: common.UInt8,
        CurrentGear: common.UInt8,
        LowGrayCompsentionStrength: common.UInt8,
        OpenDynamicTestEnable: common.Bool,
        CurrentCompsentionEnable: common.Bool,
        ConstantCurrentInflectionPoint: common.UInt8,
        DynamicEnergySavingEn: common.Bool,
        LineEliminationLength: common.UInt8,
        PerDriverMode: common.UInt8,
        PerDriverTimeSetting: common.UInt8,
        OpentestThresholdValue: common.UInt8,
        LowGrayGoodLevel: common.UInt8,
        SetWeakup: common.UInt8,
    }),
], 'ChipCFD455AExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCFD455AExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCFD455A.decompiled.cs:3371
 */
export const ChipCFD455AExtendProperty = t.intersection([
    ChipCFD455AExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCFD455AExtendProperty') }),
], 'ChipCFD455AExtendProperty');
