// @novastar/native/build/module/generated/Chip9748RGBVExtendPropey.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { Chip9748ExtendProperty } from './Chip9748ExtendProperty';
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip9748RGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:40424
 */
export const Chip9748RGBVExtendPropey = t.partial({
    RedProperty: Chip9748ExtendProperty,
    GreenProperty: Chip9748ExtendProperty,
    BlueProperty: Chip9748ExtendProperty,
    VRedProperty: Chip9748ExtendProperty,
    GrayscaleSelect: common.UInt8,
    DummyGclkEn: common.Bool,
    RedGain: common.Int32,
    BlueGain: common.Int32,
    GreenGain: common.Int32,
    RedRegValueConfigThird: common.UInt16,
    GreenRegValueConfigThird: common.UInt16,
    BlueRegValueConfigThird: common.UInt16,
    VRedRegValueConfigThird: common.UInt16,
    RedRegValueConfigSpecial: common.UInt16,
    GreenRegValueConfigSpecial: common.UInt16,
    BlueRegValueConfigSpecial: common.UInt16,
    VRedRegValueConfigSpecial: common.UInt16,
}, 'Chip9748RGBVExtendPropey');
