// @novastar/native/build/module/generated/ChipMBI5251ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipMBI5251ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        Reg_First: common.withDefault(common.UInt16, 1819),
        Reg_Second: common.withDefault(common.UInt16, 1536),
        Reg_Third: common.UInt16_65535,
        Reg_Fourth: common.UInt16_65535,
        Reg_Fifth: common.withDefault(common.UInt16, 28959),
        Reg_Sixth: common.withDefault(common.UInt16, 320),
    }),
    t.partial({}),
], 'ChipMBI5251ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5251ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5251.decompiled.cs:1069
 */
export const ChipMBI5251ExtendProperty = t.intersection([
    ChipMBI5251ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5251ExtendProperty') }),
], 'ChipMBI5251ExtendProperty');
