// @novastar/native/build/module/generated/ChipMBI5754RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5754ExtendProperty } from './ChipMBI5754ExtendProperty';
import { PointDetectParameter } from './PointDetectParameter';
export const ChipMBI5754RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_16,
    }),
    t.partial({
        RedProperty: ChipMBI5754ExtendProperty,
        GreenProperty: ChipMBI5754ExtendProperty,
        BlueProperty: ChipMBI5754ExtendProperty,
        VRedProperty: ChipMBI5754ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        PointDetectParameter,
        FailureLEDElimination: common.Bool,
        GclkAddedNumer: common.UInt8,
        EnGCLKMutiRate: common.Bool,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        ScanCount: common.UInt8,
        IsAdvancedMode: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
    }),
], 'ChipMBI5754RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5754RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5754.decompiled.cs:336
 */
export const ChipMBI5754RGBVExtendProperty = t.intersection([
    ChipMBI5754RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5754RGBVExtendProperty') }),
], 'ChipMBI5754RGBVExtendProperty');
