// @novastar/native/build/module/generated/ChipCS2017RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipCS2017ExtendProperty } from './ChipCS2017ExtendProperty';
export const ChipCS2017RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipCS2017ExtendProperty,
        GreenProperty: ChipCS2017ExtendProperty,
        BlueProperty: ChipCS2017ExtendProperty,
        VRedProperty: ChipCS2017ExtendProperty,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ConfigDataLen: common.Int32,
    }),
], 'ChipCS2017RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2017RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2017.decompiled.cs:216
 */
export const ChipCS2017RGBVExtendProperty = t.intersection([
    ChipCS2017RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2017RGBVExtendProperty') }),
], 'ChipCS2017RGBVExtendProperty');
