// @novastar/native/build/module/generated/ChipMBI5268RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5268ExtendProperty } from './ChipMBI5268ExtendProperty';
import { PointDetectParameter } from './PointDetectParameter';
export const ChipMBI5268RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_16,
    }),
    t.partial({
        RedProperty: ChipMBI5268ExtendProperty,
        GreenProperty: ChipMBI5268ExtendProperty,
        BlueProperty: ChipMBI5268ExtendProperty,
        VRedProperty: ChipMBI5268ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FailureLEDElimination: common.Bool,
        GclkAddedNumer: common.UInt8,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        ScanCount: common.UInt8,
        CompsentionTime: common.UInt16,
        GradientTransitionOptimizationTime: common.UInt16,
        M: common.UInt8,
        N: common.UInt8,
        Div: common.UInt8,
        IsAdvancedMode: common.Bool,
        IsGrayEnhncedMode: common.Bool,
        DefaultRegisterType: common.UInt8,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        PointDetectParameter,
    }),
], 'ChipMBI5268RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5268RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5268.decompiled.cs:8740
 */
export const ChipMBI5268RGBVExtendProperty = t.intersection([
    ChipMBI5268RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5268RGBVExtendProperty') }),
], 'ChipMBI5268RGBVExtendProperty');
