// @novastar/native/build/module/generated/ChipSM16017DSRGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipSM16017DSRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RedRegValue: common.UInt16,
        GreenRegValue: common.UInt16,
        BlueRegValue: common.UInt16,
        VRedRegValue: common.UInt16,
        IsUseNewModule: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
    }),
], 'ChipSM16017DSRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16017DSRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16017DS.decompiled.cs:5053
 */
export const ChipSM16017DSRGBVExtendProperty = t.intersection([
    ChipSM16017DSRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16017DSRGBVExtendProperty') }),
], 'ChipSM16017DSRGBVExtendProperty');
