// @novastar/native/build/module/generated/ChipCS2017SRGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipCS2017SExtendProperty } from './ChipCS2017SExtendProperty';
export const ChipCS2017SRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipCS2017SExtendProperty,
        GreenProperty: ChipCS2017SExtendProperty,
        BlueProperty: ChipCS2017SExtendProperty,
        VRedProperty: ChipCS2017SExtendProperty,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        ConfigDataLen: common.Int32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
    }),
], 'ChipCS2017SRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2017SRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2017S.decompiled.cs:219
 */
export const ChipCS2017SRGBVExtendProperty = t.intersection([
    ChipCS2017SRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2017SRGBVExtendProperty') }),
], 'ChipCS2017SRGBVExtendProperty');
