// @novastar/native/build/module/generated/ChipSCL8080AutoRefreshRataParam.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { AutoAdjustFreqConfigMode } from './AutoAdjustFreqConfigMode';
import { ChipAutoAdjustRefreshRateParamBase } from './ChipAutoAdjustRefreshRateParamBase';
import { ChipType } from './ChipType';
export const ChipSCL8080AutoRefreshRataParamBase = t.intersection([
    ChipAutoAdjustRefreshRateParamBase,
    t.partial({
        ChipType,
        ChipLibVersion: common.UInt8,
        ConfigMode: AutoAdjustFreqConfigMode,
        AutoAdjustParamAddress: common.UInt32,
        AutoAdjustParamDataLen: common.Int32,
    }),
], 'ChipSCL8080AutoRefreshRataParamBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSCL8080AutoRefreshRataParam}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSCL8080.decompiled.cs:82
 */
export const ChipSCL8080AutoRefreshRataParam = t.intersection([
    ChipSCL8080AutoRefreshRataParamBase,
    t.partial({ '@_xsi:type': t.literal('ChipSCL8080AutoRefreshRataParam') }),
], 'ChipSCL8080AutoRefreshRataParam');
