// @novastar/native/build/module/generated/ChipCS2033ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipCS2033ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        FirstRegValue: common.withDefault(common.UInt16, 20480),
        SecondRegValue: common.withDefault(common.UInt16, 62093),
        ThirdRegValue: common.withDefault(common.UInt16, 7265),
        FourthRegValue: common.UInt16_0,
    }),
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        LowAshEnhanceUniformity: common.UInt8,
        LineBlankStrength: common.UInt8,
        LineBlankEnhanceModeEn: common.Bool,
        CouplingStrengthAdjust: common.UInt8,
        OpenTestPointAdjust: common.UInt8,
        CouplingOptimizationEn: common.Bool,
        LowGrayCompsentionTimeSettingMode: common.UInt8,
        CouplingRangeAdjust: common.UInt8,
        PerDriverEnable: common.Bool,
        OpenProtectEnable: common.Bool,
        ScanMode: common.UInt8,
        CurrentGear: common.UInt8,
        LowGrayCompsentionStrength: common.UInt8,
        OpenDynamicTestEnable: common.Bool,
        CurrentCompsentionEnable: common.Bool,
        ConstantCurrentInflectionPoint: common.UInt8,
        DynamicEnergySavingEn: common.Bool,
        LineEliminationLength: common.UInt8,
        OutputResponse: common.UInt8,
        PerDriverMode: common.UInt8,
        PerDriverTimeSetting: common.UInt8,
        OpentestThresholdValue: common.UInt8,
        CurrentCompsentionGear: common.UInt8,
    }),
], 'ChipCS2033ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2033ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2033.decompiled.cs:3254
 */
export const ChipCS2033ExtendProperty = t.intersection([
    ChipCS2033ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2033ExtendProperty') }),
], 'ChipCS2033ExtendProperty');
