// @novastar/native/build/module/generated/ChipMBI5759ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipMBI5759CustomExtendProperty } from './ChipMBI5759CustomExtendProperty';
import { ChipMBI5759GRGBExtendProperty } from './ChipMBI5759GRGBExtendProperty';
export const ChipMBI5759ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        GRGBExtendProperty: ChipMBI5759GRGBExtendProperty,
        CustomExtendProperty: ChipMBI5759CustomExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipMBI5759ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5759ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5759.decompiled.cs:1509
 */
export const ChipMBI5759ExtendProperty = t.intersection([
    ChipMBI5759ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5759ExtendProperty') }),
], 'ChipMBI5759ExtendProperty');
