// @novastar/native/build/module/generated/ChipHX8864RGBVExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
import { ChipHX8864ExtendProperty } from './ChipHX8864ExtendProperty';
export const ChipHX8864RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_15,
    }),
    t.partial({
        RedProperty: ChipHX8864ExtendProperty,
        GreenProperty: ChipHX8864ExtendProperty,
        BlueProperty: ChipHX8864ExtendProperty,
        VRedProperty: ChipHX8864ExtendProperty,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        IsAdvancedMode: common.Bool,
        GammaTable: common.Bool,
        IsUseNewModule: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        GclkFreqP: common.UInt8,
        GclkFreqN: common.UInt8,
        GclkFreqM: common.UInt8,
        OpenCircuitDetctEnable: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        FifthDataLen: common.Int32,
        FifthStartIndex: common.Int32,
        FifthRegisterAddr: common.Int32,
        SixthDataLen: common.Int32,
        SixthStartIndex: common.Int32,
        SixthRegisterAddr: common.Int32,
        ErrRedGain: common.Int32,
    }),
], 'ChipHX8864RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipHX8864RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipHX8864.decompiled.cs:7816
 */
export const ChipHX8864RGBVExtendProperty = t.intersection([
    ChipHX8864RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipHX8864RGBVExtendProperty') }),
], 'ChipHX8864RGBVExtendProperty');
