// @novastar/native/build/module/generated/ChipMBI5264ExtendProperty.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { ChipBaseExtendPropey } from './ChipBaseExtendPropey';
export const ChipMBI5264ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_16,
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        DarkBlockCompensationOne: common.UInt8,
        DarkBlockCompensationOneEn: common.Bool,
        DarkBlockCompensationTwo: common.UInt8,
        DarkBlockCompensationThree: common.UInt8,
        FirstSweepElimination: common.UInt8,
        FirstSweepEliminationEn: common.Bool,
        UnderEliminateGhost: common.UInt8,
        GrayHorizontalStripesEliminateEn: common.Bool,
        LowAshColorCastCompensation: common.UInt8,
        AdvancedRLowAshColorCastCompensation1: common.UInt8,
        AdvancedRLowAshColorCastCompensation2: common.UInt8,
        AdvancedRLowAshColorCastCompensation3: common.UInt8,
        GradientCompensationEnable: common.Bool,
        LowAshUniformityImproved1: common.UInt8,
        LowAshUniformityImproved2: common.UInt8,
        DislodgeBadPointsGrade: common.UInt8,
        CurrentGain: common.UInt8,
    }),
], 'ChipMBI5264ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5264ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5264.decompiled.cs:103
 */
export const ChipMBI5264ExtendProperty = t.intersection([
    ChipMBI5264ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5264ExtendProperty') }),
], 'ChipMBI5264ExtendProperty');
