// @novastar/native/build/module/generated/CommonInfoCompeleteResult.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var CommonInfoCompeleteResultEnum;
(function (CommonInfoCompeleteResultEnum) {
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["OK"] = 0] = "OK";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["VerNotComparable"] = 1] = "VerNotComparable";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["CommunicationFault"] = 2] = "CommunicationFault";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["DataError"] = 3] = "DataError";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["ParseError"] = 4] = "ParseError";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["FileError"] = 5] = "FileError";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["NotSupportDevice"] = 6] = "NotSupportDevice";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["NoServerObject"] = 7] = "NoServerObject";
})(CommonInfoCompeleteResultEnum || (CommonInfoCompeleteResultEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link CommonInfoCompeleteResultEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:56479
 */
export const CommonInfoCompeleteResult = EnumFromString(CommonInfoCompeleteResultEnum, 'CommonInfoCompeleteResult');
