// @novastar/native/build/module/generated/DecodeType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var DecodeTypeEnum;
(function (DecodeTypeEnum) {
    DecodeTypeEnum[DecodeTypeEnum["StaticNoCode"] = 0] = "StaticNoCode";
    DecodeTypeEnum[DecodeTypeEnum["Decode138"] = 1] = "Decode138";
    DecodeTypeEnum[DecodeTypeEnum["DicrectCode"] = 2] = "DicrectCode";
    DecodeTypeEnum[DecodeTypeEnum["Decode595"] = 3] = "Decode595";
    DecodeTypeEnum[DecodeTypeEnum["LXY695x"] = 4] = "LXY695x";
    DecodeTypeEnum[DecodeTypeEnum["Decode5953"] = 5] = "Decode5953";
    DecodeTypeEnum[DecodeTypeEnum["Decode5958"] = 6] = "Decode5958";
    DecodeTypeEnum[DecodeTypeEnum["SM5266"] = 7] = "SM5266";
    DecodeTypeEnum[DecodeTypeEnum["ICN2012WEA"] = 8] = "ICN2012WEA";
    DecodeTypeEnum[DecodeTypeEnum["LS9739_anode"] = 9] = "LS9739_anode";
    DecodeTypeEnum[DecodeTypeEnum["LS9739_cathode"] = 10] = "LS9739_cathode";
    DecodeTypeEnum[DecodeTypeEnum["DecodeICN2018_ICN2019"] = 11] = "DecodeICN2018_ICN2019";
    DecodeTypeEnum[DecodeTypeEnum["LS9736_anode"] = 12] = "LS9736_anode";
    DecodeTypeEnum[DecodeTypeEnum["Decode5972"] = 13] = "Decode5972";
    DecodeTypeEnum[DecodeTypeEnum["LS9737_anode"] = 15] = "LS9737_anode";
    DecodeTypeEnum[DecodeTypeEnum["LS9735_anode"] = 16] = "LS9735_anode";
    DecodeTypeEnum[DecodeTypeEnum["LS2018_anode"] = 17] = "LS2018_anode";
    DecodeTypeEnum[DecodeTypeEnum["LS9737_cathode"] = 18] = "LS9737_cathode";
    DecodeTypeEnum[DecodeTypeEnum["SM5366"] = 19] = "SM5366";
    DecodeTypeEnum[DecodeTypeEnum["MY7183"] = 20] = "MY7183";
    DecodeTypeEnum[DecodeTypeEnum["LS9736_cathode"] = 21] = "LS9736_cathode";
    DecodeTypeEnum[DecodeTypeEnum["TC7261"] = 22] = "TC7261";
    DecodeTypeEnum[DecodeTypeEnum["D7266"] = 23] = "D7266";
    DecodeTypeEnum[DecodeTypeEnum["CNS3018"] = 24] = "CNS3018";
    DecodeTypeEnum[DecodeTypeEnum["LS9737_Common"] = 25] = "LS9737_Common";
    DecodeTypeEnum[DecodeTypeEnum["MBI5981"] = 26] = "MBI5981";
    DecodeTypeEnum[DecodeTypeEnum["HX6158H"] = 27] = "HX6158H";
    DecodeTypeEnum[DecodeTypeEnum["MBI5986"] = 28] = "MBI5986";
    DecodeTypeEnum[DecodeTypeEnum["DP32019"] = 29] = "DP32019";
    DecodeTypeEnum[DecodeTypeEnum["DP32020"] = 30] = "DP32020";
    DecodeTypeEnum[DecodeTypeEnum["MBI5988"] = 31] = "MBI5988";
    DecodeTypeEnum[DecodeTypeEnum["FM7239A"] = 32] = "FM7239A";
    DecodeTypeEnum[DecodeTypeEnum["SM5368"] = 33] = "SM5368";
    DecodeTypeEnum[DecodeTypeEnum["FM7519"] = 34] = "FM7519";
    DecodeTypeEnum[DecodeTypeEnum["CFD2138S"] = 35] = "CFD2138S";
    DecodeTypeEnum[DecodeTypeEnum["RT5918"] = 36] = "RT5918";
    DecodeTypeEnum[DecodeTypeEnum["Unknown"] = 255] = "Unknown";
})(DecodeTypeEnum || (DecodeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DecodeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:779
 */
export const DecodeType = EnumFromString(DecodeTypeEnum, 'DecodeType');
