// @novastar/native/build/module/generated/DetectPointType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var DetectPointTypeEnum;
(function (DetectPointTypeEnum) {
    DetectPointTypeEnum[DetectPointTypeEnum["EarlyDeath"] = 0] = "EarlyDeath";
    DetectPointTypeEnum[DetectPointTypeEnum["OpenCircuit"] = 1] = "OpenCircuit";
    DetectPointTypeEnum[DetectPointTypeEnum["ShortCircuit"] = 2] = "ShortCircuit";
    DetectPointTypeEnum[DetectPointTypeEnum["BothDivision"] = 3] = "BothDivision";
    DetectPointTypeEnum[DetectPointTypeEnum["BothNoDivision"] = 4] = "BothNoDivision";
    DetectPointTypeEnum[DetectPointTypeEnum["FourTypesCircuit"] = 5] = "FourTypesCircuit";
    DetectPointTypeEnum[DetectPointTypeEnum["FiveTypesAllIn"] = 6] = "FiveTypesAllIn";
    DetectPointTypeEnum[DetectPointTypeEnum["None"] = 7] = "None";
})(DetectPointTypeEnum || (DetectPointTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DetectPointTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:4669
 */
export const DetectPointType = EnumFromString(DetectPointTypeEnum, 'DetectPointType');
