// @novastar/native/build/module/generated/DownGhostClearType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var DownGhostClearTypeEnum;
(function (DownGhostClearTypeEnum) {
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect1"] = 0] = "Effect1";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Closed"] = 1] = "Closed";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect2"] = 2] = "Effect2";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect3"] = 3] = "Effect3";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect4"] = 4] = "Effect4";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect5"] = 5] = "Effect5";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect6"] = 6] = "Effect6";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect7"] = 7] = "Effect7";
})(DownGhostClearTypeEnum || (DownGhostClearTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DownGhostClearTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:51918
 */
export const DownGhostClearType = EnumFromString(DownGhostClearTypeEnum, 'DownGhostClearType');
