// @novastar/native/build/module/generated/FirstLineCompensationType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var FirstLineCompensationTypeEnum;
(function (FirstLineCompensationTypeEnum) {
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Closed"] = 0] = "Closed";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect1"] = 1] = "Effect1";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect2"] = 2] = "Effect2";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect3"] = 3] = "Effect3";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect4"] = 4] = "Effect4";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect5"] = 5] = "Effect5";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect6"] = 6] = "Effect6";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect7"] = 7] = "Effect7";
})(FirstLineCompensationTypeEnum || (FirstLineCompensationTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link FirstLineCompensationTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:51947
 */
export const FirstLineCompensationType = EnumFromString(FirstLineCompensationTypeEnum, 'FirstLineCompensationType');
