// @novastar/native/build/module/generated/FunctionType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var FunctionTypeEnum;
(function (FunctionTypeEnum) {
    FunctionTypeEnum[FunctionTypeEnum["NovaPro_Name"] = 0] = "NovaPro_Name";
    FunctionTypeEnum[FunctionTypeEnum["NovaPro_HD"] = 1] = "NovaPro_HD";
    FunctionTypeEnum[FunctionTypeEnum["VX2_Name"] = 2] = "VX2_Name";
    FunctionTypeEnum[FunctionTypeEnum["VX4_Name"] = 3] = "VX4_Name";
    FunctionTypeEnum[FunctionTypeEnum["K4U_Name"] = 4] = "K4U_Name";
    FunctionTypeEnum[FunctionTypeEnum["K2U_Name"] = 5] = "K2U_Name";
    FunctionTypeEnum[FunctionTypeEnum["VX4S_Name"] = 6] = "VX4S_Name";
    FunctionTypeEnum[FunctionTypeEnum["VX4S_N_Name"] = 7] = "VX4S_N_Name";
    FunctionTypeEnum[FunctionTypeEnum["VX2S_Name"] = 8] = "VX2S_Name";
    FunctionTypeEnum[FunctionTypeEnum["VX5_Name"] = 9] = "VX5_Name";
    FunctionTypeEnum[FunctionTypeEnum["K4_Name"] = 10] = "K4_Name";
    FunctionTypeEnum[FunctionTypeEnum["K4S_Name"] = 11] = "K4S_Name";
    FunctionTypeEnum[FunctionTypeEnum["K4S_N_Name"] = 12] = "K4S_N_Name";
    FunctionTypeEnum[FunctionTypeEnum["VD43_Name"] = 13] = "VD43_Name";
    FunctionTypeEnum[FunctionTypeEnum["TVctrl_230R_Name"] = 14] = "TVctrl_230R_Name";
    FunctionTypeEnum[FunctionTypeEnum["VX2U_Name"] = 15] = "VX2U_Name";
    FunctionTypeEnum[FunctionTypeEnum["VX4U_Name"] = 16] = "VX4U_Name";
    FunctionTypeEnum[FunctionTypeEnum["CVT4K_Name"] = 17] = "CVT4K_Name";
    FunctionTypeEnum[FunctionTypeEnum["Defalut"] = 18] = "Defalut";
})(FunctionTypeEnum || (FunctionTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link FunctionTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:61233
 */
export const FunctionType = EnumFromString(FunctionTypeEnum, 'FunctionType');
