// @novastar/native/build/module/generated/HWSoftwareSpaceRes.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var HWSoftwareSpaceResEnum;
(function (HWSoftwareSpaceResEnum) {
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["OK"] = 0] = "OK";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["NotSupportDevice"] = 1] = "NotSupportDevice";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["NoServerObject"] = 2] = "NoServerObject";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["FileError"] = 3] = "FileError";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["DataError"] = 4] = "DataError";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["Busy"] = 5] = "Busy";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["Unknown"] = 255] = "Unknown";
})(HWSoftwareSpaceResEnum || (HWSoftwareSpaceResEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link HWSoftwareSpaceResEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.CommonInfoAccessor.decompiled.cs:81
 */
export const HWSoftwareSpaceRes = EnumFromString(HWSoftwareSpaceResEnum, 'HWSoftwareSpaceRes');
