// @novastar/native/build/module/generated/LowGrayColorAdjType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var LowGrayColorAdjTypeEnum;
(function (LowGrayColorAdjTypeEnum) {
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Closed"] = 0] = "Closed";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect4"] = 4] = "Effect4";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect5"] = 5] = "Effect5";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect6"] = 6] = "Effect6";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect7"] = 7] = "Effect7";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect8"] = 8] = "Effect8";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect9"] = 9] = "Effect9";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect10"] = 10] = "Effect10";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect11"] = 11] = "Effect11";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect12"] = 12] = "Effect12";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect13"] = 13] = "Effect13";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect14"] = 14] = "Effect14";
    LowGrayColorAdjTypeEnum[LowGrayColorAdjTypeEnum["Effect15"] = 15] = "Effect15";
})(LowGrayColorAdjTypeEnum || (LowGrayColorAdjTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link LowGrayColorAdjTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:51930
 */
export const LowGrayColorAdjType = EnumFromString(LowGrayColorAdjTypeEnum, 'LowGrayColorAdjType');
