// @novastar/native/build/module/generated/LS9960Data.js
import * as t from 'io-ts';
import * as common from '../lib/common';
/**
 * @category Codecs
 * @desc Codec for interface {@link LS9960Data}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:7086
 */
export const LS9960Data = t.intersection([
    t.type({
        FriedFreuquance: common.Int32_60,
        ClkAndPhaseClock: common.Int32_100,
        ConnnectUnitCount: common.Int32_1,
        ICNumberOrder: common.string_empty,
        RowElimination: common.Int32_10,
        ColElimination: common.Int32_10,
        ShiftClockList: common.XMLArray(common.Numeric, 'double'),
        GammaStepupLength: common.Int32_1,
        GammaModulus: common.Numeric_28,
        TableMode: common.Int32_2,
    }),
    t.partial({
        PmData: common.Base64,
        DmFirstData: common.Base64,
        DmSecondData: common.Base64,
        DmThridData: common.Base64,
        WspData: common.Base64,
        ScanType: common.Int32,
        MaxFreshRate: common.Int32,
        MinFreshRate: common.Int32,
        MinGamaValue: common.Int32,
        MaxGamaValue: common.Int32,
        GameTable: common.Base64,
        ContrastValue: common.Int32,
        BirghtEfficty: common.Numeric,
        UnitIcCount: common.Int32,
        RedJump: common.Int32,
        GreenJump: common.Int32,
        BlueJump: common.Int32,
        RedCurrentValue: common.Int32,
        GreenCurrentValue: common.Int32,
        BlueCurrentValue: common.Int32,
        ClokPhase: common.Int32,
        ShiftClock: common.Numeric,
        ScanLineArray: common.Base64,
        IsSelfMode: common.Bool,
        MoudleWeidth: common.Int32,
        MoudleHeight: common.Int32,
        BrightMaxValue: common.Int32,
        RGBSQ: common.Int32,
        RedColPhase: common.Int32,
        GreenColPhase: common.Int32,
        BlueColPhase: common.Int32,
        ConpensationValue: common.Int32,
        ScanNum: common.Int32,
        RgbSeq: common.Int32,
        FirstDarkertoeliminateR: common.Int32,
        FirstDarkertoeliminateG: common.Int32,
        FirstDarkertoeliminateB: common.Int32,
        Vol_blkl: common.Int32,
        Mode_blkl: common.Int32,
        MinPulseR: common.Int32,
        MinPulseG: common.Int32,
        MinPulseB: common.Int32,
        BlkIC: common.Int32,
        ColBankingMode: common.Int32,
        PllMode: common.Int32,
        NVersion: common.Int32,
        CutTime: common.Int32,
        LowPower: common.Int32,
        PreVoltageRed: common.Int32,
        PreVoltageGreen: common.Int32,
        PreVoltageBlue: common.Int32,
        Pre96Port: common.Int32,
    }),
], 'LS9960Data');
