// @novastar/native/build/module/generated/MaxValue.js
/**
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:59868
 */
var MaxValue;
(function (MaxValue) {
    MaxValue[MaxValue["MAX_MODULEDATAGROUP"] = 8] = "MAX_MODULEDATAGROUP";
    MaxValue[MaxValue["MAX_SCAN"] = 16] = "MAX_SCAN";
    MaxValue[MaxValue["MAX_SCAN_NEW"] = 128] = "MAX_SCAN_NEW";
    MaxValue[MaxValue["MAX_SCANSPECIAL"] = 32] = "MAX_SCANSPECIAL";
    MaxValue[MaxValue["COLOR_COUNT"] = 4] = "COLOR_COUNT";
    MaxValue[MaxValue["MaxPortCount"] = 64] = "MaxPortCount";
    MaxValue[MaxValue["GAMMAVALUE_COUNT"] = 512] = "GAMMAVALUE_COUNT";
    MaxValue[MaxValue["COLORMATRIX_LENGTH"] = 16] = "COLORMATRIX_LENGTH";
    MaxValue[MaxValue["MAXCOUNT_OF_MODULEPOINTTABLE"] = 512] = "MAXCOUNT_OF_MODULEPOINTTABLE";
    MaxValue[MaxValue["MAXCOUNT_OF_SCANBOARDPOINTTABLE"] = 32768] = "MAXCOUNT_OF_SCANBOARDPOINTTABLE";
    MaxValue[MaxValue["Max_SCANNER_WIDTH"] = 512] = "Max_SCANNER_WIDTH";
    MaxValue[MaxValue["Max_SCANNER_HEIGHT"] = 512] = "Max_SCANNER_HEIGHT";
    MaxValue[MaxValue["MAX_SCANNER_AREA"] = 524288] = "MAX_SCANNER_AREA";
    MaxValue[MaxValue["MAX_SUBFIELDPART_COUNT"] = 8] = "MAX_SUBFIELDPART_COUNT";
    MaxValue[MaxValue["MAX_OETABLE_LENGHT"] = 512] = "MAX_OETABLE_LENGHT";
    MaxValue[MaxValue["MAX_GROUP_CONNECT_COUNT"] = 255] = "MAX_GROUP_CONNECT_COUNT";
    MaxValue[MaxValue["MAX_MULTIPLE_CHIP_TABLELEN"] = 256] = "MAX_MULTIPLE_CHIP_TABLELEN";
    MaxValue[MaxValue["MAX_MULTIPLE_TABLECOUNT"] = 5] = "MAX_MULTIPLE_TABLECOUNT";
    MaxValue[MaxValue["MAX_ASYSYS_DVISCREEN_TIME"] = 60000] = "MAX_ASYSYS_DVISCREEN_TIME";
    MaxValue[MaxValue["MAX_SCREEN_COUNT"] = 10] = "MAX_SCREEN_COUNT";
    MaxValue[MaxValue["MAX_SCANNER_DATAGROUP"] = 32] = "MAX_SCANNER_DATAGROUP";
    MaxValue[MaxValue["MAX_SCANNER_GAMMA_LENGTH"] = 512] = "MAX_SCANNER_GAMMA_LENGTH";
    MaxValue[MaxValue["MAX_MONITOR_DATAGROUP"] = 64] = "MAX_MONITOR_DATAGROUP";
    MaxValue[MaxValue["MAX_OTHER_REFRESH_NUM_PARAMS_COUNT"] = 28] = "MAX_OTHER_REFRESH_NUM_PARAMS_COUNT";
    MaxValue[MaxValue["MAX_PORT_LOADED_COUNT"] = 1024] = "MAX_PORT_LOADED_COUNT";
    MaxValue[MaxValue["Max_DCLK_NUM"] = 4] = "Max_DCLK_NUM";
    MaxValue[MaxValue["MAX_PMDATA_LENGTH"] = 4096] = "MAX_PMDATA_LENGTH";
})(MaxValue || (MaxValue = {}));
export default MaxValue;
