// @novastar/native/build/module/generated/MonitorDisplayType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var MonitorDisplayTypeEnum;
(function (MonitorDisplayTypeEnum) {
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["SBStatus"] = 0] = "SBStatus";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["Temperature"] = 1] = "Temperature";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["MCStatus"] = 2] = "MCStatus";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["Humidity"] = 3] = "Humidity";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["Smoke"] = 4] = "Smoke";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["Fan"] = 5] = "Fan";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["Power"] = 6] = "Power";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["RowLine"] = 7] = "RowLine";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["GeneralSwitch"] = 8] = "GeneralSwitch";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["SenderDVI"] = 9] = "SenderDVI";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["ExternalDevice"] = 10] = "ExternalDevice";
    MonitorDisplayTypeEnum[MonitorDisplayTypeEnum["MuduleTemperature"] = 11] = "MuduleTemperature";
})(MonitorDisplayTypeEnum || (MonitorDisplayTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link MonitorDisplayTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:56676
 */
export const MonitorDisplayType = EnumFromString(MonitorDisplayTypeEnum, 'MonitorDisplayType');
