// @novastar/native/build/module/generated/PicturePrestoreConfig.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { CabinetStoreEffect } from './CabinetStoreEffect';
import { NoSignalMode } from './NoSignalMode';
import { PictureShowMode } from './PictureShowMode';
/**
 * @category Codecs
 * @desc Codec for interface {@link PicturePrestoreConfig}
 * @see Automatically generated from ../decompiled/Nova.GigabitController.ProgramInnerData.decompiled.cs:739
 */
export const PicturePrestoreConfig = t.intersection([
    t.type({
        PictureName: common.string_empty,
        StoreEffect: common.withDefault(CabinetStoreEffect, 'FullScreen'),
        FullScreenShowMode: common.withDefault(PictureShowMode, 'Stretch'),
        OneCabinetShowMode: common.withDefault(PictureShowMode, 'Stretch'),
        IsUseStartPicture: common.Bool_false,
        StartPictureTime: common.Int32_2,
        OfflineMode: common.withDefault(NoSignalMode, 'BlackScreen'),
        NoDviMode: common.withDefault(NoSignalMode, 'BlackScreen'),
    }),
    t.partial({}),
], 'PicturePrestoreConfig');
