// @novastar/native/build/module/generated/RegisterGroupType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var RegisterGroupTypeEnum;
(function (RegisterGroupTypeEnum) {
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["FirstGroup"] = 0] = "FirstGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["SecondGroup"] = 1] = "SecondGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["ThirdGroup"] = 2] = "ThirdGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["FourthGroup"] = 3] = "FourthGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["OldChipDefaultGroup"] = 4] = "OldChipDefaultGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["SpecialGroup"] = 5] = "SpecialGroup";
})(RegisterGroupTypeEnum || (RegisterGroupTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link RegisterGroupTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:1447
 */
export const RegisterGroupType = EnumFromString(RegisterGroupTypeEnum, 'RegisterGroupType');
