// @novastar/native/build/module/generated/ResetAndSwitchCommandType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var ResetAndSwitchCommandTypeEnum;
(function (ResetAndSwitchCommandTypeEnum) {
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["DEFAULT_PARAM"] = 0] = "DEFAULT_PARAM";
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["NO_ACTION_PARAM"] = 1] = "NO_ACTION_PARAM";
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["RESET_OPERATION_PARAM"] = 2] = "RESET_OPERATION_PARAM";
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["START_ACTION_PARAM"] = 3] = "START_ACTION_PARAM";
})(ResetAndSwitchCommandTypeEnum || (ResetAndSwitchCommandTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ResetAndSwitchCommandTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1324
 */
export const ResetAndSwitchCommandType = EnumFromString(ResetAndSwitchCommandTypeEnum, 'ResetAndSwitchCommandType');
