// @novastar/native/build/module/generated/ScanBoardConnectType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var ScanBoardConnectTypeEnum;
(function (ScanBoardConnectTypeEnum) {
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftTop_Horizontal"] = 0] = "LeftTop_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftBottom_Horizontal"] = 1] = "LeftBottom_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightTop_Horizontal"] = 2] = "RightTop_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightBottom_Horizontal"] = 3] = "RightBottom_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftTop_Vertical"] = 4] = "LeftTop_Vertical";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftBottom_Vertical"] = 5] = "LeftBottom_Vertical";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightTop_Vertical"] = 6] = "RightTop_Vertical";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightBottom_Vertical"] = 7] = "RightBottom_Vertical";
})(ScanBoardConnectTypeEnum || (ScanBoardConnectTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ScanBoardConnectTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:56583
 */
export const ScanBoardConnectType = EnumFromString(ScanBoardConnectTypeEnum, 'ScanBoardConnectType');
