// @novastar/native/build/module/generated/ScannerMonitorData.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { MSRvCardStatusType } from './MSRvCardStatusType';
import { MSRvCardWorkType } from './MSRvCardWorkType';
import { OneModuleInfo } from './OneModuleInfo';
import { ScanBoardProperty } from './ScanBoardProperty';
import { ScanBoardRegionInfo } from './ScanBoardRegionInfo';
import { SmokeAlarmInfo } from './SmokeAlarmInfo';
import { ValueInfo } from './ValueInfo';
import { WorkStatusType } from './WorkStatusType';
/**
 * @category Codecs
 * @desc Codec for interface {@link ScannerMonitorData}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:60733
 */
export const ScannerMonitorData = t.intersection([
    t.type({
        IsMasterCard: common.Bool_true,
        FanSpeedOfMonitorCardCollection: common.XMLArray(ValueInfo, 'ValueInfo'),
        VoltageOfMonitorCardCollection: common.XMLArray(ValueInfo, 'ValueInfo'),
        ModuleInfoList: common.XMLArray(OneModuleInfo, 'OneModuleInfo'),
    }),
    t.partial({
        ScreenIndex: common.UInt8,
        IsConnectMC: common.Bool,
        IsConnectHubMC: common.Bool,
        ScanBdPorpertyInfo: ScanBoardProperty,
        SBRegionInfo: ScanBoardRegionInfo,
        WorkStatus: WorkStatusType,
        MSRvCardStatus: MSRvCardStatusType,
        MSRvCardWork: MSRvCardWorkType,
        TemperatureOfScanCard: ValueInfo,
        HumidityOfScanCard: ValueInfo,
        VoltageOfScanCard: ValueInfo,
        TemperatureOfMonitorCard: ValueInfo,
        HumidityOfMonitorCard: ValueInfo,
        SmokeWarn: SmokeAlarmInfo,
        GeneralStatusData: common.UInt8,
        AnalogInputBytes: common.Base64,
        ModuleStatusBytes: common.Base64,
        MonitorDataBytes: common.Base64,
        MonitorGroupSwapEnable: common.Bool,
        MonitorGroupData: common.Base64,
        ScanBordMCUVersionName: t.string,
        ScanBordFPGAVersionName: t.string,
        ScanBordMoudId: t.string,
        MonitorScanBdBackPowerDataBytes: common.Base64,
        IsRedu: common.Bool,
    }),
], 'ScannerMonitorData');
