// @novastar/native/build/module/generated/ScanBoardProperty.js
import * as t from 'io-ts';
import { DateFromISOString } from 'io-ts-types';
import * as common from '../lib/common';
import { CabinetInfomation } from './CabinetInfomation';
import { ChipCfgCurrentRange } from './ChipCfgCurrentRange';
import { ChipCfgCurrentRisingTime } from './ChipCfgCurrentRisingTime';
import { ChipOperatingMode } from './ChipOperatingMode';
import { CoefType } from './CoefType';
import { CoefficientSourceType } from './CoefficientSourceType';
import { CommonIrCabinetType } from './CommonIrCabinetType';
import { ConfigFileVersionInfo } from './ConfigFileVersionInfo';
import { ControlModeType } from './ControlModeType';
import { CorrectType } from './CorrectType';
import { CustomGammaInfo } from './CustomGammaInfo';
import { DataGroupOutPutMode } from './DataGroupOutPutMode';
import { DecodeChipParamBase } from './DecodeChipParamBase';
import { DriverCurrentSpeedType } from './DriverCurrentSpeedType';
import { FlashTopology } from './FlashTopology';
import { GhostRemoveModeType } from './GhostRemoveModeType';
import { GrayModeType } from './GrayModeType';
import { GrayRealizeType } from './GrayRealizeType';
import { IrRegularCabinetProperty } from './IrRegularCabinetProperty';
import { LEDModuleProperty } from './LEDModuleProperty';
import { LS9918Data } from './LS9918Data';
import { LS9926Data } from './LS9926Data';
import { LS9929Data } from './LS9929Data';
import { LS9930Data } from './LS9930Data';
import { LS9935Data } from './LS9935Data';
import { LS9960Data } from './LS9960Data';
import { LowGrayQueryMode } from './LowGrayQueryMode';
import MaxValue from './MaxValue';
import { ModuleCascadeDiretion } from './ModuleCascadeDiretion';
import { PriorityMode } from './PriorityMode';
import { RotateAngle } from './RotateAngle';
import { ShowTypeWhenPortDisconnected, } from './ShowTypeWhenPortDisconnected';
import { SmartSetMode } from './SmartSetMode';
import { SpecialFrameRateInfo } from './SpecialFrameRateInfo';
import { VirtualModeType } from './VirtualModeType';
import { ChipAutoAdjustRefreshRateParamBase, ChipBaseExtendPropey } from './unions';
/**
 * @category Codecs
 * @desc Codec for interface {@link ScanBoardProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:24177
 */
export const ScanBoardProperty = t.intersection([
    t.type({
        RCFGXVersion: common.withDefault(t.string, '\u672A\u77E5'),
        ProgramVersion: common.string_empty,
        GammaValue: common.UInt8_28,
        Brightness: common.Int32_255,
        RedBright: common.Int32_255,
        BlueBright: common.Int32_255,
        GreenBright: common.Int32_255,
        VRedBright: common.Int32_255,
        RedGain: common.Int32_255,
        BlueGain: common.Int32_255,
        GreenGain: common.Int32_255,
        VRedGain: common.Int32_255,
        IsChromaCorrentionLowGray: common.Bool_true,
        CoefTypeMode: common.withDefault(CoefType, 'common'),
        Width: common.Int32_32,
        Height: common.Int32_16,
        ModuleCols: common.Int32_1,
        ModuleRows: common.Int32_1,
        PhysicalDataGroupNum: common.Int32_2,
        LogicalDataGroupNum: common.Int32_8,
        RefNumPerVs: common.Int32_4,
        GrayDepth: common.UInt8_12,
        GrayMode: common.withDefault(GrayModeType, 'Normallight'),
        GrayRealize: common.withDefault(GrayRealizeType, 'HGHR'),
        LineScanTime: common.Int32_5,
        GhostRemoveType: common.withDefault(GhostRemoveModeType, 'Unknown'),
        BlankUnitNumPerScan: common.Int32_25,
        RowChangePoint: common.Int32_3,
        TinyLineFeedMoment: common.Int32_2,
        LightTimeRatio: common.Int32_100,
        LightTime: common.Int32_2650,
        LightTimeNum2: common.Int32_2650,
        ShiftUnitNum: common.Int32_32,
        PointNumberPerDriver: common.Int32_32,
        TotalUnitNum: common.withDefault(common.Int32, 267),
        DclkUnitCycle: common.UInt8_10,
        DclkPhase: common.UInt8_2,
        DclkHigh: common.UInt8_5,
        DclkHighRatio: common.withDefault(common.UInt8, 50),
        GCLKRate: common.UInt8_10,
        GCLKPhase: common.UInt8_5,
        GCLKDuty: common.UInt8_5,
        CtrlEndPoint: common.withDefault(common.Int32, 20),
        SubFields: common.Int32_4,
        TotalGclkUnitNumPerScan: common.UInt16_4096,
        AddrExtend: common.UInt8_5,
        TotalLightCdfResault: common.withDefault(common.Int32, 65535),
        LogicalShiftUnitNum: common.Int32_128,
        M1TranCntNum: common.Int32_3,
        M2TranCntNum: common.Int32_3,
        IsConnectSmartModule: common.Bool_true,
        ColorMatrix: common.XMLArray(common.Int32, 'int'),
        GclkNumPerScan: common.UInt16_4096,
        IsNoCorrectionThresholdEable: common.Bool_true,
        RedNoCorrectionAttenuation: common.UInt16_2048,
        GreenNoCorrectionAttenuation: common.UInt16_2048,
        BlueNoCorrectionAttenuation: common.UInt16_2048,
        VirRedNoCorrectionAttenuation: common.UInt16_2048,
        CabinetRotateAngle: common.withDefault(RotateAngle, 'R_360'),
        IsClearGhost: common.Bool_true,
        SerialDecodeRepeatTimes: common.UInt8_1,
        LowAshCompensationOne5253: common.UInt8_100,
        LowAshCompensationTwo5253: common.UInt8_100,
        OperatingMode: common.withDefault(ChipOperatingMode, 'Unknown'),
        IsOpenSetGainValue: common.Bool_true,
        CurrentRisingTime: common.withDefault(ChipCfgCurrentRisingTime, 'CurrentRising2'),
        RedSpeedType: common.withDefault(DriverCurrentSpeedType, 'Fast'),
        GreenSpeedType: common.withDefault(DriverCurrentSpeedType, 'Fast'),
        BlueSpeedType: common.withDefault(DriverCurrentSpeedType, 'Fast'),
        MBI515xDhT: common.UInt8_3,
        PreChargeTime: common.Int32_1,
        EliminatePotential: common.Int32_3,
        Max2053Gamma: common.withDefault(common.Numeric, 100),
        MaxGammaValue: common.withDefault(common.Int32, 4096),
        UnitIcCount: common.Int32_255,
        ICNumber: common.UInt8_5,
        BlankOptimizationLevel: common.UInt8_1,
        Support22BitModel: common.Bool_true,
        Auto5252ResetTime: common.withDefault(DateFromISOString, new Date(2017, 0, 1, 0, 0, 59)),
        GclkFreqP: common.UInt8_12,
        GclkFreqM: common.UInt8_3,
        GclkFreqN: common.UInt8_2,
        DclkNumFor2055: common.Int32_1,
        GclkNumFor2055: common.Int32_128,
        GCLKRate2055: common.withDefault(common.Numeric, 10),
        DisplayMode: common.UInt8_255,
        RealPhysicalGroupNum: common.Int32_2,
        ShadowEliminationPotentialLevel: common.UInt8_5,
        LineBlankVoltage: common.UInt8_2,
        StartScan1: common.UInt8_15,
        StartScan2: common.UInt8_31,
        StartScan3: common.withDefault(common.UInt8, 47),
        StartScan4: common.withDefault(common.UInt8, 63),
        RealBrightnessOf2053: common.UInt8_255,
        BrightPriorityMode: common.withDefault(PriorityMode, 'Contrast'),
    }),
    t.partial({
        ConfigFileVersion: ConfigFileVersionInfo,
        LEDCtrlStatus: ControlModeType,
        VirtualMode: VirtualModeType,
        IncAddrPerDrive: common.Int32,
        isSupportMulticolorGamma: common.Bool,
        RedGammaTable: common.buffer_GAMMAVALUE_COUNT,
        GreenGammaTable: common.buffer_GAMMAVALUE_COUNT,
        BlueGammaTable: common.buffer_GAMMAVALUE_COUNT,
        ExtendRedGammaTable: common.buffer_GAMMAVALUE_COUNT,
        ExtendGreenGammaTable: common.buffer_GAMMAVALUE_COUNT,
        ExtendBlueGammaTable: common.buffer_GAMMAVALUE_COUNT,
        IsEnableCalibration: common.Bool,
        CoefSourceType: CoefficientSourceType,
        CorrectionMode: CorrectType,
        ThreasholdOfPointDetect: common.Int32,
        IsIrRegular: common.Bool,
        StandardLedModuleProp: LEDModuleProperty,
        X: common.Int32,
        Y: common.Int32,
        ModCascadeType: ModuleCascadeDiretion,
        PointTableData: common.Base64,
        IsLightFrom1stGray: common.Bool,
        IsSM5266Decode: common.Bool,
        LatDelay: common.Int32,
        TinyAfterglowMoment: common.Int32,
        SubFieldPart: new common.BufferFromBase64('SubFieldPart', MaxValue.MAX_SUBFIELDPART_COUNT),
        OEData: new common.BufferFromBase64('OEData', MaxValue.MAX_OETABLE_LENGHT),
        LowGrayCompensation: common.UInt8,
        LowGrayQuery: LowGrayQueryMode,
        IsNewOERamEnable: common.Bool,
        IsLowGrayRamEnable: common.Bool,
        IsEnableTranCntNum: common.Bool,
        TempOEValue: common.Numeric,
        IsConnectHubMonitor: common.Bool,
        OfflineFrame: ShowTypeWhenPortDisconnected,
        DriverFucntion: common.UInt8,
        ABCDRollOver: common.Bool,
        ScanSequenceAdjustEn: common.Bool,
        LightFlashClose: common.Bool,
        SmartSetMode,
        RamAData: common.buffer_MAX_MULTIPLE_CHIP_TABLELEN,
        RamBData: common.buffer_MAX_MULTIPLE_CHIP_TABLELEN,
        RamCData: common.buffer_MAX_MULTIPLE_CHIP_TABLELEN,
        RamDData: common.buffer_MAX_MULTIPLE_CHIP_TABLELEN,
        IsDExtendMode: common.Bool,
        IsSymmetricalOutputMode: common.Bool,
        RedNoCorrectionThreshold: common.UInt16,
        GreenNoCorrectionThreshold: common.UInt16,
        BlueNoCorrectionThreshold: common.UInt16,
        VirRedNoCorrectionThreshold: common.UInt16,
        TwentyDataGroupEnable: common.Bool,
        GroupSwapEnable: common.Bool,
        GroupSwapInfo: new common.BufferFromBase64('GroupSwapInfo', MaxValue.MAX_SCANNER_DATAGROUP),
        MonitorGroupSwapEnable: common.Bool,
        MonitorGroupSwapInfo: new common.BufferFromBase64('MonitorGroupSwapInfo', MaxValue.MAX_MONITOR_DATAGROUP),
        Mbi5042GrayEnhanced: common.Bool,
        My9262GrayEnhanced: common.Bool,
        Mbi5166GrayEnhanced: common.Bool,
        IsLowAshOptimization: common.Bool,
        IsEnableMonitorRGB: common.Bool,
        MonitorRGBCode: common.buffer_COLOR_COUNT,
        Is24DataGroup: common.Bool,
        Is28DataGroup: common.Bool,
        OtherRefreshNumParams: new common.BufferFromBase64('OtherRefreshNumParams', MaxValue.MAX_OTHER_REFRESH_NUM_PARAMS_COUNT),
        IsEnableOtherRefreshNumParams: common.Bool,
        CommonIrCabinetMode: CommonIrCabinetType,
        IsSupportHighLoad: common.Bool,
        IsGhostSignalNegation: common.Bool,
        IsCloseGhostSignal: common.Bool,
        IsReverseScanOutput: common.Bool,
        DataGroupOutPutType: DataGroupOutPutMode,
        IsStarSwipPoint: common.Bool,
        IsOpenProtection: common.Bool,
        IsOpenClearBlankLine: common.Bool,
        IsOpenLowAshCompensation: common.Bool,
        LowAshCompensationOne: common.UInt8,
        LowAshCompensationTwo: common.UInt8,
        PowerOnBrightnessAdjustEn: common.Bool,
        OpenEMCFun: common.Bool,
        EnFrameTimeMgt: common.Bool,
        ReduceHighContrast: common.UInt8,
        IsChipSUM2117: common.Bool,
        IsStartDoubleLock: common.Bool,
        CurrentPosition: common.Int32,
        CurrentPhase: common.Int32,
        IsChipSUM2017TD: common.Bool,
        IsStartDoubleLock2017TD: common.Bool,
        CurrentPosition2017TD: common.Int32,
        CurrentPhase2017TD: common.Int32,
        RPositiveNegativeValue: common.Bool,
        GPositiveNegativeValue: common.Bool,
        BPositiveNegativeValue: common.Bool,
        CabinetInfo: CabinetInfomation,
        CurrentRange: ChipCfgCurrentRange,
        EnableEnhanceFirstScan: common.Bool,
        FirstScanCompensation: common.Int32,
        OtherScanCompenscation: common.Int32,
        ChipPropey: ChipBaseExtendPropey,
        MBI515xDeltaT: common.UInt8,
        IsOpen5958Exchange: common.Bool,
        EliminateMode: common.Int32,
        IrCabinetProp: IrRegularCabinetProperty,
        FlashPTopology: FlashTopology,
        Exceed256DataIsValid: common.Bool,
        IsKillMode: common.Bool,
        SpecialFrameRate: SpecialFrameRateInfo,
        LS9960Data,
        LS9918Data,
        LS9918orSC6618Flag: common.Int32,
        LS9929Data,
        LS9935Data,
        LS9930Data,
        LS9926Data,
        ScanShiftForImageData: common.UInt16,
        Enable18BitModel: common.Bool,
        Enable22BitModel: common.Bool,
        EnableDirectMode: common.Bool,
        ABCSignalDelayEnable: common.Bool,
        DESignalDelayEnable: common.Bool,
        ABCDESignalDelayTime: common.UInt8,
        AutoOrManual5252ResetSwitch: common.Bool,
        IntervalDate5252: common.UInt8,
        ShadowEliminationEnhancedEnable: common.Bool,
        ChipIs5124NewFlag: common.UInt8,
        DecodeChipParamExtendProp: DecodeChipParamBase,
        ChipAutoAdjustRefreshRateParam: ChipAutoAdjustRefreshRateParamBase,
        CustomGammaInfo,
        OddEvenScan: common.UInt8,
    }),
], 'ScanBoardProperty');
