// @novastar/native/build/module/generated/ScanType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var ScanTypeEnum;
(function (ScanTypeEnum) {
    ScanTypeEnum[ScanTypeEnum["Scan_static"] = 1] = "Scan_static";
    ScanTypeEnum[ScanTypeEnum["Scan_2"] = 2] = "Scan_2";
    ScanTypeEnum[ScanTypeEnum["Scan_3"] = 3] = "Scan_3";
    ScanTypeEnum[ScanTypeEnum["Scan_4"] = 4] = "Scan_4";
    ScanTypeEnum[ScanTypeEnum["Scan_5"] = 5] = "Scan_5";
    ScanTypeEnum[ScanTypeEnum["Scan_6"] = 6] = "Scan_6";
    ScanTypeEnum[ScanTypeEnum["Scan_7"] = 7] = "Scan_7";
    ScanTypeEnum[ScanTypeEnum["Scan_8"] = 8] = "Scan_8";
    ScanTypeEnum[ScanTypeEnum["Scan_9"] = 9] = "Scan_9";
    ScanTypeEnum[ScanTypeEnum["Scan_10"] = 10] = "Scan_10";
    ScanTypeEnum[ScanTypeEnum["Scan_11"] = 11] = "Scan_11";
    ScanTypeEnum[ScanTypeEnum["Scan_12"] = 12] = "Scan_12";
    ScanTypeEnum[ScanTypeEnum["Scan_13"] = 13] = "Scan_13";
    ScanTypeEnum[ScanTypeEnum["Scan_14"] = 14] = "Scan_14";
    ScanTypeEnum[ScanTypeEnum["Scan_15"] = 15] = "Scan_15";
    ScanTypeEnum[ScanTypeEnum["Scan_16"] = 16] = "Scan_16";
    ScanTypeEnum[ScanTypeEnum["Scan_17"] = 17] = "Scan_17";
    ScanTypeEnum[ScanTypeEnum["Scan_18"] = 18] = "Scan_18";
    ScanTypeEnum[ScanTypeEnum["Scan_19"] = 19] = "Scan_19";
    ScanTypeEnum[ScanTypeEnum["Scan_20"] = 20] = "Scan_20";
    ScanTypeEnum[ScanTypeEnum["Scan_21"] = 21] = "Scan_21";
    ScanTypeEnum[ScanTypeEnum["Scan_22"] = 22] = "Scan_22";
    ScanTypeEnum[ScanTypeEnum["Scan_23"] = 23] = "Scan_23";
    ScanTypeEnum[ScanTypeEnum["Scan_24"] = 24] = "Scan_24";
    ScanTypeEnum[ScanTypeEnum["Scan_25"] = 25] = "Scan_25";
    ScanTypeEnum[ScanTypeEnum["Scan_26"] = 26] = "Scan_26";
    ScanTypeEnum[ScanTypeEnum["Scan_27"] = 27] = "Scan_27";
    ScanTypeEnum[ScanTypeEnum["Scan_28"] = 28] = "Scan_28";
    ScanTypeEnum[ScanTypeEnum["Scan_29"] = 29] = "Scan_29";
    ScanTypeEnum[ScanTypeEnum["Scan_30"] = 30] = "Scan_30";
    ScanTypeEnum[ScanTypeEnum["Scan_31"] = 31] = "Scan_31";
    ScanTypeEnum[ScanTypeEnum["Scan_32"] = 32] = "Scan_32";
    ScanTypeEnum[ScanTypeEnum["Scan_33"] = 33] = "Scan_33";
    ScanTypeEnum[ScanTypeEnum["Scan_34"] = 34] = "Scan_34";
    ScanTypeEnum[ScanTypeEnum["Scan_35"] = 35] = "Scan_35";
    ScanTypeEnum[ScanTypeEnum["Scan_36"] = 36] = "Scan_36";
    ScanTypeEnum[ScanTypeEnum["Scan_37"] = 37] = "Scan_37";
    ScanTypeEnum[ScanTypeEnum["Scan_38"] = 38] = "Scan_38";
    ScanTypeEnum[ScanTypeEnum["Scan_39"] = 39] = "Scan_39";
    ScanTypeEnum[ScanTypeEnum["Scan_40"] = 40] = "Scan_40";
    ScanTypeEnum[ScanTypeEnum["Scan_41"] = 41] = "Scan_41";
    ScanTypeEnum[ScanTypeEnum["Scan_42"] = 42] = "Scan_42";
    ScanTypeEnum[ScanTypeEnum["Scan_43"] = 43] = "Scan_43";
    ScanTypeEnum[ScanTypeEnum["Scan_44"] = 44] = "Scan_44";
    ScanTypeEnum[ScanTypeEnum["Scan_45"] = 45] = "Scan_45";
    ScanTypeEnum[ScanTypeEnum["Scan_46"] = 46] = "Scan_46";
    ScanTypeEnum[ScanTypeEnum["Scan_47"] = 47] = "Scan_47";
    ScanTypeEnum[ScanTypeEnum["Scan_48"] = 48] = "Scan_48";
    ScanTypeEnum[ScanTypeEnum["Scan_49"] = 49] = "Scan_49";
    ScanTypeEnum[ScanTypeEnum["Scan_50"] = 50] = "Scan_50";
    ScanTypeEnum[ScanTypeEnum["Scan_51"] = 51] = "Scan_51";
    ScanTypeEnum[ScanTypeEnum["Scan_52"] = 52] = "Scan_52";
    ScanTypeEnum[ScanTypeEnum["Scan_53"] = 53] = "Scan_53";
    ScanTypeEnum[ScanTypeEnum["Scan_54"] = 54] = "Scan_54";
    ScanTypeEnum[ScanTypeEnum["Scan_55"] = 55] = "Scan_55";
    ScanTypeEnum[ScanTypeEnum["Scan_56"] = 56] = "Scan_56";
    ScanTypeEnum[ScanTypeEnum["Scan_57"] = 57] = "Scan_57";
    ScanTypeEnum[ScanTypeEnum["Scan_58"] = 58] = "Scan_58";
    ScanTypeEnum[ScanTypeEnum["Scan_59"] = 59] = "Scan_59";
    ScanTypeEnum[ScanTypeEnum["Scan_60"] = 60] = "Scan_60";
    ScanTypeEnum[ScanTypeEnum["Scan_61"] = 61] = "Scan_61";
    ScanTypeEnum[ScanTypeEnum["Scan_62"] = 62] = "Scan_62";
    ScanTypeEnum[ScanTypeEnum["Scan_63"] = 63] = "Scan_63";
    ScanTypeEnum[ScanTypeEnum["Scan_64"] = 64] = "Scan_64";
    ScanTypeEnum[ScanTypeEnum["Scan_65"] = 65] = "Scan_65";
    ScanTypeEnum[ScanTypeEnum["Scan_66"] = 66] = "Scan_66";
    ScanTypeEnum[ScanTypeEnum["Scan_67"] = 67] = "Scan_67";
    ScanTypeEnum[ScanTypeEnum["Scan_68"] = 68] = "Scan_68";
    ScanTypeEnum[ScanTypeEnum["Scan_69"] = 69] = "Scan_69";
    ScanTypeEnum[ScanTypeEnum["Scan_70"] = 70] = "Scan_70";
    ScanTypeEnum[ScanTypeEnum["Scan_71"] = 71] = "Scan_71";
    ScanTypeEnum[ScanTypeEnum["Scan_72"] = 72] = "Scan_72";
    ScanTypeEnum[ScanTypeEnum["Scan_73"] = 73] = "Scan_73";
    ScanTypeEnum[ScanTypeEnum["Scan_74"] = 74] = "Scan_74";
    ScanTypeEnum[ScanTypeEnum["Scan_75"] = 75] = "Scan_75";
    ScanTypeEnum[ScanTypeEnum["Scan_76"] = 76] = "Scan_76";
    ScanTypeEnum[ScanTypeEnum["Scan_77"] = 77] = "Scan_77";
    ScanTypeEnum[ScanTypeEnum["Scan_78"] = 78] = "Scan_78";
    ScanTypeEnum[ScanTypeEnum["Scan_79"] = 79] = "Scan_79";
    ScanTypeEnum[ScanTypeEnum["Scan_80"] = 80] = "Scan_80";
    ScanTypeEnum[ScanTypeEnum["Scan_81"] = 81] = "Scan_81";
    ScanTypeEnum[ScanTypeEnum["Scan_82"] = 82] = "Scan_82";
    ScanTypeEnum[ScanTypeEnum["Scan_83"] = 83] = "Scan_83";
    ScanTypeEnum[ScanTypeEnum["Scan_84"] = 84] = "Scan_84";
    ScanTypeEnum[ScanTypeEnum["Scan_85"] = 85] = "Scan_85";
    ScanTypeEnum[ScanTypeEnum["Scan_86"] = 86] = "Scan_86";
    ScanTypeEnum[ScanTypeEnum["Scan_87"] = 87] = "Scan_87";
    ScanTypeEnum[ScanTypeEnum["Scan_88"] = 88] = "Scan_88";
    ScanTypeEnum[ScanTypeEnum["Scan_89"] = 89] = "Scan_89";
    ScanTypeEnum[ScanTypeEnum["Scan_90"] = 90] = "Scan_90";
    ScanTypeEnum[ScanTypeEnum["Scan_91"] = 91] = "Scan_91";
    ScanTypeEnum[ScanTypeEnum["Scan_92"] = 92] = "Scan_92";
    ScanTypeEnum[ScanTypeEnum["Scan_93"] = 93] = "Scan_93";
    ScanTypeEnum[ScanTypeEnum["Scan_94"] = 94] = "Scan_94";
    ScanTypeEnum[ScanTypeEnum["Scan_95"] = 95] = "Scan_95";
    ScanTypeEnum[ScanTypeEnum["Scan_96"] = 96] = "Scan_96";
    ScanTypeEnum[ScanTypeEnum["Scan_97"] = 97] = "Scan_97";
    ScanTypeEnum[ScanTypeEnum["Scan_98"] = 98] = "Scan_98";
    ScanTypeEnum[ScanTypeEnum["Scan_99"] = 99] = "Scan_99";
    ScanTypeEnum[ScanTypeEnum["Scan_100"] = 100] = "Scan_100";
    ScanTypeEnum[ScanTypeEnum["Scan_101"] = 101] = "Scan_101";
    ScanTypeEnum[ScanTypeEnum["Scan_102"] = 102] = "Scan_102";
    ScanTypeEnum[ScanTypeEnum["Scan_103"] = 103] = "Scan_103";
    ScanTypeEnum[ScanTypeEnum["Scan_104"] = 104] = "Scan_104";
    ScanTypeEnum[ScanTypeEnum["Scan_105"] = 105] = "Scan_105";
    ScanTypeEnum[ScanTypeEnum["Scan_106"] = 106] = "Scan_106";
    ScanTypeEnum[ScanTypeEnum["Scan_107"] = 107] = "Scan_107";
    ScanTypeEnum[ScanTypeEnum["Scan_108"] = 108] = "Scan_108";
    ScanTypeEnum[ScanTypeEnum["Scan_109"] = 109] = "Scan_109";
    ScanTypeEnum[ScanTypeEnum["Scan_110"] = 110] = "Scan_110";
    ScanTypeEnum[ScanTypeEnum["Scan_111"] = 111] = "Scan_111";
    ScanTypeEnum[ScanTypeEnum["Scan_112"] = 112] = "Scan_112";
    ScanTypeEnum[ScanTypeEnum["Scan_113"] = 113] = "Scan_113";
    ScanTypeEnum[ScanTypeEnum["Scan_114"] = 114] = "Scan_114";
    ScanTypeEnum[ScanTypeEnum["Scan_115"] = 115] = "Scan_115";
    ScanTypeEnum[ScanTypeEnum["Scan_116"] = 116] = "Scan_116";
    ScanTypeEnum[ScanTypeEnum["Scan_117"] = 117] = "Scan_117";
    ScanTypeEnum[ScanTypeEnum["Scan_118"] = 118] = "Scan_118";
    ScanTypeEnum[ScanTypeEnum["Scan_119"] = 119] = "Scan_119";
    ScanTypeEnum[ScanTypeEnum["Scan_120"] = 120] = "Scan_120";
    ScanTypeEnum[ScanTypeEnum["Scan_121"] = 121] = "Scan_121";
    ScanTypeEnum[ScanTypeEnum["Scan_122"] = 122] = "Scan_122";
    ScanTypeEnum[ScanTypeEnum["Scan_123"] = 123] = "Scan_123";
    ScanTypeEnum[ScanTypeEnum["Scan_124"] = 124] = "Scan_124";
    ScanTypeEnum[ScanTypeEnum["Scan_125"] = 125] = "Scan_125";
    ScanTypeEnum[ScanTypeEnum["Scan_126"] = 126] = "Scan_126";
    ScanTypeEnum[ScanTypeEnum["Scan_127"] = 127] = "Scan_127";
    ScanTypeEnum[ScanTypeEnum["Scan_128"] = 128] = "Scan_128";
    ScanTypeEnum[ScanTypeEnum["Unknown"] = 255] = "Unknown";
})(ScanTypeEnum || (ScanTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ScanTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:820
 */
export const ScanType = EnumFromString(ScanTypeEnum, 'ScanType');
