// @novastar/native/build/module/generated/SimpleLEDDisplayInfo.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { LEDDisplyType } from './LEDDisplyType';
import { PortScanBoardInfo } from './PortScanBoardInfo';
import { ScreenAdjustParams } from './ScreenAdjustParams';
import { VirtualModeType } from './VirtualModeType';
/**
 * @category Codecs
 * @desc Codec for interface {@link SimpleLEDDisplayInfo}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:53801
 */
export const SimpleLEDDisplayInfo = t.intersection([
    t.type({
        PixelColsInScanBd: common.UInt16_32,
        PixelRowsInScanBd: common.UInt16_32,
        PortScanBdInfoList: common.XMLArray(PortScanBoardInfo, 'PortScanBoardInfo'),
    }),
    t.partial({
        X: common.UInt16,
        Y: common.UInt16,
        ScanBdCols: common.UInt16,
        ScanBdRows: common.UInt16,
        SenderIndex: common.UInt8,
        IsOpen18Bit: common.Bool,
        IsOpenClearview: common.Bool,
        Type: LEDDisplyType,
        VirtualMode: VirtualModeType,
        PortCols: common.UInt8,
        PortRows: common.UInt8,
        ScannerCount: common.Int32,
        ScrAdjustParams: ScreenAdjustParams,
    }),
], 'SimpleLEDDisplayInfo');
