// @novastar/native/build/module/generated/SourceSelectMode.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var SourceSelectModeEnum;
(function (SourceSelectModeEnum) {
    SourceSelectModeEnum[SourceSelectModeEnum["DVI"] = 16] = "DVI";
    SourceSelectModeEnum[SourceSelectModeEnum["DVI1"] = 17] = "DVI1";
    SourceSelectModeEnum[SourceSelectModeEnum["DVI2"] = 18] = "DVI2";
    SourceSelectModeEnum[SourceSelectModeEnum["DVI3"] = 19] = "DVI3";
    SourceSelectModeEnum[SourceSelectModeEnum["DVI4"] = 20] = "DVI4";
    SourceSelectModeEnum[SourceSelectModeEnum["HDMI"] = 32] = "HDMI";
    SourceSelectModeEnum[SourceSelectModeEnum["DP"] = 48] = "DP";
    SourceSelectModeEnum[SourceSelectModeEnum["SDI"] = 64] = "SDI";
    SourceSelectModeEnum[SourceSelectModeEnum["Dual_DVI"] = 80] = "Dual_DVI";
    SourceSelectModeEnum[SourceSelectModeEnum["Dual_DVI1"] = 81] = "Dual_DVI1";
    SourceSelectModeEnum[SourceSelectModeEnum["Dual_DVI2"] = 82] = "Dual_DVI2";
    SourceSelectModeEnum[SourceSelectModeEnum["HDMI1_4"] = 96] = "HDMI1_4";
    SourceSelectModeEnum[SourceSelectModeEnum["HDBaseT"] = 192] = "HDBaseT";
})(SourceSelectModeEnum || (SourceSelectModeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link SourceSelectModeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1217
 */
export const SourceSelectMode = EnumFromString(SourceSelectModeEnum, 'SourceSelectMode');
