// @novastar/native/build/module/generated/StandardLEDDisplayInfo.js
import * as t from 'io-ts';
import * as common from '../lib/common';
import { LEDDisplyType } from './LEDDisplyType';
import { ScanBoardRegionInfo } from './ScanBoardRegionInfo';
import { ScreenAdjustParams } from './ScreenAdjustParams';
import { VirtualModeType } from './VirtualModeType';
/**
 * @category Codecs
 * @desc Codec for interface {@link StandardLEDDisplayInfo}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:54486
 */
export const StandardLEDDisplayInfo = t.intersection([
    t.type({
        ScanBoardCols: common.UInt16_1,
        ScanBoardRows: common.UInt16_1,
        ScannerRegionList: common.XMLArray(ScanBoardRegionInfo, 'ScanBoardRegionInfo'),
    }),
    t.partial({
        X: common.UInt16,
        Y: common.UInt16,
        DVIOffest: t.record(common.Int32, common.PointFromString),
        IsOpen18Bit: common.Bool,
        IsOpenClearview: common.Bool,
        Type: LEDDisplyType,
        VirtualMode: VirtualModeType,
        ScannerCount: common.Int32,
        ScrAdjustParams: ScreenAdjustParams,
    }),
], 'StandardLEDDisplayInfo');
