// @novastar/native/build/module/generated/TestMode.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var TestModeEnum;
(function (TestModeEnum) {
    TestModeEnum[TestModeEnum["Reserved1_Mode"] = 0] = "Reserved1_Mode";
    TestModeEnum[TestModeEnum["Reserved2_Mode"] = 1] = "Reserved2_Mode";
    TestModeEnum[TestModeEnum["Red_Mode"] = 2] = "Red_Mode";
    TestModeEnum[TestModeEnum["Green_Mode"] = 3] = "Green_Mode";
    TestModeEnum[TestModeEnum["Blue_Mode"] = 4] = "Blue_Mode";
    TestModeEnum[TestModeEnum["White_Mode"] = 5] = "White_Mode";
    TestModeEnum[TestModeEnum["HorizonLine_Mode"] = 6] = "HorizonLine_Mode";
    TestModeEnum[TestModeEnum["VerticalLine_Mode"] = 7] = "VerticalLine_Mode";
    TestModeEnum[TestModeEnum["InclineLine_Mode"] = 8] = "InclineLine_Mode";
    TestModeEnum[TestModeEnum["GrayIncrease_Mode"] = 9] = "GrayIncrease_Mode";
    TestModeEnum[TestModeEnum["Age_Mode"] = 10] = "Age_Mode";
    TestModeEnum[TestModeEnum["HardwareScreen_Mode"] = 11] = "HardwareScreen_Mode";
    TestModeEnum[TestModeEnum["LogoTest"] = 16] = "LogoTest";
    TestModeEnum[TestModeEnum["ParaFreeze"] = 32] = "ParaFreeze";
    TestModeEnum[TestModeEnum["OEFreeze"] = 64] = "OEFreeze";
})(TestModeEnum || (TestModeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link TestModeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:973
 */
export const TestMode = EnumFromString(TestModeEnum, 'TestMode');
