// @novastar/native/build/module/generated/ThresholdType.js
import EnumFromString from '../lib/common/EnumFromString';
/** @category Enums */
export var ThresholdTypeEnum;
(function (ThresholdTypeEnum) {
    ThresholdTypeEnum[ThresholdTypeEnum["None"] = 0] = "None";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade1"] = 1] = "Grade1";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade2"] = 2] = "Grade2";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade3"] = 3] = "Grade3";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade4"] = 4] = "Grade4";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade5"] = 5] = "Grade5";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade6"] = 6] = "Grade6";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade7"] = 7] = "Grade7";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade8"] = 8] = "Grade8";
})(ThresholdTypeEnum || (ThresholdTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ThresholdTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:61256
 */
export const ThresholdType = EnumFromString(ThresholdTypeEnum, 'ThresholdType');
