// @novastar/native/build/main/lib/common/types.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UInt16_63 = exports.UInt16_3841 = exports.UInt16_26722 = exports.UInt16_28770 = exports.UInt16_30818 = exports.UInt16_65474 = exports.Int32_16 = exports.Int32_128 = exports.Int32_32 = exports.Int32_13 = exports.UInt16_7936 = exports.UInt16_256 = exports.UInt16_18183 = exports.UInt16_31615 = exports.UInt16_8048 = exports.buffer_6 = exports.buffer_8 = exports.UInt8_2 = exports.Int32_60 = exports.Int32_15 = exports.UInt16_15872 = exports.UInt16_508 = exports.UInt16_16384 = exports.UInt16_4096 = exports.UInt8_5 = exports.UInt16_3822 = exports.UInt16_8673 = exports.UInt16_61440 = exports.Int32_1 = exports.UInt8_28 = exports.UInt16_1 = exports.UInt8_23 = exports.UInt8_31 = exports.UInt8_15 = exports.Bool_true = exports.UInt8_32 = exports.UInt16_30976 = exports.UInt16_58369 = exports.UInt16_62465 = exports.UInt16_32569 = exports.UInt8_3 = exports.UInt8_1 = exports.UInt8_7 = exports.Int32_2650 = exports.Int32_3 = exports.Bool_false = exports.UInt8_0 = exports.UInt16_0 = exports.Int32_0 = exports.string_empty = void 0;
exports.UInt16_29696 = exports.UInt16_29440 = exports.UInt16_29184 = exports.UInt16_28928 = exports.UInt16_28672 = exports.UInt16_8732 = exports.UInt16_8449 = exports.UInt16_8200 = exports.UInt16_7940 = exports.UInt16_7746 = exports.UInt16_7434 = exports.UInt16_7376 = exports.UInt16_6928 = exports.UInt16_6687 = exports.UInt16_6400 = exports.UInt16_6175 = exports.UInt16_6128 = exports.UInt16_5632 = exports.UInt16_5376 = exports.UInt16_5120 = exports.UInt16_4609 = exports.UInt16_4385 = exports.UInt16_4226 = exports.UInt16_3588 = exports.UInt16_3329 = exports.UInt16_3073 = exports.UInt16_2944 = exports.UInt16_2568 = exports.UInt16_2312 = exports.UInt16_2080 = exports.UInt16_1824 = exports.UInt16_1538 = exports.UInt16_1292 = exports.UInt16_1027 = exports.UInt16_799 = exports.UInt16_543 = exports.UInt16_341 = exports.UInt16_85 = exports.UInt16_128 = exports.UInt8_12 = exports.UInt16_185 = exports.UInt8_8 = exports.UInt8_29 = exports.UInt8_4 = exports.Int32_3300 = exports.Int32_140 = exports.Int32_137 = exports.Int32_25 = exports.Int32_10 = exports.Int32_48 = void 0;
exports.UInt8_100 = exports.UInt16_2048 = exports.UInt8_10 = exports.Int32_255 = exports.buffer_MAX_MULTIPLE_CHIP_TABLELEN = exports.buffer_GAMMAVALUE_COUNT = exports.Int32_300 = exports.Int32_100 = exports.Int32_4559 = exports.Numeric_28 = exports.Int32_2 = exports.Int32_8 = exports.buffer_2 = exports.buffer_COLOR_COUNT = exports.buffer_MAX_MODULEDATAGROUP = exports.string_20 = exports.UInt16_768 = exports.UInt16_1024 = exports.buffer_96 = exports.UInt8_255 = exports.UInt16_7168 = exports.UInt8_56 = exports.UInt16_29951 = exports.Int32_34 = exports.UInt16_3072 = exports.UInt16_65337 = exports.Int32_6 = exports.Int32_5 = exports.UInt16_44932 = exports.UInt16_41728 = exports.Int32_196 = exports.Int32_4 = exports.UInt16_16387 = exports.UInt16_1023 = exports.UInt16_64929 = exports.UInt16_11 = exports.Int32_21 = exports.Int32_11 = exports.Int32_14 = exports.UInt16_65535 = exports.UInt16_60543 = exports.UInt16_23 = exports.UInt8_16 = exports.Int32_12 = exports.UInt16_8960 = exports.UInt16_62720 = exports.UInt16_62464 = exports.UInt16_62208 = exports.UInt16_61952 = exports.UInt16_61696 = void 0;
exports.UInt16_400 = exports.UInt16_32 = void 0;
/* Automatically generated */
const t = __importStar(require("io-ts"));
const MaxValue_1 = __importDefault(require("../../generated/MaxValue"));
const integers_1 = require("./integers");
exports.string_empty = (0, integers_1.withDefault)(t.string, '');
exports.Int32_0 = (0, integers_1.withDefault)(integers_1.Int32, 0);
exports.UInt16_0 = (0, integers_1.withDefault)(integers_1.UInt16, 0);
exports.UInt8_0 = (0, integers_1.withDefault)(integers_1.UInt8, 0);
exports.Bool_false = (0, integers_1.withDefault)(integers_1.Bool, false);
exports.Int32_3 = (0, integers_1.withDefault)(integers_1.Int32, 3);
exports.Int32_2650 = (0, integers_1.withDefault)(integers_1.Int32, 2650);
exports.UInt8_7 = (0, integers_1.withDefault)(integers_1.UInt8, 7);
exports.UInt8_1 = (0, integers_1.withDefault)(integers_1.UInt8, 1);
exports.UInt8_3 = (0, integers_1.withDefault)(integers_1.UInt8, 3);
exports.UInt16_32569 = (0, integers_1.withDefault)(integers_1.UInt16, 32569);
exports.UInt16_62465 = (0, integers_1.withDefault)(integers_1.UInt16, 62465);
exports.UInt16_58369 = (0, integers_1.withDefault)(integers_1.UInt16, 58369);
exports.UInt16_30976 = (0, integers_1.withDefault)(integers_1.UInt16, 30976);
exports.UInt8_32 = (0, integers_1.withDefault)(integers_1.UInt8, 32);
exports.Bool_true = (0, integers_1.withDefault)(integers_1.Bool, true);
exports.UInt8_15 = (0, integers_1.withDefault)(integers_1.UInt8, 15);
exports.UInt8_31 = (0, integers_1.withDefault)(integers_1.UInt8, 31);
exports.UInt8_23 = (0, integers_1.withDefault)(integers_1.UInt8, 23);
exports.UInt16_1 = (0, integers_1.withDefault)(integers_1.UInt16, 1);
exports.UInt8_28 = (0, integers_1.withDefault)(integers_1.UInt8, 28);
exports.Int32_1 = (0, integers_1.withDefault)(integers_1.Int32, 1);
exports.UInt16_61440 = (0, integers_1.withDefault)(integers_1.UInt16, 61440);
exports.UInt16_8673 = (0, integers_1.withDefault)(integers_1.UInt16, 8673);
exports.UInt16_3822 = (0, integers_1.withDefault)(integers_1.UInt16, 3822);
exports.UInt8_5 = (0, integers_1.withDefault)(integers_1.UInt8, 5);
exports.UInt16_4096 = (0, integers_1.withDefault)(integers_1.UInt16, 4096);
exports.UInt16_16384 = (0, integers_1.withDefault)(integers_1.UInt16, 16384);
exports.UInt16_508 = (0, integers_1.withDefault)(integers_1.UInt16, 508);
exports.UInt16_15872 = (0, integers_1.withDefault)(integers_1.UInt16, 15872);
exports.Int32_15 = (0, integers_1.withDefault)(integers_1.Int32, 15);
exports.Int32_60 = (0, integers_1.withDefault)(integers_1.Int32, 60);
exports.UInt8_2 = (0, integers_1.withDefault)(integers_1.UInt8, 2);
exports.buffer_8 = new integers_1.BufferFromBase64('buffer_8', 8);
exports.buffer_6 = new integers_1.BufferFromBase64('buffer_6', 6);
exports.UInt16_8048 = (0, integers_1.withDefault)(integers_1.UInt16, 8048);
exports.UInt16_31615 = (0, integers_1.withDefault)(integers_1.UInt16, 31615);
exports.UInt16_18183 = (0, integers_1.withDefault)(integers_1.UInt16, 18183);
exports.UInt16_256 = (0, integers_1.withDefault)(integers_1.UInt16, 256);
exports.UInt16_7936 = (0, integers_1.withDefault)(integers_1.UInt16, 7936);
exports.Int32_13 = (0, integers_1.withDefault)(integers_1.Int32, 13);
exports.Int32_32 = (0, integers_1.withDefault)(integers_1.Int32, 32);
exports.Int32_128 = (0, integers_1.withDefault)(integers_1.Int32, 128);
exports.Int32_16 = (0, integers_1.withDefault)(integers_1.Int32, 16);
exports.UInt16_65474 = (0, integers_1.withDefault)(integers_1.UInt16, 65474);
exports.UInt16_30818 = (0, integers_1.withDefault)(integers_1.UInt16, 30818);
exports.UInt16_28770 = (0, integers_1.withDefault)(integers_1.UInt16, 28770);
exports.UInt16_26722 = (0, integers_1.withDefault)(integers_1.UInt16, 26722);
exports.UInt16_3841 = (0, integers_1.withDefault)(integers_1.UInt16, 3841);
exports.UInt16_63 = (0, integers_1.withDefault)(integers_1.UInt16, 63);
exports.Int32_48 = (0, integers_1.withDefault)(integers_1.Int32, 48);
exports.Int32_10 = (0, integers_1.withDefault)(integers_1.Int32, 10);
exports.Int32_25 = (0, integers_1.withDefault)(integers_1.Int32, 25);
exports.Int32_137 = (0, integers_1.withDefault)(integers_1.Int32, 137);
exports.Int32_140 = (0, integers_1.withDefault)(integers_1.Int32, 140);
exports.Int32_3300 = (0, integers_1.withDefault)(integers_1.Int32, 3300);
exports.UInt8_4 = (0, integers_1.withDefault)(integers_1.UInt8, 4);
exports.UInt8_29 = (0, integers_1.withDefault)(integers_1.UInt8, 29);
exports.UInt8_8 = (0, integers_1.withDefault)(integers_1.UInt8, 8);
exports.UInt16_185 = (0, integers_1.withDefault)(integers_1.UInt16, 185);
exports.UInt8_12 = (0, integers_1.withDefault)(integers_1.UInt8, 12);
exports.UInt16_128 = (0, integers_1.withDefault)(integers_1.UInt16, 128);
exports.UInt16_85 = (0, integers_1.withDefault)(integers_1.UInt16, 85);
exports.UInt16_341 = (0, integers_1.withDefault)(integers_1.UInt16, 341);
exports.UInt16_543 = (0, integers_1.withDefault)(integers_1.UInt16, 543);
exports.UInt16_799 = (0, integers_1.withDefault)(integers_1.UInt16, 799);
exports.UInt16_1027 = (0, integers_1.withDefault)(integers_1.UInt16, 1027);
exports.UInt16_1292 = (0, integers_1.withDefault)(integers_1.UInt16, 1292);
exports.UInt16_1538 = (0, integers_1.withDefault)(integers_1.UInt16, 1538);
exports.UInt16_1824 = (0, integers_1.withDefault)(integers_1.UInt16, 1824);
exports.UInt16_2080 = (0, integers_1.withDefault)(integers_1.UInt16, 2080);
exports.UInt16_2312 = (0, integers_1.withDefault)(integers_1.UInt16, 2312);
exports.UInt16_2568 = (0, integers_1.withDefault)(integers_1.UInt16, 2568);
exports.UInt16_2944 = (0, integers_1.withDefault)(integers_1.UInt16, 2944);
exports.UInt16_3073 = (0, integers_1.withDefault)(integers_1.UInt16, 3073);
exports.UInt16_3329 = (0, integers_1.withDefault)(integers_1.UInt16, 3329);
exports.UInt16_3588 = (0, integers_1.withDefault)(integers_1.UInt16, 3588);
exports.UInt16_4226 = (0, integers_1.withDefault)(integers_1.UInt16, 4226);
exports.UInt16_4385 = (0, integers_1.withDefault)(integers_1.UInt16, 4385);
exports.UInt16_4609 = (0, integers_1.withDefault)(integers_1.UInt16, 4609);
exports.UInt16_5120 = (0, integers_1.withDefault)(integers_1.UInt16, 5120);
exports.UInt16_5376 = (0, integers_1.withDefault)(integers_1.UInt16, 5376);
exports.UInt16_5632 = (0, integers_1.withDefault)(integers_1.UInt16, 5632);
exports.UInt16_6128 = (0, integers_1.withDefault)(integers_1.UInt16, 6128);
exports.UInt16_6175 = (0, integers_1.withDefault)(integers_1.UInt16, 6175);
exports.UInt16_6400 = (0, integers_1.withDefault)(integers_1.UInt16, 6400);
exports.UInt16_6687 = (0, integers_1.withDefault)(integers_1.UInt16, 6687);
exports.UInt16_6928 = (0, integers_1.withDefault)(integers_1.UInt16, 6928);
exports.UInt16_7376 = (0, integers_1.withDefault)(integers_1.UInt16, 7376);
exports.UInt16_7434 = (0, integers_1.withDefault)(integers_1.UInt16, 7434);
exports.UInt16_7746 = (0, integers_1.withDefault)(integers_1.UInt16, 7746);
exports.UInt16_7940 = (0, integers_1.withDefault)(integers_1.UInt16, 7940);
exports.UInt16_8200 = (0, integers_1.withDefault)(integers_1.UInt16, 8200);
exports.UInt16_8449 = (0, integers_1.withDefault)(integers_1.UInt16, 8449);
exports.UInt16_8732 = (0, integers_1.withDefault)(integers_1.UInt16, 8732);
exports.UInt16_28672 = (0, integers_1.withDefault)(integers_1.UInt16, 28672);
exports.UInt16_28928 = (0, integers_1.withDefault)(integers_1.UInt16, 28928);
exports.UInt16_29184 = (0, integers_1.withDefault)(integers_1.UInt16, 29184);
exports.UInt16_29440 = (0, integers_1.withDefault)(integers_1.UInt16, 29440);
exports.UInt16_29696 = (0, integers_1.withDefault)(integers_1.UInt16, 29696);
exports.UInt16_61696 = (0, integers_1.withDefault)(integers_1.UInt16, 61696);
exports.UInt16_61952 = (0, integers_1.withDefault)(integers_1.UInt16, 61952);
exports.UInt16_62208 = (0, integers_1.withDefault)(integers_1.UInt16, 62208);
exports.UInt16_62464 = (0, integers_1.withDefault)(integers_1.UInt16, 62464);
exports.UInt16_62720 = (0, integers_1.withDefault)(integers_1.UInt16, 62720);
exports.UInt16_8960 = (0, integers_1.withDefault)(integers_1.UInt16, 8960);
exports.Int32_12 = (0, integers_1.withDefault)(integers_1.Int32, 12);
exports.UInt8_16 = (0, integers_1.withDefault)(integers_1.UInt8, 16);
exports.UInt16_23 = (0, integers_1.withDefault)(integers_1.UInt16, 23);
exports.UInt16_60543 = (0, integers_1.withDefault)(integers_1.UInt16, 60543);
exports.UInt16_65535 = (0, integers_1.withDefault)(integers_1.UInt16, 65535);
exports.Int32_14 = (0, integers_1.withDefault)(integers_1.Int32, 14);
exports.Int32_11 = (0, integers_1.withDefault)(integers_1.Int32, 11);
exports.Int32_21 = (0, integers_1.withDefault)(integers_1.Int32, 21);
exports.UInt16_11 = (0, integers_1.withDefault)(integers_1.UInt16, 11);
exports.UInt16_64929 = (0, integers_1.withDefault)(integers_1.UInt16, 64929);
exports.UInt16_1023 = (0, integers_1.withDefault)(integers_1.UInt16, 1023);
exports.UInt16_16387 = (0, integers_1.withDefault)(integers_1.UInt16, 16387);
exports.Int32_4 = (0, integers_1.withDefault)(integers_1.Int32, 4);
exports.Int32_196 = (0, integers_1.withDefault)(integers_1.Int32, 196);
exports.UInt16_41728 = (0, integers_1.withDefault)(integers_1.UInt16, 41728);
exports.UInt16_44932 = (0, integers_1.withDefault)(integers_1.UInt16, 44932);
exports.Int32_5 = (0, integers_1.withDefault)(integers_1.Int32, 5);
exports.Int32_6 = (0, integers_1.withDefault)(integers_1.Int32, 6);
exports.UInt16_65337 = (0, integers_1.withDefault)(integers_1.UInt16, 65337);
exports.UInt16_3072 = (0, integers_1.withDefault)(integers_1.UInt16, 3072);
exports.Int32_34 = (0, integers_1.withDefault)(integers_1.Int32, 34);
exports.UInt16_29951 = (0, integers_1.withDefault)(integers_1.UInt16, 29951);
exports.UInt8_56 = (0, integers_1.withDefault)(integers_1.UInt8, 56);
exports.UInt16_7168 = (0, integers_1.withDefault)(integers_1.UInt16, 7168);
exports.UInt8_255 = (0, integers_1.withDefault)(integers_1.UInt8, 255);
exports.buffer_96 = new integers_1.BufferFromBase64('buffer_96', 96);
exports.UInt16_1024 = (0, integers_1.withDefault)(integers_1.UInt16, 1024);
exports.UInt16_768 = (0, integers_1.withDefault)(integers_1.UInt16, 768);
exports.string_20 = (0, integers_1.withDefault)(t.string, '2.0');
exports.buffer_MAX_MODULEDATAGROUP = new integers_1.BufferFromBase64('buffer_MAX_MODULEDATAGROUP', MaxValue_1.default.MAX_MODULEDATAGROUP);
exports.buffer_COLOR_COUNT = new integers_1.BufferFromBase64('buffer_COLOR_COUNT', MaxValue_1.default.COLOR_COUNT);
exports.buffer_2 = new integers_1.BufferFromBase64('buffer_2', 2);
exports.Int32_8 = (0, integers_1.withDefault)(integers_1.Int32, 8);
exports.Int32_2 = (0, integers_1.withDefault)(integers_1.Int32, 2);
exports.Numeric_28 = (0, integers_1.withDefault)(integers_1.Numeric, 2.8);
exports.Int32_4559 = (0, integers_1.withDefault)(integers_1.Int32, 4559);
exports.Int32_100 = (0, integers_1.withDefault)(integers_1.Int32, 100);
exports.Int32_300 = (0, integers_1.withDefault)(integers_1.Int32, 300);
exports.buffer_GAMMAVALUE_COUNT = new integers_1.BufferFromBase64('buffer_GAMMAVALUE_COUNT', MaxValue_1.default.GAMMAVALUE_COUNT);
exports.buffer_MAX_MULTIPLE_CHIP_TABLELEN = new integers_1.BufferFromBase64('buffer_MAX_MULTIPLE_CHIP_TABLELEN', MaxValue_1.default.MAX_MULTIPLE_CHIP_TABLELEN);
exports.Int32_255 = (0, integers_1.withDefault)(integers_1.Int32, 255);
exports.UInt8_10 = (0, integers_1.withDefault)(integers_1.UInt8, 10);
exports.UInt16_2048 = (0, integers_1.withDefault)(integers_1.UInt16, 2048);
exports.UInt8_100 = (0, integers_1.withDefault)(integers_1.UInt8, 100);
exports.UInt16_32 = (0, integers_1.withDefault)(integers_1.UInt16, 32);
exports.UInt16_400 = (0, integers_1.withDefault)(integers_1.UInt16, 400);
