// @novastar/screen/build/module/ScannerCapabilitySUM2030.js
export default class ScannerCapabilitySUM2030 {
    static SUBDIFF = 0.15;
    static BLACKTIME_STATIC = 5;
    IsHasMinimumOE = false;
    MinimumOE = 0;
    LightTimePerSubField = 0;
    TotalLightCdfResault = 0;
    _totalUnitNumPerSubField = 0;
    _gclkNumPerRef = 0;
    _refNumPerVs = 0;
    _totalGclkUintNumPerScan = 0;
    _lightTimeNum2 = 0;
    _dclkUnitCycle = 0;
    _gclkUnitCycle = 0;
    _scanCount = 0;
    _fieldPeriod = 0;
    _vsFreq = 60;
    _usBlackTime = 0;
    _shiftNumPerSubField = 0;
    _maxUnitNumPerScan = 0;
    _maxGclkNumPerRef = 0;
    _brightEfficiency = 0;
    _blackingTime = 0;
    _maxLoadedPixels = 0;
    _fieldWaitTime = 0.1;
    _m1TranCntNum = 3;
    _m2TranCntNum = 3;
    _scanBdProperty = null;
    get TotalUnitNumPerSubField() {
        return this._totalUnitNumPerSubField;
    }
    get MaxLoadedPixels() {
        return this._totalUnitNumPerSubField - 3;
    }
    get BrightEfficiency() {
        return this._brightEfficiency;
    }
    get GclkNumPerRef() {
        return this._gclkNumPerRef;
    }
    get RefNumPerVs() {
        return this._refNumPerVs;
    }
    get TotalGclkUnitNumPerScan() {
        return this._totalGclkUintNumPerScan;
    }
    get LightTimeNum2() {
        return this._lightTimeNum2;
    }
    // eslint-disable-next-line class-methods-use-this
    UpdateParameters(scanBdProp, vsFreq) {
        throw new TypeError('Not implemented');
    }
}
