// @novastar/screen/build/main/ChipInherentProperties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CanManualAdjustRefreshRate = exports.DetectPointIsConfigOtherReg = exports.IsNoCurrentGainButConfig = exports.GetDetectPointType = exports.IsConfigRegisterConcernWithShiftUnitNum = exports.IsConfigRegisterConcernWithScanType = exports.IsDoubleLatChip = exports.IsSupportPhotographMode = exports.IsSupportNewOERealizeWay = exports.IsSupportDriverInterDeghostEnable = exports.IsSupportMBI5041BNewTiming = exports.IsSupport5042GrayEnhanced = exports.IsPWMAndHavePartNum = exports.IsSupportCurrentChipTypeEnum = exports.IsSupport = exports.IsChangedOETableChip = exports.IsPWMChip = exports.GetGainProportion = void 0;
// noinspection JSUnusedLocalSymbols
const ChipType_1 = require("@novastar/native/ChipType");
const DetectPointType_1 = require("@novastar/native/DetectPointType");
const ScanType_1 = require("@novastar/native/ScanType");
const chipInfo = (ChipType, isCommon, hasGain, hasPointDetect, channelCount, gainSteps) => ({
    ChipType,
    isCommon,
    hasGain,
    hasPointDetect,
    channelCount,
    gainSteps,
});
const gain16 = Array(16).map((_, index) => [
    16 * index,
    index >= 8 ? (56.25 + (index - 8) * 6.25) / 100 : (25 + 3.125 * index) / 100,
]);
const gain32 = Array(32).map((_, index) => [8 * index + 4, (index + 9) / 40]);
const gain64 = Array(64).map((_, index) => [
    4 * index + 2,
    index > 31 ? ((index - 32) * 3 + 33) / 65 : (index * 3 + 32) / 256,
]);
const gain384 = Array(384).map((_, index) => [
    index,
    index < 192 ? (index + 64) / 512 : (index - 128) / 256,
]);
const ChipInherentProperties = {
    Chip_CommonBase: chipInfo(ChipType_1.ChipTypeEnum.Chip_CommonBase, true, false, false, 16),
    Chip_MBI5167: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5167, true, false, false, 8),
    Chip_SUM2017: chipInfo(ChipType_1.ChipTypeEnum.Chip_SUM2017, true, false, false, 16),
    Chip_MBI5037: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5037, true, false, true, 16),
    Chip_MY9221: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9221, false, false, false, 12),
    Chip_MY9231: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9231, false, false, false, 12),
    Chip_TLS3001: chipInfo(ChipType_1.ChipTypeEnum.Chip_TLS3001, false, false, false, 12),
    Chip_GW6205: chipInfo(ChipType_1.ChipTypeEnum.Chip_GW6205, false, false, false, 12),
    Chip_UCS512C: chipInfo(ChipType_1.ChipTypeEnum.Chip_UCS512C, false, false, false, 12),
    Chip_UCS1912: chipInfo(ChipType_1.ChipTypeEnum.Chip_UCS1912, false, false, false, 12),
    Chip_LD1512: chipInfo(ChipType_1.ChipTypeEnum.Chip_LD1512, false, false, false, 12),
    Chip_GW6202B: chipInfo(ChipType_1.ChipTypeEnum.Chip_GW6202B, false, false, false, 12),
    Chip_MBI6030: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI6030, true, false, true, 16),
    Chip_MBI6023: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI6023, true, true, false, 16),
    Chip_MBI5124: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5124, true, false, false, 16),
    Chip_SM16017S: chipInfo(ChipType_1.ChipTypeEnum.Chip_SM16017S, true, false, false, 16),
    Chip_MBI5124A: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5124A, true, false, false, 16),
    Chip_MBI5124New: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5124New, true, false, false, 16),
    Chip_RT5924: chipInfo(ChipType_1.ChipTypeEnum.Chip_RT5924, true, false, false, 16),
    Chip_MY9868: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9868, true, false, false, 16),
    Chip_MY9868A: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9868A, true, false, false, 16),
    Chip_ICN2038: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICN2038, true, false, false, 16),
    Chip_ICN2038A: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICN2038A, true, false, false, 16),
    Chip_TLC5958: chipInfo(ChipType_1.ChipTypeEnum.Chip_TLC5958, true, false, false, 16),
    Chip_TLC59581: chipInfo(ChipType_1.ChipTypeEnum.Chip_TLC59581, true, false, false, 16),
    Chip_MBI5166: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5166, true, false, false, 16),
    Chip_MBI5266: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5266, true, false, true, 16),
    Chip_MBI5122: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5122, true, false, false, 16),
    Chip_MBI5041Q: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5041Q, true, false, false, 16),
    Chip_ICN2027: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICN2027, true, false, false, 16),
    Chip_LS9960: chipInfo(ChipType_1.ChipTypeEnum.Chip_LS9960, true, false, false, 16),
    Chip_LS9918: chipInfo(ChipType_1.ChipTypeEnum.Chip_LS9918, true, false, false, 16),
    Chip_LS9929: chipInfo(ChipType_1.ChipTypeEnum.Chip_LS9929, true, false, false, 16),
    Chip_LS9926: chipInfo(ChipType_1.ChipTypeEnum.Chip_LS9926, true, false, false, 16),
    Chip_LS9930: chipInfo(ChipType_1.ChipTypeEnum.Chip_LS9930, true, false, false, 16),
    Chip_LS9935: chipInfo(ChipType_1.ChipTypeEnum.Chip_LS9935, true, false, false, 16),
    Chip_ICND2100: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICND2100, true, false, false, 16),
    Chip_MBI5036: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5036, true, true, true, 16, gain64),
    Chip_SUM2017T: chipInfo(ChipType_1.ChipTypeEnum.Chip_SUM2017T, true, true, false, 16, gain32),
    Chip_SUM2117: chipInfo(ChipType_1.ChipTypeEnum.Chip_SUM2117, true, true, false, 16, gain32),
    Chip_SUM2017TD: chipInfo(ChipType_1.ChipTypeEnum.Chip_SUM2017TD, true, true, false, 16, gain32),
    Chip_MBI5038: chipInfo(ChipType_1.ChipTypeEnum.Chip_MBI5038, true, true, false, 16, gain64),
    Chip_MY9862: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9862, true, true, true, 16, gain64),
    Chip_MY9862A: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9862A, true, true, true, 16, gain64),
    Chip_ICN2038S: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICN2038S, true, true, true, 16, gain16),
    Chip_ICND2046: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICND2046, true, true, true, 16, gain16),
    Chip_FM6126: chipInfo(ChipType_1.ChipTypeEnum.Chip_FM6126, true, true, true, 16, gain16),
    Chip_ICN2038S_Common: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICN2038S_Common, true, true, true, 16, gain16),
    Chip_MY9866: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9866, true, true, true, 18, gain16),
    Chip_MY9866_Common: chipInfo(ChipType_1.ChipTypeEnum.Chip_MY9866_Common, true, true, true, 18, gain16),
    /** */
    Chip_ICN2053: chipInfo(ChipType_1.ChipTypeEnum.Chip_ICN2053, false, true, true, 16, gain384),
};
const GetGainProportion = (gain) => {
    if (gain < 14)
        return (12.5 + 1.4 * gain) / 100;
    if (gain < 23)
        return (31.9 + 1.4 * (gain - 14)) / 100;
    if (gain < 32)
        return (44.4 + 1.4 * (gain - 23)) / 100;
    if (gain < 41)
        return (56.9 + 1.4 * (gain - 32)) / 100;
    if (gain < 50)
        return (69.4 + 1.4 * (gain - 41)) / 100;
    if (gain < 59)
        return (81.9 + 1.4 * (gain - 50)) / 100;
    return (94.4 + 1.4 * (gain - 59)) / 100;
};
exports.GetGainProportion = GetGainProportion;
const IsPWMChip = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MBI5030,
    ChipType_1.ChipTypeEnum.Chip_MBI5040,
    ChipType_1.ChipTypeEnum.Chip_IT1505,
    ChipType_1.ChipTypeEnum.Chip_TLC5948A,
    ChipType_1.ChipTypeEnum.Chip_UCS9812,
    ChipType_1.ChipTypeEnum.Chip_SM16188B,
    ChipType_1.ChipTypeEnum.Chip_MBI5042,
    ChipType_1.ChipTypeEnum.Chip_SM16136,
    ChipType_1.ChipTypeEnum.Chip_MY9262,
    ChipType_1.ChipTypeEnum.Chip_MY9263,
    ChipType_1.ChipTypeEnum.Chip_MBI5050,
    ChipType_1.ChipTypeEnum.Chip_RFT3630,
    ChipType_1.ChipTypeEnum.Chip_SUM2030,
    ChipType_1.ChipTypeEnum.Chip_SUM2130,
    ChipType_1.ChipTypeEnum.Chip_MY9268,
    ChipType_1.ChipTypeEnum.Chip_SCL8060,
    ChipType_1.ChipTypeEnum.Chip_MBI5051,
    ChipType_1.ChipTypeEnum.Chip_MBI5052,
    ChipType_1.ChipTypeEnum.Chip_MBI5053,
    ChipType_1.ChipTypeEnum.Chip_SM16158,
    ChipType_1.ChipTypeEnum.Chip_TLC5958,
    ChipType_1.ChipTypeEnum.Chip_TLC59581,
    ChipType_1.ChipTypeEnum.Chip_MY9266,
    ChipType_1.ChipTypeEnum.Chip_MY9366,
    ChipType_1.ChipTypeEnum.Chip_MBI5042B,
    ChipType_1.ChipTypeEnum.Chip_ISA2020,
    ChipType_1.ChipTypeEnum.Chip_LS9960,
    ChipType_1.ChipTypeEnum.Chip_LS9918,
    ChipType_1.ChipTypeEnum.Chip_LS9929,
    ChipType_1.ChipTypeEnum.Chip_LS9935,
    ChipType_1.ChipTypeEnum.Chip_LS9930,
    ChipType_1.ChipTypeEnum.Chip_LS9926,
    ChipType_1.ChipTypeEnum.Chip_MBI5043,
    ChipType_1.ChipTypeEnum.Chip_MBI5041B,
    ChipType_1.ChipTypeEnum.Chip_MBIA043,
    ChipType_1.ChipTypeEnum.Chip_MBI5041Q,
    ChipType_1.ChipTypeEnum.Chip_SUM2028,
    ChipType_1.ChipTypeEnum.Chip_MY9269,
    ChipType_1.ChipTypeEnum.Chip_MBI5045,
    ChipType_1.ChipTypeEnum.Chip_SUM2032,
    ChipType_1.ChipTypeEnum.Chip_SUM2033,
    ChipType_1.ChipTypeEnum.Chip_SUM2131,
    ChipType_1.ChipTypeEnum.Chip_SUM2035,
    ChipType_1.ChipTypeEnum.Chip_SUM2135,
    ChipType_1.ChipTypeEnum.Chip_MBI5152,
    ChipType_1.ChipTypeEnum.Chip_MY9221,
    ChipType_1.ChipTypeEnum.Chip_MY9231,
    ChipType_1.ChipTypeEnum.Chip_TLS3001,
    ChipType_1.ChipTypeEnum.Chip_GW6205,
    ChipType_1.ChipTypeEnum.Chip_UCS1912,
    ChipType_1.ChipTypeEnum.Chip_LD1512,
    ChipType_1.ChipTypeEnum.Chip_GW6202B,
    ChipType_1.ChipTypeEnum.Chip_MBI6030,
    ChipType_1.ChipTypeEnum.Chip_MBI6023,
    ChipType_1.ChipTypeEnum.Chip_SM16159,
    ChipType_1.ChipTypeEnum.Chip_MBI5151,
    ChipType_1.ChipTypeEnum.Chip_MBI5155,
    ChipType_1.ChipTypeEnum.Chip_MBI5252,
    ChipType_1.ChipTypeEnum.Chip_MY9373,
    ChipType_1.ChipTypeEnum.Chip_MBI5353,
    ChipType_1.ChipTypeEnum.Chip_MBI5354,
    ChipType_1.ChipTypeEnum.Chip_MBI5253,
    ChipType_1.ChipTypeEnum.Chip_MBI5359,
    ChipType_1.ChipTypeEnum.Chip_MBI5253B,
    ChipType_1.ChipTypeEnum.Chip_ICN2053,
    ChipType_1.ChipTypeEnum.Chip_ICND2055,
    ChipType_1.ChipTypeEnum.Chip_ICN2065,
    ChipType_1.ChipTypeEnum.Chip_ICN2050,
    ChipType_1.ChipTypeEnum.Chip_MBI5051B,
    ChipType_1.ChipTypeEnum.Chip_MY9348,
    ChipType_1.ChipTypeEnum.Chip_MY9748,
    ChipType_1.ChipTypeEnum.Chip_ICND2200,
    ChipType_1.ChipTypeEnum.Chip_ICND2163,
    ChipType_1.ChipTypeEnum.Chip_VOD5153,
    ChipType_1.ChipTypeEnum.Chip_SM16169S,
    ChipType_1.ChipTypeEnum.Chip_MBI5153,
    ChipType_1.ChipTypeEnum.Chip_SM16359,
    ChipType_1.ChipTypeEnum.Chip_UCS512C,
].includes(ChipType);
exports.IsPWMChip = IsPWMChip;
const IsChangedOETableChip = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MBI5166,
    ChipType_1.ChipTypeEnum.Chip_MBI5266,
    ChipType_1.ChipTypeEnum.Chip_MBI5124A,
    ChipType_1.ChipTypeEnum.Chip_MBI5125A,
    ChipType_1.ChipTypeEnum.Chip_LS9929,
    ChipType_1.ChipTypeEnum.Chip_LS9935,
    ChipType_1.ChipTypeEnum.Chip_LS9918,
    ChipType_1.ChipTypeEnum.Chip_LS9926,
    ChipType_1.ChipTypeEnum.Chip_LS9930,
].includes(ChipType);
exports.IsChangedOETableChip = IsChangedOETableChip;
const IsSupport = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_CommonBase,
    ChipType_1.ChipTypeEnum.Chip_ICN2038S_Common,
    ChipType_1.ChipTypeEnum.Chip_ICN2027,
    ChipType_1.ChipTypeEnum.Chip_MBI5036,
    ChipType_1.ChipTypeEnum.Chip_MBI5038,
    ChipType_1.ChipTypeEnum.Chip_TLC5929,
    ChipType_1.ChipTypeEnum.Chip_RT5924,
    ChipType_1.ChipTypeEnum.Chip_MY9163,
    ChipType_1.ChipTypeEnum.Chip_MBI5042,
    ChipType_1.ChipTypeEnum.Chip_SM16136,
    ChipType_1.ChipTypeEnum.Chip_MY9262,
    ChipType_1.ChipTypeEnum.Chip_MY9263,
    ChipType_1.ChipTypeEnum.Chip_MBI5050,
    ChipType_1.ChipTypeEnum.Chip_DM13H,
    ChipType_1.ChipTypeEnum.Chip_MBI5030,
    ChipType_1.ChipTypeEnum.Chip_P2510,
    ChipType_1.ChipTypeEnum.Chip_P2518,
    ChipType_1.ChipTypeEnum.Chip_SUM2017,
    ChipType_1.ChipTypeEnum.Chip_SUM2017T,
    ChipType_1.ChipTypeEnum.Chip_SUM2117,
    ChipType_1.ChipTypeEnum.Chip_SUM2017TD,
    ChipType_1.ChipTypeEnum.Chip_SM16027,
    ChipType_1.ChipTypeEnum.Chip_SUM2030,
    ChipType_1.ChipTypeEnum.Chip_SUM2130,
    ChipType_1.ChipTypeEnum.Chip_SUM2018,
    ChipType_1.ChipTypeEnum.Chip_MBI5034,
    ChipType_1.ChipTypeEnum.Chip_MBI5122,
    ChipType_1.ChipTypeEnum.Chip_MBI5040,
    ChipType_1.ChipTypeEnum.Chip_MBI5167,
    ChipType_1.ChipTypeEnum.Chip_IT1505,
    ChipType_1.ChipTypeEnum.Chip_TLC5948A,
    ChipType_1.ChipTypeEnum.Chip_UCS9812,
    ChipType_1.ChipTypeEnum.Chip_SM16188B,
    ChipType_1.ChipTypeEnum.Chip_MY9268,
    ChipType_1.ChipTypeEnum.Chip_SCL8060,
    ChipType_1.ChipTypeEnum.Chip_MBI5224,
    ChipType_1.ChipTypeEnum.Chip_MBI5051,
    ChipType_1.ChipTypeEnum.Chip_MBI5052,
    ChipType_1.ChipTypeEnum.Chip_SM16158,
    ChipType_1.ChipTypeEnum.Chip_ISA2020,
    ChipType_1.ChipTypeEnum.Chip_MBI5053,
    ChipType_1.ChipTypeEnum.Chip_TLC5958,
    ChipType_1.ChipTypeEnum.Chip_TLC59581,
    ChipType_1.ChipTypeEnum.Chip_MY9266,
    ChipType_1.ChipTypeEnum.Chip_MY9366,
    ChipType_1.ChipTypeEnum.Chip_MBI5042B,
    ChipType_1.ChipTypeEnum.Chip_MBI5041B,
    ChipType_1.ChipTypeEnum.Chip_MBI5043,
    ChipType_1.ChipTypeEnum.Chip_MBI5041Q,
    ChipType_1.ChipTypeEnum.Chip_SUM2028,
    ChipType_1.ChipTypeEnum.Chip_MBIA043,
    ChipType_1.ChipTypeEnum.Chip_MY9269,
    ChipType_1.ChipTypeEnum.Chip_MBI5045,
    ChipType_1.ChipTypeEnum.Chip_MBI5039,
    ChipType_1.ChipTypeEnum.Chip_MBI5037,
    ChipType_1.ChipTypeEnum.Chip_SUM2032,
    ChipType_1.ChipTypeEnum.Chip_SUM2033,
    ChipType_1.ChipTypeEnum.Chip_SUM2131,
    ChipType_1.ChipTypeEnum.Chip_SUM2035,
    ChipType_1.ChipTypeEnum.Chip_SUM2135,
    ChipType_1.ChipTypeEnum.Chip_MBI5152,
    ChipType_1.ChipTypeEnum.Chip_MY9221,
    ChipType_1.ChipTypeEnum.Chip_MY9231,
    ChipType_1.ChipTypeEnum.Chip_TLS3001,
    ChipType_1.ChipTypeEnum.Chip_GW6205,
    ChipType_1.ChipTypeEnum.Chip_UCS512C,
    ChipType_1.ChipTypeEnum.Chip_UCS1912,
    ChipType_1.ChipTypeEnum.Chip_LD1512,
    ChipType_1.ChipTypeEnum.Chip_GW6202B,
    ChipType_1.ChipTypeEnum.Chip_MBI5124,
    ChipType_1.ChipTypeEnum.Chip_MBI5125,
    ChipType_1.ChipTypeEnum.Chip_SM16017S,
    ChipType_1.ChipTypeEnum.Chip_ICN2038,
    ChipType_1.ChipTypeEnum.Chip_ICN2038A,
    ChipType_1.ChipTypeEnum.Chip_MY9868,
    ChipType_1.ChipTypeEnum.Chip_MY9862,
    ChipType_1.ChipTypeEnum.Chip_SM16207S,
    ChipType_1.ChipTypeEnum.Chip_SM16237,
    ChipType_1.ChipTypeEnum.Chip_SM16227,
    ChipType_1.ChipTypeEnum.Chip_SM16237P,
    ChipType_1.ChipTypeEnum.Chip_SM16227P,
    ChipType_1.ChipTypeEnum.Chip_ICN2038S,
    ChipType_1.ChipTypeEnum.Chip_ICND2046,
    ChipType_1.ChipTypeEnum.Chip_FM6126,
    ChipType_1.ChipTypeEnum.Chip_FM6142,
    ChipType_1.ChipTypeEnum.Chip_MY9866,
    ChipType_1.ChipTypeEnum.Chip_MY9868A,
    ChipType_1.ChipTypeEnum.Chip_FM6182,
    ChipType_1.ChipTypeEnum.Chip_MY9862A,
    ChipType_1.ChipTypeEnum.Chip_MBI5166,
    ChipType_1.ChipTypeEnum.Chip_MBI5125A,
    ChipType_1.ChipTypeEnum.Chip_MBI5124A,
    ChipType_1.ChipTypeEnum.Chip_MBI5266,
    ChipType_1.ChipTypeEnum.Chip_MBI6030,
    ChipType_1.ChipTypeEnum.Chip_MBI6023,
    ChipType_1.ChipTypeEnum.Chip_SM16159,
    ChipType_1.ChipTypeEnum.Chip_SM16169S,
    ChipType_1.ChipTypeEnum.Chip_MBI5151,
    ChipType_1.ChipTypeEnum.Chip_MBI5051B,
    ChipType_1.ChipTypeEnum.Chip_MBI5155,
    ChipType_1.ChipTypeEnum.Chip_MBI5252,
    ChipType_1.ChipTypeEnum.Chip_MY9373,
    ChipType_1.ChipTypeEnum.Chip_MBI5153,
    ChipType_1.ChipTypeEnum.Chip_MBI5253,
    ChipType_1.ChipTypeEnum.Chip_MBI5359,
    ChipType_1.ChipTypeEnum.Chip_MY9348,
    ChipType_1.ChipTypeEnum.Chip_MY9748,
    ChipType_1.ChipTypeEnum.Chip_MBI5353,
    ChipType_1.ChipTypeEnum.Chip_MBI5354,
    ChipType_1.ChipTypeEnum.Chip_MBI5253B,
    ChipType_1.ChipTypeEnum.Chip_ICN2050,
    ChipType_1.ChipTypeEnum.Chip_ICND2055,
    ChipType_1.ChipTypeEnum.Chip_ICN2065,
    ChipType_1.ChipTypeEnum.Chip_ICN2053,
    ChipType_1.ChipTypeEnum.Chip_LS9960,
    ChipType_1.ChipTypeEnum.Chip_LS9929,
    ChipType_1.ChipTypeEnum.Chip_LS9935,
    ChipType_1.ChipTypeEnum.Chip_LS9926,
    ChipType_1.ChipTypeEnum.Chip_LS9930,
    ChipType_1.ChipTypeEnum.Chip_GW6808,
    ChipType_1.ChipTypeEnum.Chip_LS9918,
    ChipType_1.ChipTypeEnum.Chip_MBI5124New,
    ChipType_1.ChipTypeEnum.Chip_ICND2163,
    ChipType_1.ChipTypeEnum.Chip_VOD5153,
    ChipType_1.ChipTypeEnum.Chip_MY9866_Common,
    ChipType_1.ChipTypeEnum.Chip_SM16359,
    ChipType_1.ChipTypeEnum.Chip_SM16259,
    ChipType_1.ChipTypeEnum.Chip_SCL8080,
    ChipType_1.ChipTypeEnum.Chip_ICND2100,
    ChipType_1.ChipTypeEnum.Chip_MBI5759,
    ChipType_1.ChipTypeEnum.Chip_XM11920G,
    ChipType_1.ChipTypeEnum.Chip_XM11201G,
    ChipType_1.ChipTypeEnum.Chip_XM11202G,
    ChipType_1.ChipTypeEnum.Chip_MBI5251,
    ChipType_1.ChipTypeEnum.Chip_MBI5264,
    ChipType_1.ChipTypeEnum.Chip_MBI5254,
    ChipType_1.ChipTypeEnum.Chip_MBI5850,
    ChipType_1.ChipTypeEnum.Chip_MBI5754,
    ChipType_1.ChipTypeEnum.Chip_CS2033,
    ChipType_1.ChipTypeEnum.Chip_SCL8081,
    ChipType_1.ChipTypeEnum.Chip_SM16369,
    ChipType_1.ChipTypeEnum.Chip_DM413,
    ChipType_1.ChipTypeEnum.Chip_YY6018,
    ChipType_1.ChipTypeEnum.Chip_AXS6018,
    ChipType_1.ChipTypeEnum.Chip_RT5965,
    ChipType_1.ChipTypeEnum.Chip_FM6363,
    ChipType_1.ChipTypeEnum.Chip_FM6353,
    ChipType_1.ChipTypeEnum.Chip_FM6356,
    ChipType_1.ChipTypeEnum.Chip_FM6555,
    ChipType_1.ChipTypeEnum.Chip_FM6565,
    ChipType_1.ChipTypeEnum.Chip_FM6047,
    ChipType_1.ChipTypeEnum.Chip_LS9928,
    ChipType_1.ChipTypeEnum.Chip_LS9935B,
    ChipType_1.ChipTypeEnum.Chip_HS3257,
    ChipType_1.ChipTypeEnum.Chip_HX8055,
    ChipType_1.ChipTypeEnum.Chip_HX8864,
    ChipType_1.ChipTypeEnum.Chip_MY9758,
    ChipType_1.ChipTypeEnum.Chip_ICND2076,
    ChipType_1.ChipTypeEnum.Chip_ICND2159,
    ChipType_1.ChipTypeEnum.Chip_ICND2210,
    ChipType_1.ChipTypeEnum.Chip_CS2066,
    ChipType_1.ChipTypeEnum.Chip_CNS7253,
    ChipType_1.ChipTypeEnum.Chip_CS2017,
    ChipType_1.ChipTypeEnum.Chip_CS2017S,
    ChipType_1.ChipTypeEnum.Chip_FM6127,
    ChipType_1.ChipTypeEnum.Chip_CNS7153,
    ChipType_1.ChipTypeEnum.Chip_ICND2112,
    ChipType_1.ChipTypeEnum.Chip_UCS5603,
    ChipType_1.ChipTypeEnum.Chip_SM16017DS,
    ChipType_1.ChipTypeEnum.Chip_HX8055,
    ChipType_1.ChipTypeEnum.Chip_DP3264,
].includes(ChipType);
exports.IsSupport = IsSupport;
const IsSupportCurrentChipTypeEnum = (ChipType) => ChipType !== ChipType_1.ChipTypeEnum.Chip_SUM6082 && ChipType !== ChipType_1.ChipTypeEnum.Chip_NVS6867;
exports.IsSupportCurrentChipTypeEnum = IsSupportCurrentChipTypeEnum;
const IsPWMAndHavePartNum = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_SUM2030,
    ChipType_1.ChipTypeEnum.Chip_MY9268,
    ChipType_1.ChipTypeEnum.Chip_SUM2130,
    ChipType_1.ChipTypeEnum.Chip_MBI5030,
    ChipType_1.ChipTypeEnum.Chip_MBI5042,
    ChipType_1.ChipTypeEnum.Chip_MY9263,
    ChipType_1.ChipTypeEnum.Chip_SM16136,
    ChipType_1.ChipTypeEnum.Chip_MY9262,
    ChipType_1.ChipTypeEnum.Chip_MBI5051,
    ChipType_1.ChipTypeEnum.Chip_MBI5052,
    ChipType_1.ChipTypeEnum.Chip_SM16158,
    ChipType_1.ChipTypeEnum.Chip_MBI5053,
    ChipType_1.ChipTypeEnum.Chip_MY9266,
    ChipType_1.ChipTypeEnum.Chip_MY9366,
    ChipType_1.ChipTypeEnum.Chip_MBI5042B,
    ChipType_1.ChipTypeEnum.Chip_MBI5041B,
    ChipType_1.ChipTypeEnum.Chip_MBI5043,
    ChipType_1.ChipTypeEnum.Chip_SUM2028,
    ChipType_1.ChipTypeEnum.Chip_MBIA043,
    ChipType_1.ChipTypeEnum.Chip_MY9269,
    ChipType_1.ChipTypeEnum.Chip_MBI5045,
    ChipType_1.ChipTypeEnum.Chip_SUM2032,
    ChipType_1.ChipTypeEnum.Chip_SUM2033,
    ChipType_1.ChipTypeEnum.Chip_SUM2131,
    ChipType_1.ChipTypeEnum.Chip_SUM2035,
    ChipType_1.ChipTypeEnum.Chip_SUM2135,
    ChipType_1.ChipTypeEnum.Chip_SM16159,
    ChipType_1.ChipTypeEnum.Chip_SM16169S,
    ChipType_1.ChipTypeEnum.Chip_MBI5152,
    ChipType_1.ChipTypeEnum.Chip_MBI5151,
    ChipType_1.ChipTypeEnum.Chip_MBI5155,
    ChipType_1.ChipTypeEnum.Chip_MBI5051B,
    ChipType_1.ChipTypeEnum.Chip_MBI5252,
    ChipType_1.ChipTypeEnum.Chip_MY9373,
    ChipType_1.ChipTypeEnum.Chip_MBI5353,
    ChipType_1.ChipTypeEnum.Chip_MBI5354,
    ChipType_1.ChipTypeEnum.Chip_MBI5153,
    ChipType_1.ChipTypeEnum.Chip_MBI5253,
    ChipType_1.ChipTypeEnum.Chip_MBI5359,
    ChipType_1.ChipTypeEnum.Chip_MBI5253B,
    ChipType_1.ChipTypeEnum.Chip_MY9348,
    ChipType_1.ChipTypeEnum.Chip_MY9748,
    ChipType_1.ChipTypeEnum.Chip_ICN2050,
    ChipType_1.ChipTypeEnum.Chip_ICND2055,
    ChipType_1.ChipTypeEnum.Chip_ICN2065,
    ChipType_1.ChipTypeEnum.Chip_ICN2053,
    ChipType_1.ChipTypeEnum.Chip_SM16359,
].includes(ChipType);
exports.IsPWMAndHavePartNum = IsPWMAndHavePartNum;
const IsSupport5042GrayEnhanced = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MBI5042,
    ChipType_1.ChipTypeEnum.Chip_MBI5030,
    ChipType_1.ChipTypeEnum.Chip_SM16136,
    ChipType_1.ChipTypeEnum.Chip_MY9263,
    ChipType_1.ChipTypeEnum.Chip_MBI5042B,
    ChipType_1.ChipTypeEnum.Chip_MBI5043,
    ChipType_1.ChipTypeEnum.Chip_MBI5041B,
    ChipType_1.ChipTypeEnum.Chip_SUM2028,
    ChipType_1.ChipTypeEnum.Chip_MBIA043,
    ChipType_1.ChipTypeEnum.Chip_MBI5041Q,
    ChipType_1.ChipTypeEnum.Chip_MBI5045,
].includes(ChipType);
exports.IsSupport5042GrayEnhanced = IsSupport5042GrayEnhanced;
const IsSupportMBI5041BNewTiming = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MBI5042B,
    ChipType_1.ChipTypeEnum.Chip_SUM2028,
    ChipType_1.ChipTypeEnum.Chip_MBIA043,
    ChipType_1.ChipTypeEnum.Chip_MBI5043,
    ChipType_1.ChipTypeEnum.Chip_MBI5041B,
    ChipType_1.ChipTypeEnum.Chip_MBI5041Q,
    ChipType_1.ChipTypeEnum.Chip_MBI5045,
    ChipType_1.ChipTypeEnum.Chip_MBI5042,
].includes(ChipType);
exports.IsSupportMBI5041BNewTiming = IsSupportMBI5041BNewTiming;
const IsSupportDriverInterDeghostEnable = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MBI5051,
    ChipType_1.ChipTypeEnum.Chip_MBI5052,
    ChipType_1.ChipTypeEnum.Chip_SM16158,
    ChipType_1.ChipTypeEnum.Chip_SM16159,
    ChipType_1.ChipTypeEnum.Chip_SM16169S,
    ChipType_1.ChipTypeEnum.Chip_MBI5053,
    ChipType_1.ChipTypeEnum.Chip_MBI5152,
    ChipType_1.ChipTypeEnum.Chip_MBI5151,
    ChipType_1.ChipTypeEnum.Chip_MBI5252,
    ChipType_1.ChipTypeEnum.Chip_MBI5051B,
    ChipType_1.ChipTypeEnum.Chip_MBI5155,
    ChipType_1.ChipTypeEnum.Chip_MY9373,
    ChipType_1.ChipTypeEnum.Chip_MBI5353,
    ChipType_1.ChipTypeEnum.Chip_MBI5354,
    ChipType_1.ChipTypeEnum.Chip_MY9348,
    ChipType_1.ChipTypeEnum.Chip_MY9748,
    ChipType_1.ChipTypeEnum.Chip_MBI5359,
    ChipType_1.ChipTypeEnum.Chip_MBI5253,
].includes(ChipType);
exports.IsSupportDriverInterDeghostEnable = IsSupportDriverInterDeghostEnable;
const IsSupportNewOERealizeWay = (ChipType) => ChipType === ChipType_1.ChipTypeEnum.Chip_MBI5124New;
exports.IsSupportNewOERealizeWay = IsSupportNewOERealizeWay;
const IsSupportPhotographMode = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MY9868A,
    ChipType_1.ChipTypeEnum.Chip_MY9862A,
    ChipType_1.ChipTypeEnum.Chip_SM16207S,
    ChipType_1.ChipTypeEnum.Chip_SM16237,
    ChipType_1.ChipTypeEnum.Chip_SM16227,
    ChipType_1.ChipTypeEnum.Chip_ICN2038A,
    ChipType_1.ChipTypeEnum.Chip_ICN2038S,
    ChipType_1.ChipTypeEnum.Chip_ICND2046,
    ChipType_1.ChipTypeEnum.Chip_SM16237DS,
].includes(ChipType);
exports.IsSupportPhotographMode = IsSupportPhotographMode;
const IsDoubleLatChip = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MY9868A,
    ChipType_1.ChipTypeEnum.Chip_MY9862A,
    ChipType_1.ChipTypeEnum.Chip_SM16207S,
    ChipType_1.ChipTypeEnum.Chip_SM16237,
    ChipType_1.ChipTypeEnum.Chip_SM16227,
    ChipType_1.ChipTypeEnum.Chip_ICN2038A,
    ChipType_1.ChipTypeEnum.Chip_ICN2038S,
    ChipType_1.ChipTypeEnum.Chip_ICND2046,
    ChipType_1.ChipTypeEnum.Chip_SM16237DS,
].includes(ChipType);
exports.IsDoubleLatChip = IsDoubleLatChip;
const IsConfigRegisterConcernWithScanType = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MBI5050,
    ChipType_1.ChipTypeEnum.Chip_SUM2030,
    ChipType_1.ChipTypeEnum.Chip_SUM2130,
    ChipType_1.ChipTypeEnum.Chip_MY9268,
    ChipType_1.ChipTypeEnum.Chip_SCL8060,
    ChipType_1.ChipTypeEnum.Chip_MY9266,
    ChipType_1.ChipTypeEnum.Chip_MY9366,
    ChipType_1.ChipTypeEnum.Chip_MY9269,
    ChipType_1.ChipTypeEnum.Chip_MBI5051,
    ChipType_1.ChipTypeEnum.Chip_MBI5052,
    ChipType_1.ChipTypeEnum.Chip_SM16158,
    ChipType_1.ChipTypeEnum.Chip_SM16159,
    ChipType_1.ChipTypeEnum.Chip_SM16169S,
    ChipType_1.ChipTypeEnum.Chip_MBI5053,
    ChipType_1.ChipTypeEnum.Chip_TLC5958,
    ChipType_1.ChipTypeEnum.Chip_TLC59581,
    ChipType_1.ChipTypeEnum.Chip_SUM2032,
    ChipType_1.ChipTypeEnum.Chip_SUM2033,
    ChipType_1.ChipTypeEnum.Chip_SUM2131,
    ChipType_1.ChipTypeEnum.Chip_SUM2035,
    ChipType_1.ChipTypeEnum.Chip_SUM2135,
    ChipType_1.ChipTypeEnum.Chip_MBI5152,
    ChipType_1.ChipTypeEnum.Chip_MBI5151,
    ChipType_1.ChipTypeEnum.Chip_MBI5051B,
    ChipType_1.ChipTypeEnum.Chip_MBI5155,
    ChipType_1.ChipTypeEnum.Chip_MBI5252,
    ChipType_1.ChipTypeEnum.Chip_MY9373,
    ChipType_1.ChipTypeEnum.Chip_MBI5353,
    ChipType_1.ChipTypeEnum.Chip_MBI5354,
    ChipType_1.ChipTypeEnum.Chip_MBI5153,
    ChipType_1.ChipTypeEnum.Chip_MBI5253,
    ChipType_1.ChipTypeEnum.Chip_MBI5359,
    ChipType_1.ChipTypeEnum.Chip_MBI5253B,
    ChipType_1.ChipTypeEnum.Chip_MY9348,
    ChipType_1.ChipTypeEnum.Chip_MY9748,
    ChipType_1.ChipTypeEnum.Chip_ICN2050,
    ChipType_1.ChipTypeEnum.Chip_ICND2055,
    ChipType_1.ChipTypeEnum.Chip_ICN2065,
    ChipType_1.ChipTypeEnum.Chip_ICND2163,
    ChipType_1.ChipTypeEnum.Chip_VOD5153,
    ChipType_1.ChipTypeEnum.Chip_ICN2053,
    ChipType_1.ChipTypeEnum.Chip_SM16359,
].includes(ChipType);
exports.IsConfigRegisterConcernWithScanType = IsConfigRegisterConcernWithScanType;
const IsConfigRegisterConcernWithShiftUnitNum = (ChipType) => [ChipType_1.ChipTypeEnum.Chip_MBI6030, ChipType_1.ChipTypeEnum.Chip_MBI6023].includes(ChipType);
exports.IsConfigRegisterConcernWithShiftUnitNum = IsConfigRegisterConcernWithShiftUnitNum;
const GetDetectPointType = (ChipType) => {
    switch (ChipType) {
        case ChipType_1.ChipTypeEnum.Chip_MBI5030:
        case ChipType_1.ChipTypeEnum.Chip_MBI5034:
        case ChipType_1.ChipTypeEnum.Chip_MBI5038:
            return DetectPointType_1.DetectPointTypeEnum.OpenCircuit;
        case ChipType_1.ChipTypeEnum.Chip_MBI5036:
        case ChipType_1.ChipTypeEnum.Chip_MBI5039:
        case ChipType_1.ChipTypeEnum.Chip_RT5924:
            return DetectPointType_1.DetectPointTypeEnum.BothNoDivision;
        case ChipType_1.ChipTypeEnum.Chip_MBI5040:
        case ChipType_1.ChipTypeEnum.Chip_MY9163:
        case ChipType_1.ChipTypeEnum.Chip_TLC5929:
        case ChipType_1.ChipTypeEnum.Chip_MY9862:
        case ChipType_1.ChipTypeEnum.Chip_MBI5353:
        case ChipType_1.ChipTypeEnum.Chip_MY9263:
        case ChipType_1.ChipTypeEnum.Chip_SM16207S:
        case ChipType_1.ChipTypeEnum.Chip_SM16237:
        case ChipType_1.ChipTypeEnum.Chip_SM16227:
        case ChipType_1.ChipTypeEnum.Chip_MBI5354:
        case ChipType_1.ChipTypeEnum.Chip_MBI5359:
            return DetectPointType_1.DetectPointTypeEnum.BothDivision;
        case ChipType_1.ChipTypeEnum.Chip_DM13H:
        case ChipType_1.ChipTypeEnum.Chip_MBI5037:
            return DetectPointType_1.DetectPointTypeEnum.BothDivision;
        case ChipType_1.ChipTypeEnum.Chip_MBI5051:
        case ChipType_1.ChipTypeEnum.Chip_MBI5052:
        case ChipType_1.ChipTypeEnum.Chip_MBI5053:
        case ChipType_1.ChipTypeEnum.Chip_MBI5152:
        case ChipType_1.ChipTypeEnum.Chip_MBI5153:
        case ChipType_1.ChipTypeEnum.Chip_TLC5958:
        case ChipType_1.ChipTypeEnum.Chip_MBI5266:
        case ChipType_1.ChipTypeEnum.Chip_MBI5155:
        case ChipType_1.ChipTypeEnum.Chip_MBI5252:
        case ChipType_1.ChipTypeEnum.Chip_TLC59581:
        case ChipType_1.ChipTypeEnum.Chip_MBI5051B:
        case ChipType_1.ChipTypeEnum.Chip_MBI5253:
        case ChipType_1.ChipTypeEnum.Chip_ICN2053:
        case ChipType_1.ChipTypeEnum.Chip_MY9373:
        case ChipType_1.ChipTypeEnum.Chip_ICND2055:
        case ChipType_1.ChipTypeEnum.Chip_ICN2050:
        case ChipType_1.ChipTypeEnum.Chip_ICN2065:
        case ChipType_1.ChipTypeEnum.Chip_MBI5253B:
            return DetectPointType_1.DetectPointTypeEnum.OpenCircuit;
        case ChipType_1.ChipTypeEnum.Chip_MBI5045:
            return DetectPointType_1.DetectPointTypeEnum.OpenCircuit;
        case ChipType_1.ChipTypeEnum.Chip_IT1505:
            return DetectPointType_1.DetectPointTypeEnum.BothDivision;
        default:
            return DetectPointType_1.DetectPointTypeEnum.None;
    }
};
exports.GetDetectPointType = GetDetectPointType;
const IsNoCurrentGainButConfig = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_SUM2017,
    ChipType_1.ChipTypeEnum.Chip_SM16027,
    ChipType_1.ChipTypeEnum.Chip_MBI5037,
    ChipType_1.ChipTypeEnum.Chip_MY9221,
    ChipType_1.ChipTypeEnum.Chip_TLC5948A,
    ChipType_1.ChipTypeEnum.Chip_MY9231,
    ChipType_1.ChipTypeEnum.Chip_MBI6030,
    ChipType_1.ChipTypeEnum.Chip_MBI6023,
    ChipType_1.ChipTypeEnum.Chip_RT5924,
    ChipType_1.ChipTypeEnum.Chip_TLC5958,
    ChipType_1.ChipTypeEnum.Chip_TLC59581,
    ChipType_1.ChipTypeEnum.Chip_MBI5166,
    ChipType_1.ChipTypeEnum.Chip_MBI5125A,
    ChipType_1.ChipTypeEnum.Chip_MBI5124A,
    ChipType_1.ChipTypeEnum.Chip_MBI5266,
    ChipType_1.ChipTypeEnum.Chip_MBI5122,
    ChipType_1.ChipTypeEnum.Chip_MBI5125,
    ChipType_1.ChipTypeEnum.Chip_MBI5124,
    ChipType_1.ChipTypeEnum.Chip_SM16017S,
    ChipType_1.ChipTypeEnum.Chip_SM16207S,
    ChipType_1.ChipTypeEnum.Chip_MBI5124New,
].includes(ChipType);
exports.IsNoCurrentGainButConfig = IsNoCurrentGainButConfig;
const DetectPointIsConfigOtherReg = (ChipType) => [
    ChipType_1.ChipTypeEnum.Chip_MBI5153,
    ChipType_1.ChipTypeEnum.Chip_ICN2053,
    ChipType_1.ChipTypeEnum.Chip_ICND2055,
    ChipType_1.ChipTypeEnum.Chip_ICN2065,
    ChipType_1.ChipTypeEnum.Chip_ICN2050,
    ChipType_1.ChipTypeEnum.Chip_MBI5252,
].includes(ChipType);
exports.DetectPointIsConfigOtherReg = DetectPointIsConfigOtherReg;
const CanManualAdjustRefreshRate = (chipType, scanType) => {
    switch (chipType) {
        case ChipType_1.ChipTypeEnum.Chip_MBI5030:
        case ChipType_1.ChipTypeEnum.Chip_MBI5040:
        case ChipType_1.ChipTypeEnum.Chip_MBI5042:
        case ChipType_1.ChipTypeEnum.Chip_RFT3630:
        case ChipType_1.ChipTypeEnum.Chip_MBI5042B:
        case ChipType_1.ChipTypeEnum.Chip_MBI5045:
        case ChipType_1.ChipTypeEnum.Chip_MY9262:
        case ChipType_1.ChipTypeEnum.Chip_SM16136:
        case ChipType_1.ChipTypeEnum.Chip_SUM2028:
        case ChipType_1.ChipTypeEnum.Chip_MBI5043:
        case ChipType_1.ChipTypeEnum.Chip_MBIA043:
        case ChipType_1.ChipTypeEnum.Chip_MBI5041B:
        case ChipType_1.ChipTypeEnum.Chip_MBI5041Q:
        case ChipType_1.ChipTypeEnum.Chip_MY9263:
        case ChipType_1.ChipTypeEnum.Chip_UCS9812:
        case ChipType_1.ChipTypeEnum.Chip_TLC5948A:
        case ChipType_1.ChipTypeEnum.Chip_SM16188B:
        case ChipType_1.ChipTypeEnum.Chip_IT1505:
            return scanType !== ScanType_1.ScanTypeEnum.Scan_static;
        case ChipType_1.ChipTypeEnum.Chip_MBI5050:
        case ChipType_1.ChipTypeEnum.Chip_SCL8060:
        case ChipType_1.ChipTypeEnum.Chip_SUM2030:
        case ChipType_1.ChipTypeEnum.Chip_MY9268:
        case ChipType_1.ChipTypeEnum.Chip_MBI5051:
        case ChipType_1.ChipTypeEnum.Chip_MBI5052:
        case ChipType_1.ChipTypeEnum.Chip_MBI5053:
        case ChipType_1.ChipTypeEnum.Chip_MY9266:
        case ChipType_1.ChipTypeEnum.Chip_MY9269:
        case ChipType_1.ChipTypeEnum.Chip_SUM2032:
        case ChipType_1.ChipTypeEnum.Chip_MBI5152:
        case ChipType_1.ChipTypeEnum.Chip_MY9221:
        case ChipType_1.ChipTypeEnum.Chip_MBI5151:
        case ChipType_1.ChipTypeEnum.Chip_MBI5153:
        case ChipType_1.ChipTypeEnum.Chip_MBI6030:
        case ChipType_1.ChipTypeEnum.Chip_TLC5958:
        case ChipType_1.ChipTypeEnum.Chip_SM16158:
        case ChipType_1.ChipTypeEnum.Chip_MBI5155:
        case ChipType_1.ChipTypeEnum.Chip_MBI6023:
        case ChipType_1.ChipTypeEnum.Chip_MY9366:
        case ChipType_1.ChipTypeEnum.Chip_MBI5252:
        case ChipType_1.ChipTypeEnum.Chip_SM16159:
        case ChipType_1.ChipTypeEnum.Chip_TLS3001:
        case ChipType_1.ChipTypeEnum.Chip_TLC59581:
        case ChipType_1.ChipTypeEnum.Chip_MBI5353:
        case ChipType_1.ChipTypeEnum.Chip_GW6205:
        case ChipType_1.ChipTypeEnum.Chip_GW6202B:
        case ChipType_1.ChipTypeEnum.Chip_SUM2033:
        case ChipType_1.ChipTypeEnum.Chip_MY9231:
        case ChipType_1.ChipTypeEnum.Chip_SUM2130:
        case ChipType_1.ChipTypeEnum.Chip_SUM2131:
        case ChipType_1.ChipTypeEnum.Chip_MBI5051B:
        case ChipType_1.ChipTypeEnum.Chip_MBI5253:
        case ChipType_1.ChipTypeEnum.Chip_ICN2053:
        case ChipType_1.ChipTypeEnum.Chip_MY9373:
        case ChipType_1.ChipTypeEnum.Chip_MY9348:
        case ChipType_1.ChipTypeEnum.Chip_SUM2035:
        case ChipType_1.ChipTypeEnum.Chip_SUM2135:
        case ChipType_1.ChipTypeEnum.Chip_ICND2055:
        case ChipType_1.ChipTypeEnum.Chip_ICN2050:
        case ChipType_1.ChipTypeEnum.Chip_MBI5354:
        case ChipType_1.ChipTypeEnum.Chip_MBI5359:
        case ChipType_1.ChipTypeEnum.Chip_ICN2065:
        case ChipType_1.ChipTypeEnum.Chip_MBI5253B:
        case ChipType_1.ChipTypeEnum.Chip_MY9748:
        case ChipType_1.ChipTypeEnum.Chip_LD1512:
        case ChipType_1.ChipTypeEnum.Chip_UCS1912:
        case ChipType_1.ChipTypeEnum.Chip_UCS512C:
        case ChipType_1.ChipTypeEnum.Chip_ICND2163:
        case ChipType_1.ChipTypeEnum.Chip_SM16169S:
        case ChipType_1.ChipTypeEnum.Chip_VOD5153:
            return false;
        default:
            return true;
    }
};
exports.CanManualAdjustRefreshRate = CanManualAdjustRefreshRate;
