// @novastar/screen/build/main/FileInfo.js
"use strict";
// noinspection SpellCheckingInspection
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileInfoObjectList = exports.FileInfoObject = exports.FILE_COMPRESS_AREA_START_ADDR = exports.ScreenFileVersion = exports.ScreenFileType = void 0;
const t = __importStar(require("io-ts"));
exports.ScreenFileType = 'ScreenDataType';
exports.ScreenFileVersion = '1001';
exports.FILE_COMPRESS_AREA_START_ADDR = 83894272;
/**
 * Nova.LCT.GigabitSystem.CommonInfoAccessor.dll
 *
 * Nova.LCT.GigabitSystem.CommonInfoAccessor::FileInfoObject
 */
exports.FileInfoObject = t.type({
    FileType: t.string,
    Addr: t.number,
    SrcLength: t.number,
    DestLength: t.number,
    CheckSum: t.number,
    Version: t.string,
    DecompressProps: t.string,
});
exports.FileInfoObjectList = t.type({
    SectionFormat: t.array(exports.FileInfoObject),
});
