// @novastar/screen/build/main/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCabinetPosition = exports.getScreenLocation = exports.ZoomTypeEnum = exports.StandardLEDDisplayInfo = exports.SimpleLEDDisplayInfo = exports.ScanBoardConnectTypeEnum = exports.VedioSelectModeEnum = exports.HDEnableModeEnum = exports.DviSelectModeEnum = exports.AudioControlModeEnum = exports.itFirstNotNull = exports.itAll = exports.hasProps = exports.isTopConnection = exports.isLeftConnection = exports.isHorizontalConnection = exports.isComplexScreen = exports.isStandardScreen = exports.isSimpleScreen = void 0;
const ScreenConfigurator_1 = __importDefault(require("./ScreenConfigurator"));
__exportStar(require("./HWStatus"), exports);
var common_1 = require("./common");
Object.defineProperty(exports, "isSimpleScreen", { enumerable: true, get: function () { return common_1.isSimpleScreen; } });
Object.defineProperty(exports, "isStandardScreen", { enumerable: true, get: function () { return common_1.isStandardScreen; } });
Object.defineProperty(exports, "isComplexScreen", { enumerable: true, get: function () { return common_1.isComplexScreen; } });
Object.defineProperty(exports, "isHorizontalConnection", { enumerable: true, get: function () { return common_1.isHorizontalConnection; } });
Object.defineProperty(exports, "isLeftConnection", { enumerable: true, get: function () { return common_1.isLeftConnection; } });
Object.defineProperty(exports, "isTopConnection", { enumerable: true, get: function () { return common_1.isTopConnection; } });
Object.defineProperty(exports, "hasProps", { enumerable: true, get: function () { return common_1.hasProps; } });
Object.defineProperty(exports, "itAll", { enumerable: true, get: function () { return common_1.itAll; } });
Object.defineProperty(exports, "itFirstNotNull", { enumerable: true, get: function () { return common_1.itFirstNotNull; } });
var AudioControlMode_1 = require("@novastar/native/AudioControlMode");
Object.defineProperty(exports, "AudioControlModeEnum", { enumerable: true, get: function () { return AudioControlMode_1.AudioControlModeEnum; } });
var DviSelectMode_1 = require("@novastar/native/DviSelectMode");
Object.defineProperty(exports, "DviSelectModeEnum", { enumerable: true, get: function () { return DviSelectMode_1.DviSelectModeEnum; } });
var HDEnableMode_1 = require("@novastar/native/HDEnableMode");
Object.defineProperty(exports, "HDEnableModeEnum", { enumerable: true, get: function () { return HDEnableMode_1.HDEnableModeEnum; } });
var VedioSelectMode_1 = require("@novastar/native/VedioSelectMode");
Object.defineProperty(exports, "VedioSelectModeEnum", { enumerable: true, get: function () { return VedioSelectMode_1.VedioSelectModeEnum; } });
var ScanBoardConnectType_1 = require("@novastar/native/ScanBoardConnectType");
Object.defineProperty(exports, "ScanBoardConnectTypeEnum", { enumerable: true, get: function () { return ScanBoardConnectType_1.ScanBoardConnectTypeEnum; } });
var SimpleLEDDisplayInfo_1 = require("@novastar/native/SimpleLEDDisplayInfo");
Object.defineProperty(exports, "SimpleLEDDisplayInfo", { enumerable: true, get: function () { return SimpleLEDDisplayInfo_1.SimpleLEDDisplayInfo; } });
var StandardLEDDisplayInfo_1 = require("@novastar/native/StandardLEDDisplayInfo");
Object.defineProperty(exports, "StandardLEDDisplayInfo", { enumerable: true, get: function () { return StandardLEDDisplayInfo_1.StandardLEDDisplayInfo; } });
var ZoomType_1 = require("@novastar/native/ZoomType");
Object.defineProperty(exports, "ZoomTypeEnum", { enumerable: true, get: function () { return ZoomType_1.ZoomTypeEnum; } });
var getScreenLocation_1 = require("./getScreenLocation");
Object.defineProperty(exports, "getScreenLocation", { enumerable: true, get: function () { return __importDefault(getScreenLocation_1).default; } });
var getCabinetPosition_1 = require("./getCabinetPosition");
Object.defineProperty(exports, "getCabinetPosition", { enumerable: true, get: function () { return __importDefault(getCabinetPosition_1).default; } });
exports.default = ScreenConfigurator_1.default;
