// @novastar/screen/build/main/rcTest.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
/* eslint-disable no-await-in-loop */
const ChipType_1 = require("@novastar/native/ChipType");
const TestMode_1 = require("@novastar/native/TestMode");
const serial_1 = __importStar(require("@novastar/serial"));
const chalk_1 = __importDefault(require("chalk"));
require("./api");
const ScreenConfigurator_1 = __importDefault(require("./ScreenConfigurator"));
const cls = () => console.log('\u001bc');
const keypress = async () => {
    process.stdin.resume();
    process.stdin.setRawMode(true);
    return new Promise((resolve, reject) => {
        process.stdin.once('data', ([code]) => {
            process.stdin.setRawMode(false);
            process.stdin.pause();
            if (code === 3)
                reject(new Error('Ctrl+C'));
            else
                resolve(code);
        });
    });
};
keypress().then(code => console.log(code));
const menu = {
    title: 'Тестирование приемных карт',
    items: {
        find: {
            msg: 'Найти хост',
            code: 'SsЫы',
        },
        bright100: {
            msg: 'Яркость 100%',
            code: '1',
        },
        bright20: {
            msg: 'Яркость 20%',
            code: '2',
        },
        vert: {
            msg: 'Вертикальные линии',
            code: 'VvМм',
        },
        horiz: {
            msg: 'Горизонтальные линии',
            code: 'HhРр',
        },
        diag: {
            msg: 'Диагональные линии',
            code: 'DdВв',
        },
        chip: {
            msg: 'Запрос типа',
            code: 'CcСс',
        },
        quit: {
            msg: 'Выход',
            code: 'QqЙй',
        },
    },
};
let screen;
const delay = (ms) => new Promise(resolve => {
    setTimeout(resolve, ms);
});
const find = async () => {
    screen?.session.close();
    try {
        const [port] = await (0, serial_1.findSendingCards)();
        if (!port) {
            console.log(chalk_1.default.red('Устройства не найдены'));
            return;
        }
        const session = await serial_1.default.open({ path: port.path, baudRate: 115200 });
        screen = new ScreenConfigurator_1.default(session);
        await screen.reload();
        console.log(`Найден ${screen.devices[0]?.name}`);
    }
    catch (err) {
        console.error('Ошибка:', err.message);
        process.exit(1);
    }
};
const setBrightness = async (value) => {
    if (!(await screen?.WriteBrightness(value))) {
        console.log(chalk_1.default.red('Не удалось задать яркость'));
    }
    else {
        console.log(chalk_1.default.green(`Яркость ${value}%`));
    }
};
const setMode = async (mode) => {
    if (!(await screen?.WriteDisplayMode(mode))) {
        console.log(chalk_1.default.red('Не удалось сменить режим'));
    }
    else {
        console.log(chalk_1.default.green(TestMode_1.TestModeEnum[mode]));
    }
};
const readChipType = async () => {
    if (screen) {
        const chipType = await screen.ReadFirstChipType();
        if (chipType === null) {
            console.log(chalk_1.default.red('Не удалось считать тип чипа'));
        }
        else {
            console.log(chalk_1.default.green(ChipType_1.ChipTypeEnum[chipType]));
        }
        for await (const rem of screen.ReadReceivingCardFPGARemarks()) {
            console.log(rem ? chalk_1.default.green(rem) : chalk_1.default.red('Не удалось прочитать FPGA'));
        }
        for await (const rem of await screen.ReadReceivingCardMCURemarks()) {
            console.log(rem ? chalk_1.default.green(rem) : chalk_1.default.red('Не удалось прочитать MCU'));
        }
    }
    else {
        console.log(chalk_1.default.red('Хост не найден'));
    }
    await delay(2000);
};
const showMenu = async (m) => {
    for (let quit = false, show = true; !quit;) {
        const connected = screen?.session.isConnected;
        const entries = Object.entries(m.items).filter((_, i) => !i || connected);
        if (show) {
            cls();
            console.log(chalk_1.default.bold(m.title));
            console.log('  Выберите пункт:');
            entries.forEach(([_, { msg, code }]) => {
                console.log(`    ${chalk_1.default.inverse(code[0])}: ${msg}`);
            });
        }
        const c = String.fromCharCode(await keypress());
        const res = entries.reduce((acc, [key, { code }]) => acc ?? (code.includes(c) ? key : acc), undefined);
        show = !!res;
        if (res) {
            cls();
            console.log(chalk_1.default.bold(m.items[res].msg));
        }
        switch (res) {
            case 'quit':
                quit = true;
                break;
            case 'find':
                await find();
                break;
            case 'bright20':
                await setBrightness(20);
                break;
            case 'bright100':
                await setBrightness(100);
                break;
            case 'horiz':
                await setMode(TestMode_1.TestModeEnum.HorizonLine_Mode);
                break;
            case 'vert':
                await setMode(TestMode_1.TestModeEnum.VerticalLine_Mode);
                break;
            case 'diag':
                await setMode(TestMode_1.TestModeEnum.InclineLine_Mode);
                break;
            case 'chip':
                await readChipType();
                break;
            default:
                break;
        }
        res && res !== 'quit' && (await delay(1000));
    }
};
showMenu(menu).finally(() => {
    screen?.session.close();
    console.log('exit');
});
