// @novastar/screen/build/main/ScannerCapabilitySUM2030.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ScannerCapabilitySUM2030 {
    constructor() {
        this.IsHasMinimumOE = false;
        this.MinimumOE = 0;
        this.LightTimePerSubField = 0;
        this.TotalLightCdfResault = 0;
        this._totalUnitNumPerSubField = 0;
        this._gclkNumPerRef = 0;
        this._refNumPerVs = 0;
        this._totalGclkUintNumPerScan = 0;
        this._lightTimeNum2 = 0;
        this._dclkUnitCycle = 0;
        this._gclkUnitCycle = 0;
        this._scanCount = 0;
        this._fieldPeriod = 0;
        this._vsFreq = 60;
        this._usBlackTime = 0;
        this._shiftNumPerSubField = 0;
        this._maxUnitNumPerScan = 0;
        this._maxGclkNumPerRef = 0;
        this._brightEfficiency = 0;
        this._blackingTime = 0;
        this._maxLoadedPixels = 0;
        this._fieldWaitTime = 0.1;
        this._m1TranCntNum = 3;
        this._m2TranCntNum = 3;
        this._scanBdProperty = null;
    }
    get TotalUnitNumPerSubField() {
        return this._totalUnitNumPerSubField;
    }
    get MaxLoadedPixels() {
        return this._totalUnitNumPerSubField - 3;
    }
    get BrightEfficiency() {
        return this._brightEfficiency;
    }
    get GclkNumPerRef() {
        return this._gclkNumPerRef;
    }
    get RefNumPerVs() {
        return this._refNumPerVs;
    }
    get TotalGclkUnitNumPerScan() {
        return this._totalGclkUintNumPerScan;
    }
    get LightTimeNum2() {
        return this._lightTimeNum2;
    }
    // eslint-disable-next-line class-methods-use-this
    UpdateParameters(scanBdProp, vsFreq) {
        throw new TypeError('Not implemented');
    }
}
exports.default = ScannerCapabilitySUM2030;
ScannerCapabilitySUM2030.SUBDIFF = 0.15;
ScannerCapabilitySUM2030.BLACKTIME_STATIC = 5;
