// @novastar/native/build/main/generated/AckResults.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AckResults = exports.AckResultsEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var AckResultsEnum;
(function (AckResultsEnum) {
    AckResultsEnum[AckResultsEnum["ok"] = 0] = "ok";
    AckResultsEnum[AckResultsEnum["timeout"] = 1] = "timeout";
    AckResultsEnum[AckResultsEnum["requestCrcErr"] = 2] = "requestCrcErr";
    AckResultsEnum[AckResultsEnum["replyCrcErr"] = 3] = "replyCrcErr";
    AckResultsEnum[AckResultsEnum["unknownCmd"] = 4] = "unknownCmd";
    AckResultsEnum[AckResultsEnum["invalid"] = 255] = "invalid";
})(AckResultsEnum = exports.AckResultsEnum || (exports.AckResultsEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link AckResultsEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1122
 */
exports.AckResults = (0, EnumFromString_1.default)(AckResultsEnum, 'AckResults');
