// @novastar/native/build/main/generated/AddColorRestoreInfo.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddColorRestoreInfo = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const RGBCIE_1 = require("./RGBCIE");
const RGBCIEType_1 = require("./RGBCIEType");
/**
 * @category Codecs
 * @desc Codec for interface {@link AddColorRestoreInfo}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:57391
 */
exports.AddColorRestoreInfo = t.intersection([
    t.type({
        TagetCIEType: common.withDefault(RGBCIEType_1.RGBCIEType, 'PAL'),
        ColorRestoreName: common.string_empty,
    }),
    t.partial({
        ColorTempertureValue: common.Int32,
        OriganalCIE: RGBCIE_1.RGBCIE,
        TagetCIE: RGBCIE_1.RGBCIE,
    }),
], 'AddColorRestoreInfo');
