// @novastar/native/build/main/generated/AutoAdjustRefreshRateParameters.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoAdjustRefreshRateParameters = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const AutoAdjustRefreshRateType_1 = require("./AutoAdjustRefreshRateType");
/**
 * @category Codecs
 * @desc Codec for interface {@link AutoAdjustRefreshRateParameters}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.LEDConfigAccessorBase.decompiled.cs:29
 */
exports.AutoAdjustRefreshRateParameters = t.intersection([
    t.type({
        AdjustRateType: common.withDefault(AutoAdjustRefreshRateType_1.AutoAdjustRefreshRateType, 'R_60'),
        TotalUnitNum: common.Int32_0,
        TotalGclkUnitNumPerScan: common.UInt16_0,
        RefNumPerVs: common.Int32_0,
        GCLKRate: common.UInt8_0,
        IsEnableTranCntNum: common.Bool_false,
        M1TranCntNum: common.Int32_3,
        M2TranCntNum: common.Int32_3,
        LightTime: common.Int32_2650,
        LightTimeNum2: common.Int32_2650,
    }),
    t.partial({}),
], 'AutoAdjustRefreshRateParameters');
