// @novastar/native/build/main/generated/AutoBrightExtendData.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoBrightExtendData = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const AutoBrightCalcType_1 = require("./AutoBrightCalcType");
const DisplayAutoBrightMapping_1 = require("./DisplayAutoBrightMapping");
const OpticalProbeFailureInfo_1 = require("./OpticalProbeFailureInfo");
const PeripheralsLocation_1 = require("./PeripheralsLocation");
const SetBrightData_1 = require("./SetBrightData");
/**
 * @category Codecs
 * @desc Codec for interface {@link AutoBrightExtendData}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:58370
 */
exports.AutoBrightExtendData = t.intersection([
    t.type({
        AutoBrightMappingList: common.XMLArray(DisplayAutoBrightMapping_1.DisplayAutoBrightMapping, 'DisplayAutoBrightMapping'),
        UseLightSensorList: common.XMLArray(PeripheralsLocation_1.PeripheralsLocation, 'PeripheralsLocation'),
    }),
    t.partial({
        CalcType: AutoBrightCalcType_1.AutoBrightCalcType,
        OpticalFailureInfo: OpticalProbeFailureInfo_1.OpticalProbeFailureInfo,
        SetBrightData: SetBrightData_1.SetBrightData,
    }),
], 'AutoBrightExtendData');
