// @novastar/native/build/main/generated/AutoAdjustFreqConfigMode.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoAdjustFreqConfigMode = exports.AutoAdjustFreqConfigModeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var AutoAdjustFreqConfigModeEnum;
(function (AutoAdjustFreqConfigModeEnum) {
    AutoAdjustFreqConfigModeEnum[AutoAdjustFreqConfigModeEnum["Addr_1B00"] = 0] = "Addr_1B00";
    AutoAdjustFreqConfigModeEnum[AutoAdjustFreqConfigModeEnum["Addr_0200"] = 1] = "Addr_0200";
    AutoAdjustFreqConfigModeEnum[AutoAdjustFreqConfigModeEnum["Addr_1B00And0200"] = 2] = "Addr_1B00And0200";
    AutoAdjustFreqConfigModeEnum[AutoAdjustFreqConfigModeEnum["UnKnow"] = 255] = "UnKnow";
})(AutoAdjustFreqConfigModeEnum = exports.AutoAdjustFreqConfigModeEnum || (exports.AutoAdjustFreqConfigModeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link AutoAdjustFreqConfigModeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1477
 */
exports.AutoAdjustFreqConfigMode = (0, EnumFromString_1.default)(AutoAdjustFreqConfigModeEnum, 'AutoAdjustFreqConfigMode');
