// @novastar/native/build/main/generated/BaudRateType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaudRateType = exports.BaudRateTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var BaudRateTypeEnum;
(function (BaudRateTypeEnum) {
    BaudRateTypeEnum[BaudRateTypeEnum["BT_115200bps"] = 0] = "BT_115200bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_57600bps"] = 1] = "BT_57600bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_38400bps"] = 2] = "BT_38400bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_19200bps"] = 3] = "BT_19200bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_9600bps"] = 4] = "BT_9600bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_4800bps"] = 5] = "BT_4800bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_2400bps"] = 6] = "BT_2400bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_1200bps"] = 7] = "BT_1200bps";
    BaudRateTypeEnum[BaudRateTypeEnum["BT_Reserved"] = 8] = "BT_Reserved";
})(BaudRateTypeEnum = exports.BaudRateTypeEnum || (exports.BaudRateTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link BaudRateTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1091
 */
exports.BaudRateType = (0, EnumFromString_1.default)(BaudRateTypeEnum, 'BaudRateType');
