// @novastar/native/build/main/generated/BurrnPackType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BurrnPackType = exports.BurrnPackTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var BurrnPackTypeEnum;
(function (BurrnPackTypeEnum) {
    BurrnPackTypeEnum[BurrnPackTypeEnum["MCU"] = 0] = "MCU";
    BurrnPackTypeEnum[BurrnPackTypeEnum["FPGA"] = 1] = "FPGA";
    BurrnPackTypeEnum[BurrnPackTypeEnum["FontLib"] = 2] = "FontLib";
    BurrnPackTypeEnum[BurrnPackTypeEnum["GammaTable"] = 3] = "GammaTable";
    BurrnPackTypeEnum[BurrnPackTypeEnum["FunctionTable"] = 4] = "FunctionTable";
    BurrnPackTypeEnum[BurrnPackTypeEnum["OTHER"] = 5] = "OTHER";
})(BurrnPackTypeEnum = exports.BurrnPackTypeEnum || (exports.BurrnPackTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link BurrnPackTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1315
 */
exports.BurrnPackType = (0, EnumFromString_1.default)(BurrnPackTypeEnum, 'BurrnPackType');
