// @novastar/native/build/main/generated/Chip5125RGBVExtendPropey.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5125RGBVExtendPropey = exports.Chip5125RGBVExtendPropeyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipMBI525ExtendProperty_1 = require("./ChipMBI525ExtendProperty");
exports.Chip5125RGBVExtendPropeyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        OEPlusWidth: common.Int32_1,
        HardwareResetOne: common.Bool_true,
    }),
    t.partial({
        RedProperty: ChipMBI525ExtendProperty_1.ChipMBI525ExtendProperty,
        GreenProperty: ChipMBI525ExtendProperty_1.ChipMBI525ExtendProperty,
        BlueProperty: ChipMBI525ExtendProperty_1.ChipMBI525ExtendProperty,
        VRedProperty: ChipMBI525ExtendProperty_1.ChipMBI525ExtendProperty,
        IsAdvancedMode: common.Bool,
        HardwareResetTwo: common.Int32,
        SoftwareReset: common.Int32,
    }),
], 'Chip5125RGBVExtendPropeyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip5125RGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:42974
 */
exports.Chip5125RGBVExtendPropey = t.intersection([
    exports.Chip5125RGBVExtendPropeyBase,
    t.partial({ '@_xsi:type': t.literal('Chip5125RGBVExtendPropey') }),
], 'Chip5125RGBVExtendPropey');
