// @novastar/native/build/main/generated/ChipTLC5958ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipTLC5958ExtendProperty = exports.ChipTLC5958ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipTLC5958ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        GlobalBrightness: common.UInt8_4,
        RedGain: common.UInt16_256,
        GreenGain: common.UInt16_256,
        BlueGain: common.UInt16_256,
        SetTD0: common.UInt8_1,
        CheckVoltageSetting: common.UInt8_1,
    }),
    t.partial({
        RedLowAshEnhancedControl: common.UInt8,
        GreenLowAshEnhancedControl: common.UInt8,
        BlueLowAshEnhancedControl: common.UInt8,
        TimingControl: common.Bool,
        ImproveTheLowAsh: common.UInt8,
        PWMModelSelecte: common.Bool,
        RedInhibitoryControlEMI: common.Bool,
        GreenInhibitoryControlEMI: common.Bool,
        BlueInhibitoryControlEMI: common.Bool,
        ChargingMode: common.Bool,
        EdgeSetGCLK: common.Bool,
        PowerSavingModeSettings: common.Bool,
    }),
], 'ChipTLC5958ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipTLC5958ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:46152
 */
exports.ChipTLC5958ExtendProperty = t.intersection([
    exports.ChipTLC5958ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipTLC5958ExtendProperty') }),
], 'ChipTLC5958ExtendProperty');
