// @novastar/native/build/main/generated/ChipSCL8080RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSCL8080RGBVExtendProperty = exports.ChipSCL8080RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipSCL8080ExtendProperty_1 = require("./ChipSCL8080ExtendProperty");
exports.ChipSCL8080RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipSCL8080ExtendProperty_1.ChipSCL8080ExtendProperty,
        GreenProperty: ChipSCL8080ExtendProperty_1.ChipSCL8080ExtendProperty,
        BlueProperty: ChipSCL8080ExtendProperty_1.ChipSCL8080ExtendProperty,
        VRedProperty: ChipSCL8080ExtendProperty_1.ChipSCL8080ExtendProperty,
        ScanType: common.UInt8,
        ChipLibVersion: common.UInt8,
        IsAdvancedMode: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        IsUseNewModule: common.Bool,
    }),
], 'ChipSCL8080RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSCL8080RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSCL8080.decompiled.cs:412
 */
exports.ChipSCL8080RGBVExtendProperty = t.intersection([
    exports.ChipSCL8080RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSCL8080RGBVExtendProperty') }),
], 'ChipSCL8080RGBVExtendProperty');
