// @novastar/native/build/main/generated/ChipFM6047RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6047RGBVExtendProperty = exports.ChipFM6047RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipFM6047ExtendProperty_1 = require("./ChipFM6047ExtendProperty");
exports.ChipFM6047RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        OpenTestModel: common.UInt8_0,
    }),
    t.partial({
        RedProperty: ChipFM6047ExtendProperty_1.ChipFM6047ExtendProperty,
        GreenProperty: ChipFM6047ExtendProperty_1.ChipFM6047ExtendProperty,
        BlueProperty: ChipFM6047ExtendProperty_1.ChipFM6047ExtendProperty,
        VRedProperty: ChipFM6047ExtendProperty_1.ChipFM6047ExtendProperty,
        IsUseNewModule: common.Bool,
        IsAdvancedMode: common.Bool,
        EnGCLKMutiRate: common.Bool,
        OpenTestEnable: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
    }),
], 'ChipFM6047RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6047RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6047.decompiled.cs:6347
 */
exports.ChipFM6047RGBVExtendProperty = t.intersection([
    exports.ChipFM6047RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6047RGBVExtendProperty') }),
], 'ChipFM6047RGBVExtendProperty');
