// @novastar/native/build/main/generated/Chip5051BRGBVExtendPropey.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5051BRGBVExtendPropey = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const Chip5051BExtendPropey_1 = require("./Chip5051BExtendPropey");
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5051BRGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:28401
 */
exports.Chip5051BRGBVExtendPropey = t.partial({
    RedProperty: Chip5051BExtendPropey_1.Chip5051BExtendPropey,
    GreenProperty: Chip5051BExtendPropey_1.Chip5051BExtendPropey,
    BlueProperty: Chip5051BExtendPropey_1.Chip5051BExtendPropey,
    VRedProperty: Chip5051BExtendPropey_1.Chip5051BExtendPropey,
    RedRegValueConfigThird: common.UInt16,
    GreenRegValueConfigThird: common.UInt16,
    BlueRegValueConfigThird: common.UInt16,
    VRedRegValueConfigThird: common.UInt16,
    FailureLEDElimination: common.Bool,
}, 'Chip5051BRGBVExtendPropey');
