// @novastar/native/build/main/generated/ChipType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipType = exports.ChipTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ChipTypeEnum;
(function (ChipTypeEnum) {
    ChipTypeEnum[ChipTypeEnum["Chip_CommonBase"] = 0] = "Chip_CommonBase";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5028"] = 1] = "Chip_MBI5028";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5030"] = 2] = "Chip_MBI5030";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5036"] = 3] = "Chip_MBI5036";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5040"] = 4] = "Chip_MBI5040";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5042"] = 5] = "Chip_MBI5042";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5050"] = 6] = "Chip_MBI5050";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5167"] = 7] = "Chip_MBI5167";
    ChipTypeEnum[ChipTypeEnum["Chip_DM13H"] = 8] = "Chip_DM13H";
    ChipTypeEnum[ChipTypeEnum["Chip_P2518"] = 9] = "Chip_P2518";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2017"] = 10] = "Chip_SUM2017";
    ChipTypeEnum[ChipTypeEnum["Chip_P2510"] = 11] = "Chip_P2510";
    ChipTypeEnum[ChipTypeEnum["Chip_RFT3630"] = 12] = "Chip_RFT3630";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5034"] = 13] = "Chip_MBI5034";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2018"] = 14] = "Chip_SUM2018";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2030"] = 15] = "Chip_SUM2030";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5037"] = 16] = "Chip_MBI5037";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5039"] = 17] = "Chip_MBI5039";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9268"] = 18] = "Chip_MY9268";
    ChipTypeEnum[ChipTypeEnum["Chip_SCL8060"] = 19] = "Chip_SCL8060";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5224"] = 20] = "Chip_MBI5224";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5051"] = 21] = "Chip_MBI5051";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5052"] = 22] = "Chip_MBI5052";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5053"] = 23] = "Chip_MBI5053";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9266"] = 24] = "Chip_MY9266";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9269"] = 25] = "Chip_MY9269";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5042B"] = 26] = "Chip_MBI5042B";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5045"] = 27] = "Chip_MBI5045";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2032"] = 28] = "Chip_SUM2032";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5152"] = 29] = "Chip_MBI5152";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9221"] = 30] = "Chip_MY9221";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5124"] = 31] = "Chip_MBI5124";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5151"] = 32] = "Chip_MBI5151";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5153"] = 33] = "Chip_MBI5153";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9163"] = 34] = "Chip_MY9163";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI6030"] = 35] = "Chip_MBI6030";
    ChipTypeEnum[ChipTypeEnum["Chip_RT5924"] = 36] = "Chip_RT5924";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9262"] = 37] = "Chip_MY9262";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16136"] = 38] = "Chip_SM16136";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9868"] = 39] = "Chip_MY9868";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2028"] = 40] = "Chip_SUM2028";
    ChipTypeEnum[ChipTypeEnum["Chip_TLC5958"] = 41] = "Chip_TLC5958";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5266"] = 42] = "Chip_MBI5266";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16158"] = 43] = "Chip_SM16158";
    ChipTypeEnum[ChipTypeEnum["Chip_TLC5929"] = 44] = "Chip_TLC5929";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5166"] = 45] = "Chip_MBI5166";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5155"] = 46] = "Chip_MBI5155";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5122"] = 47] = "Chip_MBI5122";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI6023"] = 48] = "Chip_MBI6023";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5124A"] = 49] = "Chip_MBI5124A";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9366"] = 50] = "Chip_MY9366";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5252"] = 51] = "Chip_MBI5252";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5043"] = 52] = "Chip_MBI5043";
    ChipTypeEnum[ChipTypeEnum["Chip_MBIA043"] = 53] = "Chip_MBIA043";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9862"] = 54] = "Chip_MY9862";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5041B"] = 55] = "Chip_MBI5041B";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5041Q"] = 56] = "Chip_MBI5041Q";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5038"] = 57] = "Chip_MBI5038";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2038"] = 58] = "Chip_ICN2038";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2027"] = 59] = "Chip_ICN2027";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16027"] = 60] = "Chip_SM16027";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16159"] = 61] = "Chip_SM16159";
    ChipTypeEnum[ChipTypeEnum["Chip_TLS3001"] = 62] = "Chip_TLS3001";
    ChipTypeEnum[ChipTypeEnum["Chip_TLC59581"] = 63] = "Chip_TLC59581";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5353"] = 64] = "Chip_MBI5353";
    ChipTypeEnum[ChipTypeEnum["Chip_GW6205"] = 65] = "Chip_GW6205";
    ChipTypeEnum[ChipTypeEnum["Chip_GW6202B"] = 66] = "Chip_GW6202B";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2017T"] = 67] = "Chip_SUM2017T";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2033"] = 68] = "Chip_SUM2033";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9231"] = 69] = "Chip_MY9231";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2130"] = 70] = "Chip_SUM2130";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2131"] = 71] = "Chip_SUM2131";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9868A"] = 72] = "Chip_MY9868A";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9862A"] = 73] = "Chip_MY9862A";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2038A"] = 74] = "Chip_ICN2038A";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9263"] = 75] = "Chip_MY9263";
    ChipTypeEnum[ChipTypeEnum["Chip_UCS9812"] = 76] = "Chip_UCS9812";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5051B"] = 77] = "Chip_MBI5051B";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI6020"] = 78] = "Chip_MBI6020";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5253"] = 79] = "Chip_MBI5253";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI6120"] = 80] = "Chip_MBI6120";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2053"] = 81] = "Chip_ICN2053";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9373"] = 82] = "Chip_MY9373";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16017S"] = 83] = "Chip_SM16017S";
    ChipTypeEnum[ChipTypeEnum["Chip_SC6618"] = 84] = "Chip_SC6618";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16207S"] = 85] = "Chip_SM16207S";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2117"] = 86] = "Chip_SUM2117";
    ChipTypeEnum[ChipTypeEnum["Chip_TLC5948A"] = 87] = "Chip_TLC5948A";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2038S"] = 88] = "Chip_ICN2038S";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9348"] = 89] = "Chip_MY9348";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2035"] = 90] = "Chip_SUM2035";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2135"] = 91] = "Chip_SUM2135";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM20167"] = 92] = "Chip_SUM20167";
    ChipTypeEnum[ChipTypeEnum["Chip_TM1916"] = 93] = "Chip_TM1916";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2030T"] = 94] = "Chip_SUM2030T";
    ChipTypeEnum[ChipTypeEnum["Chip_GW6808"] = 95] = "Chip_GW6808";
    ChipTypeEnum[ChipTypeEnum["Chip_GW98XX"] = 96] = "Chip_GW98XX";
    ChipTypeEnum[ChipTypeEnum["Chip_SC6660"] = 97] = "Chip_SC6660";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9960"] = 98] = "Chip_LS9960";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16237"] = 99] = "Chip_SM16237";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2038S_Common"] = 100] = "Chip_ICN2038S_Common";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9918"] = 101] = "Chip_LS9918";
    ChipTypeEnum[ChipTypeEnum["Chip_UCS5603"] = 102] = "Chip_UCS5603";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5125"] = 103] = "Chip_MBI5125";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5125A"] = 104] = "Chip_MBI5125A";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2100"] = 105] = "Chip_ICND2100";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2110"] = 106] = "Chip_ICND2110";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2055"] = 107] = "Chip_ICND2055";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2045"] = 108] = "Chip_ICN2045";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2013"] = 109] = "Chip_ICN2013";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2050"] = 110] = "Chip_ICN2050";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN205X"] = 111] = "Chip_ICN205X";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16227"] = 112] = "Chip_SM16227";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9929"] = 113] = "Chip_LS9929";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5354"] = 114] = "Chip_MBI5354";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16188B"] = 115] = "Chip_SM16188B";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6182"] = 116] = "Chip_FM6182";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5359"] = 117] = "Chip_MBI5359";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2065"] = 118] = "Chip_ICN2065";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5253B"] = 119] = "Chip_MBI5253B";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9866"] = 120] = "Chip_MY9866";
    ChipTypeEnum[ChipTypeEnum["Chip_SCL8080"] = 121] = "Chip_SCL8080";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16259"] = 122] = "Chip_SM16259";
    ChipTypeEnum[ChipTypeEnum["Chip_NVS6867"] = 123] = "Chip_NVS6867";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM6082"] = 124] = "Chip_SUM6082";
    ChipTypeEnum[ChipTypeEnum["Chip_UCS8902"] = 125] = "Chip_UCS8902";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16237P"] = 126] = "Chip_SM16237P";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16227P"] = 127] = "Chip_SM16227P";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9748"] = 128] = "Chip_MY9748";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6142"] = 129] = "Chip_FM6142";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6153"] = 130] = "Chip_FM6153";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2200"] = 131] = "Chip_ICND2200";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2017TD"] = 132] = "Chip_SUM2017TD";
    ChipTypeEnum[ChipTypeEnum["Chip_ISA2020"] = 133] = "Chip_ISA2020";
    ChipTypeEnum[ChipTypeEnum["Chip_IT1505"] = 134] = "Chip_IT1505";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16237DS"] = 135] = "Chip_SM16237DS";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6126"] = 136] = "Chip_FM6126";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9930"] = 137] = "Chip_LS9930";
    ChipTypeEnum[ChipTypeEnum["Chip_LD1512"] = 138] = "Chip_LD1512";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5124New"] = 139] = "Chip_MBI5124New";
    ChipTypeEnum[ChipTypeEnum["Chip_UCS1912"] = 140] = "Chip_UCS1912";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9961"] = 141] = "Chip_LS9961";
    ChipTypeEnum[ChipTypeEnum["Chip_TLC6946"] = 142] = "Chip_TLC6946";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2112"] = 143] = "Chip_ICND2112";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2046"] = 144] = "Chip_ICND2046";
    ChipTypeEnum[ChipTypeEnum["Chip_ICN2038S_\u901A\u7528"] = 145] = "Chip_ICN2038S_\u901A\u7528";
    ChipTypeEnum[ChipTypeEnum["Chip_UCS512C"] = 146] = "Chip_UCS512C";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16359"] = 147] = "Chip_SM16359";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5759"] = 148] = "Chip_MBI5759";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9866_Common"] = 149] = "Chip_MY9866_Common";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9919"] = 150] = "Chip_LS9919";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5850"] = 151] = "Chip_MBI5850";
    ChipTypeEnum[ChipTypeEnum["Chip_CNS7153"] = 152] = "Chip_CNS7153";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2163"] = 153] = "Chip_ICND2163";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9917"] = 154] = "Chip_LS9917";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5325"] = 155] = "Chip_MBI5325";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5325D_DPWM"] = 156] = "Chip_MBI5325D_DPWM";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16218"] = 157] = "Chip_SM16218";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2153"] = 158] = "Chip_ICND2153";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2150"] = 159] = "Chip_ICND2150";
    ChipTypeEnum[ChipTypeEnum["Chip_SCIC2023"] = 160] = "Chip_SCIC2023";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9553"] = 161] = "Chip_MY9553";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9553_Common"] = 162] = "Chip_MY9553_Common";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16169S"] = 163] = "Chip_SM16169S";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9920"] = 164] = "Chip_LS9920";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9926"] = 165] = "Chip_LS9926";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9935"] = 166] = "Chip_LS9935";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2076"] = 167] = "Chip_ICND2076";
    ChipTypeEnum[ChipTypeEnum["Chip_HX8055"] = 168] = "Chip_HX8055";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2047"] = 169] = "Chip_ICND2047";
    ChipTypeEnum[ChipTypeEnum["Chip_SUM2036"] = 170] = "Chip_SUM2036";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6648"] = 171] = "Chip_FM6648";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6127"] = 172] = "Chip_FM6127";
    ChipTypeEnum[ChipTypeEnum["Chip_CNS7253"] = 173] = "Chip_CNS7253";
    ChipTypeEnum[ChipTypeEnum["Chip_XM11920G"] = 174] = "Chip_XM11920G";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5251"] = 175] = "Chip_MBI5251";
    ChipTypeEnum[ChipTypeEnum["Chip_SCL8081"] = 176] = "Chip_SCL8081";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5754"] = 177] = "Chip_MBI5754";
    ChipTypeEnum[ChipTypeEnum["Chip_VOD5153"] = 178] = "Chip_VOD5153";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5254"] = 179] = "Chip_MBI5254";
    ChipTypeEnum[ChipTypeEnum["Chip_CS2017"] = 180] = "Chip_CS2017";
    ChipTypeEnum[ChipTypeEnum["Chip_CS2017S"] = 181] = "Chip_CS2017S";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6353"] = 182] = "Chip_FM6353";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6356"] = 183] = "Chip_FM6356";
    ChipTypeEnum[ChipTypeEnum["Chip_YY6018"] = 184] = "Chip_YY6018";
    ChipTypeEnum[ChipTypeEnum["Chip_DP5525"] = 185] = "Chip_DP5525";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5864"] = 186] = "Chip_MBI5864";
    ChipTypeEnum[ChipTypeEnum["Chip_CS2033"] = 187] = "Chip_CS2033";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5264"] = 188] = "Chip_MBI5264";
    ChipTypeEnum[ChipTypeEnum["Chip_HS3257"] = 189] = "Chip_HS3257";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9935B"] = 190] = "Chip_LS9935B";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9928"] = 191] = "Chip_LS9928";
    ChipTypeEnum[ChipTypeEnum["Chip_TLC6983"] = 192] = "Chip_TLC6983";
    ChipTypeEnum[ChipTypeEnum["Chip_XM11201G"] = 193] = "Chip_XM11201G";
    ChipTypeEnum[ChipTypeEnum["Chip_XM11202G"] = 194] = "Chip_XM11202G";
    ChipTypeEnum[ChipTypeEnum["Chip_AXS6018"] = 195] = "Chip_AXS6018";
    ChipTypeEnum[ChipTypeEnum["Chip_CS2066"] = 196] = "Chip_CS2066";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16369"] = 197] = "Chip_SM16369";
    ChipTypeEnum[ChipTypeEnum["Chip_HX8864"] = 198] = "Chip_HX8864";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6565"] = 199] = "Chip_FM6565";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6555"] = 200] = "Chip_FM6555";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI6033"] = 201] = "Chip_MBI6033";
    ChipTypeEnum[ChipTypeEnum["Chip_SCL8061"] = 202] = "Chip_SCL8061";
    ChipTypeEnum[ChipTypeEnum["Chip_RT5965"] = 203] = "Chip_RT5965";
    ChipTypeEnum[ChipTypeEnum["Chip_MY9758"] = 204] = "Chip_MY9758";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2210"] = 205] = "Chip_ICND2210";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6047"] = 206] = "Chip_FM6047";
    ChipTypeEnum[ChipTypeEnum["Chip_DM413"] = 207] = "Chip_DM413";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6363"] = 208] = "Chip_FM6363";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2159"] = 209] = "Chip_ICND2159";
    ChipTypeEnum[ChipTypeEnum["Chip_RM1A00"] = 210] = "Chip_RM1A00";
    ChipTypeEnum[ChipTypeEnum["Chip_RT7920"] = 211] = "Chip_RT7920";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16017DS"] = 212] = "Chip_SM16017DS";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9903"] = 213] = "Chip_LS9903";
    ChipTypeEnum[ChipTypeEnum["Chip_D8023"] = 214] = "Chip_D8023";
    ChipTypeEnum[ChipTypeEnum["Chip_FM6518"] = 216] = "Chip_FM6518";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5268"] = 215] = "Chip_MBI5268";
    ChipTypeEnum[ChipTypeEnum["Chip_TL3102UD"] = 217] = "Chip_TL3102UD";
    ChipTypeEnum[ChipTypeEnum["Chip_HBS1910"] = 218] = "Chip_HBS1910";
    ChipTypeEnum[ChipTypeEnum["Chip_DP3264"] = 219] = "Chip_DP3264";
    ChipTypeEnum[ChipTypeEnum["Chip_DP3246"] = 220] = "Chip_DP3246";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16389"] = 221] = "Chip_SM16389";
    ChipTypeEnum[ChipTypeEnum["Chip_D2618"] = 222] = "Chip_D2618";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5252B"] = 223] = "Chip_MBI5252B";
    ChipTypeEnum[ChipTypeEnum["Chip_RZ"] = 224] = "Chip_RZ";
    ChipTypeEnum[ChipTypeEnum["Chip_HX5011"] = 225] = "Chip_HX5011";
    ChipTypeEnum[ChipTypeEnum["Chip_ICND2126"] = 226] = "Chip_ICND2126";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5253NewB"] = 227] = "Chip_MBI5253NewB";
    ChipTypeEnum[ChipTypeEnum["Chip_CFD455A"] = 228] = "Chip_CFD455A";
    ChipTypeEnum[ChipTypeEnum["Chip_DP3230"] = 229] = "Chip_DP3230";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9936"] = 230] = "Chip_LS9936";
    ChipTypeEnum[ChipTypeEnum["Chip_CFD435A"] = 231] = "Chip_CFD435A";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16380"] = 232] = "Chip_SM16380";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5034B"] = 233] = "Chip_MBI5034B";
    ChipTypeEnum[ChipTypeEnum["Chip_TL3103UD"] = 234] = "Chip_TL3103UD";
    ChipTypeEnum[ChipTypeEnum["Chip_HB200B"] = 235] = "Chip_HB200B";
    ChipTypeEnum[ChipTypeEnum["Chip_CFD555A"] = 236] = "Chip_CFD555A";
    ChipTypeEnum[ChipTypeEnum["Chip_RT5938SS"] = 237] = "Chip_RT5938SS";
    ChipTypeEnum[ChipTypeEnum["Chip_LS9931"] = 238] = "Chip_LS9931";
    ChipTypeEnum[ChipTypeEnum["Chip_NT27002S"] = 239] = "Chip_NT27002S";
    ChipTypeEnum[ChipTypeEnum["Chip_SM16388"] = 240] = "Chip_SM16388";
    ChipTypeEnum[ChipTypeEnum["Chip_DP3265"] = 241] = "Chip_DP3265";
    ChipTypeEnum[ChipTypeEnum["Chip_MBI5269"] = 242] = "Chip_MBI5269";
    ChipTypeEnum[ChipTypeEnum["Chip_CFD335A"] = 243] = "Chip_CFD335A";
    ChipTypeEnum[ChipTypeEnum["Chip_RT5938SS_Common"] = 244] = "Chip_RT5938SS_Common";
    ChipTypeEnum[ChipTypeEnum["Chip_TBS3268"] = 245] = "Chip_TBS3268";
    ChipTypeEnum[ChipTypeEnum["Chip_MTS01"] = 246] = "Chip_MTS01";
    ChipTypeEnum[ChipTypeEnum["Chip_XM11480G"] = 247] = "Chip_XM11480G";
    ChipTypeEnum[ChipTypeEnum["Unknown"] = 255] = "Unknown";
})(ChipTypeEnum = exports.ChipTypeEnum || (exports.ChipTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ChipTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:483
 */
exports.ChipType = (0, EnumFromString_1.default)(ChipTypeEnum, 'ChipType');
