// @novastar/native/build/main/generated/ChipICND2126RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipICND2126RGBVExtendProperty = exports.ChipICND2126RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipICND2126ExtendProperty_1 = require("./ChipICND2126ExtendProperty");
const PointDetectParameter_1 = require("./PointDetectParameter");
exports.ChipICND2126RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipICND2126ExtendProperty_1.ChipICND2126ExtendProperty,
        GreenProperty: ChipICND2126ExtendProperty_1.ChipICND2126ExtendProperty,
        BlueProperty: ChipICND2126ExtendProperty_1.ChipICND2126ExtendProperty,
        VRedProperty: ChipICND2126ExtendProperty_1.ChipICND2126ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        IsAdvancedMode: common.Bool,
        PointDetectType: common.UInt8,
        SetSpecialBit: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        PointDetectParameter: PointDetectParameter_1.PointDetectParameter,
    }),
], 'ChipICND2126RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICND2126RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICN2126.decompiled.cs:1400
 */
exports.ChipICND2126RGBVExtendProperty = t.intersection([
    exports.ChipICND2126RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICND2126RGBVExtendProperty') }),
], 'ChipICND2126RGBVExtendProperty');
