// @novastar/native/build/main/generated/ChipXM11202GRGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipXM11202GRGBVExtendProperty = exports.ChipXM11202GRGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipXM11202GRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegisterArray: common.XMLArray(common.UInt64, 'ulong'),
        ConfigRegisterNum: common.Int32_16,
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        IsUseNewModule: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipXM11202GRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipXM11202GRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipXM11202G.decompiled.cs:1003
 */
exports.ChipXM11202GRGBVExtendProperty = t.intersection([
    exports.ChipXM11202GRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipXM11202GRGBVExtendProperty') }),
], 'ChipXM11202GRGBVExtendProperty');
