// @novastar/native/build/main/generated/ChipMY9862RGBExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMY9862RGBExtendProperty = exports.ChipMY9862RGBExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipMY9862ExtendProperty_1 = require("./ChipMY9862ExtendProperty");
exports.ChipMY9862RGBExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipMY9862ExtendProperty_1.ChipMY9862ExtendProperty,
        GreenProperty: ChipMY9862ExtendProperty_1.ChipMY9862ExtendProperty,
        BlueProperty: ChipMY9862ExtendProperty_1.ChipMY9862ExtendProperty,
        VRedProperty: ChipMY9862ExtendProperty_1.ChipMY9862ExtendProperty,
    }),
], 'ChipMY9862RGBExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMY9862RGBExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:41886
 */
exports.ChipMY9862RGBExtendProperty = t.intersection([
    exports.ChipMY9862RGBExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMY9862RGBExtendProperty') }),
], 'ChipMY9862RGBExtendProperty');
