// @novastar/native/build/main/generated/ChipCS2017SExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCS2017SExtendProperty = exports.ChipCS2017SExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipCS2017SExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        FirstRegistor: common.UInt16_7936,
    }),
    t.partial({
        LineEliminationEnable: common.Bool,
        ShadowEliminationLevel: common.UInt8,
        ResponseSpeed: common.UInt8,
        WorkMode: common.UInt8,
        SetCurrentPosition: common.UInt8,
        InflectionCurrent: common.UInt8,
        DMAWriteMode: common.UInt8,
        CurrentGain: common.UInt8,
    }),
], 'ChipCS2017SExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2017SExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2017S.decompiled.cs:81
 */
exports.ChipCS2017SExtendProperty = t.intersection([
    exports.ChipCS2017SExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2017SExtendProperty') }),
], 'ChipCS2017SExtendProperty');
