// @novastar/native/build/main/generated/Chip16169sExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip16169sExtendProperty = exports.Chip16169sExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ScanType_1 = require("./ScanType");
exports.Chip16169sExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        FailureLEDEliminationGainLevel: common.UInt8_32,
    }),
    t.partial({
        GrayHorizontalStripesEliminateLow: common.Bool,
        LowGayCompensation: common.UInt8,
        FailureLEDElimination: common.Bool,
        OpenTestGrade: common.UInt8,
        ShadowVoltage: common.UInt8,
        IsClearGhostEnable: common.Bool,
        SubFields: common.UInt8,
        ScanType: ScanType_1.ScanType,
        GrayDepth: common.UInt8,
        LowGayCompensationEnhance: common.Bool,
        Gain: common.UInt8,
    }),
], 'Chip16169sExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip16169sExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:51618
 */
exports.Chip16169sExtendProperty = t.intersection([
    exports.Chip16169sExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('Chip16169sExtendProperty') }),
], 'Chip16169sExtendProperty');
