// @novastar/native/build/main/generated/Chip5359RGBExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5359RGBExtendProperty = exports.Chip5359RGBExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const Chip5359ExtendProperty_1 = require("./Chip5359ExtendProperty");
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.Chip5359RGBExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        Global1_Register1: common.withDefault(common.UInt16, 16448),
        Global1_Register2: common.withDefault(common.UInt16, 8252),
        Global1_Register3: common.withDefault(common.UInt16, 3132),
        Global2_Register2: common.UInt16_4096,
        Global2_Register3: common.withDefault(common.UInt16, 16319),
        R1_Register1: common.UInt16_16384,
        G1_Register1: common.UInt16_16384,
        B1_Register1: common.UInt16_16384,
        R2_Register1: common.UInt16_508,
        R2_Register3: common.UInt16_15872,
        G2_Register1: common.UInt16_508,
        G2_Register3: common.UInt16_15872,
        B2_Register1: common.UInt16_508,
        B2_Register3: common.UInt16_15872,
        IsAmplifyCurrent: common.Bool_true,
        M: common.withDefault(common.Int32, 127),
        N: common.Int32_15,
        Dvi: common.Int32_3,
        ExcessiveTime: common.Int32_60,
    }),
    t.partial({
        RextendProperty: Chip5359ExtendProperty_1.Chip5359ExtendProperty,
        GextendProperty: Chip5359ExtendProperty_1.Chip5359ExtendProperty,
        BextendProperty: Chip5359ExtendProperty_1.Chip5359ExtendProperty,
        IsAdvancedMode: common.Bool,
        Global2_Register1: common.UInt16,
        Global3_Register1: common.UInt16,
        Global3_Register2: common.UInt16,
        Global3_Register3: common.UInt16,
        R1_Register2: common.UInt16,
        R1_Register3: common.UInt16,
        G1_Register2: common.UInt16,
        G1_Register3: common.UInt16,
        B1_Register2: common.UInt16,
        B1_Register3: common.UInt16,
        R2_Register2: common.UInt16,
        G2_Register2: common.UInt16,
        B2_Register2: common.UInt16,
        IstwoWayTransmission: common.Bool,
        IsGhostRemovalEnhance: common.Bool,
        IsSavePower: common.Bool,
        IsSavePowerPlus: common.Bool,
        FineGrade: common.Int32,
        GlobalGain: common.Int32,
        Gclk: common.Int32,
        EnDoubleRefresh: common.Bool,
        IsDoublePower: common.Bool,
        RowChangePoint5359: common.UInt16,
    }),
], 'Chip5359RGBExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip5359RGBExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:32687
 */
exports.Chip5359RGBExtendProperty = t.intersection([
    exports.Chip5359RGBExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('Chip5359RGBExtendProperty') }),
], 'Chip5359RGBExtendProperty');
