// @novastar/native/build/main/generated/ChipICND2126ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipICND2126ExtendProperty = exports.ChipICND2126ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipICND2126ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ShortDetection: common.UInt8,
        ErrorSurvey_OnTime: common.Bool,
        SavingPowerMode: common.UInt8,
        CurrentGain: common.UInt16,
        ShadowEliminationEn: common.Bool,
        InflectionPiont: common.UInt8,
        ShadowEliminationElectric: common.UInt8,
    }),
], 'ChipICND2126ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICND2126ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICN2126.decompiled.cs:1268
 */
exports.ChipICND2126ExtendProperty = t.intersection([
    exports.ChipICND2126ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICND2126ExtendProperty') }),
], 'ChipICND2126ExtendProperty');
