// @novastar/native/build/main/generated/ChipCNS7153RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCNS7153RGBVExtendProperty = exports.ChipCNS7153RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipCNS7153ExtendProperty_1 = require("./ChipCNS7153ExtendProperty");
exports.ChipCNS7153RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        RedProperty: ChipCNS7153ExtendProperty_1.ChipCNS7153ExtendProperty,
        GreenProperty: ChipCNS7153ExtendProperty_1.ChipCNS7153ExtendProperty,
        BlueProperty: ChipCNS7153ExtendProperty_1.ChipCNS7153ExtendProperty,
        VRedProperty: ChipCNS7153ExtendProperty_1.ChipCNS7153ExtendProperty,
        IsUseNewModule: common.Bool,
    }),
], 'ChipCNS7153RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCNS7153RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCNS7153.decompiled.cs:1972
 */
exports.ChipCNS7153RGBVExtendProperty = t.intersection([
    exports.ChipCNS7153RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCNS7153RGBVExtendProperty') }),
], 'ChipCNS7153RGBVExtendProperty');
