// @novastar/native/build/main/generated/Chip5155RGBVExtendPropey.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5155RGBVExtendPropey = exports.Chip5155RGBVExtendPropeyBase = void 0;
const t = __importStar(require("io-ts"));
const Chip5155ExtendProperty_1 = require("./Chip5155ExtendProperty");
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.Chip5155RGBVExtendPropeyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: Chip5155ExtendProperty_1.Chip5155ExtendProperty,
        GreenProperty: Chip5155ExtendProperty_1.Chip5155ExtendProperty,
        BlueProperty: Chip5155ExtendProperty_1.Chip5155ExtendProperty,
        VRedProperty: Chip5155ExtendProperty_1.Chip5155ExtendProperty,
    }),
], 'Chip5155RGBVExtendPropeyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip5155RGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:41777
 */
exports.Chip5155RGBVExtendPropey = t.intersection([
    exports.Chip5155RGBVExtendPropeyBase,
    t.partial({ '@_xsi:type': t.literal('Chip5155RGBVExtendPropey') }),
], 'Chip5155RGBVExtendPropey');
