// @novastar/native/build/main/generated/ChipSM16380RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16380RGBVExtendProperty = exports.ChipSM16380RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipSM16380ExtendProperty_1 = require("./ChipSM16380ExtendProperty");
exports.ChipSM16380RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        RedExtendProperty: ChipSM16380ExtendProperty_1.ChipSM16380ExtendProperty,
        GreenExtendProperty: ChipSM16380ExtendProperty_1.ChipSM16380ExtendProperty,
        BlueExtendProperty: ChipSM16380ExtendProperty_1.ChipSM16380ExtendProperty,
        VRedExtendProperty: ChipSM16380ExtendProperty_1.ChipSM16380ExtendProperty,
        IsUseNewModule: common.Bool,
        ScanCount: common.UInt8,
        MutiRate: common.UInt8,
        FailureEliminationEn: common.Bool,
        CurrentGainGrade: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipSM16380RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16380RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16380.decompiled.cs:1196
 */
exports.ChipSM16380RGBVExtendProperty = t.intersection([
    exports.ChipSM16380RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16380RGBVExtendProperty') }),
], 'ChipSM16380RGBVExtendProperty');
