// @novastar/native/build/main/generated/Chip2028RGBVExtendPropey.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip2028RGBVExtendPropey = exports.Chip2028RGBVExtendPropeyBase = void 0;
const t = __importStar(require("io-ts"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipSUM2028ExtendProperty_1 = require("./ChipSUM2028ExtendProperty");
exports.Chip2028RGBVExtendPropeyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipSUM2028ExtendProperty_1.ChipSUM2028ExtendProperty,
        GreenProperty: ChipSUM2028ExtendProperty_1.ChipSUM2028ExtendProperty,
        BlueProperty: ChipSUM2028ExtendProperty_1.ChipSUM2028ExtendProperty,
        VRedProperty: ChipSUM2028ExtendProperty_1.ChipSUM2028ExtendProperty,
    }),
], 'Chip2028RGBVExtendPropeyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip2028RGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:47971
 */
exports.Chip2028RGBVExtendPropey = t.intersection([
    exports.Chip2028RGBVExtendPropeyBase,
    t.partial({ '@_xsi:type': t.literal('Chip2028RGBVExtendPropey') }),
], 'Chip2028RGBVExtendPropey');
