// @novastar/native/build/main/generated/ChipCFD555ARGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCFD555ARGBVExtendProperty = exports.ChipCFD555ARGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipCFD555AExtendProperty_1 = require("./ChipCFD555AExtendProperty");
exports.ChipCFD555ARGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipCFD555AExtendProperty_1.ChipCFD555AExtendProperty,
        GreenProperty: ChipCFD555AExtendProperty_1.ChipCFD555AExtendProperty,
        BlueProperty: ChipCFD555AExtendProperty_1.ChipCFD555AExtendProperty,
        VRedProperty: ChipCFD555AExtendProperty_1.ChipCFD555AExtendProperty,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        PLLFreqDivision: common.UInt8,
        PLLFreqDoubling: common.UInt8,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipCFD555ARGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCFD555ARGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCFD555A.decompiled.cs:8408
 */
exports.ChipCFD555ARGBVExtendProperty = t.intersection([
    exports.ChipCFD555ARGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCFD555ARGBVExtendProperty') }),
], 'ChipCFD555ARGBVExtendProperty');
