// @novastar/native/build/main/generated/Chip5253BExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5253BExtendProperty = exports.Chip5253BExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.Chip5253BExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        FirstSweepElimination: common.UInt8_31,
        UnderEliminateGhost: common.UInt8_31,
        UnderEliminateGhostingEnhancedMode: common.Bool_true,
        LowAshEnhanceUniformity: common.Bool_true,
        AdvancedRLowAshColorCastCompensation1: common.UInt8_31,
        AdvancedRLowAshColorCastCompensation3: common.UInt8_31,
        DislodgeBadPointsGrade: common.UInt8_1,
        TextGhostElimination: common.UInt8_31,
        CrossCouplingOptimization: common.UInt8_31,
        BrightBlockCouplingOptimization: common.UInt8_31,
        FirstSweepBiasElimination: common.UInt8_31,
    }),
    t.partial({
        GrayHorizontalStripesEliminateLow: common.Bool,
        LowAshColorCastCompensation: common.UInt8,
        AdvancedRLowAshColorCastCompensation2: common.UInt8,
        GradientCompensation: common.UInt8,
        GradientCompensationOne: common.UInt8,
    }),
], 'Chip5253BExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip5253BExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:32405
 */
exports.Chip5253BExtendProperty = t.intersection([exports.Chip5253BExtendPropertyBase, t.partial({ '@_xsi:type': t.literal('Chip5253BExtendProperty') })], 'Chip5253BExtendProperty');
