// @novastar/native/build/main/generated/ChipFM6182ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6182ExtendProperty = exports.ChipFM6182ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipFM6182ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegValueConfigFirst: common.UInt16_63,
        RegValueConfigSecond: common.UInt16_63,
    }),
    t.partial({
        CurrentLowCompsention: common.Int32,
        LowCompsention: common.Int32,
        FirstCompsentionOne: common.Int32,
        ErrorDetection: common.Int32,
        GhostElimination: common.Int32,
        DetectionEable: common.Int32,
        FirstCompsentionTwo: common.Int32,
        CurrentPoint: common.Int32,
        OpenUpTime: common.Int32,
        PDMOpenTime: common.Int32,
        ParityTimeSet: common.Int32,
    }),
], 'ChipFM6182ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6182ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:42673
 */
exports.ChipFM6182ExtendProperty = t.intersection([
    exports.ChipFM6182ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6182ExtendProperty') }),
], 'ChipFM6182ExtendProperty');
