// @novastar/native/build/main/generated/ChipFM6363RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6363RGBVExtendProperty = exports.ChipFM6363RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipFM6363ExtendProperty_1 = require("./ChipFM6363ExtendProperty");
exports.ChipFM6363RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipFM6363ExtendProperty_1.ChipFM6363ExtendProperty,
        GreenProperty: ChipFM6363ExtendProperty_1.ChipFM6363ExtendProperty,
        BlueProperty: ChipFM6363ExtendProperty_1.ChipFM6363ExtendProperty,
        VRedProperty: ChipFM6363ExtendProperty_1.ChipFM6363ExtendProperty,
        ScanCount: common.UInt8,
        LowAshPitsImprove: common.UInt8,
        IsAberrationOptimize: common.Bool,
        IsCouplingOptimize: common.Bool,
        IsStarSwipPoint: common.Bool,
        IsAdvancedModel: common.Bool,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        FifthDataLen: common.Int32,
        FifthStartIndex: common.Int32,
        FifthRegisterAddr: common.Int32,
    }),
], 'ChipFM6363RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6363RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6363.decompiled.cs:1032
 */
exports.ChipFM6363RGBVExtendProperty = t.intersection([
    exports.ChipFM6363RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6363RGBVExtendProperty') }),
], 'ChipFM6363RGBVExtendProperty');
