// @novastar/native/build/main/generated/ChipFM6353ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6353ExtendProperty = exports.ChipFM6353ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipFM6353ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegFirst: common.withDefault(common.UInt16, 7536),
        RegSecond: common.withDefault(common.UInt16, 64853),
        RegThird: common.withDefault(common.UInt16, 16391),
        RegFourth: common.withDefault(common.UInt16, 64),
    }),
    t.partial({
        LowAshPitsImprove: common.UInt8,
        IsElimateShadow: common.Bool,
        Gain: common.UInt16,
        LowGrayAdjustGrade: common.UInt8,
        IsLowGrayAdjust: common.Bool,
        IsLowGrayCompensate: common.Bool,
        LowGrayCompensateGrade: common.UInt8,
        IsFirstLineSlantsDark: common.Bool,
        FirstLineSlantsDarkOptimization: common.UInt8,
        IsUseNewModule: common.Bool,
    }),
], 'ChipFM6353ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6353ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6353.decompiled.cs:946
 */
exports.ChipFM6353ExtendProperty = t.intersection([
    exports.ChipFM6353ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6353ExtendProperty') }),
], 'ChipFM6353ExtendProperty');
