// @novastar/native/build/main/generated/ChipFM6047ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6047ExtendProperty = exports.ChipFM6047ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipFM6047ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ThirdRegValue: common.UInt16,
        FourthRegValue: common.UInt16,
        OEWidth: common.UInt8,
        OEDelay: common.UInt8,
        Inflexion: common.UInt8,
        HiddenMode: common.UInt8,
        CurrentGain: common.UInt16,
        ShadowEliminationLevel: common.UInt8,
        OpenTestEnable: common.Bool,
        ErrCurrentGain: common.UInt16,
    }),
], 'ChipFM6047ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6047ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6047.decompiled.cs:6202
 */
exports.ChipFM6047ExtendProperty = t.intersection([
    exports.ChipFM6047ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6047ExtendProperty') }),
], 'ChipFM6047ExtendProperty');
