// @novastar/native/build/main/generated/ChipCNS7253ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCNS7253ExtendProperty = exports.ChipCNS7253ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipCNS7253ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RegFirst: common.UInt16,
        RegSecond: common.UInt16,
        RegThird: common.UInt16,
        RegFourth: common.UInt16,
        OPEN_DET: common.Bool,
        SCAN_LINE: common.UInt8,
        PWM_OPT: common.UInt8,
        GCLK_DOUBLE: common.Bool,
        PWM_GROUP: common.UInt16,
        PWM_REV: common.Bool,
        PartNumPerRefValue: common.UInt8,
        ADJ: common.UInt8,
        IGAIN: common.UInt16,
        ADJ_Enable: common.Bool,
        KeenPointVoltage: common.UInt8,
        FirstLineOffsetCompensation: common.UInt8,
        PWM_ALL: common.UInt8,
        PWM_ALL_Enable: common.Bool,
        FirstLineOffsetCompensation_Enable: common.Bool,
        OPEN_RST: common.Bool,
    }),
], 'ChipCNS7253ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCNS7253ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCNS7253.decompiled.cs:87
 */
exports.ChipCNS7253ExtendProperty = t.intersection([
    exports.ChipCNS7253ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCNS7253ExtendProperty') }),
], 'ChipCNS7253ExtendProperty');
