// @novastar/native/build/main/generated/ChipTLC6983RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipTLC6983RGBVExtendProperty = exports.ChipTLC6983RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipTLC6983ExtendProperty_1 = require("./ChipTLC6983ExtendProperty");
exports.ChipTLC6983RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_12,
    }),
    t.partial({
        ExtendProperty: ChipTLC6983ExtendProperty_1.ChipTLC6983ExtendProperty,
        TempSubField: common.UInt8,
        TempICNum: common.UInt8,
        TempGclkFreqR: common.UInt8,
        TempBit3: common.UInt8,
        SubField: common.UInt8,
        GclkFreqR: common.UInt8,
        Bit3: common.UInt8,
        ICNum: common.UInt8,
        PWMNum: common.UInt16,
        HHGCLKNum: common.UInt8,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipTLC6983RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipTLC6983RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipTLC6983.decompiled.cs:4847
 */
exports.ChipTLC6983RGBVExtendProperty = t.intersection([
    exports.ChipTLC6983RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipTLC6983RGBVExtendProperty') }),
], 'ChipTLC6983RGBVExtendProperty');
