// @novastar/native/build/main/generated/ChipTLC59581ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipTLC59581ExtendProperty = exports.ChipTLC59581ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipTLC59581ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        GlobalBrightness: common.UInt8_4,
        RedGain: common.UInt16_256,
        GreenGain: common.UInt16_256,
        BlueGain: common.UInt16_256,
        SetTD0: common.UInt8_1,
        GamaValue: common.UInt8_255,
        RedNegVoltageCtrl: common.UInt8_3,
        GreenNegVoltageCtrl: common.UInt8_3,
        BlueNegVoltageCtrl: common.UInt8_3,
    }),
    t.partial({
        RedInhibitoryControlEMI: common.Bool,
        GreenInhibitoryControlEMI: common.Bool,
        BlueInhibitoryControlEMI: common.Bool,
        PrechargeModeControl: common.UInt8,
        CaterpillarElimination: common.UInt8,
        EdgeSetGCLK: common.UInt8,
        PWMModelSelecte: common.Bool,
        RedInterference: common.UInt8,
        GreenInterference: common.UInt8,
        BlueInterference: common.UInt8,
        HardwarePar: common.UInt8,
        PowerSavingModeSettings: common.Bool,
        RedLowGrayscaleUniformity: common.UInt8,
        GreenLowGrayscaleUniformity: common.UInt8,
        BlueLowGrayscaleUniformity: common.UInt8,
        HighLightInterference: common.Bool,
    }),
], 'ChipTLC59581ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipTLC59581ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:46466
 */
exports.ChipTLC59581ExtendProperty = t.intersection([
    exports.ChipTLC59581ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipTLC59581ExtendProperty') }),
], 'ChipTLC59581ExtendProperty');
