// @novastar/native/build/main/generated/ChipMBI5253NewBRGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMBI5253NewBRGBVExtendProperty = exports.ChipMBI5253NewBRGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipMBI5253NewBExtendProperty_1 = require("./ChipMBI5253NewBExtendProperty");
const PointDetectParameter_1 = require("./PointDetectParameter");
exports.ChipMBI5253NewBRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_14,
    }),
    t.partial({
        RedProperty: ChipMBI5253NewBExtendProperty_1.ChipMBI5253NewBExtendProperty,
        GreenProperty: ChipMBI5253NewBExtendProperty_1.ChipMBI5253NewBExtendProperty,
        BlueProperty: ChipMBI5253NewBExtendProperty_1.ChipMBI5253NewBExtendProperty,
        VRedProperty: ChipMBI5253NewBExtendProperty_1.ChipMBI5253NewBExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FailureLEDElimination: common.Bool,
        GclkAddedNumer: common.UInt8,
        EnGCLKMutiRate: common.Bool,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        ScanCount: common.UInt8,
        IsAdvancedMode: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        PointDetectParameter: PointDetectParameter_1.PointDetectParameter,
    }),
], 'ChipMBI5253NewBRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5253NewBRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5253NewB.decompiled.cs:1484
 */
exports.ChipMBI5253NewBRGBVExtendProperty = t.intersection([
    exports.ChipMBI5253NewBRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5253NewBRGBVExtendProperty') }),
], 'ChipMBI5253NewBRGBVExtendProperty');
