// @novastar/native/build/main/generated/ChipFM6565ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6565ExtendProperty = exports.ChipFM6565ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipFM6565ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LEN: common.Int32_48,
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        ShadowEliminationLevel: common.UInt8,
        ShadowEliminationTime: common.UInt8,
        LowAshColorCompensation: common.UInt8,
        LowAshFirstLineDarkCompensationTimeFineTuningEn: common.Bool,
        LowAshFirstLineDarkCompensation: common.UInt8,
        LowAshFirstLineDarkCompensationTime: common.UInt8,
        CouplingOptimization1: common.UInt8,
        CouplingOptimization2: common.UInt8,
        CouplingOptimization2_Enable: common.Bool,
        SlowlyOpen: common.UInt8,
        InflectionPointVoltage: common.UInt8,
        CurrentGain: common.UInt16,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
    }),
], 'ChipFM6565ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6565ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6565.decompiled.cs:1100
 */
exports.ChipFM6565ExtendProperty = t.intersection([
    exports.ChipFM6565ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6565ExtendProperty') }),
], 'ChipFM6565ExtendProperty');
