// @novastar/native/build/main/generated/ChipCfgCurrentRisingTime.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCfgCurrentRisingTime = exports.ChipCfgCurrentRisingTimeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ChipCfgCurrentRisingTimeEnum;
(function (ChipCfgCurrentRisingTimeEnum) {
    ChipCfgCurrentRisingTimeEnum[ChipCfgCurrentRisingTimeEnum["CurrentRising1"] = 0] = "CurrentRising1";
    ChipCfgCurrentRisingTimeEnum[ChipCfgCurrentRisingTimeEnum["CurrentRising2"] = 1] = "CurrentRising2";
    ChipCfgCurrentRisingTimeEnum[ChipCfgCurrentRisingTimeEnum["CurrentRising3"] = 2] = "CurrentRising3";
    ChipCfgCurrentRisingTimeEnum[ChipCfgCurrentRisingTimeEnum["CurrentRising4"] = 3] = "CurrentRising4";
})(ChipCfgCurrentRisingTimeEnum = exports.ChipCfgCurrentRisingTimeEnum || (exports.ChipCfgCurrentRisingTimeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ChipCfgCurrentRisingTimeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1174
 */
exports.ChipCfgCurrentRisingTime = (0, EnumFromString_1.default)(ChipCfgCurrentRisingTimeEnum, 'ChipCfgCurrentRisingTime');
