// @novastar/native/build/main/generated/ChipTLC5948AExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipTLC5948AExtendProperty = exports.ChipTLC5948AExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipTLC5948AExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RepeatDisplayMode: common.Bool_true,
        PWMDisplayMode: common.Int32_1,
    }),
    t.partial({
        IsAdvancedMode: common.Bool,
        OpenCheckVoltageRed: common.Int32,
        OpenCheckVoltageGreen: common.Int32,
        OpenCheckVoltageBlue: common.Int32,
        ShortCheckVoltageRed: common.Int32,
        ShortCheckVoltageGreen: common.Int32,
        ShortCheckVoltageBlue: common.Int32,
        HiddenCheck: common.Int32,
        CheckLostGain: common.Int32,
        GlobalBrightRed: common.Int32,
        GlobalBrightGreen: common.Int32,
        GlobalBrightBlue: common.Int32,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
        RedRegValueConfigSecond: common.UInt16,
        GreenRegValueConfigSecond: common.UInt16,
        BlueRegValueConfigSecond: common.UInt16,
        VRedRegValueConfigSecond: common.UInt16,
    }),
], 'ChipTLC5948AExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipTLC5948AExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:43609
 */
exports.ChipTLC5948AExtendProperty = t.intersection([
    exports.ChipTLC5948AExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipTLC5948AExtendProperty') }),
], 'ChipTLC5948AExtendProperty');
