// @novastar/native/build/main/generated/Chip2053ExtendPropety.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip2053ExtendPropety = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip2053ExtendPropety}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:38979
 */
exports.Chip2053ExtendPropety = t.intersection([
    t.type({
        EnhancedModeSwitch: common.Bool_true,
        RLowAshPitsImprove: common.UInt8_1,
        GLowAshPitsImprove: common.UInt8_1,
        BLowAshPitsImprove: common.UInt8_1,
        RColorAdjust: common.UInt8_15,
        GColorAdjust: common.UInt8_15,
        BColorAdjust: common.UInt8_15,
        RVanishingStrength: common.UInt8_31,
        GVanishingStrength: common.withDefault(common.UInt8, 25),
        BVanishingStrength: common.UInt8_23,
        RdeGhost: common.Bool_true,
        GdeGhost: common.Bool_true,
        BdeGhost: common.Bool_true,
        RBlackingEnhance: common.Bool_true,
        GBlackingEnhance: common.Bool_true,
        BBlackingEnhance: common.Bool_true,
        RHeadDarkLineOpt: common.UInt8_1,
        GHeadDarkLineOpt: common.UInt8_1,
        BHeadDarkLineOpt: common.UInt8_1,
        RFirstLineSlantsDarkOptimization: common.UInt8_1,
        GFirstLineSlantsDarkOptimization: common.UInt8_1,
        BFirstLineSlantsDarkOptimization: common.UInt8_1,
        RRemoveBadPointGain: common.UInt16_1,
        GRemoveBadPointGain: common.UInt16_1,
        BRemoveBadPointGain: common.UInt16_1,
    }),
    t.partial({
        EnhancedMode: common.UInt8,
        RAberrationOptimize: common.Bool,
        GAberrationOptimize: common.Bool,
        BAberrationOptimize: common.Bool,
        FailureLEDElimination: common.Bool,
        RColorAdjustEn: common.Bool,
        GColorAdjustEn: common.Bool,
        BColorAdjustEn: common.Bool,
        RedRegValueConfigThird: common.UInt16,
        GreenRegValueConfigThird: common.UInt16,
        BlueRegValueConfigThird: common.UInt16,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigSpecial: common.UInt16,
        GreenRegValueConfigSpecial: common.UInt16,
        BlueRegValueConfigSpecial: common.UInt16,
        VRedRegValueConfigSpecial: common.UInt16,
        IsCustomMode: common.Bool,
        RedRegValueConfigFifth: common.UInt16,
        GreenRegValueConfigFifth: common.UInt16,
        BlueRegValueConfigFifth: common.UInt16,
    }),
], 'Chip2053ExtendPropety');
