// @novastar/native/build/main/generated/ChipAXS6018RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipAXS6018RGBVExtendProperty = exports.ChipAXS6018RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipAXS6018RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        WeightValueConfig: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        FirstRegData: common.buffer_8,
        SecondRegData: common.buffer_8,
        ThirdRegData: common.buffer_8,
        FourthRegData: common.buffer_6,
        FifthRegData: common.buffer_6,
        SixthRegData: common.buffer_6,
        FirstRegisterAddr: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthRegisterAddr: common.Int32,
        FifthRegisterAddr: common.Int32,
        SixthRegisterAddr: common.Int32,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FifthDataLen: common.Int32,
        FifthStartIndex: common.Int32,
        SixthDataLen: common.Int32,
        SixthStartIndex: common.Int32,
        IsUseNewModule: common.Bool,
        FirstLineDark: common.UInt8,
        ChargingOften: common.UInt8,
        LineShadowTime: common.UInt8,
        ShadowChargingMode: common.UInt8,
        AdditionalField: common.UInt8,
        HighGrayGain: common.UInt8,
        MultilevelCurrent: common.Bool,
        BlackLine: common.UInt8,
        DischargeTime: common.UInt8,
        DischargeR: common.Bool,
        DischargeG: common.Bool,
        DischargeB: common.Bool,
        RextR: common.UInt8,
        RextG: common.UInt8,
        RextB: common.UInt8,
        TubeDischargeEnable: common.Bool,
        LineTubeDischargeEnable: common.Bool,
        Contrast: common.UInt8,
        GammaGloablBright: common.UInt8,
        GammaRBright: common.UInt8,
        GammaGBright: common.UInt8,
        GammaBBright: common.UInt8,
        LineCP: common.UInt8,
        CtrlEndPoint: common.UInt8,
    }),
], 'ChipAXS6018RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipAXS6018RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipAXS6018.decompiled.cs:2231
 */
exports.ChipAXS6018RGBVExtendProperty = t.intersection([
    exports.ChipAXS6018RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipAXS6018RGBVExtendProperty') }),
], 'ChipAXS6018RGBVExtendProperty');
