// @novastar/native/build/main/generated/ChipHX8864RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipHX8864RGBVExtendProperty = exports.ChipHX8864RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipHX8864ExtendProperty_1 = require("./ChipHX8864ExtendProperty");
exports.ChipHX8864RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_15,
    }),
    t.partial({
        RedProperty: ChipHX8864ExtendProperty_1.ChipHX8864ExtendProperty,
        GreenProperty: ChipHX8864ExtendProperty_1.ChipHX8864ExtendProperty,
        BlueProperty: ChipHX8864ExtendProperty_1.ChipHX8864ExtendProperty,
        VRedProperty: ChipHX8864ExtendProperty_1.ChipHX8864ExtendProperty,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        IsAdvancedMode: common.Bool,
        GammaTable: common.Bool,
        IsUseNewModule: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        GclkFreqP: common.UInt8,
        GclkFreqN: common.UInt8,
        GclkFreqM: common.UInt8,
        OpenCircuitDetctEnable: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        FifthDataLen: common.Int32,
        FifthStartIndex: common.Int32,
        FifthRegisterAddr: common.Int32,
        SixthDataLen: common.Int32,
        SixthStartIndex: common.Int32,
        SixthRegisterAddr: common.Int32,
        ErrRedGain: common.Int32,
    }),
], 'ChipHX8864RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipHX8864RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipHX8864.decompiled.cs:7816
 */
exports.ChipHX8864RGBVExtendProperty = t.intersection([
    exports.ChipHX8864RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipHX8864RGBVExtendProperty') }),
], 'ChipHX8864RGBVExtendProperty');
