// @novastar/native/build/main/generated/ChipSM16369ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16369ExtendProperty = exports.ChipSM16369ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSM16369ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        ShadowEliminationLevel_1: common.UInt8,
        ShadowEliminationTime_1: common.UInt8,
        ShadowEliminationEnable_2: common.Bool,
        ShadowEliminationLevel_2: common.UInt8,
        ShadowEliminationEnable_3: common.Bool,
        ShadowEliminationLevel_3: common.UInt8,
        FirstLineDarkCompensationLevel: common.UInt8,
        FirstLineDarkCompensationTime: common.UInt8,
        LowGrayCompsention1: common.UInt8,
        LowGrayCompsention2: common.UInt8,
        LowGrayHorizontalStripesImproved: common.UInt8,
        CoupledEnhancementModeEn: common.Bool,
        CouplingOptimization_1: common.UInt8,
        CouplingOptimization_2: common.UInt16,
        EnergySavingMode: common.UInt8,
        GrayScaleUniformity: common.UInt8,
        CrossEliminationLevel: common.UInt8,
    }),
], 'ChipSM16369ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16369ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16369.decompiled.cs:1005
 */
exports.ChipSM16369ExtendProperty = t.intersection([
    exports.ChipSM16369ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16369ExtendProperty') }),
], 'ChipSM16369ExtendProperty');
