// @novastar/native/build/main/generated/ChipFM6518RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6518RGBVExtendProperty = exports.ChipFM6518RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipFM6518ExtendProperty_1 = require("./ChipFM6518ExtendProperty");
exports.ChipFM6518RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipFM6518ExtendProperty_1.ChipFM6518ExtendProperty,
        GreenProperty: ChipFM6518ExtendProperty_1.ChipFM6518ExtendProperty,
        BlueProperty: ChipFM6518ExtendProperty_1.ChipFM6518ExtendProperty,
        VRedProperty: ChipFM6518ExtendProperty_1.ChipFM6518ExtendProperty,
        ScanCount: common.UInt8,
        IsStarSwipPoint: common.Bool,
        IsAdvancedModel: common.Bool,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
    }),
], 'ChipFM6518RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6518RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6518.decompiled.cs:6602
 */
exports.ChipFM6518RGBVExtendProperty = t.intersection([
    exports.ChipFM6518RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6518RGBVExtendProperty') }),
], 'ChipFM6518RGBVExtendProperty');
