// @novastar/native/build/main/generated/ChipFM6182RGBVExtendPropey.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6182RGBVExtendPropey = exports.ChipFM6182RGBVExtendPropeyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipFM6182ExtendProperty_1 = require("./ChipFM6182ExtendProperty");
exports.ChipFM6182RGBVExtendPropeyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipFM6182ExtendProperty_1.ChipFM6182ExtendProperty,
        GreenProperty: ChipFM6182ExtendProperty_1.ChipFM6182ExtendProperty,
        BlueProperty: ChipFM6182ExtendProperty_1.ChipFM6182ExtendProperty,
        VRedProperty: ChipFM6182ExtendProperty_1.ChipFM6182ExtendProperty,
        IsAdvancedMode: common.Bool,
    }),
], 'ChipFM6182RGBVExtendPropeyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6182RGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:42907
 */
exports.ChipFM6182RGBVExtendPropey = t.intersection([
    exports.ChipFM6182RGBVExtendPropeyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6182RGBVExtendPropey') }),
], 'ChipFM6182RGBVExtendPropey');
