// @novastar/native/build/main/generated/Chip5153ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5153ExtendProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5153ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:31144
 */
exports.Chip5153ExtendProperty = t.partial({
    EliminateAfterLow: common.UInt8,
    FirstSweepEliminationMaster: common.UInt8,
    FirstSweepEliminationMedium: common.Bool,
    FirstSweepEliminationSecond: common.UInt8,
    LowAshColorCompensationOne: common.UInt8,
    LowAshColorCompensationTwo: common.UInt8,
    LowAshEnhanceUniformity: common.Bool,
    GrayHorizontalStripesEliminate: common.Bool,
    EnhancedModeOne: common.Bool,
    SlowOpen: common.Bool,
    SwapErrorPoint: common.UInt8,
    EnhancedModeTwo: common.Bool,
    RedRegValueConfigThird: common.UInt16,
    GreenRegValueConfigThird: common.UInt16,
    BlueRegValueConfigThird: common.UInt16,
    VRedRegValueConfigThird: common.UInt16,
}, 'Chip5153ExtendProperty');
