// @novastar/native/build/main/generated/ChipHX8864ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipHX8864ExtendProperty = exports.ChipHX8864ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipHX8864ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ThirdRegValue: common.UInt16,
        FourthRegValue: common.UInt16,
        FifthRegValue: common.UInt16,
        SixthRegValue: common.UInt16,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        LowGreyOrderOptModel: common.UInt8,
        LowGreyOrderOpt2: common.UInt8,
        CouplingStartTime: common.UInt8,
        CurrentGain: common.UInt16,
        ErrorCurrentGain: common.UInt16,
        InflectionPointVoltage: common.UInt8,
        FirstLineCompensationTime: common.UInt8,
        ShadowEliminationTime: common.UInt8,
        FirstLineCompensationLevel: common.UInt8,
        ShadowEliminationLevel: common.UInt8,
        LowGrayColorCompensation: common.UInt8,
        CouplingOptimization1: common.UInt8,
        CouplingOptimization2Enable: common.Bool,
        CouplingOptimization2: common.UInt8,
        CouplingReguEnable: common.Bool,
        CouplingEnhan: common.UInt8,
    }),
], 'ChipHX8864ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipHX8864ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipHX8864.decompiled.cs:7537
 */
exports.ChipHX8864ExtendProperty = t.intersection([
    exports.ChipHX8864ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipHX8864ExtendProperty') }),
], 'ChipHX8864ExtendProperty');
