// @novastar/native/build/main/generated/ChipSM16227ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16227ExtendProperty = exports.ChipSM16227ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSM16227ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RUnderEliminateGhostOne: common.Int32,
        GUnderEliminateGhostOne: common.Int32,
        BUnderEliminateGhostOne: common.Int32,
        RUnderEliminateGhostTwo: common.Int32,
        GUnderEliminateGhostTwo: common.Int32,
        BUnderEliminateGhostTwo: common.Int32,
        ROffsetCompsation: common.Int32,
        GOffsetCompsation: common.Int32,
        BOffsetCompsation: common.Int32,
        RFirstCompsation: common.Int32,
        GFirstCompsation: common.Int32,
        BFirstCompsation: common.Int32,
        IsAdvancedMode: common.Bool,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
    }),
], 'ChipSM16227ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16227ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:43987
 */
exports.ChipSM16227ExtendProperty = t.intersection([
    exports.ChipSM16227ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16227ExtendProperty') }),
], 'ChipSM16227ExtendProperty');
