// @novastar/native/build/main/generated/ChipDP3264ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipDP3264ExtendProperty = exports.ChipDP3264ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipDP3264ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        ConstantCurrentOutputInflectionPointGrade: common.UInt8,
        LowGreyPittingOptimization: common.UInt8,
        GammaSmoothingLevel: common.UInt8,
        ShadowEliminationLevelEnable: common.Bool,
        ShadowEliminationLevel: common.UInt8,
        ShadowEliminationTime: common.UInt8,
        ShadowEliminationMoment: common.UInt8,
        CouplingOptimizationEnable: common.Bool,
        CouplingOptimizationEnhance: common.Bool,
        CouplingOptimizationGrade: common.UInt8,
        CouplingCoefficient: common.UInt8,
        CurrentGain: common.UInt16,
    }),
], 'ChipDP3264ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipDP3264ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipDP3264.decompiled.cs:6833
 */
exports.ChipDP3264ExtendProperty = t.intersection([
    exports.ChipDP3264ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipDP3264ExtendProperty') }),
], 'ChipDP3264ExtendProperty');
