// @novastar/native/build/main/generated/ChipMY9366ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMY9366ExtendProperty = exports.ChipMY9366ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipMY9366ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RIsClearGhost: common.Bool_true,
        GIsClearGhost: common.Bool_true,
        BIsClearGhost: common.Bool_true,
        VRIsClearGhost: common.Bool_true,
    }),
    t.partial({
        RFirstScancompensation: common.UInt8,
        GFirstScancompensation: common.UInt8,
        BFirstScancompensation: common.UInt8,
        VRFirstScancompensation: common.UInt8,
        RSmallAdjust: common.UInt8,
        GSmallAdjust: common.UInt8,
        BSmallAdjust: common.UInt8,
        VRSmallAdjust: common.UInt8,
        RCurrentCompensation: common.Bool,
        GCurrentCompensation: common.Bool,
        BCurrentCompensation: common.Bool,
        VRCurrentCompensation: common.Bool,
        RLowGrayValue: common.UInt8,
        GLowGrayValue: common.UInt8,
        BLowGrayValue: common.UInt8,
        VRLowGrayValue: common.UInt8,
    }),
], 'ChipMY9366ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMY9366ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:48073
 */
exports.ChipMY9366ExtendProperty = t.intersection([
    exports.ChipMY9366ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMY9366ExtendProperty') }),
], 'ChipMY9366ExtendProperty');
