// @novastar/native/build/main/generated/ChipICN2065AutoRefreshRataParam.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipICN2065AutoRefreshRataParam = exports.ChipICN2065AutoRefreshRataParamBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const AutoAdjustFreqConfigMode_1 = require("./AutoAdjustFreqConfigMode");
const ChipAutoAdjustRefreshRateParamBase_1 = require("./ChipAutoAdjustRefreshRateParamBase");
const ChipType_1 = require("./ChipType");
exports.ChipICN2065AutoRefreshRataParamBase = t.intersection([
    ChipAutoAdjustRefreshRateParamBase_1.ChipAutoAdjustRefreshRateParamBase,
    t.partial({
        ChipType: ChipType_1.ChipType,
        ChipLibVersion: common.UInt8,
        ConfigMode: AutoAdjustFreqConfigMode_1.AutoAdjustFreqConfigMode,
        AutoAdjustParamAddress: common.UInt32,
        AutoAdjustParamDataLen: common.Int32,
    }),
], 'ChipICN2065AutoRefreshRataParamBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICN2065AutoRefreshRataParam}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICND2065.decompiled.cs:47
 */
exports.ChipICN2065AutoRefreshRataParam = t.intersection([
    exports.ChipICN2065AutoRefreshRataParamBase,
    t.partial({ '@_xsi:type': t.literal('ChipICN2065AutoRefreshRataParam') }),
], 'ChipICN2065AutoRefreshRataParam');
