// @novastar/native/build/main/generated/Chip9348ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip9348ExtendProperty = exports.Chip9348ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.Chip9348ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        CurrentSet1: common.UInt8,
        CurrentSet2: common.UInt8,
        CurrentGain: common.UInt8,
        LowGrayscaleCompensation: common.UInt8,
        FirstScanCompensation: common.UInt8,
        CompensationMode: common.UInt8,
        CurrentSet3: common.UInt8,
        NextGhostAbatement: common.UInt8,
        Adjustment: common.UInt8,
    }),
], 'Chip9348ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip9348ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:39711
 */
exports.Chip9348ExtendProperty = t.intersection([exports.Chip9348ExtendPropertyBase, t.partial({ '@_xsi:type': t.literal('Chip9348ExtendProperty') })], 'Chip9348ExtendProperty');
