// @novastar/native/build/main/generated/ChipCS2066RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCS2066RGBVExtendProperty = exports.ChipCS2066RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipCS2066ExtendProperty_1 = require("./ChipCS2066ExtendProperty");
exports.ChipCS2066RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipCS2066ExtendProperty_1.ChipCS2066ExtendProperty,
        GreenProperty: ChipCS2066ExtendProperty_1.ChipCS2066ExtendProperty,
        BlueProperty: ChipCS2066ExtendProperty_1.ChipCS2066ExtendProperty,
        VRedProperty: ChipCS2066ExtendProperty_1.ChipCS2066ExtendProperty,
        SubField: common.UInt8,
        GrayDepth: common.UInt8,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        PLLFreqDivision: common.UInt8,
        PLLFreqDoubling: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        FifthDataLen: common.Int32,
        FifthStartIndex: common.Int32,
        FifthRegisterAddr: common.Int32,
        SixthDataLen: common.Int32,
        SixthStartIndex: common.Int32,
        SixthRegisterAddr: common.Int32,
        SeventhDataLen: common.Int32,
        SeventhStartIndex: common.Int32,
        SeventhRegisterAddr: common.Int32,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipCS2066RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2066RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2066.decompiled.cs:8337
 */
exports.ChipCS2066RGBVExtendProperty = t.intersection([
    exports.ChipCS2066RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2066RGBVExtendProperty') }),
], 'ChipCS2066RGBVExtendProperty');
