// @novastar/native/build/main/generated/ChipIT1505ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipIT1505ExtendProperty = exports.ChipIT1505ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipIT1505ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RedRegValueConfigFirst: common.UInt16_60543,
        GreenRegValueConfigFirst: common.UInt16_60543,
        BlueRegValueConfigFirst: common.UInt16_60543,
        VRedRegValueConfigFirst: common.UInt16_60543,
    }),
    t.partial({
        IsAdvancedMode: common.Bool,
        CureentRedGain: common.UInt16,
        CureentGreenGain: common.UInt16,
        CureentBlueGain: common.UInt16,
        RedShortCirult: common.Int32,
        GreenShortCirult: common.Int32,
        BlueShortCirult: common.Int32,
        PowerSavingMode: common.Int32,
        PWMAlgorithm: common.Int32,
    }),
], 'ChipIT1505ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipIT1505ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:50395
 */
exports.ChipIT1505ExtendProperty = t.intersection([
    exports.ChipIT1505ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipIT1505ExtendProperty') }),
], 'ChipIT1505ExtendProperty');
