// @novastar/native/build/main/generated/ChipSUM2033ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSUM2033ExtendProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipSUM2033BaseExtendProperty_1 = require("./ChipSUM2033BaseExtendProperty");
/**
 * @category Codecs
 * @desc Codec for interface {@link ChipSUM2033ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:49514
 */
exports.ChipSUM2033ExtendProperty = t.intersection([
    t.type({
        RedVanishingRating: common.UInt8_15,
        GreenVanishingRating: common.withDefault(common.UInt8, 13),
        BlueVanishingRating: common.UInt8_5,
        VRedVanishingRating: common.UInt8_15,
        RedRegValueConfigFourth: common.UInt16_7168,
        GreenRegValueConfigFourth: common.UInt16_7168,
        BlueRegValueConfigFourth: common.UInt16_7168,
        VRedRegValueConfigFourth: common.UInt16_7168,
    }),
    t.partial({
        RedProperty: ChipSUM2033BaseExtendProperty_1.ChipSUM2033BaseExtendProperty,
        GreenProperty: ChipSUM2033BaseExtendProperty_1.ChipSUM2033BaseExtendProperty,
        BlueProperty: ChipSUM2033BaseExtendProperty_1.ChipSUM2033BaseExtendProperty,
        VRedProperty: ChipSUM2033BaseExtendProperty_1.ChipSUM2033BaseExtendProperty,
        UsingBright: common.Bool,
        UsingSUM2033Gamma: common.Bool,
        IsDoubleRate: common.Bool,
        RedRegValueConfigThird: common.UInt16,
        GreenRegValueConfigThird: common.UInt16,
        BlueRegValueConfigThird: common.UInt16,
        VRedRegValueConfigThird: common.UInt16,
    }),
], 'ChipSUM2033ExtendProperty');
