// @novastar/native/build/main/generated/ChipSM16389RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16389RGBVExtendProperty = exports.ChipSM16389RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipSM16389ExtendProperty_1 = require("./ChipSM16389ExtendProperty");
exports.ChipSM16389RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_34,
    }),
    t.partial({
        RedProperty: ChipSM16389ExtendProperty_1.ChipSM16389ExtendProperty,
        GreenProperty: ChipSM16389ExtendProperty_1.ChipSM16389ExtendProperty,
        BlueProperty: ChipSM16389ExtendProperty_1.ChipSM16389ExtendProperty,
        VRedProperty: ChipSM16389ExtendProperty_1.ChipSM16389ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        IsAdvancedMode: common.Bool,
        TestGain: common.UInt8,
        IsErrOpen: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        LittlePeriodCount: common.UInt8,
        RefreshMagnificationP1: common.UInt8,
        RefreshMagnificationP2: common.UInt8,
        RefreshMagnificationP3: common.UInt8,
        LineScanGrayScale: common.Int32,
    }),
], 'ChipSM16389RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16389RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16389.decompiled.cs:1321
 */
exports.ChipSM16389RGBVExtendProperty = t.intersection([
    exports.ChipSM16389RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16389RGBVExtendProperty') }),
], 'ChipSM16389RGBVExtendProperty');
