// @novastar/native/build/main/generated/ChipRT5965RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipRT5965RGBVExtendProperty = exports.ChipRT5965RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipRT5965ExtendProperty_1 = require("./ChipRT5965ExtendProperty");
exports.ChipRT5965RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipRT5965ExtendProperty_1.ChipRT5965ExtendProperty,
        GreenProperty: ChipRT5965ExtendProperty_1.ChipRT5965ExtendProperty,
        BlueProperty: ChipRT5965ExtendProperty_1.ChipRT5965ExtendProperty,
        VRedProperty: ChipRT5965ExtendProperty_1.ChipRT5965ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        ScanCount: common.UInt8,
        SubField: common.UInt8,
        BitMode: common.Bool,
        OpenCircuitDetectionEn: common.Bool,
        IsAdvancedMode: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        SecondDataLen: common.Int32,
        SecondStartIndex: common.Int32,
        SecondRegisterAddr: common.Int32,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
    }),
], 'ChipRT5965RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipRT5965RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipRT5965.decompiled.cs:1202
 */
exports.ChipRT5965RGBVExtendProperty = t.intersection([
    exports.ChipRT5965RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipRT5965RGBVExtendProperty') }),
], 'ChipRT5965RGBVExtendProperty');
