// @novastar/native/build/main/generated/ChipRT5938SSExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipRT5938SSExtendProperty = exports.ChipRT5938SSExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipRT5938SSExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        EliminationLevel: common.UInt8,
        EliminateEnable: common.Bool,
        OpenCircuitLevel: common.UInt8,
        ShortCircuitLevel: common.UInt8,
        SlowOpen: common.Bool,
        PowerSavingModeEn: common.Bool,
        CurrentGain: common.UInt16,
    }),
], 'ChipRT5938SSExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipRT5938SSExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipRT5938SS.decompiled.cs:6143
 */
exports.ChipRT5938SSExtendProperty = t.intersection([
    exports.ChipRT5938SSExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipRT5938SSExtendProperty') }),
], 'ChipRT5938SSExtendProperty');
