// @novastar/native/build/main/generated/ChipICND2055RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipICND2055RGBVExtendProperty = exports.ChipICND2055RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipICND2055ExtendProperty_1 = require("./ChipICND2055ExtendProperty");
exports.ChipICND2055RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipICND2055ExtendProperty_1.ChipICND2055ExtendProperty,
        GreenProperty: ChipICND2055ExtendProperty_1.ChipICND2055ExtendProperty,
        BlueProperty: ChipICND2055ExtendProperty_1.ChipICND2055ExtendProperty,
        VRedProperty: ChipICND2055ExtendProperty_1.ChipICND2055ExtendProperty,
    }),
], 'ChipICND2055RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICND2055RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:33515
 */
exports.ChipICND2055RGBVExtendProperty = t.intersection([
    exports.ChipICND2055RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICND2055RGBVExtendProperty') }),
], 'ChipICND2055RGBVExtendProperty');
