// @novastar/native/build/main/generated/ChipCS2017RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCS2017RGBVExtendProperty = exports.ChipCS2017RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipCS2017ExtendProperty_1 = require("./ChipCS2017ExtendProperty");
exports.ChipCS2017RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipCS2017ExtendProperty_1.ChipCS2017ExtendProperty,
        GreenProperty: ChipCS2017ExtendProperty_1.ChipCS2017ExtendProperty,
        BlueProperty: ChipCS2017ExtendProperty_1.ChipCS2017ExtendProperty,
        VRedProperty: ChipCS2017ExtendProperty_1.ChipCS2017ExtendProperty,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        ConfigDataLen: common.Int32,
    }),
], 'ChipCS2017RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2017RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2017.decompiled.cs:216
 */
exports.ChipCS2017RGBVExtendProperty = t.intersection([
    exports.ChipCS2017RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2017RGBVExtendProperty') }),
], 'ChipCS2017RGBVExtendProperty');
