// @novastar/native/build/main/generated/ChipMBI5759ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMBI5759ExtendProperty = exports.ChipMBI5759ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipMBI5759CustomExtendProperty_1 = require("./ChipMBI5759CustomExtendProperty");
const ChipMBI5759GRGBExtendProperty_1 = require("./ChipMBI5759GRGBExtendProperty");
exports.ChipMBI5759ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        GRGBExtendProperty: ChipMBI5759GRGBExtendProperty_1.ChipMBI5759GRGBExtendProperty,
        CustomExtendProperty: ChipMBI5759CustomExtendProperty_1.ChipMBI5759CustomExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipMBI5759ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5759ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5759.decompiled.cs:1509
 */
exports.ChipMBI5759ExtendProperty = t.intersection([
    exports.ChipMBI5759ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5759ExtendProperty') }),
], 'ChipMBI5759ExtendProperty');
