// @novastar/native/build/main/generated/ChipICND2112RGBVExtendPropey.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipICND2112RGBVExtendPropey = exports.ChipICND2112RGBVExtendPropeyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipICND2112ExtendProperty_1 = require("./ChipICND2112ExtendProperty");
exports.ChipICND2112RGBVExtendPropeyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipICND2112ExtendProperty_1.ChipICND2112ExtendProperty,
        GreenProperty: ChipICND2112ExtendProperty_1.ChipICND2112ExtendProperty,
        BlueProperty: ChipICND2112ExtendProperty_1.ChipICND2112ExtendProperty,
        VRedProperty: ChipICND2112ExtendProperty_1.ChipICND2112ExtendProperty,
        IsAdvancedMode: common.Bool,
        MaxUnitChipPerScan: common.Int32,
    }),
], 'ChipICND2112RGBVExtendPropeyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICND2112RGBVExtendPropey}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipICND2112.decompiled.cs:169
 */
exports.ChipICND2112RGBVExtendPropey = t.intersection([
    exports.ChipICND2112RGBVExtendPropeyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICND2112RGBVExtendPropey') }),
], 'ChipICND2112RGBVExtendPropey');
