// @novastar/native/build/main/generated/ChipCS2066ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCS2066ExtendProperty = exports.ChipCS2066ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipCS2066ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ThirdRegValue: common.UInt16,
        FourthRegValue: common.UInt16,
        FifthRegValue: common.UInt16,
        SixthRegValue: common.UInt16,
        SeventhRegValue: common.UInt16,
        RefreshRate: common.UInt8,
        ScanNumSetting: common.UInt8,
        CurrentGain: common.UInt8,
        LowAshEnhanceUniformity: common.UInt8,
        LineBlankStrength: common.UInt8,
        OutputResponse: common.UInt8,
        PerDriverCycleNumSetting: common.UInt8,
        LineBlankEnhanceModeEn: common.Bool,
        CoupledOutputShift: common.UInt8,
        OpenTestPointAdjust: common.UInt8,
        CouplingOptimizationEn: common.Bool,
        LowGrayCompsentionTimeSettingMode: common.UInt8,
        CouplingRangeAdjust: common.UInt8,
        PerDriverEnable: common.Bool,
        PWMGraySetting: common.UInt8,
        OpenProtectEnable: common.Bool,
        ScanMode: common.UInt8,
        LowGrayPulseWidthCompensation: common.UInt8,
        PLLFreqDivision: common.UInt8,
        PLLFreqDoubling: common.UInt8,
        CurrentGear: common.UInt8,
        LowGrayCompsentionStrength: common.UInt8,
        OpenDynamicTestEnable: common.Bool,
        CurrentCompsentionEnable: common.Bool,
        ConstantCurrentInflectionPoint: common.UInt8,
        DynamicEnergySavingEn: common.Bool,
        LineEliminationLength: common.UInt8,
        PerDriverMode: common.UInt8,
        PerDriverTimeSetting: common.UInt8,
        OpentestThresholdValue: common.UInt8,
        CurrentCompsentionGear: common.UInt8,
        WakeUpSettings: common.UInt8,
    }),
], 'ChipCS2066ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCS2066ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCS2066.decompiled.cs:7863
 */
exports.ChipCS2066ExtendProperty = t.intersection([
    exports.ChipCS2066ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCS2066ExtendProperty') }),
], 'ChipCS2066ExtendProperty');
