// @novastar/native/build/main/generated/ChipHS3257RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipHS3257RGBVExtendProperty = exports.ChipHS3257RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipHS3257ExtendProperty_1 = require("./ChipHS3257ExtendProperty");
exports.ChipHS3257RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        ModeSelect: common.UInt8_0,
    }),
    t.partial({
        RedProperty: ChipHS3257ExtendProperty_1.ChipHS3257ExtendProperty,
        GreenProperty: ChipHS3257ExtendProperty_1.ChipHS3257ExtendProperty,
        BlueProperty: ChipHS3257ExtendProperty_1.ChipHS3257ExtendProperty,
        VRedProperty: ChipHS3257ExtendProperty_1.ChipHS3257ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        IsAdvancedMode: common.Bool,
        ThirdDataLen: common.Int32,
        ThirdDataStartIndex: common.Int32,
        ThirdRegisterAddr: common.Int32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
    }),
], 'ChipHS3257RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipHS3257RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipHS3257.decompiled.cs:995
 */
exports.ChipHS3257RGBVExtendProperty = t.intersection([
    exports.ChipHS3257RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipHS3257RGBVExtendProperty') }),
], 'ChipHS3257RGBVExtendProperty');
