// @novastar/native/build/main/generated/ChipRT5938SSRGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipRT5938SSRGBVExtendProperty = exports.ChipRT5938SSRGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipRT5938SSExtendProperty_1 = require("./ChipRT5938SSExtendProperty");
const PointDetectParameter_1 = require("./PointDetectParameter");
exports.ChipRT5938SSRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_1,
    }),
    t.partial({
        RedProperty: ChipRT5938SSExtendProperty_1.ChipRT5938SSExtendProperty,
        GreenProperty: ChipRT5938SSExtendProperty_1.ChipRT5938SSExtendProperty,
        BlueProperty: ChipRT5938SSExtendProperty_1.ChipRT5938SSExtendProperty,
        VRedProperty: ChipRT5938SSExtendProperty_1.ChipRT5938SSExtendProperty,
        IsUseNewModule: common.Bool,
        IsAdvancedMode: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        PointDetectParameter: PointDetectParameter_1.PointDetectParameter,
    }),
], 'ChipRT5938SSRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipRT5938SSRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipRT5938SS.decompiled.cs:6754
 */
exports.ChipRT5938SSRGBVExtendProperty = t.intersection([
    exports.ChipRT5938SSRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipRT5938SSRGBVExtendProperty') }),
], 'ChipRT5938SSRGBVExtendProperty');
