// @novastar/native/build/main/generated/ChipSM16017SExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16017SExtendProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
/**
 * @category Codecs
 * @desc Codec for interface {@link ChipSM16017SExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:43154
 */
exports.ChipSM16017SExtendProperty = t.intersection([
    t.type({
        REliminateAfterLow: common.UInt8_5,
        GEliminateAfterLow: common.UInt8_5,
        BEliminateAfterLow: common.UInt8_5,
        GDimCompensation: common.UInt8_3,
        BDimCompensation: common.UInt8_3,
        RLowAshImprovement: common.withDefault(common.UInt8, 6),
        GLowAshImprovement: common.UInt8_7,
        BLowAshImprovement: common.UInt8_7,
        RedRegValueConfigFirst: common.UInt16_41728,
        GreenRegValueConfigFirst: common.UInt16_44932,
        BlueRegValueConfigFirst: common.UInt16_44932,
        VRedRegValueConfigFirst: common.UInt16_41728,
    }),
    t.partial({
        RDimCompensation: common.UInt8,
        IsAdvancedMode: common.Bool,
    }),
], 'ChipSM16017SExtendProperty');
