// @novastar/native/build/main/generated/Chip5253BRGBExtendPropety.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5253BRGBExtendPropety = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const Chip5253BExtendProperty_1 = require("./Chip5253BExtendProperty");
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5253BRGBExtendPropety}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:31904
 */
exports.Chip5253BRGBExtendPropety = t.intersection([
    t.type({
        EnableLowImprovement: common.Bool_true,
        EnShadeTransitOptimize: common.Bool_true,
        EnGCLKMutiRate: common.Bool_true,
        RedRegValueConfigThird: common.withDefault(common.UInt16, 49251),
        GreenRegValueConfigThird: common.withDefault(common.UInt16, 28771),
        BlueRegValueConfigThird: common.withDefault(common.UInt16, 24675),
        RedRegValueConfigFour: common.UInt16_61440,
        GreenRegValueConfigFour: common.UInt16_61440,
        BlueRegValueConfigFour: common.UInt16_61440,
        VRedRegValueConfigFour: common.UInt16_61440,
        RedRegValueConfigFive: common.UInt16_8673,
        GreenRegValueConfigFive: common.UInt16_8673,
        BlueRegValueConfigFive: common.UInt16_8673,
        VRedRegValueConfigFive: common.UInt16_8673,
        RedRegValueConfigSix: common.UInt16_3822,
        GreenRegValueConfigSix: common.UInt16_3822,
        BlueRegValueConfigSix: common.UInt16_3822,
        VRedRegValueConfigSix: common.UInt16_3822,
    }),
    t.partial({
        RedProperty: Chip5253BExtendProperty_1.Chip5253BExtendProperty,
        GreenProperty: Chip5253BExtendProperty_1.Chip5253BExtendProperty,
        BlueProperty: Chip5253BExtendProperty_1.Chip5253BExtendProperty,
        VRedProperty: Chip5253BExtendProperty_1.Chip5253BExtendProperty,
        EnDoubleRefresh: common.Bool,
        FailureLEDElimination: common.Bool,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigSeven: common.UInt16,
        GreenRegValueConfigSeven: common.UInt16,
        BlueRegValueConfigSeven: common.UInt16,
        VRedRegValueConfigSeven: common.UInt16,
        RedRegValueConfigSpecial: common.UInt16,
        GreenRegValueConfigSpecial: common.UInt16,
        BlueRegValueConfigSpecial: common.UInt16,
        VRedRegValueConfigSpecial: common.UInt16,
    }),
], 'Chip5253BRGBExtendPropety');
