// @novastar/native/build/main/generated/ChipICN2065RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipICN2065RGBVExtendProperty = exports.ChipICN2065RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipICN2065ExtendProperty_1 = require("./ChipICN2065ExtendProperty");
exports.ChipICN2065RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipICN2065ExtendProperty_1.ChipICN2065ExtendProperty,
        GreenProperty: ChipICN2065ExtendProperty_1.ChipICN2065ExtendProperty,
        BlueProperty: ChipICN2065ExtendProperty_1.ChipICN2065ExtendProperty,
        VRedProperty: ChipICN2065ExtendProperty_1.ChipICN2065ExtendProperty,
    }),
], 'ChipICN2065RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipICN2065RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:36288
 */
exports.ChipICN2065RGBVExtendProperty = t.intersection([
    exports.ChipICN2065RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipICN2065RGBVExtendProperty') }),
], 'ChipICN2065RGBVExtendProperty');
