// @novastar/native/build/main/generated/ChipMY9221ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMY9221ExtendProperty = exports.ChipMY9221ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const DriverCurrentSpeedType_1 = require("./DriverCurrentSpeedType");
exports.ChipMY9221ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        DriverSpeed: common.withDefault(DriverCurrentSpeedType_1.DriverCurrentSpeedType, 'Fast'),
        IsCurrentBreakUp: common.Bool_true,
        IsInternalOscillator: common.Bool_true,
        IsDriverLED: common.Bool_true,
        IsAutoRefreshPicture: common.Bool_true,
        IsPictrueRepeat: common.Bool_true,
    }),
    t.partial({}),
], 'ChipMY9221ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMY9221ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:43403
 */
exports.ChipMY9221ExtendProperty = t.intersection([
    exports.ChipMY9221ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMY9221ExtendProperty') }),
], 'ChipMY9221ExtendProperty');
