// @novastar/native/build/main/generated/ChipCFD455AExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipCFD455AExtendProperty = exports.ChipCFD455AExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipCFD455AExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ThirdRegValue: common.UInt16,
        FourthRegValue: common.UInt16,
        LowAshEnhanceUniformity: common.UInt8,
        LineBlankStrength: common.UInt8,
        LineBlankEnhanceModeEn: common.Bool,
        CouplingStrengthAdjust: common.UInt8,
        OpenTestPointAdjust: common.UInt8,
        CouplingOptimizationEn: common.Bool,
        LowGrayCompsentionTimeSettingMode: common.UInt8,
        CouplingRangeAdjust: common.UInt8,
        PerDriverEnable: common.Bool,
        OpenProtectEnable: common.Bool,
        ScanMode: common.UInt8,
        CurrentGear: common.UInt8,
        LowGrayCompsentionStrength: common.UInt8,
        OpenDynamicTestEnable: common.Bool,
        CurrentCompsentionEnable: common.Bool,
        ConstantCurrentInflectionPoint: common.UInt8,
        DynamicEnergySavingEn: common.Bool,
        LineEliminationLength: common.UInt8,
        PerDriverMode: common.UInt8,
        PerDriverTimeSetting: common.UInt8,
        OpentestThresholdValue: common.UInt8,
        LowGrayGoodLevel: common.UInt8,
        SetWeakup: common.UInt8,
    }),
], 'ChipCFD455AExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipCFD455AExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipCFD455A.decompiled.cs:3371
 */
exports.ChipCFD455AExtendProperty = t.intersection([
    exports.ChipCFD455AExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipCFD455AExtendProperty') }),
], 'ChipCFD455AExtendProperty');
