// @novastar/native/build/main/generated/ChipSM16380ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16380ExtendProperty = exports.ChipSM16380ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSM16380ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        FirstRegValue: common.UInt16,
        SecondRegValue: common.UInt16,
        ShadowEliminationLevel: common.UInt8,
        ShadowEliminationTime: common.UInt8,
        ShadowEliminationEnable: common.Bool,
        FirstLineDarkCompensationLevel: common.UInt8,
        LowGrayCompsention: common.UInt8,
        CrossEliminationLevel: common.UInt8,
        GrayScaleUniformity: common.UInt8,
        CouplingOptimization: common.Int32,
        EnergySavingMode: common.UInt8,
        EnergySavingEnable: common.Bool,
    }),
], 'ChipSM16380ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16380ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16380.decompiled.cs:1028
 */
exports.ChipSM16380ExtendProperty = t.intersection([
    exports.ChipSM16380ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16380ExtendProperty') }),
], 'ChipSM16380ExtendProperty');
