// @novastar/native/build/main/generated/ChipMBI5251RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMBI5251RGBVExtendProperty = exports.ChipMBI5251RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipMBI5251ExtendProperty_1 = require("./ChipMBI5251ExtendProperty");
exports.ChipMBI5251RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipMBI5251ExtendProperty_1.ChipMBI5251ExtendProperty,
        GreenProperty: ChipMBI5251ExtendProperty_1.ChipMBI5251ExtendProperty,
        BlueProperty: ChipMBI5251ExtendProperty_1.ChipMBI5251ExtendProperty,
        VRedProperty: ChipMBI5251ExtendProperty_1.ChipMBI5251ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        IsAdvancedMode: common.Bool,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
        RedRegValueConfigSecond: common.UInt16,
        GreenRegValueConfigSecond: common.UInt16,
        BlueRegValueConfigSecond: common.UInt16,
        VRedRegValueConfigSecond: common.UInt16,
        RedRegValueConfigThird: common.UInt16,
        GreenRegValueConfigThird: common.UInt16,
        BlueRegValueConfigThird: common.UInt16,
        VRedRegValueConfigThird: common.UInt16,
        RedRegValueConfigFour: common.UInt16,
        GreenRegValueConfigFour: common.UInt16,
        BlueRegValueConfigFour: common.UInt16,
        VRedRegValueConfigFour: common.UInt16,
        RedRegValueConfigFive: common.UInt16,
        GreenRegValueConfigFive: common.UInt16,
        BlueRegValueConfigFive: common.UInt16,
        VRedRegValueConfigFive: common.UInt16,
        RedRegValueConfigSix: common.UInt16,
        GreenRegValueConfigSix: common.UInt16,
        BlueRegValueConfigSix: common.UInt16,
        VRedRegValueConfigSix: common.UInt16,
    }),
], 'ChipMBI5251RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5251RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5251.decompiled.cs:1238
 */
exports.ChipMBI5251RGBVExtendProperty = t.intersection([
    exports.ChipMBI5251RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5251RGBVExtendProperty') }),
], 'ChipMBI5251RGBVExtendProperty');
