// @novastar/native/build/main/generated/Chip5354ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5354ExtendProperty = exports.Chip5354ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.Chip5354ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        DeGhostLevel: common.UInt8,
        FirstSweepEliminateFirst: common.UInt8,
        FirstSweepEliminateTwo: common.UInt8,
        LowAshImprovementFirst: common.UInt8,
        LowAshImprovementTwo: common.UInt8,
        LowAshImprovementThrid: common.Bool,
        EnhancedMode: common.Bool,
        LowAshdimLineConpensentionY: common.Bool,
        OpenDetectPower: common.UInt8,
    }),
], 'Chip5354ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link Chip5354ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:41297
 */
exports.Chip5354ExtendProperty = t.intersection([exports.Chip5354ExtendPropertyBase, t.partial({ '@_xsi:type': t.literal('Chip5354ExtendProperty') })], 'Chip5354ExtendProperty');
