// @novastar/native/build/main/generated/ChipTL3103UDRGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipTL3103UDRGBVExtendProperty = exports.ChipTL3103UDRGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipTL3103UDExtendProperty_1 = require("./ChipTL3103UDExtendProperty");
exports.ChipTL3103UDRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_11,
    }),
    t.partial({
        RedProperty: ChipTL3103UDExtendProperty_1.ChipTL3103UDExtendProperty,
        GreenProperty: ChipTL3103UDExtendProperty_1.ChipTL3103UDExtendProperty,
        BlueProperty: ChipTL3103UDExtendProperty_1.ChipTL3103UDExtendProperty,
        VRedProperty: ChipTL3103UDExtendProperty_1.ChipTL3103UDExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        GrayDepth: common.UInt8,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipTL3103UDRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipTL3103UDRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipTL3103UD.decompiled.cs:1057
 */
exports.ChipTL3103UDRGBVExtendProperty = t.intersection([
    exports.ChipTL3103UDRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipTL3103UDRGBVExtendProperty') }),
], 'ChipTL3103UDRGBVExtendProperty');
