// @novastar/native/build/main/generated/ChipSM16389ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16389ExtendProperty = exports.ChipSM16389ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSM16389ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_34,
        RegisterArray: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        EnergyPattern: common.UInt8,
        LowGrayHomogeneity: common.UInt8,
        CouplingOptimize2: common.Int32,
        BlankingTime1: common.UInt8,
        FirstLineCompensation: common.UInt8,
        LowGrayOptimize2: common.UInt8,
        OpenTest: common.Bool,
        EliminateOpenCrossGrade: common.UInt8,
        CouplingEnhancePattern: common.Bool,
        LowGrayOptimize1: common.UInt8,
        LowGrayAcrossStripeOptimize: common.UInt8,
        CouplingGrade: common.UInt8,
        Gain: common.Int32,
        BlankingGrade1: common.UInt8,
        FirstLineDarkCompensationGrade: common.UInt8,
        BlankingOpen2: common.Bool,
        BlankingGrade2: common.UInt8,
        BlankingOpen3: common.Bool,
        BlankingGrade3: common.UInt8,
        CouplingOptimize1: common.UInt8,
    }),
], 'ChipSM16389ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16389ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16389.decompiled.cs:1029
 */
exports.ChipSM16389ExtendProperty = t.intersection([
    exports.ChipSM16389ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16389ExtendProperty') }),
], 'ChipSM16389ExtendProperty');
