// @novastar/native/build/main/generated/ChipRT5924ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipRT5924ExtendProperty = exports.ChipRT5924ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipRT5924ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RLowAshRange: common.UInt8_4,
        GLowAshRange: common.UInt8_4,
        BLowAshRange: common.UInt8_4,
        VRLowAshRange: common.UInt8_4,
        RPrechargeFun: common.Bool_true,
        GPrechargeFun: common.Bool_true,
        BPrechargeFun: common.Bool_true,
        VRPrechargeFun: common.Bool_true,
    }),
    t.partial({}),
], 'ChipRT5924ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipRT5924ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:48706
 */
exports.ChipRT5924ExtendProperty = t.intersection([
    exports.ChipRT5924ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipRT5924ExtendProperty') }),
], 'ChipRT5924ExtendProperty');
