// @novastar/native/build/main/generated/ChipSM16207SExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16207SExtendProperty = exports.ChipSM16207SExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSM16207SExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REliminateAfterLow1: common.UInt8_5,
        GEliminateAfterLow1: common.UInt8_5,
        BEliminateAfterLow1: common.UInt8_5,
        RDimCompensation: common.UInt8_3,
        GDimCompensation: common.UInt8_3,
        BDimCompensation: common.UInt8_3,
        REliminateAfterLow2: common.UInt8_5,
        GEliminateAfterLow2: common.UInt8_5,
        BEliminateAfterLow2: common.UInt8_5,
    }),
    t.partial({
        RLowAshImprovement: common.UInt8,
        GLowAshImprovement: common.UInt8,
        BLowAshImprovement: common.UInt8,
        IsAdvancedMode: common.Bool,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
    }),
], 'ChipSM16207SExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16207SExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:44711
 */
exports.ChipSM16207SExtendProperty = t.intersection([
    exports.ChipSM16207SExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16207SExtendProperty') }),
], 'ChipSM16207SExtendProperty');
