// @novastar/native/build/main/generated/Chip5253RGBExtendPropety.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip5253RGBExtendPropety = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const Chip5253ExtendProperty_1 = require("./Chip5253ExtendProperty");
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip5253RGBExtendPropety}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:31426
 */
exports.Chip5253RGBExtendPropety = t.partial({
    RedProperty: Chip5253ExtendProperty_1.Chip5253ExtendProperty,
    GreenProperty: Chip5253ExtendProperty_1.Chip5253ExtendProperty,
    BlueProperty: Chip5253ExtendProperty_1.Chip5253ExtendProperty,
    VRedProperty: Chip5253ExtendProperty_1.Chip5253ExtendProperty,
    EnableLowImprovement: common.Bool,
    EnShadeTransitOptimize: common.Bool,
    EnDoubleRefresh: common.Bool,
    EnGCLKMutiRate: common.Bool,
    FailureLEDElimination: common.Bool,
    RedRegValueConfigThird: common.UInt16,
    GreenRegValueConfigThird: common.UInt16,
    BlueRegValueConfigThird: common.UInt16,
    VRedRegValueConfigThird: common.UInt16,
    RedRegValueConfigFour: common.UInt16,
    GreenRegValueConfigFour: common.UInt16,
    BlueRegValueConfigFour: common.UInt16,
    VRedRegValueConfigFour: common.UInt16,
    RedRegValueConfigFive: common.UInt16,
    GreenRegValueConfigFive: common.UInt16,
    BlueRegValueConfigFive: common.UInt16,
    VRedRegValueConfigFive: common.UInt16,
    RedRegValueConfigSix: common.UInt16,
    GreenRegValueConfigSix: common.UInt16,
    BlueRegValueConfigSix: common.UInt16,
    VRedRegValueConfigSix: common.UInt16,
    RedRegValueConfigSpecial: common.UInt16,
    GreenRegValueConfigSpecial: common.UInt16,
    BlueRegValueConfigSpecial: common.UInt16,
    VRedRegValueConfigSpecial: common.UInt16,
}, 'Chip5253RGBExtendPropety');
