// @novastar/native/build/main/generated/ChipRT5965ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipRT5965ExtendProperty = exports.ChipRT5965ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipRT5965ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        FirstReg: common.UInt16,
        SecondReg: common.UInt16,
        ThirdReg: common.UInt16,
        CurrentGain: common.UInt8,
        CouplingEliminationLevel: common.UInt8,
        CouplingEliminationEn: common.Bool,
        FirstScanDarkEliminationLevel: common.UInt8,
        LowCompsentionOne: common.UInt8,
        LowCompsentionTwo: common.UInt8,
        GhostEliminationLevel: common.UInt8,
        ShadowEhancedModeEn: common.Bool,
        SlowOpen: common.Bool,
        OpenCircuitDetectionLevel: common.UInt8,
        PowerSavingModeEn: common.Bool,
        ChannelPowerSavingModeEn: common.Bool,
    }),
], 'ChipRT5965ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipRT5965ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipRT5965.decompiled.cs:1017
 */
exports.ChipRT5965ExtendProperty = t.intersection([
    exports.ChipRT5965ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipRT5965ExtendProperty') }),
], 'ChipRT5965ExtendProperty');
