// @novastar/native/build/main/generated/ChipSM16259RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16259RGBVExtendProperty = exports.ChipSM16259RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipSM16259ExtendProperty_1 = require("./ChipSM16259ExtendProperty");
exports.ChipSM16259RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipSM16259ExtendProperty_1.ChipSM16259ExtendProperty,
        GreenProperty: ChipSM16259ExtendProperty_1.ChipSM16259ExtendProperty,
        BlueProperty: ChipSM16259ExtendProperty_1.ChipSM16259ExtendProperty,
        VRedProperty: ChipSM16259ExtendProperty_1.ChipSM16259ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
    }),
], 'ChipSM16259RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16259RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16259.decompiled.cs:432
 */
exports.ChipSM16259RGBVExtendProperty = t.intersection([
    exports.ChipSM16259RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16259RGBVExtendProperty') }),
], 'ChipSM16259RGBVExtendProperty');
