// @novastar/native/build/main/generated/ChipSUM2030ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSUM2030ExtendProperty = exports.ChipSUM2030ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSUM2030ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        IsOpenCircuitProtection: common.Bool_true,
    }),
    t.partial({
        IsCustomSecondChipReg: common.Bool,
        RedRegLowValue: common.UInt8,
        RedRegHighValue: common.UInt8,
        GreenRegLowValue: common.UInt8,
        GreenRegHighValue: common.UInt8,
        BlueRegLowValue: common.UInt8,
        BlueRegHighValue: common.UInt8,
        VRedRegLowValue: common.UInt8,
        VRedRegHighValue: common.UInt8,
    }),
], 'ChipSUM2030ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSUM2030ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:47546
 */
exports.ChipSUM2030ExtendProperty = t.intersection([
    exports.ChipSUM2030ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSUM2030ExtendProperty') }),
], 'ChipSUM2030ExtendProperty');
