// @novastar/native/build/main/generated/ChipMBI5850ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMBI5850ExtendProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
/**
 * @category Codecs
 * @desc Codec for interface {@link ChipMBI5850ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5850.decompiled.cs:702
 */
exports.ChipMBI5850ExtendProperty = t.intersection([
    t.type({
        REG_NUMBER: common.withDefault(common.Int32, 24),
        MBI5850REG: common.XMLArray(common.UInt16, 'ushort'),
    }),
    t.partial({
        GrayScale: common.UInt16,
        EnDoubleRefresh: common.Bool,
        BlankUnitNumPerScan: common.UInt16,
        ExcessiveTime: common.UInt16,
        RowChangePoint: common.UInt16,
        M: common.UInt16,
        N: common.UInt16,
        DVI: common.UInt16,
        ShowsTheGhostGear_R: common.UInt16,
        ShowsTheGhostGear_G: common.UInt16,
        ShowsTheGhostGear_B: common.UInt16,
        ShowsTheGeargradualGhost_R: common.UInt16,
        ShowsTheGeargradualGhost_G: common.UInt16,
        ShowsTheGeargradualGhost_B: common.UInt16,
        HiddenGhostGear_1_R: common.UInt16,
        HiddenGhostGear_1_G: common.UInt16,
        HiddenGhostGear_1_B: common.UInt16,
        HiddenGhostGear_2_R: common.UInt16,
        HiddenGhostGear_2_G: common.UInt16,
        HiddenGhostGear_2_B: common.UInt16,
        HiddenGhostGear_3_R: common.UInt16,
        HiddenGhostGear_3_G: common.UInt16,
        HiddenGhostGear_3_B: common.UInt16,
        LowAshCompensation_R: common.UInt16,
        LowAshCompensation_G: common.UInt16,
        LowAshCompensation_B: common.UInt16,
        FailureLEDElimination: common.Bool,
        DislodgeBadPointsGrade_R: common.UInt8,
        DislodgeBadPointsGrade_G: common.UInt8,
        DislodgeBadPointsGrade_B: common.UInt8,
        Gain_R: common.UInt16,
        Gain_G: common.UInt16,
        Gain_B: common.UInt16,
    }),
], 'ChipMBI5850ExtendProperty');
