// @novastar/native/build/main/generated/ChipMBI525ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMBI525ExtendProperty = exports.ChipMBI525ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipMBI525ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegValueConfigFirst: common.withDefault(common.UInt16, 60735),
        LowCompsention: common.Bool_true,
        RefreshUp: common.Bool_true,
        GhostStall: common.Int32_11,
        GhostModle: common.Int32_1,
    }),
    t.partial({
        SettingNoise: common.Int32,
    }),
], 'ChipMBI525ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI525ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:42551
 */
exports.ChipMBI525ExtendProperty = t.intersection([
    exports.ChipMBI525ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI525ExtendProperty') }),
], 'ChipMBI525ExtendProperty');
