// @novastar/native/build/main/generated/ChipSCL8060ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSCL8060ExtendProperty = exports.ChipSCL8060ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSCL8060ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        EnLEDOpenShortDet: common.UInt8_3,
        LongWaitB: common.Bool_true,
    }),
    t.partial({
        RedDtVoltage: common.Bool,
        GreenDtVoltage: common.Bool,
        BlueDtVoltage: common.Bool,
        AnodeSideFastCharge: common.UInt8,
        DeadTimeGCLKVsync: common.UInt8,
        SLowFrame: common.Bool,
        OutPutCurRange: common.Bool,
        F50_25Hz: common.Bool,
        Dtbig: common.Bool,
        GclkSkewValue: common.UInt8,
        EnLEDShortProtect: common.Bool,
        GclkE1A0A2Source: common.Bool,
        EnOverTemprDect: common.Bool,
        GrayScaleMode: common.Bool,
    }),
], 'ChipSCL8060ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSCL8060ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:45870
 */
exports.ChipSCL8060ExtendProperty = t.intersection([
    exports.ChipSCL8060ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSCL8060ExtendProperty') }),
], 'ChipSCL8060ExtendProperty');
