// @novastar/native/build/main/generated/ChipHX8055ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipHX8055ExtendProperty = exports.ChipHX8055ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipHX8055ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegFirst: common.UInt16_8048,
        RegSecond: common.UInt16_31615,
        RegThird: common.UInt16_18183,
        RegFourth: common.UInt16_256,
    }),
    t.partial({}),
], 'ChipHX8055ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipHX8055ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipHX8055.decompiled.cs:1763
 */
exports.ChipHX8055ExtendProperty = t.intersection([
    exports.ChipHX8055ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipHX8055ExtendProperty') }),
], 'ChipHX8055ExtendProperty');
