// @novastar/native/build/main/generated/Chip16159RGBExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chip16159RGBExtendProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const Chip16159ExtendProperty_1 = require("./Chip16159ExtendProperty");
/**
 * @category Codecs
 * @desc Codec for interface {@link Chip16159RGBExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:48942
 */
exports.Chip16159RGBExtendProperty = t.intersection([
    t.type({
        RedRegValueConfigFirst: common.UInt16_32569,
        GreenRegValueConfigFirst: common.UInt16_32569,
        BlueRegValueConfigFirst: common.UInt16_32569,
        VRedRegValueConfigFirst: common.UInt16_32569,
        RedRegValueConfigSecond: common.UInt16_62465,
        GreenRegValueConfigSecond: common.UInt16_58369,
        BlueRegValueConfigSecond: common.UInt16_58369,
        VRedRegValueConfigSecond: common.UInt16_62465,
        RedRegValueConfigThird: common.UInt16_30976,
        GreenRegValueConfigThird: common.UInt16_30976,
        BlueRegValueConfigThird: common.UInt16_30976,
        VRedRegValueConfigThird: common.UInt16_30976,
    }),
    t.partial({
        RedProperty: Chip16159ExtendProperty_1.Chip16159ExtendProperty,
        GreenProperty: Chip16159ExtendProperty_1.Chip16159ExtendProperty,
        BlueProperty: Chip16159ExtendProperty_1.Chip16159ExtendProperty,
        VRedProperty: Chip16159ExtendProperty_1.Chip16159ExtendProperty,
        IsOpenTest: common.Bool,
        OpenTestGrade: common.UInt8,
    }),
], 'Chip16159RGBExtendProperty');
