// @novastar/native/build/main/generated/ChipDP3264RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipDP3264RGBVExtendProperty = exports.ChipDP3264RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipDP3264ExtendProperty_1 = require("./ChipDP3264ExtendProperty");
exports.ChipDP3264RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_13,
        RefreshNum: common.Int32_32,
        MaxRefreshNum: common.Int32_128,
    }),
    t.partial({
        RedProperty: ChipDP3264ExtendProperty_1.ChipDP3264ExtendProperty,
        GreenProperty: ChipDP3264ExtendProperty_1.ChipDP3264ExtendProperty,
        BlueProperty: ChipDP3264ExtendProperty_1.ChipDP3264ExtendProperty,
        VRedProperty: ChipDP3264ExtendProperty_1.ChipDP3264ExtendProperty,
        RefreshNumPerVs: common.UInt8,
        LineGclkNum: common.UInt8,
        R: common.UInt8,
        LowGrayDisplayEnhance: common.Bool,
        BlackScreenDynamicEnergySaving: common.UInt8,
        HighGrayDataIndependentRefresh: common.UInt8,
        EnableToRemoveBadPoints: common.Bool,
        IsAdvancedMode: common.Bool,
        IsUseNewModule: common.Bool,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
        FourthDataLen: common.Int32,
        FourthStartIndex: common.Int32,
        FourthRegisterAddr: common.Int32,
        ErrRedGain: common.Int32,
        ErrGreenGain: common.Int32,
        ErrBlueGain: common.Int32,
        ErrVRedGain: common.Int32,
    }),
], 'ChipDP3264RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipDP3264RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipDP3264.decompiled.cs:7020
 */
exports.ChipDP3264RGBVExtendProperty = t.intersection([
    exports.ChipDP3264RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipDP3264RGBVExtendProperty') }),
], 'ChipDP3264RGBVExtendProperty');
