// @novastar/native/build/main/generated/ChipSCL8081RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSCL8081RGBVExtendProperty = exports.ChipSCL8081RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipSCL8081ExtendProperty_1 = require("./ChipSCL8081ExtendProperty");
exports.ChipSCL8081RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        REG_LENGTH: common.Int32_4,
    }),
    t.partial({
        RedProperty: ChipSCL8081ExtendProperty_1.ChipSCL8081ExtendProperty,
        GreenProperty: ChipSCL8081ExtendProperty_1.ChipSCL8081ExtendProperty,
        BlueProperty: ChipSCL8081ExtendProperty_1.ChipSCL8081ExtendProperty,
        VRedProperty: ChipSCL8081ExtendProperty_1.ChipSCL8081ExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        ScanCount: common.UInt8,
        SpecialDataLen: common.Int32,
        SpecialRegisterAddr: common.UInt32,
    }),
], 'ChipSCL8081RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSCL8081RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSCL8081.decompiled.cs:908
 */
exports.ChipSCL8081RGBVExtendProperty = t.intersection([
    exports.ChipSCL8081RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSCL8081RGBVExtendProperty') }),
], 'ChipSCL8081RGBVExtendProperty');
