// @novastar/native/build/main/generated/ChipMBI5034BRGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipMBI5034BRGBVExtendProperty = exports.ChipMBI5034BRGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipMBI5034BExtendProperty_1 = require("./ChipMBI5034BExtendProperty");
const PointDetectParameter_1 = require("./PointDetectParameter");
exports.ChipMBI5034BRGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipMBI5034BExtendProperty_1.ChipMBI5034BExtendProperty,
        GreenProperty: ChipMBI5034BExtendProperty_1.ChipMBI5034BExtendProperty,
        BlueProperty: ChipMBI5034BExtendProperty_1.ChipMBI5034BExtendProperty,
        VRedProperty: ChipMBI5034BExtendProperty_1.ChipMBI5034BExtendProperty,
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
        IsAdvancedMode: common.Bool,
        PointDetectType: common.UInt8,
        FirstDataLen: common.Int32,
        FirstStartIndex: common.Int32,
        FirstRegisterAddr: common.Int32,
        PointDetectParameter: PointDetectParameter_1.PointDetectParameter,
    }),
], 'ChipMBI5034BRGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipMBI5034BRGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipMBI5034B.decompiled.cs:1370
 */
exports.ChipMBI5034BRGBVExtendProperty = t.intersection([
    exports.ChipMBI5034BRGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipMBI5034BRGBVExtendProperty') }),
], 'ChipMBI5034BRGBVExtendProperty');
