// @novastar/native/build/main/generated/ChipHX8055RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipHX8055RGBVExtendProperty = exports.ChipHX8055RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipHX8055ExtendProperty_1 = require("./ChipHX8055ExtendProperty");
exports.ChipHX8055RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        RedProperty: ChipHX8055ExtendProperty_1.ChipHX8055ExtendProperty,
        GreenProperty: ChipHX8055ExtendProperty_1.ChipHX8055ExtendProperty,
        BlueProperty: ChipHX8055ExtendProperty_1.ChipHX8055ExtendProperty,
        VRedProperty: ChipHX8055ExtendProperty_1.ChipHX8055ExtendProperty,
        IsUseNewModule: common.Bool,
    }),
], 'ChipHX8055RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipHX8055RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipHX8055.decompiled.cs:1973
 */
exports.ChipHX8055RGBVExtendProperty = t.intersection([
    exports.ChipHX8055RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipHX8055RGBVExtendProperty') }),
], 'ChipHX8055RGBVExtendProperty');
