// @novastar/native/build/main/generated/ChipSM16359RGBVExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16359RGBVExtendProperty = exports.ChipSM16359RGBVExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
const ChipSM16359ExtendProperty_1 = require("./ChipSM16359ExtendProperty");
exports.ChipSM16359RGBVExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        RedProperty: ChipSM16359ExtendProperty_1.ChipSM16359ExtendProperty,
        GreenProperty: ChipSM16359ExtendProperty_1.ChipSM16359ExtendProperty,
        BlueProperty: ChipSM16359ExtendProperty_1.ChipSM16359ExtendProperty,
        VRedProperty: ChipSM16359ExtendProperty_1.ChipSM16359ExtendProperty,
    }),
], 'ChipSM16359RGBVExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16359RGBVExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:51301
 */
exports.ChipSM16359RGBVExtendProperty = t.intersection([
    exports.ChipSM16359RGBVExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16359RGBVExtendProperty') }),
], 'ChipSM16359RGBVExtendProperty');
