// @novastar/native/build/main/generated/ChipSUM2032ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSUM2032ExtendProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
/**
 * @category Codecs
 * @desc Codec for interface {@link ChipSUM2032ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:48419
 */
exports.ChipSUM2032ExtendProperty = t.intersection([
    t.type({
        RColumnGhostCancelingMode: common.UInt8_56,
        GColumnGhostCancelingMode: common.UInt8_56,
        BColumnGhostCancelingMode: common.UInt8_56,
        VRColumnGhostCancelingMode: common.UInt8_56,
        RColumnGhostCanceling: common.UInt8_7,
        GColumnGhostCanceling: common.UInt8_5,
        BColumnGhostCanceling: common.UInt8_3,
        VRColumnGhostCanceling: common.UInt8_7,
        RLowGrayMode: common.UInt8_7,
        GLowGrayMode: common.UInt8_1,
        BLowGrayMode: common.UInt8_1,
        VRLowGrayMode: common.UInt8_7,
        RLowGrayCompensation: common.UInt8_1,
        GLowGrayCompensation: common.UInt8_4,
        BLowGrayCompensation: common.UInt8_4,
        VRLowGrayCompensation: common.UInt8_1,
    }),
    t.partial({}),
], 'ChipSUM2032ExtendProperty');
