// @novastar/native/build/main/generated/ChipSM16158ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipSM16158ExtendProperty = exports.ChipSM16158ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipSM16158ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.partial({
        IsAdvancedMode: common.Bool,
        RLowAshImprovement: common.UInt8,
        GLowAshImprovement: common.UInt8,
        BLowAshImprovement: common.UInt8,
        OpenDetection: common.Bool,
        REnableOpenInspection: common.Bool,
        GEnableOpenInspection: common.Bool,
        BEnableOpenInspection: common.Bool,
        RLowAshPitsImprove: common.UInt8,
        GLowAshPitsImprove: common.UInt8,
        BLowAshPitsImprove: common.UInt8,
        RDimInterval: common.UInt8,
        GDimInterval: common.UInt8,
        BDimInterval: common.UInt8,
        RDimCompensation: common.UInt8,
        GDimCompensation: common.UInt8,
        BDimCompensation: common.UInt8,
        RedRegValueConfigFirst: common.UInt16,
        GreenRegValueConfigFirst: common.UInt16,
        BlueRegValueConfigFirst: common.UInt16,
        VRedRegValueConfigFirst: common.UInt16,
        RedRegValueConfigSecond: common.UInt16,
        GreenRegValueConfigSecond: common.UInt16,
        BlueRegValueConfigSecond: common.UInt16,
        VRedRegValueConfigSecond: common.UInt16,
    }),
], 'ChipSM16158ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipSM16158ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:45009
 */
exports.ChipSM16158ExtendProperty = t.intersection([
    exports.ChipSM16158ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipSM16158ExtendProperty') }),
], 'ChipSM16158ExtendProperty');
