// @novastar/native/build/main/generated/ChipFM6127ExtendProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChipFM6127ExtendProperty = exports.ChipFM6127ExtendPropertyBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.ChipFM6127ExtendPropertyBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RedRegValueConfigFirst: common.UInt16_65474,
        GreenRegValueConfigFirst: common.UInt16_65474,
        BlueRegValueConfigFirst: common.UInt16_65474,
        VRedRegValueConfigFirst: common.UInt16_65474,
        RedRegValueConfigSecond: common.UInt16_30818,
        GreenRegValueConfigSecond: common.UInt16_28770,
        BlueRegValueConfigSecond: common.UInt16_26722,
        VRedRegValueConfigSecond: common.UInt16_30818,
        RedRegValueConfigThird: common.UInt16_3841,
        GreenRegValueConfigThird: common.UInt16_3841,
        BlueRegValueConfigThird: common.UInt16_3841,
        VRedRegValueConfigThird: common.UInt16_3841,
        RedRegValueConfigFourth: common.UInt16_65474,
        GreenRegValueConfigFourth: common.UInt16_65474,
        BlueRegValueConfigFourth: common.UInt16_65474,
        VRedRegValueConfigFourth: common.UInt16_65474,
        IsAdvancedMode: common.Bool_true,
    }),
    t.partial({
        IsUseNewModule: common.Bool,
        ChipLibVersion: common.UInt8,
    }),
], 'ChipFM6127ExtendPropertyBase');
/**
 * @category Codecs
 * @desc Codec for {@link ChipFM6127ExtendProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipFM6127.decompiled.cs:6135
 */
exports.ChipFM6127ExtendProperty = t.intersection([
    exports.ChipFM6127ExtendPropertyBase,
    t.partial({ '@_xsi:type': t.literal('ChipFM6127ExtendProperty') }),
], 'ChipFM6127ExtendProperty');
