// @novastar/native/build/main/generated/CodeTypes.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeTypes = exports.CodeTypesEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var CodeTypesEnum;
(function (CodeTypesEnum) {
    CodeTypesEnum[CodeTypesEnum["requestRead"] = 0] = "requestRead";
    CodeTypesEnum[CodeTypesEnum["replyRead"] = 1] = "replyRead";
    CodeTypesEnum[CodeTypesEnum["requestWrite"] = 2] = "requestWrite";
    CodeTypesEnum[CodeTypesEnum["replyWrite"] = 3] = "replyWrite";
})(CodeTypesEnum = exports.CodeTypesEnum || (exports.CodeTypesEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link CodeTypesEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1114
 */
exports.CodeTypes = (0, EnumFromString_1.default)(CodeTypesEnum, 'CodeTypes');
