// @novastar/native/build/main/generated/CommonInfoCompeleteResult.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommonInfoCompeleteResult = exports.CommonInfoCompeleteResultEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var CommonInfoCompeleteResultEnum;
(function (CommonInfoCompeleteResultEnum) {
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["OK"] = 0] = "OK";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["VerNotComparable"] = 1] = "VerNotComparable";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["CommunicationFault"] = 2] = "CommunicationFault";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["DataError"] = 3] = "DataError";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["ParseError"] = 4] = "ParseError";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["FileError"] = 5] = "FileError";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["NotSupportDevice"] = 6] = "NotSupportDevice";
    CommonInfoCompeleteResultEnum[CommonInfoCompeleteResultEnum["NoServerObject"] = 7] = "NoServerObject";
})(CommonInfoCompeleteResultEnum = exports.CommonInfoCompeleteResultEnum || (exports.CommonInfoCompeleteResultEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link CommonInfoCompeleteResultEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:56479
 */
exports.CommonInfoCompeleteResult = (0, EnumFromString_1.default)(CommonInfoCompeleteResultEnum, 'CommonInfoCompeleteResult');
