// @novastar/native/build/main/generated/ComplexLEDDisplayInfo.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplexLEDDisplayInfo = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const LEDDisplyType_1 = require("./LEDDisplyType");
const ScanBoardRegionInfo_1 = require("./ScanBoardRegionInfo");
const ScreenAdjustParams_1 = require("./ScreenAdjustParams");
const VirtualModeType_1 = require("./VirtualModeType");
/**
 * @category Codecs
 * @desc Codec for interface {@link ComplexLEDDisplayInfo}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:55281
 */
exports.ComplexLEDDisplayInfo = t.intersection([
    t.type({
        ScanBoardRegionInfoList: common.XMLArray(ScanBoardRegionInfo_1.ScanBoardRegionInfo, 'ScanBoardRegionInfo'),
    }),
    t.partial({
        IsOpen18Bit: common.Bool,
        IsOpenClearview: common.Bool,
        Type: LEDDisplyType_1.LEDDisplyType,
        VirtualMode: VirtualModeType_1.VirtualModeType,
        ScannerCount: common.Int32,
        ScrAdjustParams: ScreenAdjustParams_1.ScreenAdjustParams,
    }),
], 'ComplexLEDDisplayInfo');
