// @novastar/native/build/main/generated/ControlModeType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ControlModeType = exports.ControlModeTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ControlModeTypeEnum;
(function (ControlModeTypeEnum) {
    ControlModeTypeEnum[ControlModeTypeEnum["NormalMode"] = 0] = "NormalMode";
    ControlModeTypeEnum[ControlModeTypeEnum["BlackMode"] = 1] = "BlackMode";
    ControlModeTypeEnum[ControlModeTypeEnum["LockMode"] = 2] = "LockMode";
    ControlModeTypeEnum[ControlModeTypeEnum["Unknown"] = 255] = "Unknown";
})(ControlModeTypeEnum = exports.ControlModeTypeEnum || (exports.ControlModeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ControlModeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:455
 */
exports.ControlModeType = (0, EnumFromString_1.default)(ControlModeTypeEnum, 'ControlModeType');
