// @novastar/native/build/main/generated/CustomRegExtendPropertyBase.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomRegExtendPropertyBase = exports.CustomRegExtendPropertyBaseBase = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipBaseExtendPropey_1 = require("./ChipBaseExtendPropey");
exports.CustomRegExtendPropertyBaseBase = t.intersection([
    ChipBaseExtendPropey_1.ChipBaseExtendPropey,
    t.type({
        RegValueConfigThird: common.UInt16_30976,
    }),
    t.partial({
        IsAdvancedMode: common.Bool,
        RegValueConfigFirst: common.UInt16,
        RegValueConfigSecond: common.UInt16,
    }),
], 'CustomRegExtendPropertyBaseBase');
/**
 * @category Codecs
 * @desc Codec for {@link CustomRegExtendPropertyBase}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:49222
 */
exports.CustomRegExtendPropertyBase = t.intersection([
    exports.CustomRegExtendPropertyBaseBase,
    t.partial({ '@_xsi:type': t.literal('CustomRegExtendPropertyBase') }),
], 'CustomRegExtendPropertyBase');
