// @novastar/native/build/main/generated/DataGroupOutPutMode.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGroupOutPutMode = exports.DataGroupOutPutModeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var DataGroupOutPutModeEnum;
(function (DataGroupOutPutModeEnum) {
    DataGroupOutPutModeEnum[DataGroupOutPutModeEnum["Normal"] = 0] = "Normal";
    DataGroupOutPutModeEnum[DataGroupOutPutModeEnum["Group20"] = 1] = "Group20";
    DataGroupOutPutModeEnum[DataGroupOutPutModeEnum["Group24"] = 2] = "Group24";
    DataGroupOutPutModeEnum[DataGroupOutPutModeEnum["Group28"] = 3] = "Group28";
})(DataGroupOutPutModeEnum = exports.DataGroupOutPutModeEnum || (exports.DataGroupOutPutModeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DataGroupOutPutModeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1281
 */
exports.DataGroupOutPutMode = (0, EnumFromString_1.default)(DataGroupOutPutModeEnum, 'DataGroupOutPutMode');
