// @novastar/native/build/main/generated/DetectPointType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DetectPointType = exports.DetectPointTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var DetectPointTypeEnum;
(function (DetectPointTypeEnum) {
    DetectPointTypeEnum[DetectPointTypeEnum["EarlyDeath"] = 0] = "EarlyDeath";
    DetectPointTypeEnum[DetectPointTypeEnum["OpenCircuit"] = 1] = "OpenCircuit";
    DetectPointTypeEnum[DetectPointTypeEnum["ShortCircuit"] = 2] = "ShortCircuit";
    DetectPointTypeEnum[DetectPointTypeEnum["BothDivision"] = 3] = "BothDivision";
    DetectPointTypeEnum[DetectPointTypeEnum["BothNoDivision"] = 4] = "BothNoDivision";
    DetectPointTypeEnum[DetectPointTypeEnum["FourTypesCircuit"] = 5] = "FourTypesCircuit";
    DetectPointTypeEnum[DetectPointTypeEnum["FiveTypesAllIn"] = 6] = "FiveTypesAllIn";
    DetectPointTypeEnum[DetectPointTypeEnum["None"] = 7] = "None";
})(DetectPointTypeEnum = exports.DetectPointTypeEnum || (exports.DetectPointTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DetectPointTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:4669
 */
exports.DetectPointType = (0, EnumFromString_1.default)(DetectPointTypeEnum, 'DetectPointType');
