// @novastar/native/build/main/generated/DeviceTypes.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceTypes = exports.DeviceTypesEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var DeviceTypesEnum;
(function (DeviceTypesEnum) {
    DeviceTypesEnum[DeviceTypesEnum["FirstScaleDevice"] = 0] = "FirstScaleDevice";
    DeviceTypesEnum[DeviceTypesEnum["Scanner"] = 1] = "Scanner";
    DeviceTypesEnum[DeviceTypesEnum["FunctionCard"] = 2] = "FunctionCard";
    DeviceTypesEnum[DeviceTypesEnum["Distributor"] = 3] = "Distributor";
    DeviceTypesEnum[DeviceTypesEnum["Sender3D"] = 4] = "Sender3D";
})(DeviceTypesEnum = exports.DeviceTypesEnum || (exports.DeviceTypesEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DeviceTypesEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1132
 */
exports.DeviceTypes = (0, EnumFromString_1.default)(DeviceTypesEnum, 'DeviceTypes');
