// @novastar/native/build/main/generated/DisplayModeType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisplayModeType = exports.DisplayModeTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var DisplayModeTypeEnum;
(function (DisplayModeTypeEnum) {
    DisplayModeTypeEnum[DisplayModeTypeEnum["BlackScreen"] = 0] = "BlackScreen";
    DisplayModeTypeEnum[DisplayModeTypeEnum["LastFrame"] = 1] = "LastFrame";
    DisplayModeTypeEnum[DisplayModeTypeEnum["ShowLogo"] = 16] = "ShowLogo";
    DisplayModeTypeEnum[DisplayModeTypeEnum["Unknown"] = 255] = "Unknown";
})(DisplayModeTypeEnum = exports.DisplayModeTypeEnum || (exports.DisplayModeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DisplayModeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1045
 */
exports.DisplayModeType = (0, EnumFromString_1.default)(DisplayModeTypeEnum, 'DisplayModeType');
