// @novastar/native/build/main/generated/DownGhostClearType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DownGhostClearType = exports.DownGhostClearTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var DownGhostClearTypeEnum;
(function (DownGhostClearTypeEnum) {
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect1"] = 0] = "Effect1";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Closed"] = 1] = "Closed";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect2"] = 2] = "Effect2";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect3"] = 3] = "Effect3";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect4"] = 4] = "Effect4";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect5"] = 5] = "Effect5";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect6"] = 6] = "Effect6";
    DownGhostClearTypeEnum[DownGhostClearTypeEnum["Effect7"] = 7] = "Effect7";
})(DownGhostClearTypeEnum = exports.DownGhostClearTypeEnum || (exports.DownGhostClearTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DownGhostClearTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:51918
 */
exports.DownGhostClearType = (0, EnumFromString_1.default)(DownGhostClearTypeEnum, 'DownGhostClearType');
