// @novastar/native/build/main/generated/DviSelectMode.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DviSelectMode = exports.DviSelectModeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var DviSelectModeEnum;
(function (DviSelectModeEnum) {
    DviSelectModeEnum[DviSelectModeEnum["DVI"] = 88] = "DVI";
    DviSelectModeEnum[DviSelectModeEnum["HDMI"] = 5] = "HDMI";
    DviSelectModeEnum[DviSelectModeEnum["SDI"] = 1] = "SDI";
    DviSelectModeEnum[DviSelectModeEnum["DP"] = 95] = "DP";
    DviSelectModeEnum[DviSelectModeEnum["Dual_DVI"] = 97] = "Dual_DVI";
    DviSelectModeEnum[DviSelectModeEnum["HDBaseT"] = 63] = "HDBaseT";
    DviSelectModeEnum[DviSelectModeEnum["LVDS"] = 107] = "LVDS";
})(DviSelectModeEnum = exports.DviSelectModeEnum || (exports.DviSelectModeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link DviSelectModeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1207
 */
exports.DviSelectMode = (0, EnumFromString_1.default)(DviSelectModeEnum, 'DviSelectMode');
