// @novastar/native/build/main/generated/EVersionType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EVersionType = exports.EVersionTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var EVersionTypeEnum;
(function (EVersionTypeEnum) {
    EVersionTypeEnum[EVersionTypeEnum["Sender"] = 0] = "Sender";
    EVersionTypeEnum[EVersionTypeEnum["ScannerMCU"] = 1] = "ScannerMCU";
    EVersionTypeEnum[EVersionTypeEnum["ScannerFPGA"] = 2] = "ScannerFPGA";
    EVersionTypeEnum[EVersionTypeEnum["MoudeMCU"] = 3] = "MoudeMCU";
    EVersionTypeEnum[EVersionTypeEnum["SenderFPGA"] = 4] = "SenderFPGA";
    EVersionTypeEnum[EVersionTypeEnum["Sender3DFPGA"] = 5] = "Sender3DFPGA";
    EVersionTypeEnum[EVersionTypeEnum["Sender3DMCU"] = 6] = "Sender3DMCU";
})(EVersionTypeEnum = exports.EVersionTypeEnum || (exports.EVersionTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link EVersionTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:59242
 */
exports.EVersionType = (0, EnumFromString_1.default)(EVersionTypeEnum, 'EVersionType');
