// @novastar/native/build/main/generated/FirstLineCompensationType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirstLineCompensationType = exports.FirstLineCompensationTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var FirstLineCompensationTypeEnum;
(function (FirstLineCompensationTypeEnum) {
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Closed"] = 0] = "Closed";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect1"] = 1] = "Effect1";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect2"] = 2] = "Effect2";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect3"] = 3] = "Effect3";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect4"] = 4] = "Effect4";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect5"] = 5] = "Effect5";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect6"] = 6] = "Effect6";
    FirstLineCompensationTypeEnum[FirstLineCompensationTypeEnum["Effect7"] = 7] = "Effect7";
})(FirstLineCompensationTypeEnum = exports.FirstLineCompensationTypeEnum || (exports.FirstLineCompensationTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link FirstLineCompensationTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:51947
 */
exports.FirstLineCompensationType = (0, EnumFromString_1.default)(FirstLineCompensationTypeEnum, 'FirstLineCompensationType');
