// @novastar/native/build/main/generated/FlashCheckErrorType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlashCheckErrorType = exports.FlashCheckErrorTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var FlashCheckErrorTypeEnum;
(function (FlashCheckErrorTypeEnum) {
    FlashCheckErrorTypeEnum[FlashCheckErrorTypeEnum["CommunicationError"] = 0] = "CommunicationError";
    FlashCheckErrorTypeEnum[FlashCheckErrorTypeEnum["HardwareError"] = 1] = "HardwareError";
    FlashCheckErrorTypeEnum[FlashCheckErrorTypeEnum["CorrectionCodeError"] = 2] = "CorrectionCodeError";
    FlashCheckErrorTypeEnum[FlashCheckErrorTypeEnum["FlashTopologyError"] = 3] = "FlashTopologyError";
})(FlashCheckErrorTypeEnum = exports.FlashCheckErrorTypeEnum || (exports.FlashCheckErrorTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link FlashCheckErrorTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:57858
 */
exports.FlashCheckErrorType = (0, EnumFromString_1.default)(FlashCheckErrorTypeEnum, 'FlashCheckErrorType');
