// @novastar/native/build/main/generated/FlashTopologyAndMeaasge.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlashTopologyAndMeaasge = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
/**
 * @category Codecs
 * @desc Codec for interface {@link FlashTopologyAndMeaasge}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:52910
 */
exports.FlashTopologyAndMeaasge = t.intersection([
    t.type({
        ThisFlashExist: common.Bool_true,
    }),
    t.partial({
        SerialNumOfThisBus: common.Int32,
        FlashSerialNumOnThisBus: common.Int32,
        StartXOfThisFlash: common.Int32,
        StartYOfThisFlash: common.Int32,
        TotalFlashCascadedOnThisBus: common.Int32,
        PixelColsOfThisFlash: common.Int32,
        PixelRowsOfThisFlash: common.Int32,
        GroupNumInModule: common.Int32,
    }),
], 'FlashTopologyAndMeaasge');
