// @novastar/native/build/main/generated/GetSenderFunctionInfoRes.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSenderFunctionInfoRes = exports.GetSenderFunctionInfoResEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var GetSenderFunctionInfoResEnum;
(function (GetSenderFunctionInfoResEnum) {
    GetSenderFunctionInfoResEnum[GetSenderFunctionInfoResEnum["ReadTimeOut"] = 0] = "ReadTimeOut";
    GetSenderFunctionInfoResEnum[GetSenderFunctionInfoResEnum["CheckFail"] = 1] = "CheckFail";
    GetSenderFunctionInfoResEnum[GetSenderFunctionInfoResEnum["ServerLinkError"] = 2] = "ServerLinkError";
    GetSenderFunctionInfoResEnum[GetSenderFunctionInfoResEnum["Succeed"] = 3] = "Succeed";
})(GetSenderFunctionInfoResEnum = exports.GetSenderFunctionInfoResEnum || (exports.GetSenderFunctionInfoResEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link GetSenderFunctionInfoResEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:16656
 */
exports.GetSenderFunctionInfoRes = (0, EnumFromString_1.default)(GetSenderFunctionInfoResEnum, 'GetSenderFunctionInfoRes');
