// @novastar/native/build/main/generated/GrayRealizeType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrayRealizeType = exports.GrayRealizeTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var GrayRealizeTypeEnum;
(function (GrayRealizeTypeEnum) {
    GrayRealizeTypeEnum[GrayRealizeTypeEnum["Traditional"] = 0] = "Traditional";
    GrayRealizeTypeEnum[GrayRealizeTypeEnum["HGHR"] = 1] = "HGHR";
    GrayRealizeTypeEnum[GrayRealizeTypeEnum["GrayFrist"] = 2] = "GrayFrist";
    GrayRealizeTypeEnum[GrayRealizeTypeEnum["Blance"] = 3] = "Blance";
    GrayRealizeTypeEnum[GrayRealizeTypeEnum["Photograph"] = 4] = "Photograph";
    GrayRealizeTypeEnum[GrayRealizeTypeEnum["BrightEfficiency"] = 5] = "BrightEfficiency";
})(GrayRealizeTypeEnum = exports.GrayRealizeTypeEnum || (exports.GrayRealizeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link GrayRealizeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1008
 */
exports.GrayRealizeType = (0, EnumFromString_1.default)(GrayRealizeTypeEnum, 'GrayRealizeType');
