// @novastar/native/build/main/generated/GrayModeType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrayModeType = exports.GrayModeTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var GrayModeTypeEnum;
(function (GrayModeTypeEnum) {
    GrayModeTypeEnum[GrayModeTypeEnum["Lowlight"] = 0] = "Lowlight";
    GrayModeTypeEnum[GrayModeTypeEnum["Normallight"] = 1] = "Normallight";
    GrayModeTypeEnum[GrayModeTypeEnum["Highlight"] = 2] = "Highlight";
    GrayModeTypeEnum[GrayModeTypeEnum["LowerLowLight"] = 3] = "LowerLowLight";
    GrayModeTypeEnum[GrayModeTypeEnum["Unknown"] = 255] = "Unknown";
})(GrayModeTypeEnum = exports.GrayModeTypeEnum || (exports.GrayModeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link GrayModeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:999
 */
exports.GrayModeType = (0, EnumFromString_1.default)(GrayModeTypeEnum, 'GrayModeType');
