// @novastar/native/build/main/generated/HighLowBitsType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HighLowBitsType = exports.HighLowBitsTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var HighLowBitsTypeEnum;
(function (HighLowBitsTypeEnum) {
    HighLowBitsTypeEnum[HighLowBitsTypeEnum["Horizontal_LeftHigh"] = 0] = "Horizontal_LeftHigh";
    HighLowBitsTypeEnum[HighLowBitsTypeEnum["Horizontal_RightHigh"] = 1] = "Horizontal_RightHigh";
    HighLowBitsTypeEnum[HighLowBitsTypeEnum["Vertical_UpHigh"] = 2] = "Vertical_UpHigh";
    HighLowBitsTypeEnum[HighLowBitsTypeEnum["Vertical_DownHigh"] = 3] = "Vertical_DownHigh";
})(HighLowBitsTypeEnum = exports.HighLowBitsTypeEnum || (exports.HighLowBitsTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link HighLowBitsTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:58653
 */
exports.HighLowBitsType = (0, EnumFromString_1.default)(HighLowBitsTypeEnum, 'HighLowBitsType');
