// @novastar/native/build/main/generated/HLGModel.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HLGModel = exports.HLGModelEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var HLGModelEnum;
(function (HLGModelEnum) {
    HLGModelEnum[HLGModelEnum["HLG_10"] = 0] = "HLG_10";
    HLGModelEnum[HLGModelEnum["HLG_11"] = 1] = "HLG_11";
    HLGModelEnum[HLGModelEnum["HLG_12"] = 2] = "HLG_12";
    HLGModelEnum[HLGModelEnum["HLG_13"] = 3] = "HLG_13";
    HLGModelEnum[HLGModelEnum["HLG_14"] = 4] = "HLG_14";
    HLGModelEnum[HLGModelEnum["HLG_15"] = 5] = "HLG_15";
    HLGModelEnum[HLGModelEnum["HLG_16"] = 6] = "HLG_16";
    HLGModelEnum[HLGModelEnum["HLG_17"] = 7] = "HLG_17";
    HLGModelEnum[HLGModelEnum["Unknow"] = 8] = "Unknow";
})(HLGModelEnum = exports.HLGModelEnum || (exports.HLGModelEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link HLGModelEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1391
 */
exports.HLGModel = (0, EnumFromString_1.default)(HLGModelEnum, 'HLGModel');
