// @novastar/native/build/main/generated/HWSoftwareSpaceRes.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HWSoftwareSpaceRes = exports.HWSoftwareSpaceResEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var HWSoftwareSpaceResEnum;
(function (HWSoftwareSpaceResEnum) {
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["OK"] = 0] = "OK";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["NotSupportDevice"] = 1] = "NotSupportDevice";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["NoServerObject"] = 2] = "NoServerObject";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["FileError"] = 3] = "FileError";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["DataError"] = 4] = "DataError";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["Busy"] = 5] = "Busy";
    HWSoftwareSpaceResEnum[HWSoftwareSpaceResEnum["Unknown"] = 255] = "Unknown";
})(HWSoftwareSpaceResEnum = exports.HWSoftwareSpaceResEnum || (exports.HWSoftwareSpaceResEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link HWSoftwareSpaceResEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.CommonInfoAccessor.decompiled.cs:81
 */
exports.HWSoftwareSpaceRes = (0, EnumFromString_1.default)(HWSoftwareSpaceResEnum, 'HWSoftwareSpaceRes');
