// @novastar/native/build/main/generated/IrRegularCabinetProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IrRegularCabinetProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const CabinetType_1 = require("./CabinetType");
const ModuleInIrRegularCabinet_1 = require("./ModuleInIrRegularCabinet");
/**
 * @category Codecs
 * @desc Codec for interface {@link IrRegularCabinetProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:15275
 */
exports.IrRegularCabinetProperty = t.intersection([
    t.type({
        Name: common.string_empty,
        CabinetVersion: common.string_20,
        ModuleListInCabinet: common.XMLArray(ModuleInIrRegularCabinet_1.ModuleInIrRegularCabinet, 'ModuleInIrRegularCabinet'),
    }),
    t.partial({
        PhysicalDataGroupNum: common.Int32,
        CabinetType: CabinetType_1.CabinetType,
        Width: common.Int32,
        Height: common.Int32,
        MaxSumOfScanPointInGroup: common.Int32,
        PointTableData: common.Base64,
        PumpModel: common.Int32,
        pumpGroupsLines: common.Base64,
        PumpWidthReduction: common.Int32,
        PumpHeightReduction: common.Int32,
    }),
], 'IrRegularCabinetProperty');
