// @novastar/native/build/main/generated/LowAshGrade.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LowAshGrade = exports.LowAshGradeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var LowAshGradeEnum;
(function (LowAshGradeEnum) {
    LowAshGradeEnum[LowAshGradeEnum["NotEffective"] = 0] = "NotEffective";
    LowAshGradeEnum[LowAshGradeEnum["OneGrade"] = 1] = "OneGrade";
    LowAshGradeEnum[LowAshGradeEnum["TwoGrade"] = 2] = "TwoGrade";
    LowAshGradeEnum[LowAshGradeEnum["ThreeGrade"] = 3] = "ThreeGrade";
})(LowAshGradeEnum = exports.LowAshGradeEnum || (exports.LowAshGradeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link LowAshGradeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16259.decompiled.cs:103
 */
exports.LowAshGrade = (0, EnumFromString_1.default)(LowAshGradeEnum, 'LowAshGrade');
