// @novastar/native/build/main/generated/MCTRL1600DVIEnum.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MCTRL1600DVIEnum = exports.MCTRL1600DVIEnumEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var MCTRL1600DVIEnumEnum;
(function (MCTRL1600DVIEnumEnum) {
    MCTRL1600DVIEnumEnum[MCTRL1600DVIEnumEnum["DVI1"] = 0] = "DVI1";
    MCTRL1600DVIEnumEnum[MCTRL1600DVIEnumEnum["DVI2"] = 1] = "DVI2";
    MCTRL1600DVIEnumEnum[MCTRL1600DVIEnumEnum["DVI3"] = 2] = "DVI3";
    MCTRL1600DVIEnumEnum[MCTRL1600DVIEnumEnum["DVI4"] = 3] = "DVI4";
})(MCTRL1600DVIEnumEnum = exports.MCTRL1600DVIEnumEnum || (exports.MCTRL1600DVIEnumEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link MCTRL1600DVIEnumEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1344
 */
exports.MCTRL1600DVIEnum = (0, EnumFromString_1.default)(MCTRL1600DVIEnumEnum, 'MCTRL1600DVIEnum');
