// @novastar/native/build/main/generated/MessageType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageType = exports.MessageTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var MessageTypeEnum;
(function (MessageTypeEnum) {
    MessageTypeEnum[MessageTypeEnum["SCREEN_AREA_ACQUIRER_TOPOLOGY_SHOW"] = 1] = "SCREEN_AREA_ACQUIRER_TOPOLOGY_SHOW";
    MessageTypeEnum[MessageTypeEnum["SCREEN_AREA_ACQUIRER_TOPOLOGY_CLOSE"] = 2] = "SCREEN_AREA_ACQUIRER_TOPOLOGY_CLOSE";
    MessageTypeEnum[MessageTypeEnum["DVI_COEFFICIENT_UPLOAD_START"] = 3] = "DVI_COEFFICIENT_UPLOAD_START";
    MessageTypeEnum[MessageTypeEnum["DVI_COEFFICIENT_UPLOAD_END"] = 4] = "DVI_COEFFICIENT_UPLOAD_END";
    MessageTypeEnum[MessageTypeEnum["UNKNOWN"] = 0] = "UNKNOWN";
})(MessageTypeEnum = exports.MessageTypeEnum || (exports.MessageTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link MessageTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:16554
 */
exports.MessageType = (0, EnumFromString_1.default)(MessageTypeEnum, 'MessageType');
