// @novastar/native/build/main/generated/ModuleCascadeDiretion.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModuleCascadeDiretion = exports.ModuleCascadeDiretionEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ModuleCascadeDiretionEnum;
(function (ModuleCascadeDiretionEnum) {
    ModuleCascadeDiretionEnum[ModuleCascadeDiretionEnum["RightLeft"] = 0] = "RightLeft";
    ModuleCascadeDiretionEnum[ModuleCascadeDiretionEnum["LeftRight"] = 1] = "LeftRight";
    ModuleCascadeDiretionEnum[ModuleCascadeDiretionEnum["DownUp"] = 2] = "DownUp";
    ModuleCascadeDiretionEnum[ModuleCascadeDiretionEnum["UpDown"] = 3] = "UpDown";
    ModuleCascadeDiretionEnum[ModuleCascadeDiretionEnum["Unknown"] = 255] = "Unknown";
})(ModuleCascadeDiretionEnum = exports.ModuleCascadeDiretionEnum || (exports.ModuleCascadeDiretionEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ModuleCascadeDiretionEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:746
 */
exports.ModuleCascadeDiretion = (0, EnumFromString_1.default)(ModuleCascadeDiretionEnum, 'ModuleCascadeDiretion');
