// @novastar/native/build/main/generated/ModulationModeType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModulationModeType = exports.ModulationModeTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ModulationModeTypeEnum;
(function (ModulationModeTypeEnum) {
    ModulationModeTypeEnum[ModulationModeTypeEnum["NoDistributor"] = 0] = "NoDistributor";
    ModulationModeTypeEnum[ModulationModeTypeEnum["TwoToFour"] = 1] = "TwoToFour";
    ModulationModeTypeEnum[ModulationModeTypeEnum["OneToEight"] = 2] = "OneToEight";
    ModulationModeTypeEnum[ModulationModeTypeEnum["Unknown"] = 255] = "Unknown";
})(ModulationModeTypeEnum = exports.ModulationModeTypeEnum || (exports.ModulationModeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ModulationModeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1294
 */
exports.ModulationModeType = (0, EnumFromString_1.default)(ModulationModeTypeEnum, 'ModulationModeType');
