// @novastar/native/build/main/generated/MSRvCardStatusType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MSRvCardStatusType = exports.MSRvCardStatusTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var MSRvCardStatusTypeEnum;
(function (MSRvCardStatusTypeEnum) {
    MSRvCardStatusTypeEnum[MSRvCardStatusTypeEnum["CardOk"] = 0] = "CardOk";
    MSRvCardStatusTypeEnum[MSRvCardStatusTypeEnum["MasterOK"] = 1] = "MasterOK";
    MSRvCardStatusTypeEnum[MSRvCardStatusTypeEnum["SideOK"] = 2] = "SideOK";
    MSRvCardStatusTypeEnum[MSRvCardStatusTypeEnum["CardError"] = 3] = "CardError";
    MSRvCardStatusTypeEnum[MSRvCardStatusTypeEnum["UnConnect"] = 4] = "UnConnect";
    MSRvCardStatusTypeEnum[MSRvCardStatusTypeEnum["UnKnown"] = 5] = "UnKnown";
})(MSRvCardStatusTypeEnum = exports.MSRvCardStatusTypeEnum || (exports.MSRvCardStatusTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link MSRvCardStatusTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:60699
 */
exports.MSRvCardStatusType = (0, EnumFromString_1.default)(MSRvCardStatusTypeEnum, 'MSRvCardStatusType');
