// @novastar/native/build/main/generated/MultiplierGrade.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiplierGrade = exports.MultiplierGradeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var MultiplierGradeEnum;
(function (MultiplierGradeEnum) {
    MultiplierGradeEnum[MultiplierGradeEnum["OneGrade"] = 0] = "OneGrade";
    MultiplierGradeEnum[MultiplierGradeEnum["TwoGrade"] = 2] = "TwoGrade";
    MultiplierGradeEnum[MultiplierGradeEnum["ThreeGrade"] = 3] = "ThreeGrade";
    MultiplierGradeEnum[MultiplierGradeEnum["FourGrade"] = 4] = "FourGrade";
    MultiplierGradeEnum[MultiplierGradeEnum["SevenGrade"] = 5] = "SevenGrade";
    MultiplierGradeEnum[MultiplierGradeEnum["EightGrade"] = 6] = "EightGrade";
})(MultiplierGradeEnum = exports.MultiplierGradeEnum || (exports.MultiplierGradeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link MultiplierGradeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ChipSM16259.decompiled.cs:82
 */
exports.MultiplierGrade = (0, EnumFromString_1.default)(MultiplierGradeEnum, 'MultiplierGrade');
