// @novastar/native/build/main/generated/NSCardType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NSCardType = exports.NSCardTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var NSCardTypeEnum;
(function (NSCardTypeEnum) {
    NSCardTypeEnum[NSCardTypeEnum["Sender"] = 1] = "Sender";
    NSCardTypeEnum[NSCardTypeEnum["Controller"] = 257] = "Controller";
    NSCardTypeEnum[NSCardTypeEnum["HDMICard"] = 4353] = "HDMICard";
    NSCardTypeEnum[NSCardTypeEnum["Scanner"] = 16641] = "Scanner";
    NSCardTypeEnum[NSCardTypeEnum["TVCard"] = 8449] = "TVCard";
    NSCardTypeEnum[NSCardTypeEnum["FunctionCard"] = 33025] = "FunctionCard";
    NSCardTypeEnum[NSCardTypeEnum["VirtualSender"] = 12545] = "VirtualSender";
    NSCardTypeEnum[NSCardTypeEnum["NovaProHD"] = 24833] = "NovaProHD";
    NSCardTypeEnum[NSCardTypeEnum["NovaProHD_II"] = 24865] = "NovaProHD_II";
    NSCardTypeEnum[NSCardTypeEnum["VP200"] = 24834] = "VP200";
    NSCardTypeEnum[NSCardTypeEnum["V800"] = 24835] = "V800";
    NSCardTypeEnum[NSCardTypeEnum["V900"] = 24836] = "V900";
    NSCardTypeEnum[NSCardTypeEnum["VX2"] = 24837] = "VX2";
    NSCardTypeEnum[NSCardTypeEnum["VX4"] = 24838] = "VX4";
    NSCardTypeEnum[NSCardTypeEnum["VX4S"] = 24839] = "VX4S";
    NSCardTypeEnum[NSCardTypeEnum["V700"] = 24840] = "V700";
    NSCardTypeEnum[NSCardTypeEnum["Nova3DHD"] = 24848] = "Nova3DHD";
    NSCardTypeEnum[NSCardTypeEnum["HDMICard6P"] = 37121] = "HDMICard6P";
    NSCardTypeEnum[NSCardTypeEnum["E500"] = 4354] = "E500";
    NSCardTypeEnum[NSCardTypeEnum["VX5"] = 24841] = "VX5";
    NSCardTypeEnum[NSCardTypeEnum["MCTRL4K"] = 4355] = "MCTRL4K";
    NSCardTypeEnum[NSCardTypeEnum["K4"] = 24850] = "K4";
    NSCardTypeEnum[NSCardTypeEnum["K4S"] = 24851] = "K4S";
    NSCardTypeEnum[NSCardTypeEnum["K4A"] = 24866] = "K4A";
    NSCardTypeEnum[NSCardTypeEnum["VD43"] = 24852] = "VD43";
    NSCardTypeEnum[NSCardTypeEnum["VS1"] = 28929] = "VS1";
    NSCardTypeEnum[NSCardTypeEnum["VS2"] = 28930] = "VS2";
    NSCardTypeEnum[NSCardTypeEnum["VS3"] = 28931] = "VS3";
    NSCardTypeEnum[NSCardTypeEnum["VS4"] = 28932] = "VS4";
    NSCardTypeEnum[NSCardTypeEnum["TVctrl_230R"] = 24853] = "TVctrl_230R";
    NSCardTypeEnum[NSCardTypeEnum["D_DV18"] = 29185] = "D_DV18";
    NSCardTypeEnum[NSCardTypeEnum["VP200U"] = 28933] = "VP200U";
    NSCardTypeEnum[NSCardTypeEnum["VX2U"] = 24854] = "VX2U";
    NSCardTypeEnum[NSCardTypeEnum["VX4U"] = 24855] = "VX4U";
    NSCardTypeEnum[NSCardTypeEnum["CVT4K"] = 29187] = "CVT4K";
    NSCardTypeEnum[NSCardTypeEnum["CVT10"] = 29192] = "CVT10";
    NSCardTypeEnum[NSCardTypeEnum["MCTRLR5"] = 4356] = "MCTRLR5";
    NSCardTypeEnum[NSCardTypeEnum["K2U"] = 24856] = "K2U";
    NSCardTypeEnum[NSCardTypeEnum["K4U"] = 24857] = "K4U";
    NSCardTypeEnum[NSCardTypeEnum["VX2S"] = 24864] = "VX2S";
    NSCardTypeEnum[NSCardTypeEnum["D_HM14_4K"] = 29186] = "D_HM14_4K";
    NSCardTypeEnum[NSCardTypeEnum["JZ_AX100"] = 38401] = "JZ_AX100";
    NSCardTypeEnum[NSCardTypeEnum["T3"] = 38402] = "T3";
    NSCardTypeEnum[NSCardTypeEnum["T6"] = 38404] = "T6";
    NSCardTypeEnum[NSCardTypeEnum["T1"] = 38405] = "T1";
    NSCardTypeEnum[NSCardTypeEnum["T2"] = 38406] = "T2";
    NSCardTypeEnum[NSCardTypeEnum["T8"] = 38408] = "T8";
    NSCardTypeEnum[NSCardTypeEnum["JT100"] = 38409] = "JT100";
    NSCardTypeEnum[NSCardTypeEnum["T1_4G"] = 38410] = "T1_4G";
    NSCardTypeEnum[NSCardTypeEnum["T2_4G"] = 38411] = "T2_4G";
    NSCardTypeEnum[NSCardTypeEnum["T4"] = 38412] = "T4";
    NSCardTypeEnum[NSCardTypeEnum["T40"] = 38427] = "T40";
    NSCardTypeEnum[NSCardTypeEnum["T50"] = 38428] = "T50";
    NSCardTypeEnum[NSCardTypeEnum["T60"] = 38429] = "T60";
    NSCardTypeEnum[NSCardTypeEnum["T4A"] = 38420] = "T4A";
    NSCardTypeEnum[NSCardTypeEnum["TCC70"] = 38419] = "TCC70";
    NSCardTypeEnum[NSCardTypeEnum["LCC70"] = 38421] = "LCC70";
    NSCardTypeEnum[NSCardTypeEnum["JT20"] = 38430] = "JT20";
    NSCardTypeEnum[NSCardTypeEnum["TCC60_YJ"] = 38415] = "TCC60_YJ";
    NSCardTypeEnum[NSCardTypeEnum["TCC50"] = 38413] = "TCC50";
    NSCardTypeEnum[NSCardTypeEnum["Thundervier_S1"] = 29442] = "Thundervier_S1";
    NSCardTypeEnum[NSCardTypeEnum["KT3"] = 4358] = "KT3";
    NSCardTypeEnum[NSCardTypeEnum["MCTRL1600"] = 4357] = "MCTRL1600";
    NSCardTypeEnum[NSCardTypeEnum["SD2000E"] = 4866] = "SD2000E";
    NSCardTypeEnum[NSCardTypeEnum["MCTRL660_Pro"] = 4359] = "MCTRL660_Pro";
    NSCardTypeEnum[NSCardTypeEnum["MCTRL660_ROE"] = 4360] = "MCTRL660_ROE";
    NSCardTypeEnum[NSCardTypeEnum["MCTRL600_A"] = 4867] = "MCTRL600_A";
    NSCardTypeEnum[NSCardTypeEnum["VX6"] = 25089] = "VX6";
    NSCardTypeEnum[NSCardTypeEnum["J10"] = 28939] = "J10";
    NSCardTypeEnum[NSCardTypeEnum["N6"] = 28940] = "N6";
    NSCardTypeEnum[NSCardTypeEnum["N9"] = 28941] = "N9";
    NSCardTypeEnum[NSCardTypeEnum["VS7"] = 28935] = "VS7";
    NSCardTypeEnum[NSCardTypeEnum["VS8"] = 28936] = "VS8";
    NSCardTypeEnum[NSCardTypeEnum["VX5S"] = 25090] = "VX5S";
    NSCardTypeEnum[NSCardTypeEnum["VX6S"] = 25091] = "VX6S";
    NSCardTypeEnum[NSCardTypeEnum["VM_Dual"] = 28938] = "VM_Dual";
    NSCardTypeEnum[NSCardTypeEnum["V760"] = 24868] = "V760";
    NSCardTypeEnum[NSCardTypeEnum["VS0"] = 24867] = "VS0";
    NSCardTypeEnum[NSCardTypeEnum["PJ1"] = 4361] = "PJ1";
    NSCardTypeEnum[NSCardTypeEnum["K16"] = 25092] = "K16";
    NSCardTypeEnum[NSCardTypeEnum["ProUHD_Jr"] = 25093] = "ProUHD_Jr";
    NSCardTypeEnum[NSCardTypeEnum["K6s"] = 25094] = "K6s";
    NSCardTypeEnum[NSCardTypeEnum["C1"] = 29697] = "C1";
    NSCardTypeEnum[NSCardTypeEnum["D_DP24_4K"] = 29188] = "D_DP24_4K";
    NSCardTypeEnum[NSCardTypeEnum["MCTRL4K_ViewPro"] = 28937] = "MCTRL4K_ViewPro";
    NSCardTypeEnum[NSCardTypeEnum["C3"] = 116] = "C3";
    NSCardTypeEnum[NSCardTypeEnum["VD28"] = 24869] = "VD28";
    NSCardTypeEnum[NSCardTypeEnum["ProUHD"] = 29956] = "ProUHD";
    NSCardTypeEnum[NSCardTypeEnum["ATM"] = 29190] = "ATM";
    NSCardTypeEnum[NSCardTypeEnum["E3000Ctrl"] = 30370] = "E3000Ctrl";
    NSCardTypeEnum[NSCardTypeEnum["NovaPro_UHD"] = 34052] = "NovaPro_UHD";
    NSCardTypeEnum[NSCardTypeEnum["SgLkDVIIp"] = 30209] = "SgLkDVIIp";
    NSCardTypeEnum[NSCardTypeEnum["SgLkDVIOp"] = 30273] = "SgLkDVIOp";
    NSCardTypeEnum[NSCardTypeEnum["E3000BkBd"] = 30402] = "E3000BkBd";
    NSCardTypeEnum[NSCardTypeEnum["ThreeGSDIIp"] = 30215] = "ThreeGSDIIp";
    NSCardTypeEnum[NSCardTypeEnum["DP11Ip"] = 30216] = "DP11Ip";
    NSCardTypeEnum[NSCardTypeEnum["HDMI13Op"] = 30277] = "HDMI13Op";
    NSCardTypeEnum[NSCardTypeEnum["HDMI13Ip"] = 30214] = "HDMI13Ip";
    NSCardTypeEnum[NSCardTypeEnum["F2Ctrl"] = 29955] = "F2Ctrl";
    NSCardTypeEnum[NSCardTypeEnum["DLDVIIp"] = 30211] = "DLDVIIp";
    NSCardTypeEnum[NSCardTypeEnum["DP12Ip"] = 30213] = "DP12Ip";
    NSCardTypeEnum[NSCardTypeEnum["D12"] = 29958] = "D12";
    NSCardTypeEnum[NSCardTypeEnum["VE7"] = 29189] = "VE7";
    NSCardTypeEnum[NSCardTypeEnum["HDMI20DP12IP"] = 30217] = "HDMI20DP12IP";
    NSCardTypeEnum[NSCardTypeEnum["E8000_1"] = 29957] = "E8000_1";
    NSCardTypeEnum[NSCardTypeEnum["E8000_2"] = 30279] = "E8000_2";
    NSCardTypeEnum[NSCardTypeEnum["E8000_3"] = 30280] = "E8000_3";
    NSCardTypeEnum[NSCardTypeEnum["V960"] = 24870] = "V960";
    NSCardTypeEnum[NSCardTypeEnum["V1060"] = 25095] = "V1060";
    NSCardTypeEnum[NSCardTypeEnum["UTA3000"] = 25098] = "UTA3000";
    NSCardTypeEnum[NSCardTypeEnum["TVSend_4K"] = 8450] = "TVSend_4K";
    NSCardTypeEnum[NSCardTypeEnum["SX100"] = 20993] = "SX100";
    NSCardTypeEnum[NSCardTypeEnum["KT8"] = 4362] = "KT8";
    NSCardTypeEnum[NSCardTypeEnum["VX1"] = 24871] = "VX1";
    NSCardTypeEnum[NSCardTypeEnum["VX400S"] = 24872] = "VX400S";
    NSCardTypeEnum[NSCardTypeEnum["V1260"] = 25096] = "V1260";
    NSCardTypeEnum[NSCardTypeEnum["Prime_4K"] = 25097] = "Prime_4K";
    NSCardTypeEnum[NSCardTypeEnum["H9"] = 31558] = "H9";
    NSCardTypeEnum[NSCardTypeEnum["E3000_ControlCard"] = 29960] = "E3000_ControlCard";
    NSCardTypeEnum[NSCardTypeEnum["V10601"] = 24615] = "V10601";
    NSCardTypeEnum[NSCardTypeEnum["D_4Input"] = 30465] = "D_4Input";
    NSCardTypeEnum[NSCardTypeEnum["D_4HDMIIput"] = 30470] = "D_4HDMIIput";
    NSCardTypeEnum[NSCardTypeEnum["D_43GSDIIput"] = 30471] = "D_43GSDIIput";
    NSCardTypeEnum[NSCardTypeEnum["D_4DP11Iput"] = 30472] = "D_4DP11Iput";
    NSCardTypeEnum[NSCardTypeEnum["D_1HDMI20_1DP12Iput"] = 30473] = "D_1HDMI20_1DP12Iput";
    NSCardTypeEnum[NSCardTypeEnum["D_1HDMI20_1DP12OutPut"] = 30532] = "D_1HDMI20_1DP12OutPut";
    NSCardTypeEnum[NSCardTypeEnum["D_4HDMIOutPut"] = 30533] = "D_4HDMIOutPut";
    NSCardTypeEnum[NSCardTypeEnum["D_4DVIOutPut"] = 30534] = "D_4DVIOutPut";
    NSCardTypeEnum[NSCardTypeEnum["F_4DVIInput"] = 30721] = "F_4DVIInput";
    NSCardTypeEnum[NSCardTypeEnum["F_4HDMI13Iput"] = 30726] = "F_4HDMI13Iput";
    NSCardTypeEnum[NSCardTypeEnum["F_43GSDIIput"] = 30727] = "F_43GSDIIput";
    NSCardTypeEnum[NSCardTypeEnum["F_4DP11Iput"] = 30728] = "F_4DP11Iput";
    NSCardTypeEnum[NSCardTypeEnum["F_1HDMI20_1DP12Iput"] = 30729] = "F_1HDMI20_1DP12Iput";
    NSCardTypeEnum[NSCardTypeEnum["F_1HDMI20_1DP12OutPut"] = 30788] = "F_1HDMI20_1DP12OutPut";
    NSCardTypeEnum[NSCardTypeEnum["F_4HDMI13OutPut"] = 30789] = "F_4HDMI13OutPut";
    NSCardTypeEnum[NSCardTypeEnum["F_4DVIOutPut"] = 30790] = "F_4DVIOutPut";
    NSCardTypeEnum[NSCardTypeEnum["CR4K_3D"] = 4364] = "CR4K_3D";
    NSCardTypeEnum[NSCardTypeEnum["H_4xSLDVI_I"] = 31488] = "H_4xSLDVI_I";
    NSCardTypeEnum[NSCardTypeEnum["H_4xDVI_O"] = 31552] = "H_4xDVI_O";
    NSCardTypeEnum[NSCardTypeEnum["H_4xHDMI_O"] = 31553] = "H_4xHDMI_O";
    NSCardTypeEnum[NSCardTypeEnum["H_AUX_9U"] = 31616] = "H_AUX_9U";
    NSCardTypeEnum[NSCardTypeEnum["LP10SD1600"] = 4365] = "LP10SD1600";
    NSCardTypeEnum[NSCardTypeEnum["RSDIHDMIDPIpt"] = 31233] = "RSDIHDMIDPIpt";
    NSCardTypeEnum[NSCardTypeEnum["RHDMIFiberOpt"] = 31296] = "RHDMIFiberOpt";
    NSCardTypeEnum[NSCardTypeEnum["RSDIHDMIProcess"] = 31297] = "RSDIHDMIProcess";
    NSCardTypeEnum[NSCardTypeEnum["H9CtrlCard"] = 29962] = "H9CtrlCard";
    NSCardTypeEnum[NSCardTypeEnum["H5CtrlCard"] = 29963] = "H5CtrlCard";
    NSCardTypeEnum[NSCardTypeEnum["HMVRCard"] = 31559] = "HMVRCard";
    NSCardTypeEnum[NSCardTypeEnum["HIPCCard"] = 31493] = "HIPCCard";
    NSCardTypeEnum[NSCardTypeEnum["F4Lite"] = 29964] = "F4Lite";
    NSCardTypeEnum[NSCardTypeEnum["HHDMI20DP12IMCard"] = 31491] = "HHDMI20DP12IMCard";
    NSCardTypeEnum[NSCardTypeEnum["VX16s"] = 25099] = "VX16s";
    NSCardTypeEnum[NSCardTypeEnum["Mee200"] = 8451] = "Mee200";
    NSCardTypeEnum[NSCardTypeEnum["Mee400"] = 8452] = "Mee400";
    NSCardTypeEnum[NSCardTypeEnum["Mee400C"] = 8453] = "Mee400C";
    NSCardTypeEnum[NSCardTypeEnum["KT16E"] = 4371] = "KT16E";
    NSCardTypeEnum[NSCardTypeEnum["KT8E"] = 4368] = "KT8E";
    NSCardTypeEnum[NSCardTypeEnum["K4S_N"] = 24873] = "K4S_N";
    NSCardTypeEnum[NSCardTypeEnum["VX4S_N"] = 24874] = "VX4S_N";
    NSCardTypeEnum[NSCardTypeEnum["H_20"] = 31560] = "H_20";
    NSCardTypeEnum[NSCardTypeEnum["T30"] = 38432] = "T30";
    NSCardTypeEnum[NSCardTypeEnum["T80"] = 38417] = "T80";
    NSCardTypeEnum[NSCardTypeEnum["JT50"] = 38418] = "JT50";
    NSCardTypeEnum[NSCardTypeEnum["V1160"] = 25101] = "V1160";
    NSCardTypeEnum[NSCardTypeEnum["Vunit2000"] = 29193] = "Vunit2000";
    NSCardTypeEnum[NSCardTypeEnum["VX1000"] = 25100] = "VX1000";
    NSCardTypeEnum[NSCardTypeEnum["VX600"] = 25102] = "VX600";
    NSCardTypeEnum[NSCardTypeEnum["V1060n"] = 25103] = "V1060n";
    NSCardTypeEnum[NSCardTypeEnum["KT16C"] = 4370] = "KT16C";
    NSCardTypeEnum[NSCardTypeEnum["KT16E_CE"] = 4870] = "KT16E_CE";
    NSCardTypeEnum[NSCardTypeEnum["T3H"] = 38422] = "T3H";
    NSCardTypeEnum[NSCardTypeEnum["T4H"] = 38423] = "T4H";
    NSCardTypeEnum[NSCardTypeEnum["T6H"] = 38424] = "T6H";
    NSCardTypeEnum[NSCardTypeEnum["T8H"] = 38425] = "T8H";
    NSCardTypeEnum[NSCardTypeEnum["JT100H"] = 38426] = "JT100H";
    NSCardTypeEnum[NSCardTypeEnum["MP8_JF"] = 4871] = "MP8_JF";
    NSCardTypeEnum[NSCardTypeEnum["MTX4000"] = 4480] = "MTX4000";
    NSCardTypeEnum[NSCardTypeEnum["TwoInOne"] = 39321] = "TwoInOne";
    NSCardTypeEnum[NSCardTypeEnum["JT200"] = 38433] = "JT200";
    NSCardTypeEnum[NSCardTypeEnum["Unknown"] = 255] = "Unknown";
})(NSCardTypeEnum = exports.NSCardTypeEnum || (exports.NSCardTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link NSCardTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:61049
 */
exports.NSCardType = (0, EnumFromString_1.default)(NSCardTypeEnum, 'NSCardType');
