// @novastar/native/build/main/generated/OneSmartBrightEasyConfig.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OneSmartBrightEasyConfig = void 0;
const t = __importStar(require("io-ts"));
const io_ts_types_1 = require("io-ts-types");
const common = __importStar(require("../lib/common"));
const SmartBrightAdjustType_1 = require("./SmartBrightAdjustType");
/**
 * @category Codecs
 * @desc Codec for interface {@link OneSmartBrightEasyConfig}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:58176
 */
exports.OneSmartBrightEasyConfig = t.intersection([
    t.type({
        IsConfigEnable: common.Bool_true,
        CustomDayCollection: common.XMLArray(t.string, 'DayOfWeek'),
    }),
    t.partial({
        StartTime: io_ts_types_1.DateFromISOString,
        ScheduleType: SmartBrightAdjustType_1.SmartBrightAdjustType,
        IsColorTemp: common.Bool,
        ColorTempValue: t.string,
        BrightPercent: common.Numeric,
        IsEnableGamma: common.Bool,
        GammaValue: common.Numeric,
        IsHasAdjusted: common.Bool,
    }),
], 'OneSmartBrightEasyConfig');
