// @novastar/native/build/main/generated/OutDeviceType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutDeviceType = exports.OutDeviceTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var OutDeviceTypeEnum;
(function (OutDeviceTypeEnum) {
    OutDeviceTypeEnum[OutDeviceTypeEnum["LightSensor"] = 0] = "LightSensor";
    OutDeviceTypeEnum[OutDeviceTypeEnum["TemAndHumSesor"] = 1] = "TemAndHumSesor";
    OutDeviceTypeEnum[OutDeviceTypeEnum["Emitter_3D"] = 2] = "Emitter_3D";
    OutDeviceTypeEnum[OutDeviceTypeEnum["TemperatureSensor"] = 3] = "TemperatureSensor";
    OutDeviceTypeEnum[OutDeviceTypeEnum["HumiditySensor"] = 4] = "HumiditySensor";
    OutDeviceTypeEnum[OutDeviceTypeEnum["None"] = 255] = "None";
})(OutDeviceTypeEnum = exports.OutDeviceTypeEnum || (exports.OutDeviceTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link OutDeviceTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1104
 */
exports.OutDeviceType = (0, EnumFromString_1.default)(OutDeviceTypeEnum, 'OutDeviceType');
