// @novastar/native/build/main/generated/PicturePrestoreConfig.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PicturePrestoreConfig = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const CabinetStoreEffect_1 = require("./CabinetStoreEffect");
const NoSignalMode_1 = require("./NoSignalMode");
const PictureShowMode_1 = require("./PictureShowMode");
/**
 * @category Codecs
 * @desc Codec for interface {@link PicturePrestoreConfig}
 * @see Automatically generated from ../decompiled/Nova.GigabitController.ProgramInnerData.decompiled.cs:739
 */
exports.PicturePrestoreConfig = t.intersection([
    t.type({
        PictureName: common.string_empty,
        StoreEffect: common.withDefault(CabinetStoreEffect_1.CabinetStoreEffect, 'FullScreen'),
        FullScreenShowMode: common.withDefault(PictureShowMode_1.PictureShowMode, 'Stretch'),
        OneCabinetShowMode: common.withDefault(PictureShowMode_1.PictureShowMode, 'Stretch'),
        IsUseStartPicture: common.Bool_false,
        StartPictureTime: common.Int32_2,
        OfflineMode: common.withDefault(NoSignalMode_1.NoSignalMode, 'BlackScreen'),
        NoDviMode: common.withDefault(NoSignalMode_1.NoSignalMode, 'BlackScreen'),
    }),
    t.partial({}),
], 'PicturePrestoreConfig');
