// @novastar/native/build/main/generated/PointDetectConfig.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PointDetectConfig = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ChipType_1 = require("./ChipType");
const DetectPointType_1 = require("./DetectPointType");
const ThresholdType_1 = require("./ThresholdType");
/**
 * @category Codecs
 * @desc Codec for interface {@link PointDetectConfig}
 * @see Automatically generated from ../decompiled/Nova.GigabitController.ProgramInnerData.decompiled.cs:497
 */
exports.PointDetectConfig = t.intersection([
    t.type({
        ScreenSN: common.string_empty,
        ComPort: common.string_empty,
        ScreenIndex: common.Int32_0,
        ScreenChipType: common.withDefault(ChipType_1.ChipType, 'Unknown'),
        DetectType: common.withDefault(DetectPointType_1.DetectPointType, 'None'),
        DetectThreshold: common.withDefault(ThresholdType_1.ThresholdType, 'None'),
        IsUseGain: common.Bool_true,
        DetectRedGainStep: common.Int32_32,
        DetectGreenGainStep: common.Int32_32,
        DetectBlueGainStep: common.Int32_32,
        IsSynchronousGain: common.Bool_true,
        IsUsedResistor: common.Bool_false,
        RedResistor: common.Int32_300,
        GreenResistor: common.Int32_300,
        BlueResistor: common.Int32_300,
    }),
    t.partial({}),
], 'PointDetectConfig');
