// @novastar/native/build/main/generated/PortScanBoardInfo.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PortScanBoardInfo = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ScanBoardConnectType_1 = require("./ScanBoardConnectType");
/**
 * @category Codecs
 * @desc Codec for interface {@link PortScanBoardInfo}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:53804
 */
exports.PortScanBoardInfo = t.partial({
    ConnectType: ScanBoardConnectType_1.ScanBoardConnectType,
    PortIndex: common.UInt8,
    ScanBdBegColNo: common.UInt16,
    ScanBdBegRowNo: common.UInt16,
    ScanBdEndColNo: common.UInt16,
    ScanBdEndRowNo: common.UInt16,
}, 'PortScanBoardInfo');
