// @novastar/native/build/main/generated/RegisterGroupType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisterGroupType = exports.RegisterGroupTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var RegisterGroupTypeEnum;
(function (RegisterGroupTypeEnum) {
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["FirstGroup"] = 0] = "FirstGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["SecondGroup"] = 1] = "SecondGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["ThirdGroup"] = 2] = "ThirdGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["FourthGroup"] = 3] = "FourthGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["OldChipDefaultGroup"] = 4] = "OldChipDefaultGroup";
    RegisterGroupTypeEnum[RegisterGroupTypeEnum["SpecialGroup"] = 5] = "SpecialGroup";
})(RegisterGroupTypeEnum = exports.RegisterGroupTypeEnum || (exports.RegisterGroupTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link RegisterGroupTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:1447
 */
exports.RegisterGroupType = (0, EnumFromString_1.default)(RegisterGroupTypeEnum, 'RegisterGroupType');
