// @novastar/native/build/main/generated/RegistorCfgGroupType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistorCfgGroupType = exports.RegistorCfgGroupTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var RegistorCfgGroupTypeEnum;
(function (RegistorCfgGroupTypeEnum) {
    RegistorCfgGroupTypeEnum[RegistorCfgGroupTypeEnum["Group1"] = 0] = "Group1";
    RegistorCfgGroupTypeEnum[RegistorCfgGroupTypeEnum["Group2"] = 1] = "Group2";
    RegistorCfgGroupTypeEnum[RegistorCfgGroupTypeEnum["Group3"] = 2] = "Group3";
    RegistorCfgGroupTypeEnum[RegistorCfgGroupTypeEnum["Group4"] = 3] = "Group4";
    RegistorCfgGroupTypeEnum[RegistorCfgGroupTypeEnum["Group5"] = 4] = "Group5";
    RegistorCfgGroupTypeEnum[RegistorCfgGroupTypeEnum["Group6"] = 5] = "Group6";
})(RegistorCfgGroupTypeEnum = exports.RegistorCfgGroupTypeEnum || (exports.RegistorCfgGroupTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link RegistorCfgGroupTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:61269
 */
exports.RegistorCfgGroupType = (0, EnumFromString_1.default)(RegistorCfgGroupTypeEnum, 'RegistorCfgGroupType');
