// @novastar/native/build/main/generated/RegularCabinetProperty.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegularCabinetProperty = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const CabinetType_1 = require("./CabinetType");
const LEDModuleProperty_1 = require("./LEDModuleProperty");
const ModuleCascadeDiretion_1 = require("./ModuleCascadeDiretion");
/**
 * @category Codecs
 * @desc Codec for interface {@link RegularCabinetProperty}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:23989
 */
exports.RegularCabinetProperty = t.intersection([
    t.type({
        Name: common.string_empty,
        CabinetVersion: common.string_20,
    }),
    t.partial({
        ModCascadeType: ModuleCascadeDiretion_1.ModuleCascadeDiretion,
        CabinetType: CabinetType_1.CabinetType,
        Width: common.Int32,
        Height: common.Int32,
        PointTableData: common.Base64,
        ModColsInCabinet: common.UInt8,
        ModRowsInCabinet: common.UInt8,
        ModuleProperty: LEDModuleProperty_1.LEDModuleProperty,
    }),
], 'RegularCabinetProperty');
