// @novastar/native/build/main/generated/ResetAndSwitchCommandType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResetAndSwitchCommandType = exports.ResetAndSwitchCommandTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ResetAndSwitchCommandTypeEnum;
(function (ResetAndSwitchCommandTypeEnum) {
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["DEFAULT_PARAM"] = 0] = "DEFAULT_PARAM";
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["NO_ACTION_PARAM"] = 1] = "NO_ACTION_PARAM";
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["RESET_OPERATION_PARAM"] = 2] = "RESET_OPERATION_PARAM";
    ResetAndSwitchCommandTypeEnum[ResetAndSwitchCommandTypeEnum["START_ACTION_PARAM"] = 3] = "START_ACTION_PARAM";
})(ResetAndSwitchCommandTypeEnum = exports.ResetAndSwitchCommandTypeEnum || (exports.ResetAndSwitchCommandTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ResetAndSwitchCommandTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1324
 */
exports.ResetAndSwitchCommandType = (0, EnumFromString_1.default)(ResetAndSwitchCommandTypeEnum, 'ResetAndSwitchCommandType');
