// @novastar/native/build/main/generated/RestoreType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestoreType = exports.RestoreTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var RestoreTypeEnum;
(function (RestoreTypeEnum) {
    RestoreTypeEnum[RestoreTypeEnum["Sender"] = 0] = "Sender";
    RestoreTypeEnum[RestoreTypeEnum["Scander"] = 1] = "Scander";
    RestoreTypeEnum[RestoreTypeEnum["AllSystem"] = 2] = "AllSystem";
    RestoreTypeEnum[RestoreTypeEnum["None"] = 3] = "None";
})(RestoreTypeEnum = exports.RestoreTypeEnum || (exports.RestoreTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link RestoreTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:59502
 */
exports.RestoreType = (0, EnumFromString_1.default)(RestoreTypeEnum, 'RestoreType');
