// @novastar/native/build/main/generated/RotateAngle.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RotateAngle = exports.RotateAngleEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var RotateAngleEnum;
(function (RotateAngleEnum) {
    RotateAngleEnum[RotateAngleEnum["R_90"] = 0] = "R_90";
    RotateAngleEnum[RotateAngleEnum["R_180"] = 1] = "R_180";
    RotateAngleEnum[RotateAngleEnum["R_270"] = 2] = "R_270";
    RotateAngleEnum[RotateAngleEnum["R_360"] = 3] = "R_360";
})(RotateAngleEnum = exports.RotateAngleEnum || (exports.RotateAngleEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link RotateAngleEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1194
 */
exports.RotateAngle = (0, EnumFromString_1.default)(RotateAngleEnum, 'RotateAngle');
