// @novastar/native/build/main/generated/ScanBoardRegionInfo.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScanBoardRegionInfo = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const CabinetCorrectLocationSize_1 = require("./CabinetCorrectLocationSize");
const DviSelectMode_1 = require("./DviSelectMode");
/**
 * @category Codecs
 * @desc Codec for interface {@link ScanBoardRegionInfo}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:53392
 */
exports.ScanBoardRegionInfo = t.intersection([
    t.type({
        DviSelect: common.withDefault(DviSelectMode_1.DviSelectMode, 'DVI'),
        DVIIndex: common.UInt8_255,
        Width: common.UInt16_32,
        Height: common.UInt16_32,
        TagInfoColor: common.withDefault(t.string, 'get_Red()'),
    }),
    t.partial({
        XInPort: common.UInt16,
        YInPort: common.UInt16,
        SenderIndex: common.UInt8,
        PortIndex: common.UInt8,
        ConnectIndex: common.UInt16,
        X: common.UInt16,
        Y: common.UInt16,
        IsHasMrvID: common.Bool,
        MrvID: t.string,
        FreeRoutConnect: common.UInt16,
        MrvIDObject: t.UnknownRecord,
        CabinetCorrectLocationSize: CabinetCorrectLocationSize_1.CabinetCorrectLocationSize,
        RowIndexInScreen: common.Int32,
        ColIndexInScreen: common.Int32,
        TagInfo: common.Bool,
    }),
], 'ScanBoardRegionInfo');
