// @novastar/native/build/main/generated/ScannerFunctionListEnum.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScannerFunctionListEnum = exports.ScannerFunctionListEnumEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ScannerFunctionListEnumEnum;
(function (ScannerFunctionListEnumEnum) {
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["NoSenderCardMode"] = 0] = "NoSenderCardMode";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ErrorBitDetect"] = 1] = "ErrorBitDetect";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["DobulePowerBackUp"] = 2] = "DobulePowerBackUp";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["DobuleCardBackUp"] = 3] = "DobuleCardBackUp";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["LinkMonitorCard"] = 4] = "LinkMonitorCard";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["HubMonitor"] = 5] = "HubMonitor";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["SmartModule"] = 6] = "SmartModule";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ModuleFlash"] = 7] = "ModuleFlash";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["LargeLoad"] = 8] = "LargeLoad";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ClearViewMode"] = 9] = "ClearViewMode";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["Is18BitMode"] = 10] = "Is18BitMode";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["MappingMode"] = 11] = "MappingMode";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["NewAdjustLine"] = 12] = "NewAdjustLine";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["CorrectionCofeDobuleBackUp"] = 13] = "CorrectionCofeDobuleBackUp";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["AutoCorrection"] = 14] = "AutoCorrection";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["CorrectionMode"] = 15] = "CorrectionMode";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["IsSupport18BitGamma"] = 16] = "IsSupport18BitGamma";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["SingleGamma"] = 17] = "SingleGamma";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ConfigParameterDobuleBackUp"] = 18] = "ConfigParameterDobuleBackUp";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ProgramRead"] = 19] = "ProgramRead";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["HDRMode"] = 20] = "HDRMode";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["LowDelay"] = 21] = "LowDelay";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["AnyAngleRotation"] = 22] = "AnyAngleRotation";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["Rotation90"] = 23] = "Rotation90";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["FineGrayAdjust"] = 24] = "FineGrayAdjust";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["GrayFirstAdjustBright"] = 25] = "GrayFirstAdjustBright";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["UpLoadCorrectionCofeSpeedUp"] = 26] = "UpLoadCorrectionCofeSpeedUp";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["MatrixColorMangement"] = 27] = "MatrixColorMangement";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["XBitMode"] = 28] = "XBitMode";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["HardwarePlayScreenAdjustLine"] = 29] = "HardwarePlayScreenAdjustLine";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ScannerParameterSegmentSolidified"] = 30] = "ScannerParameterSegmentSolidified";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ProgramIsSimplyVersion"] = 31] = "ProgramIsSimplyVersion";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["FineGrayAdjustVer3"] = 32] = "FineGrayAdjustVer3";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["MatrixColorMangement2"] = 33] = "MatrixColorMangement2";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["FactoryBackUp"] = 34] = "FactoryBackUp";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["CorrectionCompress"] = 35] = "CorrectionCompress";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["ABL"] = 36] = "ABL";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["MagicGray"] = 37] = "MagicGray";
    ScannerFunctionListEnumEnum[ScannerFunctionListEnumEnum["UnKnow"] = 65535] = "UnKnow";
})(ScannerFunctionListEnumEnum = exports.ScannerFunctionListEnumEnum || (exports.ScannerFunctionListEnumEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ScannerFunctionListEnumEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1410
 */
exports.ScannerFunctionListEnum = (0, EnumFromString_1.default)(ScannerFunctionListEnumEnum, 'ScannerFunctionListEnum');
