// @novastar/native/build/main/generated/ScanBoardConnectType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScanBoardConnectType = exports.ScanBoardConnectTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ScanBoardConnectTypeEnum;
(function (ScanBoardConnectTypeEnum) {
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftTop_Horizontal"] = 0] = "LeftTop_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftBottom_Horizontal"] = 1] = "LeftBottom_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightTop_Horizontal"] = 2] = "RightTop_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightBottom_Horizontal"] = 3] = "RightBottom_Horizontal";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftTop_Vertical"] = 4] = "LeftTop_Vertical";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["LeftBottom_Vertical"] = 5] = "LeftBottom_Vertical";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightTop_Vertical"] = 6] = "RightTop_Vertical";
    ScanBoardConnectTypeEnum[ScanBoardConnectTypeEnum["RightBottom_Vertical"] = 7] = "RightBottom_Vertical";
})(ScanBoardConnectTypeEnum = exports.ScanBoardConnectTypeEnum || (exports.ScanBoardConnectTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ScanBoardConnectTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:56583
 */
exports.ScanBoardConnectType = (0, EnumFromString_1.default)(ScanBoardConnectTypeEnum, 'ScanBoardConnectType');
