// @novastar/native/build/main/generated/ScreenDataInSoftSpace.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScreenDataInSoftSpace = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const CabinetInDevice_1 = require("./CabinetInDevice");
const DviSelectMode_1 = require("./DviSelectMode");
const LEDDisplyType_1 = require("./LEDDisplyType");
const OnePortLoadInfo_1 = require("./OnePortLoadInfo");
const VirtualModeType_1 = require("./VirtualModeType");
/**
 * @category Codecs
 * @desc Codec for interface {@link ScreenDataInSoftSpace}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.CommonInfoAccessor.decompiled.cs:1019
 */
exports.ScreenDataInSoftSpace = t.intersection([
    t.type({
        UUID: common.string_empty,
        DviSelect: common.withDefault(DviSelectMode_1.DviSelectMode, 'DVI'),
        OnePortLoadInfo: common.XMLArray(OnePortLoadInfo_1.OnePortLoadInfo, 'OnePortLoadInfo'),
        CabinetInDevice: common.XMLArray(CabinetInDevice_1.CabinetInDevice, 'CabinetInDevice'),
    }),
    t.partial({
        ScrType: LEDDisplyType_1.LEDDisplyType,
        VirMode: VirtualModeType_1.VirtualModeType,
        ScrX: common.Int32,
        ScrY: common.Int32,
        CabinetCol: common.UInt16,
        CabinetRow: common.UInt16,
        PortCols: common.UInt8,
        PortRows: common.UInt8,
        DeviceID: common.UInt8,
        CabinetWidth: common.UInt16,
        CabinetHeight: common.UInt16,
        ScreenIndex: common.Int32,
        DVIlist: t.record(common.Int32, common.PointFromString),
    }),
], 'ScreenDataInSoftSpace');
