// @novastar/native/build/main/generated/SenderFunctionListEnum.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SenderFunctionListEnum = exports.SenderFunctionListEnumEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var SenderFunctionListEnumEnum;
(function (SenderFunctionListEnumEnum) {
    SenderFunctionListEnumEnum[SenderFunctionListEnumEnum["ABL"] = 0] = "ABL";
    SenderFunctionListEnumEnum[SenderFunctionListEnumEnum["ITMO"] = 1] = "ITMO";
    SenderFunctionListEnumEnum[SenderFunctionListEnumEnum["EDE"] = 2] = "EDE";
    SenderFunctionListEnumEnum[SenderFunctionListEnumEnum["FreeRoute"] = 3] = "FreeRoute";
    SenderFunctionListEnumEnum[SenderFunctionListEnumEnum["UnKnow"] = 65535] = "UnKnow";
})(SenderFunctionListEnumEnum = exports.SenderFunctionListEnumEnum || (exports.SenderFunctionListEnumEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link SenderFunctionListEnumEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1453
 */
exports.SenderFunctionListEnum = (0, EnumFromString_1.default)(SenderFunctionListEnumEnum, 'SenderFunctionListEnum');
