// @novastar/native/build/main/generated/SetBrightData.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SetBrightData = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const AdjustMode_1 = require("./AdjustMode");
const LightSensorData_1 = require("./LightSensorData");
/**
 * @category Codecs
 * @desc Codec for interface {@link SetBrightData}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:31
 */
exports.SetBrightData = t.intersection([
    t.type({
        PosNumber: common.UInt8_255,
        SectionNumber: common.UInt8_255,
        MaxLuxData: common.UInt16_65535,
        MinLuxData: common.UInt16_65535,
        MaxAdjustBright: common.UInt8_255,
        MinAdjustBright: common.UInt8_255,
        LightSensorDataList: common.XMLArray(LightSensorData_1.LightSensorData, 'LightSensorData'),
    }),
    t.partial({
        IsHWAutoBright: common.Bool,
        IsConfigurationSetted: common.Bool,
        SelectAdjustMode: AdjustMode_1.AdjustMode,
    }),
], 'SetBrightData');
