// @novastar/native/build/main/generated/SocketModeType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SocketModeType = exports.SocketModeTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var SocketModeTypeEnum;
(function (SocketModeTypeEnum) {
    SocketModeTypeEnum[SocketModeTypeEnum["LessOrEqual4_Data2"] = 0] = "LessOrEqual4_Data2";
    SocketModeTypeEnum[SocketModeTypeEnum["LessOrEqual4_Data1"] = 1] = "LessOrEqual4_Data1";
    SocketModeTypeEnum[SocketModeTypeEnum["LessOrEqual8_Data1"] = 2] = "LessOrEqual8_Data1";
    SocketModeTypeEnum[SocketModeTypeEnum["LessOrEqual8_Data2"] = 3] = "LessOrEqual8_Data2";
    SocketModeTypeEnum[SocketModeTypeEnum["MoreThan8"] = 4] = "MoreThan8";
    SocketModeTypeEnum[SocketModeTypeEnum["Unknown"] = 255] = "Unknown";
})(SocketModeTypeEnum = exports.SocketModeTypeEnum || (exports.SocketModeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link SocketModeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:736
 */
exports.SocketModeType = (0, EnumFromString_1.default)(SocketModeTypeEnum, 'SocketModeType');
