// @novastar/native/build/main/generated/SoftwareSpaceBaseAddress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:59842
 */
var SoftwareSpaceBaseAddress;
(function (SoftwareSpaceBaseAddress) {
    SoftwareSpaceBaseAddress[SoftwareSpaceBaseAddress["BASE_ADDRESS"] = 0] = "BASE_ADDRESS";
    SoftwareSpaceBaseAddress[SoftwareSpaceBaseAddress["DVI_BASE_ADDRESS"] = 54] = "DVI_BASE_ADDRESS";
    SoftwareSpaceBaseAddress[SoftwareSpaceBaseAddress["SCREEN_BASE_ADDRESS"] = 182] = "SCREEN_BASE_ADDRESS";
    SoftwareSpaceBaseAddress[SoftwareSpaceBaseAddress["REDUNDANCY_BASE_ADDRESS"] = 413696] = "REDUNDANCY_BASE_ADDRESS";
    SoftwareSpaceBaseAddress[SoftwareSpaceBaseAddress["MODULATION_BASE_ADDRESS"] = 417792] = "MODULATION_BASE_ADDRESS";
    SoftwareSpaceBaseAddress[SoftwareSpaceBaseAddress["CORRECTION_BASE_ADDRESS"] = 0] = "CORRECTION_BASE_ADDRESS";
})(SoftwareSpaceBaseAddress || (SoftwareSpaceBaseAddress = {}));
exports.default = SoftwareSpaceBaseAddress;
