// @novastar/native/build/main/generated/SystemParameterConfig.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemParameterConfig = void 0;
const t = __importStar(require("io-ts"));
const common = __importStar(require("../lib/common"));
const ScanBoardProperty_1 = require("./ScanBoardProperty");
const SenderRedundancyInfo_1 = require("./SenderRedundancyInfo");
/**
 * @category Codecs
 * @desc Codec for interface {@link SystemParameterConfig}
 * @see Automatically generated from ../decompiled/Nova.GigabitController.ProgramInnerData.decompiled.cs:905
 */
exports.SystemParameterConfig = t.intersection([
    t.type({
        ReduInfoList: common.XMLArray(SenderRedundancyInfo_1.SenderRedundancyInfo, 'SenderRedundancyInfo'),
        CaptureScreenFrame: common.withDefault(common.Int32, 30),
    }),
    t.partial({
        SacnBdProp: ScanBoardProperty_1.ScanBoardProperty,
        ScreenCfgInfo: common.Base64,
        SyncVideoData: common.Base64,
    }),
], 'SystemParameterConfig');
