// @novastar/native/build/main/generated/ThresholdType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThresholdType = exports.ThresholdTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var ThresholdTypeEnum;
(function (ThresholdTypeEnum) {
    ThresholdTypeEnum[ThresholdTypeEnum["None"] = 0] = "None";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade1"] = 1] = "Grade1";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade2"] = 2] = "Grade2";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade3"] = 3] = "Grade3";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade4"] = 4] = "Grade4";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade5"] = 5] = "Grade5";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade6"] = 6] = "Grade6";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade7"] = 7] = "Grade7";
    ThresholdTypeEnum[ThresholdTypeEnum["Grade8"] = 8] = "Grade8";
})(ThresholdTypeEnum = exports.ThresholdTypeEnum || (exports.ThresholdTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link ThresholdTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:61256
 */
exports.ThresholdType = (0, EnumFromString_1.default)(ThresholdTypeEnum, 'ThresholdType');
