// @novastar/native/build/main/generated/UserLoginType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserLoginType = exports.UserLoginTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var UserLoginTypeEnum;
(function (UserLoginTypeEnum) {
    UserLoginTypeEnum[UserLoginTypeEnum["UnLogin"] = 0] = "UnLogin";
    UserLoginTypeEnum[UserLoginTypeEnum["Admin"] = 1] = "Admin";
    UserLoginTypeEnum[UserLoginTypeEnum["Debuger"] = 2] = "Debuger";
    UserLoginTypeEnum[UserLoginTypeEnum["Windicator"] = 3] = "Windicator";
    UserLoginTypeEnum[UserLoginTypeEnum["AsyncLogin"] = 4] = "AsyncLogin";
})(UserLoginTypeEnum = exports.UserLoginTypeEnum || (exports.UserLoginTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link UserLoginTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.DataClass.decompiled.cs:56849
 */
exports.UserLoginType = (0, EnumFromString_1.default)(UserLoginTypeEnum, 'UserLoginType');
