// @novastar/native/build/main/generated/VirtualModeType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualModeType = exports.VirtualModeTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var VirtualModeTypeEnum;
(function (VirtualModeTypeEnum) {
    VirtualModeTypeEnum[VirtualModeTypeEnum["Disable"] = 0] = "Disable";
    VirtualModeTypeEnum[VirtualModeTypeEnum["Led4Mode1"] = 135] = "Led4Mode1";
    VirtualModeTypeEnum[VirtualModeTypeEnum["Led4Mode2"] = 136] = "Led4Mode2";
    VirtualModeTypeEnum[VirtualModeTypeEnum["Led3"] = 129] = "Led3";
    VirtualModeTypeEnum[VirtualModeTypeEnum["Led31"] = 130] = "Led31";
    VirtualModeTypeEnum[VirtualModeTypeEnum["Unknown"] = 255] = "Unknown";
})(VirtualModeTypeEnum = exports.VirtualModeTypeEnum || (exports.VirtualModeTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link VirtualModeTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:769
 */
exports.VirtualModeType = (0, EnumFromString_1.default)(VirtualModeTypeEnum, 'VirtualModeType');
