// @novastar/native/build/main/generated/WorkStatusType.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkStatusType = exports.WorkStatusTypeEnum = void 0;
const EnumFromString_1 = __importDefault(require("../lib/common/EnumFromString"));
/** @category Enums */
var WorkStatusTypeEnum;
(function (WorkStatusTypeEnum) {
    WorkStatusTypeEnum[WorkStatusTypeEnum["OK"] = 0] = "OK";
    WorkStatusTypeEnum[WorkStatusTypeEnum["Error"] = 1] = "Error";
    WorkStatusTypeEnum[WorkStatusTypeEnum["SenderCardError"] = 2] = "SenderCardError";
    WorkStatusTypeEnum[WorkStatusTypeEnum["Unknown"] = 255] = "Unknown";
})(WorkStatusTypeEnum = exports.WorkStatusTypeEnum || (exports.WorkStatusTypeEnum = {}));
/**
 * @category Codecs
 * @desc Codec for {@link WorkStatusTypeEnum}
 * @see Automatically generated from ../decompiled/Nova.LCT.GigabitSystem.ProtocolEnum.decompiled.cs:1053
 */
exports.WorkStatusType = (0, EnumFromString_1.default)(WorkStatusTypeEnum, 'WorkStatusType');
